// @multiversx/sdk-core/out/delegation/delegationTransactionsFactory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DelegationTransactionsFactory = void 0;
const abi_1 = require("../abi");
const address_1 = require("../core/address");
const constants_1 = require("../core/constants");
const errors_1 = require("../core/errors");
const transactionBuilder_1 = require("../core/transactionBuilder");
/**
 * Use this class to create delegation related transactions like creating a new delegation contract or adding nodes.
 */
class DelegationTransactionsFactory {
    constructor(options) {
        this.config = options.config;
        this.argSerializer = new abi_1.ArgSerializer();
        this.delegationManagerAddress = address_1.Address.newFromHex(constants_1.DELEGATION_MANAGER_SC_ADDRESS_HEX, this.config.addressHrp);
    }
    createTransactionForNewDelegationContract(sender, options) {
        const dataParts = [
            "createNewDelegationContract",
            ...this.argSerializer.valuesToStrings([
                new abi_1.BigUIntValue(options.totalDelegationCap),
                new abi_1.BigUIntValue(options.serviceFee),
            ]),
        ];
        const executionGasLimit = this.config.gasLimitCreateDelegationContract + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.delegationManagerAddress,
            dataParts: dataParts,
            gasLimit: executionGasLimit,
            addDataMovementGas: true,
            amount: options.amount,
        }).build();
    }
    createTransactionForAddingNodes(sender, options) {
        if (options.publicKeys.length !== options.signedMessages.length) {
            throw new errors_1.Err("The number of public keys should match the number of signed messages");
        }
        const signedMessagesAsTypedValues = options.signedMessages.map((message) => new abi_1.BytesValue(Buffer.from(message)));
        const messagesAsStrings = this.argSerializer.valuesToStrings(signedMessagesAsTypedValues);
        const numNodes = options.publicKeys.length;
        const dataParts = ["addNodes"];
        for (let i = 0; i < numNodes; i++) {
            dataParts.push(...[options.publicKeys[i].hex(), messagesAsStrings[i]]);
        }
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: this.computeExecutionGasLimitForNodesManagement(numNodes),
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForRemovingNodes(sender, options) {
        const dataParts = ["removeNodes"];
        for (const key of options.publicKeys) {
            dataParts.push(key.hex());
        }
        const numNodes = options.publicKeys.length;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: this.computeExecutionGasLimitForNodesManagement(numNodes),
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForStakingNodes(sender, options) {
        let dataParts = ["stakeNodes"];
        for (const key of options.publicKeys) {
            dataParts = dataParts.concat(key.hex());
        }
        const numNodes = options.publicKeys.length;
        const additionalGasForAllNodes = BigInt(numNodes) * this.config.additionalGasLimitPerValidatorNode;
        const executionGasLimit = additionalGasForAllNodes + this.config.gasLimitStake + this.config.gasLimitDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: executionGasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnbondingNodes(sender, options) {
        let dataParts = ["unBondNodes"];
        for (const key of options.publicKeys) {
            dataParts = dataParts.concat(key.hex());
        }
        const numNodes = options.publicKeys.length;
        const executionGasLimit = BigInt(numNodes) * this.config.additionalGasLimitPerValidatorNode +
            this.config.gasLimitUnbond +
            this.config.gasLimitDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: executionGasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnstakingNodes(sender, options) {
        let dataParts = ["unStakeNodes"];
        for (const key of options.publicKeys) {
            dataParts = dataParts.concat(key.hex());
        }
        const numNodes = options.publicKeys.length;
        const executionGasLimit = BigInt(numNodes) * this.config.additionalGasLimitPerValidatorNode +
            this.config.gasLimitUnstake +
            this.config.gasLimitDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: executionGasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnjailingNodes(sender, options) {
        const dataParts = ["unJailNodes"];
        for (const key of options.publicKeys) {
            dataParts.push(key.hex());
        }
        const numNodes = options.publicKeys.length;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: this.computeExecutionGasLimitForNodesManagement(numNodes),
            addDataMovementGas: true,
            amount: options.amount,
        }).build();
    }
    createTransactionForChangingServiceFee(sender, options) {
        const dataParts = [
            "changeServiceFee",
            this.argSerializer.valuesToStrings([new abi_1.BigUIntValue(options.serviceFee)])[0],
        ];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForModifyingDelegationCap(sender, options) {
        const dataParts = [
            "modifyTotalDelegationCap",
            this.argSerializer.valuesToStrings([new abi_1.BigUIntValue(options.delegationCap)])[0],
        ];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingAutomaticActivation(sender, options) {
        const dataParts = ["setAutomaticActivation", this.argSerializer.valuesToStrings([new abi_1.StringValue("true")])[0]];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnsettingAutomaticActivation(sender, options) {
        const dataParts = ["setAutomaticActivation", this.argSerializer.valuesToStrings([new abi_1.StringValue("false")])[0]];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingCapCheckOnRedelegateRewards(sender, options) {
        const dataParts = [
            "setCheckCapOnReDelegateRewards",
            this.argSerializer.valuesToStrings([new abi_1.StringValue("true")])[0],
        ];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnsettingCapCheckOnRedelegateRewards(sender, options) {
        const dataParts = [
            "setCheckCapOnReDelegateRewards",
            this.argSerializer.valuesToStrings([new abi_1.StringValue("false")])[0],
        ];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingMetadata(sender, options) {
        const dataParts = [
            "setMetaData",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.name),
                new abi_1.StringValue(options.website),
                new abi_1.StringValue(options.identifier),
            ]),
        ];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForDelegating(sender, options) {
        const dataParts = ["delegate"];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            amount: options.amount,
            addDataMovementGas: false,
        }).build();
    }
    createTransactionForClaimingRewards(sender, options) {
        const dataParts = ["claimRewards"];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: false,
        }).build();
    }
    createTransactionForRedelegatingRewards(sender, options) {
        const dataParts = ["reDelegateRewards"];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: false,
        }).build();
    }
    createTransactionForUndelegating(sender, options) {
        const dataParts = ["unDelegate", this.argSerializer.valuesToStrings([new abi_1.BigUIntValue(options.amount)])[0]];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: false,
        }).build();
    }
    createTransactionForWithdrawing(sender, options) {
        const dataParts = ["withdraw"];
        const gasLimit = this.config.gasLimitDelegationOperations + this.config.additionalGasLimitForDelegationOperations;
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: options.delegationContract,
            dataParts: dataParts,
            gasLimit: gasLimit,
            addDataMovementGas: false,
        }).build();
    }
    computeExecutionGasLimitForNodesManagement(numNodes) {
        const additionalGasForAllNodes = this.config.additionalGasLimitPerValidatorNode * BigInt(numNodes);
        return this.config.gasLimitDelegationOperations + additionalGasForAllNodes;
    }
}
exports.DelegationTransactionsFactory = DelegationTransactionsFactory;
