// @multiversx/sdk-dapp/__commonjs/UI/operaWallet/OperaWalletLoginButton/index.js
"use strict";var Am=Object.create;var Lo=Object.defineProperty,km=Object.defineProperties,Lm=Object.getOwnPropertyDescriptor,Pm=Object.getOwnPropertyDescriptors,Em=Object.getOwnPropertyNames,en=Object.getOwnPropertySymbols,Cm=Object.getPrototypeOf,tt=Object.prototype.hasOwnProperty,Rr=Object.prototype.propertyIsEnumerable;var Nr=(e,o,t)=>o in e?Lo(e,o,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[o]=t,u=(e,o)=>{for(var t in o||(o={}))tt.call(o,t)&&Nr(e,t,o[t]);if(en)for(var t of en(o))Rr.call(o,t)&&Nr(e,t,o[t]);return e},v=(e,o)=>km(e,Pm(o));var Ue=(e,o)=>{var t={};for(var s in e)tt.call(e,s)&&o.indexOf(s)<0&&(t[s]=e[s]);if(e!=null&&en)for(var s of en(e))o.indexOf(s)<0&&Rr.call(e,s)&&(t[s]=e[s]);return t};var h=(e,o)=>()=>(e&&(o=e(e=0)),o);var b=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),ie=(e,o)=>{for(var t in o)Lo(e,t,{get:o[t],enumerable:!0})},Dr=(e,o,t,s)=>{if(o&&typeof o=="object"||typeof o=="function")for(let c of Em(o))!tt.call(e,c)&&c!==t&&Lo(e,c,{get:()=>o[c],enumerable:!(s=Lm(o,c))||s.enumerable});return e};var x=(e,o,t)=>(t=e!=null?Am(Cm(e)):{},Dr(o||!e||!e.__esModule?Lo(t,"default",{value:e,enumerable:!0}):t,e)),ee=e=>Dr(Lo({},"__esModule",{value:!0}),e);var g=(e,o,t)=>new Promise((s,c)=>{var p=l=>{try{d(t.next(l))}catch(f){c(f)}},m=l=>{try{d(t.throw(l))}catch(f){c(f)}},d=l=>l.done?s(l.value):Promise.resolve(l.value).then(p,m);d((t=t.apply(e,o)).next())});var Br=b(on=>{"use strict";n();on.byteLength=Rm;on.toByteArray=Mm;on.fromByteArray=Fm;var ce=[],oe=[],Nm=typeof Uint8Array!="undefined"?Uint8Array:Array,rt="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Ge=0,Mr=rt.length;Ge<Mr;++Ge)ce[Ge]=rt[Ge],oe[rt.charCodeAt(Ge)]=Ge;var Ge,Mr;oe["-".charCodeAt(0)]=62;oe["_".charCodeAt(0)]=63;function Or(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var t=e.indexOf("=");t===-1&&(t=o);var s=t===o?0:4-t%4;return[t,s]}function Rm(e){var o=Or(e),t=o[0],s=o[1];return(t+s)*3/4-s}function Dm(e,o,t){return(o+t)*3/4-t}function Mm(e){var o,t=Or(e),s=t[0],c=t[1],p=new Nm(Dm(e,s,c)),m=0,d=c>0?s-4:s,l;for(l=0;l<d;l+=4)o=oe[e.charCodeAt(l)]<<18|oe[e.charCodeAt(l+1)]<<12|oe[e.charCodeAt(l+2)]<<6|oe[e.charCodeAt(l+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return c===2&&(o=oe[e.charCodeAt(l)]<<2|oe[e.charCodeAt(l+1)]>>4,p[m++]=o&255),c===1&&(o=oe[e.charCodeAt(l)]<<10|oe[e.charCodeAt(l+1)]<<4|oe[e.charCodeAt(l+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function Om(e){return ce[e>>18&63]+ce[e>>12&63]+ce[e>>6&63]+ce[e&63]}function Bm(e,o,t){for(var s,c=[],p=o;p<t;p+=3)s=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),c.push(Om(s));return c.join("")}function Fm(e){for(var o,t=e.length,s=t%3,c=[],p=16383,m=0,d=t-s;m<d;m+=p)c.push(Bm(e,m,m+p>d?d:m+p));return s===1?(o=e[t-1],c.push(ce[o>>2]+ce[o<<4&63]+"==")):s===2&&(o=(e[t-2]<<8)+e[t-1],c.push(ce[o>>10]+ce[o>>4&63]+ce[o<<2&63]+"=")),c.join("")}});var Fr=b(at=>{n();at.read=function(e,o,t,s,c){var p,m,d=c*8-s-1,l=(1<<d)-1,f=l>>1,S=-7,T=t?c-1:0,w=t?-1:1,E=e[o+T];for(T+=w,p=E&(1<<-S)-1,E>>=-S,S+=d;S>0;p=p*256+e[o+T],T+=w,S-=8);for(m=p&(1<<-S)-1,p>>=-S,S+=s;S>0;m=m*256+e[o+T],T+=w,S-=8);if(p===0)p=1-f;else{if(p===l)return m?NaN:(E?-1:1)*(1/0);m=m+Math.pow(2,s),p=p-f}return(E?-1:1)*m*Math.pow(2,p-s)};at.write=function(e,o,t,s,c,p){var m,d,l,f=p*8-c-1,S=(1<<f)-1,T=S>>1,w=c===23?Math.pow(2,-24)-Math.pow(2,-77):0,E=s?0:p-1,P=s?1:-1,N=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(d=isNaN(o)?1:0,m=S):(m=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+T>=1?o+=w/l:o+=w*Math.pow(2,1-T),o*l>=2&&(m++,l/=2),m+T>=S?(d=0,m=S):m+T>=1?(d=(o*l-1)*Math.pow(2,c),m=m+T):(d=o*Math.pow(2,T-1)*Math.pow(2,c),m=0));c>=8;e[t+E]=d&255,E+=P,d/=256,c-=8);for(m=m<<c|d,f+=c;f>0;e[t+E]=m&255,E+=P,m/=256,f-=8);e[t+E-P]|=N*128}});var Zr=b(ao=>{"use strict";n();var it=Br(),ro=Fr(),Wr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;ao.Buffer=_;ao.SlowBuffer=Vm;ao.INSPECT_MAX_BYTES=50;var nn=2147483647;ao.kMaxLength=nn;_.TYPED_ARRAY_SUPPORT=Wm();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Wm(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(t){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function fe(e){if(e>nn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,_.prototype),o}function _(e,o,t){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return mt(e)}return Hr(e,o,t)}_.poolSize=8192;function Hr(e,o,t){if(typeof e=="string")return Gm(e,o);if(ArrayBuffer.isView(e))return Hm(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(pe(e,ArrayBuffer)||e&&pe(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(pe(e,SharedArrayBuffer)||e&&pe(e.buffer,SharedArrayBuffer)))return ct(e,o,t);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var s=e.valueOf&&e.valueOf();if(s!=null&&s!==e)return _.from(s,o,t);var c=$m(e);if(c)return c;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return _.from(e[Symbol.toPrimitive]("string"),o,t);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}_.from=function(e,o,t){return Hr(e,o,t)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function $r(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Um(e,o,t){return $r(e),e<=0?fe(e):o!==void 0?typeof t=="string"?fe(e).fill(o,t):fe(e).fill(o):fe(e)}_.alloc=function(e,o,t){return Um(e,o,t)};function mt(e){return $r(e),fe(e<0?0:dt(e)|0)}_.allocUnsafe=function(e){return mt(e)};_.allocUnsafeSlow=function(e){return mt(e)};function Gm(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!_.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var t=Vr(e,o)|0,s=fe(t),c=s.write(e,o);return c!==t&&(s=s.slice(0,c)),s}function st(e){for(var o=e.length<0?0:dt(e.length)|0,t=fe(o),s=0;s<o;s+=1)t[s]=e[s]&255;return t}function Hm(e){if(pe(e,Uint8Array)){var o=new Uint8Array(e);return ct(o.buffer,o.byteOffset,o.byteLength)}return st(e)}function ct(e,o,t){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(t||0))throw new RangeError('"length" is outside of buffer bounds');var s;return o===void 0&&t===void 0?s=new Uint8Array(e):t===void 0?s=new Uint8Array(e,o):s=new Uint8Array(e,o,t),Object.setPrototypeOf(s,_.prototype),s}function $m(e){if(_.isBuffer(e)){var o=dt(e.length)|0,t=fe(o);return t.length===0||e.copy(t,0,0,o),t}if(e.length!==void 0)return typeof e.length!="number"||lt(e.length)?fe(0):st(e);if(e.type==="Buffer"&&Array.isArray(e.data))return st(e.data)}function dt(e){if(e>=nn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+nn.toString(16)+" bytes");return e|0}function Vm(e){return+e!=e&&(e=0),_.alloc(+e)}_.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==_.prototype};_.compare=function(o,t){if(pe(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),pe(t,Uint8Array)&&(t=_.from(t,t.offset,t.byteLength)),!_.isBuffer(o)||!_.isBuffer(t))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===t)return 0;for(var s=o.length,c=t.length,p=0,m=Math.min(s,c);p<m;++p)if(o[p]!==t[p]){s=o[p],c=t[p];break}return s<c?-1:c<s?1:0};_.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(o,t){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return _.alloc(0);var s;if(t===void 0)for(t=0,s=0;s<o.length;++s)t+=o[s].length;var c=_.allocUnsafe(t),p=0;for(s=0;s<o.length;++s){var m=o[s];if(pe(m,Uint8Array))p+m.length>c.length?_.from(m).copy(c,p):Uint8Array.prototype.set.call(c,m,p);else if(_.isBuffer(m))m.copy(c,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return c};function Vr(e,o){if(_.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||pe(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var t=e.length,s=arguments.length>2&&arguments[2]===!0;if(!s&&t===0)return 0;for(var c=!1;;)switch(o){case"ascii":case"latin1":case"binary":return t;case"utf8":case"utf-8":return pt(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return t*2;case"hex":return t>>>1;case"base64":return Yr(e).length;default:if(c)return s?-1:pt(e).length;o=(""+o).toLowerCase(),c=!0}}_.byteLength=Vr;function qm(e,o,t){var s=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((t===void 0||t>this.length)&&(t=this.length),t<=0)||(t>>>=0,o>>>=0,t<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return od(this,o,t);case"utf8":case"utf-8":return jr(this,o,t);case"ascii":return Qm(this,o,t);case"latin1":case"binary":return ed(this,o,t);case"base64":return Zm(this,o,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return nd(this,o,t);default:if(s)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),s=!0}}_.prototype._isBuffer=!0;function He(e,o,t){var s=e[o];e[o]=e[t],e[t]=s}_.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var t=0;t<o;t+=2)He(this,t,t+1);return this};_.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var t=0;t<o;t+=4)He(this,t,t+3),He(this,t+1,t+2);return this};_.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var t=0;t<o;t+=8)He(this,t,t+7),He(this,t+1,t+6),He(this,t+2,t+5),He(this,t+3,t+4);return this};_.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?jr(this,0,o):qm.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(o){if(!_.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:_.compare(this,o)===0};_.prototype.inspect=function(){var o="",t=ao.INSPECT_MAX_BYTES;return o=this.toString("hex",0,t).replace(/(.{2})/g,"$1 ").trim(),this.length>t&&(o+=" ... "),"<Buffer "+o+">"};Wr&&(_.prototype[Wr]=_.prototype.inspect);_.prototype.compare=function(o,t,s,c,p){if(pe(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),!_.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(t===void 0&&(t=0),s===void 0&&(s=o?o.length:0),c===void 0&&(c=0),p===void 0&&(p=this.length),t<0||s>o.length||c<0||p>this.length)throw new RangeError("out of range index");if(c>=p&&t>=s)return 0;if(c>=p)return-1;if(t>=s)return 1;if(t>>>=0,s>>>=0,c>>>=0,p>>>=0,this===o)return 0;for(var m=p-c,d=s-t,l=Math.min(m,d),f=this.slice(c,p),S=o.slice(t,s),T=0;T<l;++T)if(f[T]!==S[T]){m=f[T],d=S[T];break}return m<d?-1:d<m?1:0};function qr(e,o,t,s,c){if(e.length===0)return-1;if(typeof t=="string"?(s=t,t=0):t>2147483647?t=2147483647:t<-2147483648&&(t=-2147483648),t=+t,lt(t)&&(t=c?0:e.length-1),t<0&&(t=e.length+t),t>=e.length){if(c)return-1;t=e.length-1}else if(t<0)if(c)t=0;else return-1;if(typeof o=="string"&&(o=_.from(o,s)),_.isBuffer(o))return o.length===0?-1:Ur(e,o,t,s,c);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?c?Uint8Array.prototype.indexOf.call(e,o,t):Uint8Array.prototype.lastIndexOf.call(e,o,t):Ur(e,[o],t,s,c);throw new TypeError("val must be string, number or Buffer")}function Ur(e,o,t,s,c){var p=1,m=e.length,d=o.length;if(s!==void 0&&(s=String(s).toLowerCase(),s==="ucs2"||s==="ucs-2"||s==="utf16le"||s==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,d/=2,t/=2}function l(E,P){return p===1?E[P]:E.readUInt16BE(P*p)}var f;if(c){var S=-1;for(f=t;f<m;f++)if(l(e,f)===l(o,S===-1?0:f-S)){if(S===-1&&(S=f),f-S+1===d)return S*p}else S!==-1&&(f-=f-S),S=-1}else for(t+d>m&&(t=m-d),f=t;f>=0;f--){for(var T=!0,w=0;w<d;w++)if(l(e,f+w)!==l(o,w)){T=!1;break}if(T)return f}return-1}_.prototype.includes=function(o,t,s){return this.indexOf(o,t,s)!==-1};_.prototype.indexOf=function(o,t,s){return qr(this,o,t,s,!0)};_.prototype.lastIndexOf=function(o,t,s){return qr(this,o,t,s,!1)};function jm(e,o,t,s){t=Number(t)||0;var c=e.length-t;s?(s=Number(s),s>c&&(s=c)):s=c;var p=o.length;s>p/2&&(s=p/2);for(var m=0;m<s;++m){var d=parseInt(o.substr(m*2,2),16);if(lt(d))return m;e[t+m]=d}return m}function zm(e,o,t,s){return tn(pt(o,e.length-t),e,t,s)}function Km(e,o,t,s){return tn(ad(o),e,t,s)}function Xm(e,o,t,s){return tn(Yr(o),e,t,s)}function Ym(e,o,t,s){return tn(id(o,e.length-t),e,t,s)}_.prototype.write=function(o,t,s,c){if(t===void 0)c="utf8",s=this.length,t=0;else if(s===void 0&&typeof t=="string")c=t,s=this.length,t=0;else if(isFinite(t))t=t>>>0,isFinite(s)?(s=s>>>0,c===void 0&&(c="utf8")):(c=s,s=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-t;if((s===void 0||s>p)&&(s=p),o.length>0&&(s<0||t<0)||t>this.length)throw new RangeError("Attempt to write outside buffer bounds");c||(c="utf8");for(var m=!1;;)switch(c){case"hex":return jm(this,o,t,s);case"utf8":case"utf-8":return zm(this,o,t,s);case"ascii":case"latin1":case"binary":return Km(this,o,t,s);case"base64":return Xm(this,o,t,s);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Ym(this,o,t,s);default:if(m)throw new TypeError("Unknown encoding: "+c);c=(""+c).toLowerCase(),m=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Zm(e,o,t){return o===0&&t===e.length?it.fromByteArray(e):it.fromByteArray(e.slice(o,t))}function jr(e,o,t){t=Math.min(e.length,t);for(var s=[],c=o;c<t;){var p=e[c],m=null,d=p>239?4:p>223?3:p>191?2:1;if(c+d<=t){var l,f,S,T;switch(d){case 1:p<128&&(m=p);break;case 2:l=e[c+1],(l&192)===128&&(T=(p&31)<<6|l&63,T>127&&(m=T));break;case 3:l=e[c+1],f=e[c+2],(l&192)===128&&(f&192)===128&&(T=(p&15)<<12|(l&63)<<6|f&63,T>2047&&(T<55296||T>57343)&&(m=T));break;case 4:l=e[c+1],f=e[c+2],S=e[c+3],(l&192)===128&&(f&192)===128&&(S&192)===128&&(T=(p&15)<<18|(l&63)<<12|(f&63)<<6|S&63,T>65535&&T<1114112&&(m=T))}}m===null?(m=65533,d=1):m>65535&&(m-=65536,s.push(m>>>10&1023|55296),m=56320|m&1023),s.push(m),c+=d}return Jm(s)}var Gr=4096;function Jm(e){var o=e.length;if(o<=Gr)return String.fromCharCode.apply(String,e);for(var t="",s=0;s<o;)t+=String.fromCharCode.apply(String,e.slice(s,s+=Gr));return t}function Qm(e,o,t){var s="";t=Math.min(e.length,t);for(var c=o;c<t;++c)s+=String.fromCharCode(e[c]&127);return s}function ed(e,o,t){var s="";t=Math.min(e.length,t);for(var c=o;c<t;++c)s+=String.fromCharCode(e[c]);return s}function od(e,o,t){var s=e.length;(!o||o<0)&&(o=0),(!t||t<0||t>s)&&(t=s);for(var c="",p=o;p<t;++p)c+=sd[e[p]];return c}function nd(e,o,t){for(var s=e.slice(o,t),c="",p=0;p<s.length-1;p+=2)c+=String.fromCharCode(s[p]+s[p+1]*256);return c}_.prototype.slice=function(o,t){var s=this.length;o=~~o,t=t===void 0?s:~~t,o<0?(o+=s,o<0&&(o=0)):o>s&&(o=s),t<0?(t+=s,t<0&&(t=0)):t>s&&(t=s),t<o&&(t=o);var c=this.subarray(o,t);return Object.setPrototypeOf(c,_.prototype),c};function U(e,o,t){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>t)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=this[o],p=1,m=0;++m<t&&(p*=256);)c+=this[o+m]*p;return c};_.prototype.readUintBE=_.prototype.readUIntBE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=this[o+--t],p=1;t>0&&(p*=256);)c+=this[o+--t]*p;return c};_.prototype.readUint8=_.prototype.readUInt8=function(o,t){return o=o>>>0,t||U(o,1,this.length),this[o]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(o,t){return o=o>>>0,t||U(o,2,this.length),this[o]|this[o+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(o,t){return o=o>>>0,t||U(o,2,this.length),this[o]<<8|this[o+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(o,t){return o=o>>>0,t||U(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(o,t){return o=o>>>0,t||U(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};_.prototype.readIntLE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=this[o],p=1,m=0;++m<t&&(p*=256);)c+=this[o+m]*p;return p*=128,c>=p&&(c-=Math.pow(2,8*t)),c};_.prototype.readIntBE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=t,p=1,m=this[o+--c];c>0&&(p*=256);)m+=this[o+--c]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*t)),m};_.prototype.readInt8=function(o,t){return o=o>>>0,t||U(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};_.prototype.readInt16LE=function(o,t){o=o>>>0,t||U(o,2,this.length);var s=this[o]|this[o+1]<<8;return s&32768?s|4294901760:s};_.prototype.readInt16BE=function(o,t){o=o>>>0,t||U(o,2,this.length);var s=this[o+1]|this[o]<<8;return s&32768?s|4294901760:s};_.prototype.readInt32LE=function(o,t){return o=o>>>0,t||U(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};_.prototype.readInt32BE=function(o,t){return o=o>>>0,t||U(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};_.prototype.readFloatLE=function(o,t){return o=o>>>0,t||U(o,4,this.length),ro.read(this,o,!0,23,4)};_.prototype.readFloatBE=function(o,t){return o=o>>>0,t||U(o,4,this.length),ro.read(this,o,!1,23,4)};_.prototype.readDoubleLE=function(o,t){return o=o>>>0,t||U(o,8,this.length),ro.read(this,o,!0,52,8)};_.prototype.readDoubleBE=function(o,t){return o=o>>>0,t||U(o,8,this.length),ro.read(this,o,!1,52,8)};function j(e,o,t,s,c,p){if(!_.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>c||o<p)throw new RangeError('"value" argument is out of bounds');if(t+s>e.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(o,t,s,c){if(o=+o,t=t>>>0,s=s>>>0,!c){var p=Math.pow(2,8*s)-1;j(this,o,t,s,p,0)}var m=1,d=0;for(this[t]=o&255;++d<s&&(m*=256);)this[t+d]=o/m&255;return t+s};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(o,t,s,c){if(o=+o,t=t>>>0,s=s>>>0,!c){var p=Math.pow(2,8*s)-1;j(this,o,t,s,p,0)}var m=s-1,d=1;for(this[t+m]=o&255;--m>=0&&(d*=256);)this[t+m]=o/d&255;return t+s};_.prototype.writeUint8=_.prototype.writeUInt8=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,1,255,0),this[t]=o&255,t+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,65535,0),this[t]=o&255,this[t+1]=o>>>8,t+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,65535,0),this[t]=o>>>8,this[t+1]=o&255,t+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,4294967295,0),this[t+3]=o>>>24,this[t+2]=o>>>16,this[t+1]=o>>>8,this[t]=o&255,t+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,4294967295,0),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};_.prototype.writeIntLE=function(o,t,s,c){if(o=+o,t=t>>>0,!c){var p=Math.pow(2,8*s-1);j(this,o,t,s,p-1,-p)}var m=0,d=1,l=0;for(this[t]=o&255;++m<s&&(d*=256);)o<0&&l===0&&this[t+m-1]!==0&&(l=1),this[t+m]=(o/d>>0)-l&255;return t+s};_.prototype.writeIntBE=function(o,t,s,c){if(o=+o,t=t>>>0,!c){var p=Math.pow(2,8*s-1);j(this,o,t,s,p-1,-p)}var m=s-1,d=1,l=0;for(this[t+m]=o&255;--m>=0&&(d*=256);)o<0&&l===0&&this[t+m+1]!==0&&(l=1),this[t+m]=(o/d>>0)-l&255;return t+s};_.prototype.writeInt8=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,1,127,-128),o<0&&(o=255+o+1),this[t]=o&255,t+1};_.prototype.writeInt16LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,32767,-32768),this[t]=o&255,this[t+1]=o>>>8,t+2};_.prototype.writeInt16BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,32767,-32768),this[t]=o>>>8,this[t+1]=o&255,t+2};_.prototype.writeInt32LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,2147483647,-2147483648),this[t]=o&255,this[t+1]=o>>>8,this[t+2]=o>>>16,this[t+3]=o>>>24,t+4};_.prototype.writeInt32BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};function zr(e,o,t,s,c,p){if(t+s>e.length)throw new RangeError("Index out of range");if(t<0)throw new RangeError("Index out of range")}function Kr(e,o,t,s,c){return o=+o,t=t>>>0,c||zr(e,o,t,4,34028234663852886e22,-34028234663852886e22),ro.write(e,o,t,s,23,4),t+4}_.prototype.writeFloatLE=function(o,t,s){return Kr(this,o,t,!0,s)};_.prototype.writeFloatBE=function(o,t,s){return Kr(this,o,t,!1,s)};function Xr(e,o,t,s,c){return o=+o,t=t>>>0,c||zr(e,o,t,8,17976931348623157e292,-17976931348623157e292),ro.write(e,o,t,s,52,8),t+8}_.prototype.writeDoubleLE=function(o,t,s){return Xr(this,o,t,!0,s)};_.prototype.writeDoubleBE=function(o,t,s){return Xr(this,o,t,!1,s)};_.prototype.copy=function(o,t,s,c){if(!_.isBuffer(o))throw new TypeError("argument should be a Buffer");if(s||(s=0),!c&&c!==0&&(c=this.length),t>=o.length&&(t=o.length),t||(t=0),c>0&&c<s&&(c=s),c===s||o.length===0||this.length===0)return 0;if(t<0)throw new RangeError("targetStart out of bounds");if(s<0||s>=this.length)throw new RangeError("Index out of range");if(c<0)throw new RangeError("sourceEnd out of bounds");c>this.length&&(c=this.length),o.length-t<c-s&&(c=o.length-t+s);var p=c-s;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(t,s,c):Uint8Array.prototype.set.call(o,this.subarray(s,c),t),p};_.prototype.fill=function(o,t,s,c){if(typeof o=="string"){if(typeof t=="string"?(c=t,t=0,s=this.length):typeof s=="string"&&(c=s,s=this.length),c!==void 0&&typeof c!="string")throw new TypeError("encoding must be a string");if(typeof c=="string"&&!_.isEncoding(c))throw new TypeError("Unknown encoding: "+c);if(o.length===1){var p=o.charCodeAt(0);(c==="utf8"&&p<128||c==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(t<0||this.length<t||this.length<s)throw new RangeError("Out of range index");if(s<=t)return this;t=t>>>0,s=s===void 0?this.length:s>>>0,o||(o=0);var m;if(typeof o=="number")for(m=t;m<s;++m)this[m]=o;else{var d=_.isBuffer(o)?o:_.from(o,c),l=d.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<s-t;++m)this[m+t]=d[m%l]}return this};var td=/[^+/0-9A-Za-z-_]/g;function rd(e){if(e=e.split("=")[0],e=e.trim().replace(td,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function pt(e,o){o=o||1/0;for(var t,s=e.length,c=null,p=[],m=0;m<s;++m){if(t=e.charCodeAt(m),t>55295&&t<57344){if(!c){if(t>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===s){(o-=3)>-1&&p.push(239,191,189);continue}c=t;continue}if(t<56320){(o-=3)>-1&&p.push(239,191,189),c=t;continue}t=(c-55296<<10|t-56320)+65536}else c&&(o-=3)>-1&&p.push(239,191,189);if(c=null,t<128){if((o-=1)<0)break;p.push(t)}else if(t<2048){if((o-=2)<0)break;p.push(t>>6|192,t&63|128)}else if(t<65536){if((o-=3)<0)break;p.push(t>>12|224,t>>6&63|128,t&63|128)}else if(t<1114112){if((o-=4)<0)break;p.push(t>>18|240,t>>12&63|128,t>>6&63|128,t&63|128)}else throw new Error("Invalid code point")}return p}function ad(e){for(var o=[],t=0;t<e.length;++t)o.push(e.charCodeAt(t)&255);return o}function id(e,o){for(var t,s,c,p=[],m=0;m<e.length&&!((o-=2)<0);++m)t=e.charCodeAt(m),s=t>>8,c=t%256,p.push(c),p.push(s);return p}function Yr(e){return it.toByteArray(rd(e))}function tn(e,o,t,s){for(var c=0;c<s&&!(c+t>=o.length||c>=e.length);++c)o[c+t]=e[c];return c}function pe(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function lt(e){return e!==e}var sd=function(){for(var e="0123456789abcdef",o=new Array(256),t=0;t<16;++t)for(var s=t*16,c=0;c<16;++c)o[s+c]=e[t]+e[c];return o}()});var na=b((rx,oa)=>{n();var O=oa.exports={},me,de;function _t(){throw new Error("setTimeout has not been defined")}function ut(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?me=setTimeout:me=_t}catch(e){me=_t}try{typeof clearTimeout=="function"?de=clearTimeout:de=ut}catch(e){de=ut}})();function Jr(e){if(me===setTimeout)return setTimeout(e,0);if((me===_t||!me)&&setTimeout)return me=setTimeout,setTimeout(e,0);try{return me(e,0)}catch(o){try{return me.call(null,e,0)}catch(t){return me.call(this,e,0)}}}function cd(e){if(de===clearTimeout)return clearTimeout(e);if((de===ut||!de)&&clearTimeout)return de=clearTimeout,clearTimeout(e);try{return de(e)}catch(o){try{return de.call(null,e)}catch(t){return de.call(this,e)}}}var Te=[],io=!1,$e,rn=-1;function pd(){!io||!$e||(io=!1,$e.length?Te=$e.concat(Te):rn=-1,Te.length&&Qr())}function Qr(){if(!io){var e=Jr(pd);io=!0;for(var o=Te.length;o;){for($e=Te,Te=[];++rn<o;)$e&&$e[rn].run();rn=-1,o=Te.length}$e=null,io=!1,cd(e)}}O.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var t=1;t<arguments.length;t++)o[t-1]=arguments[t];Te.push(new ea(e,o)),Te.length===1&&!io&&Jr(Qr)};function ea(e,o){this.fun=e,this.array=o}ea.prototype.run=function(){this.fun.apply(null,this.array)};O.title="browser";O.browser=!0;O.env={};O.argv=[];O.version="";O.versions={};function he(){}O.on=he;O.addListener=he;O.once=he;O.off=he;O.removeListener=he;O.removeAllListeners=he;O.emit=he;O.prependListener=he;O.prependOnceListener=he;O.listeners=function(e){return[]};O.binding=function(e){throw new Error("process.binding is not supported")};O.cwd=function(){return"/"};O.chdir=function(e){throw new Error("process.chdir is not supported")};O.umask=function(){return 0}});var r,i,md,a,n=h(()=>{r=x(Zr()),i=x(na()),md=function(e){function o(){var s=this||self;return delete e.prototype.__magic__,s}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var t=__magic__;return t}(Object),a=md});var ta=h(()=>{"use strict";n()});var xe,le=h(()=>{"use strict";n();xe="Action not allowed. User is logged in. Call logout() first"});var ra=h(()=>{"use strict";n()});var gt,aa=h(()=>{"use strict";n();gt=(s=>(s.SetGuardian="SetGuardian",s.GuardAccount="GuardAccount",s.UnGuardAccount="UnGuardAccount",s))(gt||{})});var ia=h(()=>{"use strict";n()});var ft=h(()=>{"use strict";n()});var sa=h(()=>{"use strict";n()});var Tt=h(()=>{"use strict";n()});var ca=h(()=>{"use strict";n()});var pa=h(()=>{"use strict";n()});var Ne,so,Re=h(()=>{"use strict";n();Ne=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),so=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var F,ma,fx,da,Tx,C=h(()=>{"use strict";n();Re();F=(w=>(w.ledger="ledger",w.walletconnect="walletconnect",w.walletconnectv2="walletconnectv2",w.wallet="wallet",w.crossWindow="crossWindow",w.iframe="iframe",w.extension="extension",w.passkey="passkey",w.metamask="metamask",w.opera="opera",w.extra="extra",w.none="",w))(F||{}),ma=(t=>(t.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",t.reloginRequest="RELOGIN_REQUEST",t))(ma||{}),fx=u(u({},so.WindowProviderRequestEnums),ma),da=(o=>(o.reloginResponse="RELOGIN_RESPONSE",o))(da||{}),Tx=u(u({},so.WindowProviderResponseEnums),da)});var la=h(()=>{"use strict";n()});var _a=h(()=>{"use strict";n()});var z=h(()=>{"use strict";n()});var ua=h(()=>{"use strict";n()});var ga=h(()=>{"use strict";n()});var fa=h(()=>{"use strict";n()});var B=h(()=>{"use strict";n();Tt();ca();pa();C();la();_a();z();ua();ga();fa()});var co,Ta,Dx,ha,Mx,xa,Ox,Bx,dd,ya,po=h(()=>{"use strict";n();B();co={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:Ta,egldLabel:Dx}=co["devnet"],{chainId:ha,egldLabel:Mx}=co["testnet"],{chainId:xa,egldLabel:Ox}=co["mainnet"],Bx={["devnet"]:Ta,["testnet"]:ha,["mainnet"]:xa},dd={[Ta]:"devnet",[ha]:"testnet",[xa]:"mainnet"},ya="multiversx://"});var Po=h(()=>{"use strict";n()});var _e,Sa=h(()=>{"use strict";n();_e=require("@multiversx/sdk-web-wallet-provider")});var q,Ve=h(()=>{"use strict";n();q=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var ba,wa,ht,$x,Vx,xt=h(()=>{"use strict";n();Ve();ht=String((wa=(ba=q.safeWindow)==null?void 0:ba.navigator)==null?void 0:wa.userAgent),$x=/^((?!chrome|android).)*safari/i.test(ht),Vx=/firefox/i.test(ht)&&/windows/i.test(ht)});var an,sn,va,cn,Ia,mo,k=h(()=>{"use strict";n();ta();le();ra();aa();ia();ft();sa();po();Po();Sa();xt();an=5e4,sn=1e9,va=1,cn="logout",Ia="login",mo="0"});var lo,pn=h(()=>{"use strict";n();lo=()=>Date.now()/1e3});var Aa=h(()=>{"use strict";n()});var ka=h(()=>{"use strict";n()});var yt=h(()=>{"use strict";n();pn();Aa();ka()});var St={};ie(St,{clear:()=>gd,getItem:()=>_d,localStorageKeys:()=>ye,removeItem:()=>ud,setItem:()=>ld});var ye,mn,ld,_d,ud,gd,qe=h(()=>{"use strict";n();yt();ye={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},mn=typeof localStorage!="undefined",ld=({key:e,data:o,expires:t})=>{!mn||localStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},_d=e=>{if(!mn)return;let o=localStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:lo()>=t.expires?(localStorage.removeItem(String(e)),null):t.data},ud=e=>{!mn||localStorage.removeItem(String(e))},gd=()=>{!mn||localStorage.clear()}});var bt={};ie(bt,{clear:()=>Ca,getItem:()=>Pa,removeItem:()=>Ea,setItem:()=>La,storage:()=>fd});var La,Pa,Ea,Ca,fd,Na=h(()=>{"use strict";n();La=({key:e,data:o,expires:t})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},Pa=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Date.now()>=t.expires?(sessionStorage.removeItem(String(e)),null):t.data},Ea=e=>sessionStorage.removeItem(String(e)),Ca=()=>sessionStorage.clear(),fd={setItem:La,getItem:Pa,removeItem:Ea,clear:Ca}});var De,je=h(()=>{"use strict";n();qe();Na();De={session:bt,local:St}});var wt,R,K,D=h(()=>{"use strict";n();wt=require("@reduxjs/toolkit");k();R=(0,wt.createAction)(cn),K=(0,wt.createAction)(Ia,e=>({payload:e}))});var It,Ra,Da,X,vt,Ma,ne,Se,At,Sy,Td,hd,by,wy,vy,xd,yd,dn,ln=h(()=>{"use strict";n();It=require("@multiversx/sdk-core"),Ra=require("@reduxjs/toolkit"),Da=require("redux-persist");k();je();qe();D();X={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:mo},vt={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":X},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Ma=(0,Ra.createSlice)({name:"accountInfoSlice",initialState:vt,reducers:{setAddress:(e,o)=>{let t=o.payload;e.address=t,e.publicKey=t?new It.Address(t).hex():""},setAccount:(e,o)=>{let t=e.address===o.payload.address;e.accounts={[e.address]:t?o.payload:X},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:t}=e;e.accounts[t].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}},setWebsocketBatchEvent:(e,o)=>{e.websocketBatchEvent={timestamp:Date.now(),data:o.payload}}},extraReducers:e=>{e.addCase(R,()=>(De.local.removeItem(ye.loginExpiresAt),vt)),e.addCase(K,(o,t)=>{let{address:s}=t.payload;o.address=s,o.publicKey=new It.Address(s).toHex()}),e.addCase(Da.REHYDRATE,(o,t)=>{var f;if(!((f=t.payload)!=null&&f.account))return;let{account:s}=t.payload,{address:c,shard:p,accounts:m,publicKey:d}=s;o.address=c,o.shard=p;let l=m&&c in m;o.accounts=l?m:vt.accounts,o.publicKey=d})}}),{setAccount:ne,setAddress:Se,setAccountNonce:At,setAccountShard:Sy,setLedgerAccount:Td,updateLedgerAccount:hd,setWalletConnectAccount:by,setIsAccountLoading:wy,setAccountLoadingError:vy,setWebsocketEvent:xd,setWebsocketBatchEvent:yd}=Ma.actions,dn=Ma.reducer});function Eo(){return new Date().setHours(new Date().getHours()+24)}function Co(e){De.local.setItem({key:ye.loginExpiresAt,data:e,expires:e})}var kt=h(()=>{"use strict";n();je();qe()});var Ba,Oa,Fa,Dy,Sd,bd,No,My,wd,Wa,Oy,By,Fy,_n,un=h(()=>{"use strict";n();Ba=require("@reduxjs/toolkit");kt();C();D();Oa={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},Fa=(0,Ba.createSlice)({name:"loginInfoSlice",initialState:Oa,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},setWebviewLogin:(e,o)=>{e.webviewLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload},setIsWalletConnectV2Initialized:(e,o)=>{e.isWalletConnectV2Initialized=o.payload}},extraReducers:e=>{e.addCase(R,()=>Oa),e.addCase(K,(o,t)=>{o.isLoginSessionInvalid=!1,o.loginMethod=t.payload.loginMethod,o.iframeLoginType=t.payload.iframeLoginType,Co(Eo())})}}),{setLoginMethod:Dy,setWalletConnectLogin:Sd,setLedgerLogin:bd,setTokenLogin:No,setTokenLoginSignature:My,setWalletLogin:wd,invalidateLoginSession:Wa,setLogoutRoute:Oy,setIsWalletConnectV2Initialized:By,setWebviewLogin:Fy}=Fa.actions,_n=Fa.reducer});var Ga,Ua,Ha,Hy,vd,$y,Id,gn,fn=h(()=>{"use strict";n();Ga=require("@reduxjs/toolkit");D();Ua={},Ha=(0,Ga.createSlice)({name:"modalsSlice",initialState:Ua,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(R,()=>Ua)}}),{setTxSubmittedModal:Hy,setNotificationModal:vd,clearTxSubmittedModal:$y,clearNotificationModal:Id}=Ha.actions,gn=Ha.reducer});var ze,_o=h(()=>{"use strict";n();ze=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var M,H=h(()=>{"use strict";n();_o();M=()=>{if(!ze())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:o,hash:t,origin:s,href:c,search:p}}=window;return{pathname:o,hash:t,origin:s,href:c,search:p}}});var $a=h(()=>{"use strict";n();H()});var Ke,Va=h(()=>{"use strict";n();$();Ke=()=>{let{pathname:e,search:o,hash:t}=M();return`${e!=null?e:""}${o!=null?o:""}${t!=null?t:""}`}});var qa,ja=h(()=>{"use strict";n();Ve();qa=e=>{if(!q.safeWindow.location)return;if(!new URL(q.safeWindow.location.href).protocol.startsWith("http"))return q.safeWindow.location.reload();q.safeWindow.location.assign(e)}});var $=h(()=>{"use strict";n();$a();Va();H();ja()});var Ad,Tn,Ro=h(()=>{"use strict";n();$();Ad=!1,Tn=({timeout:e=0,url:o})=>{Ad||setTimeout(()=>{if(!!window)return qa(o)},e)}});function Me({callbackRoute:e,onLoginRedirect:o,options:t}){let s=Boolean(e),c=typeof o=="function";if(s&&e!=null){if(c)return o(e,t);Tn({url:e,timeout:kd})}}var kd,za=h(()=>{"use strict";n();Ro();kd=200});function Ka(e){return e[Math.floor(Math.random()*e.length)]}var Xa=h(()=>{"use strict";n()});var Lt=h(()=>{"use strict";n();ft()});var be=h(()=>{"use strict";n();za();Xa();Lt()});var Ya,Za,Ja,Pt,Pd,Qa,SS,bS,ei,hn,xn=h(()=>{"use strict";n();Ya=require("@reduxjs/toolkit"),Za=x(require("lodash.omit")),Ja=require("redux-persist");Po();D();be();Pt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},Pd={network:Pt},Qa=(0,Ya.createSlice)({name:"appConfig",initialState:Pd,reducers:{initializeNetworkConfig:(e,o)=>{let t=Ka(o.payload.walletConnectV2RelayAddresses),s=(0,Za.default)(o.payload,"walletConnectV2RelayAddresses");e.network=v(u(u({},e.network),s),{walletConnectV2RelayAddress:t})},updateNetworkConfig:(e,o)=>{e.network=u(u({},e.network),o.payload)},setCustomWalletAddress:(e,o)=>{e.network.customWalletAddress=o.payload}},extraReducers:e=>{e.addCase(R,o=>{o.network.customWalletAddress=void 0}),e.addCase(Ja.REHYDRATE,(o,t)=>{var c,p;if(!((p=(c=t.payload)==null?void 0:c.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:s}}=t.payload;o.network.customWalletAddress=s})}}),{initializeNetworkConfig:SS,updateNetworkConfig:bS,setCustomWalletAddress:ei}=Qa.actions,hn=Qa.reducer});var oi,Et,ni,LS,PS,ES,yn,Sn=h(()=>{"use strict";n();oi=require("@reduxjs/toolkit");B();D();Et={isSigning:!1,signedSessions:{}},ni=(0,oi.createSlice)({name:"signedMessageInfoSliceState",initialState:Et,reducers:{setSignSession:(e,o)=>{let{sessionId:t,signedSession:s,errorMessage:c}=o.payload;c&&(e.errorMessage=c),e.isSigning=s.status==="pending",e.signedSessions[t]=u(u({},e.signedSessions[t]),s)},setSignSessionState:(e,o)=>u(u({},e),o.payload),clearSignedMessageInfo:()=>Et},extraReducers:e=>{e.addCase(R,()=>Et)}}),{setSignSession:LS,clearSignedMessageInfo:PS,setSignSessionState:ES}=ni.actions,yn=ni.reducer});var ri,ai,ti,ii,Ed,Cd,BS,Nd,bn,wn=h(()=>{"use strict";n();ri=require("@reduxjs/toolkit"),ai=require("redux-persist");B();pn();D();ti={customToasts:[],transactionToasts:[]},ii=(0,ri.createSlice)({name:"toastsSlice",initialState:ti,reducers:{addCustomToast:(e,o)=>{let t=o.payload.toastId||`custom-toast-${e.customToasts.length+1}`,s=e.customToasts.findIndex(c=>c.toastId===t);if(s!==-1){e.customToasts[s]=v(u(u({},e.customToasts[s]),o.payload),{type:"custom",toastId:t});return}e.customToasts.push(v(u({},o.payload),{type:"custom",toastId:t}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(t=>t.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:lo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(t=>t.toastId!==o.payload)}},extraReducers:e=>{e.addCase(R,()=>ti),e.addCase(ai.REHYDRATE,(o,t)=>{var c,p;let s=(p=(c=t.customToasts)==null?void 0:c.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=s})}}),{addCustomToast:Ed,removeCustomToast:Cd,addTransactionToast:BS,removeTransactionToast:Nd}=ii.actions,bn=ii.reducer});var si,Nt,Rt,Dt,Rd,Ct,ci,GS,Dd,Mt,vn,In=h(()=>{"use strict";n();si=require("@reduxjs/toolkit");D();Nt="Transaction failed",Rt="Transaction successful",Dt="Processing transaction",Rd="Transaction submitted",Ct={},ci=(0,si.createSlice)({name:"transactionsInfo",initialState:Ct,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:t,transactionsDisplayInfo:s}=o.payload;t!=null&&(e[t]={errorMessage:(s==null?void 0:s.errorMessage)||Nt,successMessage:(s==null?void 0:s.successMessage)||Rt,processingMessage:(s==null?void 0:s.processingMessage)||Dt,submittedMessage:(s==null?void 0:s.submittedMessage)||Rd,transactionDuration:s==null?void 0:s.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>Ct},extraReducers:e=>{e.addCase(R,()=>Ct)}}),{clearTransactionsInfo:GS,setTransactionsDisplayInfo:Dd,clearTransactionsInfoForSessionId:Mt}=ci.actions,vn=ci.reducer});function go(e){return e!=null&&(Vd(e)||Xd(e))}function fo(e){return e!=null&&(qd(e)||Yd(e))}function To(e){return e!=null&&(jd(e)||Zd(e))}function pi(e){return e!=null&&(zd(e)||Jd(e))}function Ot(e){return e!=null&&Kd(e)}function Vd(e){return e!=null&&Md.includes(e)}function qd(e){return e!=null&&Od.includes(e)}function jd(e){return e!=null&&Bd.includes(e)}function zd(e){return e!=null&&Fd.includes(e)}function Kd(e){return e!=null&&Wd.includes(e)}function Xd(e){return e!=null&&Ud.includes(e)}function Yd(e){return e!=null&&Gd.includes(e)}function Zd(e){return e!=null&&Hd.includes(e)}function Jd(e){return e!=null&&$d.includes(e)}var Md,Od,Bd,Fd,Wd,Ud,Gd,Hd,$d,ho=h(()=>{"use strict";n();C();Md=["sent"],Od=["success"],Bd=["fail","cancelled","timedOut"],Fd=["invalid"],Wd=["timedOut"],Ud=["pending"],Gd=["success"],Hd=["fail","invalid"],$d=["not executed"]});var mi,di,Do,Qd,li,_i,ui,el,An,ol,nl,YS,tl,Mo,Bt,ZS,kn,Ln=h(()=>{"use strict";n();mi=require("@reduxjs/toolkit"),di=require("redux-persist");C();ho();D();Do={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Qd={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},li=(0,mi.createSlice)({name:"transactionsSlice",initialState:Do,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,f;let{sessionId:t,transactions:s,errorMessage:c,status:p,redirectRoute:m,customTransactionInformation:d}=o.payload;e.customTransactionInformationForSessionId[t]=u(u(u({},Qd),((l=e.signedTransactions[t])==null?void 0:l.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[t]||{}),e.signedTransactions[t]={transactions:s,status:p,errorMessage:c,redirectRoute:m,customTransactionInformation:u(u({},e.customTransactionInformationForSessionId[t]),d!=null?d:{})},((f=e==null?void 0:e.transactionsToSign)==null?void 0:f.sessionId)===t&&(e.transactionsToSign=Do.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=u(u({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:t,status:s,errorMessage:c,transactions:p}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].status=s,c!=null&&(e.signedTransactions[t].errorMessage=c),p!=null&&(e.signedTransactions[t].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var f,S,T,w,E,P,N,V;let{sessionId:t,status:s,errorMessage:c,transactionHash:p,serverTransaction:m,inTransit:d}=o.payload,l=(S=(f=e.signedTransactions)==null?void 0:f[t])==null?void 0:S.transactions;if(l!=null){e.signedTransactions[t].transactions=l.map(W=>W.hash===p?v(u(u({},m!=null?m:{}),W),{status:s,errorMessage:c,inTransit:d}):W);let Q=(w=(T=e.signedTransactions[t])==null?void 0:T.transactions)==null?void 0:w.every(W=>fo(W.status)),ae=(P=(E=e.signedTransactions[t])==null?void 0:E.transactions)==null?void 0:P.some(W=>To(W.status)),Ce=(V=(N=e.signedTransactions[t])==null?void 0:N.transactions)==null?void 0:V.every(W=>pi(W.status));Q&&(e.signedTransactions[t].status="success"),ae&&(e.signedTransactions[t].status="fail"),Ce&&(e.signedTransactions[t].status="invalid")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:t,customTransactionInformation:s}=o.payload;e.customTransactionInformationForSessionId[t]=s,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Do.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Do.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload},updateSignedTransactionsCustomTransactionInformation:(e,o)=>{let{sessionId:t,customTransactionInformationOverrides:s}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].customTransactionInformation=u(u({},e.signedTransactions[t].customTransactionInformation),s))}},extraReducers:e=>{e.addCase(R,()=>Do),e.addCase(di.REHYDRATE,(o,t)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=t.payload)!=null&&m.transactions))return;let{signedTransactions:s,customTransactionInformationForSessionId:c}=t.payload.transactions,p=Object.entries(s).reduce((d,[l,f])=>{let S=new Date(l),T=new Date;return T.setHours(T.getHours()+5),T-S>0||(d[l]=f),d},{});c!=null&&(o.customTransactionInformationForSessionId=c),s!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:_i,updateSignedTransactions:ui,setTransactionsToSign:el,clearAllTransactionsToSign:An,clearAllSignedTransactions:ol,clearSignedTransaction:nl,clearTransactionToSign:YS,setSignTransactionsError:tl,setSignTransactionsCancelMessage:Mo,moveTransactionsToSignedState:Bt,updateSignedTransactionsCustomTransactionInformation:ZS}=li.actions,kn=li.reducer});var gi,Ft,fi,ob,nb,rl,tb,Pn,En=h(()=>{"use strict";n();gi=require("@reduxjs/toolkit");D();Ft={},fi=(0,gi.createSlice)({name:"batchTransactionsSlice",initialState:Ft,reducers:{setBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},updateBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},clearBatchTransactions:(e,o)=>{delete e[o.payload.batchId]},clearAllBatchTransactions:()=>Ft},extraReducers:e=>{e.addCase(R,()=>Ft)}}),{setBatchTransactions:ob,updateBatchTransactions:nb,clearBatchTransactions:rl,clearAllBatchTransactions:tb}=fi.actions,Pn=fi.reducer});var hi,Ti,xi,sb,yi,Wt=h(()=>{"use strict";n();hi=require("@reduxjs/toolkit");D();Ti={},xi=(0,hi.createSlice)({name:"dappConfigSlice",initialState:Ti,reducers:{setDappConfig:(e,o)=>o.payload},extraReducers:e=>{e.addCase(R,()=>Ti)}}),{setDappConfig:sb}=xi.actions,yi=xi.reducer});var L=h(()=>{"use strict";n();ln();un();fn();xn();Sn();wn();In();Ln();En();Wt()});var Ut=h(()=>{"use strict";n()});var wi,cl,pl,Xe,Rn=h(()=>{"use strict";n();wi=require("@reduxjs/toolkit");Ut();ln();En();Wt();un();fn();xn();Sn();wn();In();Ln();cl={["account"]:dn,["dappConfig"]:yi,["loginInfo"]:_n,["modals"]:gn,["networkConfig"]:hn,["signedMessageInfo"]:yn,["toasts"]:bn,["transactionsInfo"]:vn,["transactions"]:kn,["batchTransactions"]:Pn},pl=(e={})=>(0,wi.combineReducers)(u(u({},cl),e)),Xe=pl});var ki={};ie(ki,{default:()=>Il,sessionStorageReducers:()=>Gt});function we(e,o=[]){return{key:e,version:1,storage:Ii.default,blacklist:o}}var Y,vi,Ii,ml,Oo,dl,ll,_l,ul,gl,fl,Tl,hl,xl,yl,Ai,Sl,Gt,bl,wl,vl,Il,Li=h(()=>{"use strict";n();Y=require("redux-persist"),vi=x(require("redux-persist/lib/storage")),Ii=x(require("redux-persist/lib/storage/session"));Rn();L();ln();En();un();fn();xn();Sn();wn();In();Ln();Ut();ml={persistReducersStorageType:"localStorage"},Oo={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},dl=we(Oo["account"]),ll=we(Oo["loginInfo"]),_l=we(Oo["modals"]),ul=we(Oo["networkConfig"]),gl={2:e=>v(u({},e),{networkConfig:Pt})};fl=we("sdk-dapp-transactionsInfo"),Tl=we("sdk-dapp-transactions",["transactionsToSign"]),hl=we("sdk-dapp-batchTransactions",["batchTransactions"]),xl=we("sdk-dapp-toasts"),yl=we("sdk-dapp-signedMessageInfo"),Ai={key:"sdk-dapp-store",version:2,storage:vi.default,whitelist:Object.keys(Oo),migrate:(0,Y.createMigrate)(gl,{debug:!1})},Sl=v(u({},Ai),{whitelist:[]}),Gt={["toasts"]:(0,Y.persistReducer)(xl,bn),["transactions"]:(0,Y.persistReducer)(Tl,kn),["transactionsInfo"]:(0,Y.persistReducer)(fl,vn),["batchTransactions"]:(0,Y.persistReducer)(hl,Pn),["signedMessageInfo"]:(0,Y.persistReducer)(yl,yn)},bl=v(u({},Gt),{["account"]:(0,Y.persistReducer)(dl,dn),["loginInfo"]:(0,Y.persistReducer)(ll,_n),["modals"]:(0,Y.persistReducer)(_l,gn),["networkConfig"]:(0,Y.persistReducer)(ul,hn)}),wl=ml.persistReducersStorageType==="localStorage",vl=wl?(0,Y.persistReducer)(Ai,Xe(Gt)):(0,Y.persistReducer)(Sl,Xe(bl)),Il=vl});var Pi={};ie(Pi,{default:()=>Al});var Al,Ei=h(()=>{"use strict";n();Rn();Al=Xe()});var Ci={};ie(Ci,{default:()=>Ll});var te,kl,Ll,Ni=h(()=>{"use strict";n();te=require("redux-persist"),kl=[te.FLUSH,te.REHYDRATE,te.PAUSE,te.PERSIST,te.PURGE,te.REGISTER],Ll=kl});var Mi={};ie(Mi,{default:()=>Di});function Di(e){return(0,Ri.persistStore)(e)}var Ri,Oi=h(()=>{"use strict";n();Ri=require("redux-persist")});var Ls=b((s2,ks)=>{n();var Lu=typeof a=="object"&&a&&a.Object===Object&&a;ks.exports=Lu});var $o=b((c2,Ps)=>{n();var Pu=Ls(),Eu=typeof self=="object"&&self&&self.Object===Object&&self,Cu=Pu||Eu||Function("return this")();Ps.exports=Cu});var rr=b((p2,Es)=>{n();var Nu=$o(),Ru=Nu.Symbol;Es.exports=Ru});var Ds=b((m2,Rs)=>{n();var Cs=rr(),Ns=Object.prototype,Du=Ns.hasOwnProperty,Mu=Ns.toString,Vo=Cs?Cs.toStringTag:void 0;function Ou(e){var o=Du.call(e,Vo),t=e[Vo];try{e[Vo]=void 0;var s=!0}catch(p){}var c=Mu.call(e);return s&&(o?e[Vo]=t:delete e[Vo]),c}Rs.exports=Ou});var Os=b((d2,Ms)=>{n();var Bu=Object.prototype,Fu=Bu.toString;function Wu(e){return Fu.call(e)}Ms.exports=Wu});var ar=b((l2,Ws)=>{n();var Bs=rr(),Uu=Ds(),Gu=Os(),Hu="[object Null]",$u="[object Undefined]",Fs=Bs?Bs.toStringTag:void 0;function Vu(e){return e==null?e===void 0?$u:Hu:Fs&&Fs in Object(e)?Uu(e):Gu(e)}Ws.exports=Vu});var Gs=b((_2,Us)=>{n();var qu=Array.isArray;Us.exports=qu});var $s=b((u2,Hs)=>{n();function ju(e){return e!=null&&typeof e=="object"}Hs.exports=ju});var qs=b((g2,Vs)=>{n();var zu=ar(),Ku=Gs(),Xu=$s(),Yu="[object String]";function Zu(e){return typeof e=="string"||!Ku(e)&&Xu(e)&&zu(e)==Yu}Vs.exports=Zu});var pr={};ie(pr,{css:()=>ec,default:()=>rg});var ec,rg,mr=h(()=>{"use strict";n();ec=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ec));rg={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var lr={};ie(lr,{css:()=>tc,default:()=>sg});var tc,sg,_r=h(()=>{"use strict";n();tc=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(tc));sg={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var Jn=b((F3,sc)=>{n();function bg(e){var o=typeof e;return e!=null&&(o=="object"||o=="function")}sc.exports=bg});var pc=b((W3,cc)=>{n();var wg=ar(),vg=Jn(),Ig="[object AsyncFunction]",Ag="[object Function]",kg="[object GeneratorFunction]",Lg="[object Proxy]";function Pg(e){if(!vg(e))return!1;var o=wg(e);return o==Ag||o==kg||o==Ig||o==Lg}cc.exports=Pg});var dc=b((U3,mc)=>{n();var Eg=$o(),Cg=Eg["__core-js_shared__"];mc.exports=Cg});var uc=b((G3,_c)=>{n();var ur=dc(),lc=function(){var e=/[^.]+$/.exec(ur&&ur.keys&&ur.keys.IE_PROTO||"");return e?"Symbol(src)_1."+e:""}();function Ng(e){return!!lc&&lc in e}_c.exports=Ng});var fc=b((H3,gc)=>{n();var Rg=Function.prototype,Dg=Rg.toString;function Mg(e){if(e!=null){try{return Dg.call(e)}catch(o){}try{return e+""}catch(o){}}return""}gc.exports=Mg});var hc=b(($3,Tc)=>{n();var Og=pc(),Bg=uc(),Fg=Jn(),Wg=fc(),Ug=/[\\^$.*+?()[\]{}|]/g,Gg=/^\[object .+?Constructor\]$/,Hg=Function.prototype,$g=Object.prototype,Vg=Hg.toString,qg=$g.hasOwnProperty,jg=RegExp("^"+Vg.call(qg).replace(Ug,"\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g,"$1.*?")+"$");function zg(e){if(!Fg(e)||Bg(e))return!1;var o=Og(e)?jg:Gg;return o.test(Wg(e))}Tc.exports=zg});var yc=b((V3,xc)=>{n();function Kg(e,o){return e==null?void 0:e[o]}xc.exports=Kg});var Qn=b((q3,Sc)=>{n();var Xg=hc(),Yg=yc();function Zg(e,o){var t=Yg(e,o);return Xg(t)?t:void 0}Sc.exports=Zg});var Ko=b((j3,bc)=>{n();var Jg=Qn(),Qg=Jg(Object,"create");bc.exports=Qg});var Ic=b((z3,vc)=>{n();var wc=Ko();function ef(){this.__data__=wc?wc(null):{},this.size=0}vc.exports=ef});var kc=b((K3,Ac)=>{n();function of(e){var o=this.has(e)&&delete this.__data__[e];return this.size-=o?1:0,o}Ac.exports=of});var Pc=b((X3,Lc)=>{n();var nf=Ko(),tf="__lodash_hash_undefined__",rf=Object.prototype,af=rf.hasOwnProperty;function sf(e){var o=this.__data__;if(nf){var t=o[e];return t===tf?void 0:t}return af.call(o,e)?o[e]:void 0}Lc.exports=sf});var Cc=b((Y3,Ec)=>{n();var cf=Ko(),pf=Object.prototype,mf=pf.hasOwnProperty;function df(e){var o=this.__data__;return cf?o[e]!==void 0:mf.call(o,e)}Ec.exports=df});var Rc=b((Z3,Nc)=>{n();var lf=Ko(),_f="__lodash_hash_undefined__";function uf(e,o){var t=this.__data__;return this.size+=this.has(e)?0:1,t[e]=lf&&o===void 0?_f:o,this}Nc.exports=uf});var Mc=b((J3,Dc)=>{n();var gf=Ic(),ff=kc(),Tf=Pc(),hf=Cc(),xf=Rc();function vo(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}vo.prototype.clear=gf;vo.prototype.delete=ff;vo.prototype.get=Tf;vo.prototype.has=hf;vo.prototype.set=xf;Dc.exports=vo});var Bc=b((Q3,Oc)=>{n();function yf(){this.__data__=[],this.size=0}Oc.exports=yf});var Wc=b((eO,Fc)=>{n();function Sf(e,o){return e===o||e!==e&&o!==o}Fc.exports=Sf});var Xo=b((oO,Uc)=>{n();var bf=Wc();function wf(e,o){for(var t=e.length;t--;)if(bf(e[t][0],o))return t;return-1}Uc.exports=wf});var Hc=b((nO,Gc)=>{n();var vf=Xo(),If=Array.prototype,Af=If.splice;function kf(e){var o=this.__data__,t=vf(o,e);if(t<0)return!1;var s=o.length-1;return t==s?o.pop():Af.call(o,t,1),--this.size,!0}Gc.exports=kf});var Vc=b((tO,$c)=>{n();var Lf=Xo();function Pf(e){var o=this.__data__,t=Lf(o,e);return t<0?void 0:o[t][1]}$c.exports=Pf});var jc=b((rO,qc)=>{n();var Ef=Xo();function Cf(e){return Ef(this.__data__,e)>-1}qc.exports=Cf});var Kc=b((aO,zc)=>{n();var Nf=Xo();function Rf(e,o){var t=this.__data__,s=Nf(t,e);return s<0?(++this.size,t.push([e,o])):t[s][1]=o,this}zc.exports=Rf});var Yc=b((iO,Xc)=>{n();var Df=Bc(),Mf=Hc(),Of=Vc(),Bf=jc(),Ff=Kc();function Io(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Io.prototype.clear=Df;Io.prototype.delete=Mf;Io.prototype.get=Of;Io.prototype.has=Bf;Io.prototype.set=Ff;Xc.exports=Io});var Jc=b((sO,Zc)=>{n();var Wf=Qn(),Uf=$o(),Gf=Wf(Uf,"Map");Zc.exports=Gf});var op=b((cO,ep)=>{n();var Qc=Mc(),Hf=Yc(),$f=Jc();function Vf(){this.size=0,this.__data__={hash:new Qc,map:new($f||Hf),string:new Qc}}ep.exports=Vf});var tp=b((pO,np)=>{n();function qf(e){var o=typeof e;return o=="string"||o=="number"||o=="symbol"||o=="boolean"?e!=="__proto__":e===null}np.exports=qf});var Yo=b((mO,rp)=>{n();var jf=tp();function zf(e,o){var t=e.__data__;return jf(o)?t[typeof o=="string"?"string":"hash"]:t.map}rp.exports=zf});var ip=b((dO,ap)=>{n();var Kf=Yo();function Xf(e){var o=Kf(this,e).delete(e);return this.size-=o?1:0,o}ap.exports=Xf});var cp=b((lO,sp)=>{n();var Yf=Yo();function Zf(e){return Yf(this,e).get(e)}sp.exports=Zf});var mp=b((_O,pp)=>{n();var Jf=Yo();function Qf(e){return Jf(this,e).has(e)}pp.exports=Qf});var lp=b((uO,dp)=>{n();var eT=Yo();function oT(e,o){var t=eT(this,e),s=t.size;return t.set(e,o),this.size+=t.size==s?0:1,this}dp.exports=oT});var up=b((gO,_p)=>{n();var nT=op(),tT=ip(),rT=cp(),aT=mp(),iT=lp();function Ao(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Ao.prototype.clear=nT;Ao.prototype.delete=tT;Ao.prototype.get=rT;Ao.prototype.has=aT;Ao.prototype.set=iT;_p.exports=Ao});var fp=b((fO,gp)=>{n();var sT="__lodash_hash_undefined__";function cT(e){return this.__data__.set(e,sT),this}gp.exports=cT});var hp=b((TO,Tp)=>{n();function pT(e){return this.__data__.has(e)}Tp.exports=pT});var yp=b((hO,xp)=>{n();var mT=up(),dT=fp(),lT=hp();function et(e){var o=-1,t=e==null?0:e.length;for(this.__data__=new mT;++o<t;)this.add(e[o])}et.prototype.add=et.prototype.push=dT;et.prototype.has=lT;xp.exports=et});var bp=b((xO,Sp)=>{n();function _T(e,o,t,s){for(var c=e.length,p=t+(s?1:-1);s?p--:++p<c;)if(o(e[p],p,e))return p;return-1}Sp.exports=_T});var vp=b((yO,wp)=>{n();function uT(e){return e!==e}wp.exports=uT});var Ap=b((SO,Ip)=>{n();function gT(e,o,t){for(var s=t-1,c=e.length;++s<c;)if(e[s]===o)return s;return-1}Ip.exports=gT});var Lp=b((bO,kp)=>{n();var fT=bp(),TT=vp(),hT=Ap();function xT(e,o,t){return o===o?hT(e,o,t):fT(e,TT,t)}kp.exports=xT});var Ep=b((wO,Pp)=>{n();var yT=Lp();function ST(e,o){var t=e==null?0:e.length;return!!t&&yT(e,o,0)>-1}Pp.exports=ST});var Np=b((vO,Cp)=>{n();function bT(e,o,t){for(var s=-1,c=e==null?0:e.length;++s<c;)if(t(o,e[s]))return!0;return!1}Cp.exports=bT});var Dp=b((IO,Rp)=>{n();function wT(e,o){return e.has(o)}Rp.exports=wT});var Op=b((AO,Mp)=>{n();var vT=Qn(),IT=$o(),AT=vT(IT,"Set");Mp.exports=AT});var Fp=b((kO,Bp)=>{n();function kT(){}Bp.exports=kT});var gr=b((LO,Wp)=>{n();function LT(e){var o=-1,t=Array(e.size);return e.forEach(function(s){t[++o]=s}),t}Wp.exports=LT});var Gp=b((PO,Up)=>{n();var fr=Op(),PT=Fp(),ET=gr(),CT=1/0,NT=fr&&1/ET(new fr([,-0]))[1]==CT?function(e){return new fr(e)}:PT;Up.exports=NT});var $p=b((EO,Hp)=>{n();var RT=yp(),DT=Ep(),MT=Np(),OT=Dp(),BT=Gp(),FT=gr(),WT=200;function UT(e,o,t){var s=-1,c=DT,p=e.length,m=!0,d=[],l=d;if(t)m=!1,c=MT;else if(p>=WT){var f=o?null:BT(e);if(f)return FT(f);m=!1,c=OT,l=new RT}else l=o?[]:d;e:for(;++s<p;){var S=e[s],T=o?o(S):S;if(S=t||S!==0?S:0,m&&T===T){for(var w=l.length;w--;)if(l[w]===T)continue e;o&&l.push(T),d.push(S)}else c(l,T,t)||(l!==d&&l.push(T),d.push(S))}return d}Hp.exports=UT});var qp=b((CO,Vp)=>{n();var GT=$p();function HT(e){return e&&e.length?GT(e):[]}Vp.exports=HT});var Pe={};ie(Pe,{css:()=>im,default:()=>Eh});var im,Eh,Ee=h(()=>{"use strict";n();im=`.dapp-core-component__dappModalStyles__dappModal {
  position: fixed;
  z-index: 999;
  padding-top: 100px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
}

.dapp-core-component__dappModalStyles__dappModalContent {
  background: #fff;
  border: 2px solid #aaa;
  border-radius: 5px;
  z-index: 999;
  max-width: 500px;
  margin: auto;
  padding: 0;
  position: relative;
  width: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.dapp-core-component__dappModalStyles__dappModalHeader {
  padding: 2px 16px;
  background-color: #fff;
  font-size: 1.25rem;
}

.dapp-core-component__dappModalStyles__dappModalHeaderText {
  padding: 0 1rem;
}

.dapp-core-component__dappModalStyles__dappModalBody {
  padding: 16px 0;
}

.dapp-core-component__dappModalStyles__dappModalFooter {
  background-color: #fff;
  padding: 1rem 0.5rem;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton {
  position: absolute;
  top: 0;
  right: 0;
  float: right;
  font-weight: bold;
  border-radius: 5px;
  font-size: 1rem;
  padding: 0.5rem 1rem;
  border: 0;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton:hover,
.dapp-core-component__dappModalStyles__dappModalCloseButton:focus {
  text-decoration: none;
  cursor: pointer;
  color: #212529;
  background-color: #dee2e6;
  border-color: #e9ecef;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(im));Eh={dappModal:"dapp-core-component__dappModalStyles__dappModal",dappModalContent:"dapp-core-component__dappModalStyles__dappModalContent",dappModalHeader:"dapp-core-component__dappModalStyles__dappModalHeader",dappModalHeaderText:"dapp-core-component__dappModalStyles__dappModalHeaderText",dappModalBody:"dapp-core-component__dappModalStyles__dappModalBody",dappModalFooter:"dapp-core-component__dappModalStyles__dappModalFooter",dappModalCloseButton:"dapp-core-component__dappModalStyles__dappModalCloseButton"}});var Er={};ie(Er,{css:()=>hm,default:()=>Xh});var hm,Xh,Cr=h(()=>{"use strict";n();hm=`.dapp-core-component__loginButtonStyles__default-login-button {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__wrapper {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__login-text {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-wrapper {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-content {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-title {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-icon {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(hm));Xh={"default-login-button":"dapp-core-component__loginButtonStyles__default-login-button",defaultLoginButton:"dapp-core-component__loginButtonStyles__default-login-button",wrapper:"dapp-core-component__loginButtonStyles__wrapper","login-text":"dapp-core-component__loginButtonStyles__login-text",loginText:"dapp-core-component__loginButtonStyles__login-text","button-wrapper":"dapp-core-component__loginButtonStyles__button-wrapper",buttonWrapper:"dapp-core-component__loginButtonStyles__button-wrapper","button-content":"dapp-core-component__loginButtonStyles__button-content",buttonContent:"dapp-core-component__loginButtonStyles__button-content","button-title":"dapp-core-component__loginButtonStyles__button-title",buttonTitle:"dapp-core-component__loginButtonStyles__button-title","button-icon":"dapp-core-component__loginButtonStyles__button-icon",buttonIcon:"dapp-core-component__loginButtonStyles__button-icon"}});var Qh={};ie(Qh,{OperaWalletLoginButton:()=>Jh});module.exports=ee(Qh);n();n();var nt=x(require("react"));k();n();n();n();var Hi=require("react"),xo=require("react-redux");n();var Ht=require("@reduxjs/toolkit"),Ui=require("react-redux/lib/utils/Subscription");L();n();var Si=x(require("lodash.throttle"));k();L();kt();je();qe();var al=[cn],Cn=!1,il=(0,Si.default)(()=>{Co(Eo())},5e3),bi=e=>o=>t=>{if(al.includes(t.type))return o(t);let s=e.getState(),c=De.local.getItem(ye.loginExpiresAt);if(!Boolean(s==null?void 0:s.account.address))return o(t);if(c==null)return Co(Eo());let m=Date.now();return c-m<0&&!Cn?setTimeout(()=>{Cn=!0,e.dispatch(Wa())},1e3):(Cn&&(Cn=!1),il()),o(t)};n();n();function Nn(){return typeof sessionStorage!="undefined"}var Bi=Nn()?(Li(),ee(ki)).default:(Ei(),ee(Pi)).default,Fi=Nn()?(Ni(),ee(Ci)).default:[],Wi=Nn()?(Oi(),ee(Mi)).default:e=>e;Rn();var I=(0,Ht.configureStore)({reducer:Bi,middleware:e=>e({serializableCheck:{ignoredActions:[...Fi,At.type,ne.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(bi)}),Gi=(0,Ui.createSubscription)(I),a0=Wi(I),i0=(0,Ht.configureStore)({reducer:Xe});var Pl={store:I,subscription:Gi},$t=(0,Hi.createContext)(Pl),m0=(0,xo.createStoreHook)($t),G=(0,xo.createDispatchHook)($t),A=(0,xo.createSelectorHook)($t);n();n();L();n();var $i=x(require("lodash.isequal")),Dn=require("reselect"),y=(0,Dn.createSelectorCreator)(Dn.defaultMemoize,$i.default);var ue=e=>e.account,Ye=y(ue,e=>e.address),ve=y(ue,Ye,(e,o)=>o in e.accounts?e.accounts[o]:X),El=y(ue,ve,(e,o)=>{let c=e,{accounts:t}=c,s=Ue(c,["accounts"]);return v(u({},s),{address:o.address,account:o})}),g0=y(ve,e=>e.balance),Cl=y(ve,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),f0=y(ue,e=>e.shard),Nl=y(ue,e=>e.ledgerAccount),T0=y(ue,e=>e.walletConnectAccount),h0=y(ue,e=>e.isAccountLoading),x0=y(ue,e=>e.accountLoadingError),Rl=y(ue,e=>e.websocketEvent),Dl=y(ue,e=>e.websocketBatchEvent);n();var Vi=e=>e.dappConfig,b0=y(Vi,e=>e.shouldUseWebViewProvider);n();var J=e=>e.loginInfo,Ml=y(J,e=>e.loginMethod),Bo=y(J,Ye,(e,o)=>Boolean(o)),A0=y(J,e=>e.walletConnectLogin),Ol=y(J,e=>e.ledgerLogin),Bl=y(J,e=>e.walletLogin),k0=y(J,e=>e.isLoginSessionInvalid),Fo=y(J,e=>e.tokenLogin),qi=y(J,e=>e.logoutRoute),Fl=y(J,e=>e.isWalletConnectV2Initialized);n();var ji=e=>e.modals,E0=y(ji,e=>e.txSubmittedModal),Wl=y(ji,e=>e.notificationModal);n();var ge=e=>e.networkConfig,Mn=y(ge,e=>e.network.chainId),Ul=y(ge,e=>e.network.roundDuration),R0=y(ge,e=>e.network.walletConnectBridgeAddress),Gl=y(ge,e=>e.network.walletConnectV2RelayAddress),Hl=y(ge,e=>e.network.walletConnectV2ProjectId),$l=y(ge,e=>e.network.walletConnectV2Options),Vl=y(ge,e=>e.network.walletConnectDeepLink),Z=y(ge,e=>e.network),zi=y(Z,e=>e.apiAddress),Ki=y(Z,e=>e.explorerAddress),Xi=y(Z,e=>{var o;return(o=e.customWalletAddress)!=null?o:e.walletAddress}),ql=y(Z,e=>e.xAliasAddress),Yi=y(Z,e=>e.egldLabel);n();var On=e=>e.signedMessageInfo,O0=y(On,e=>e.isSigning),B0=y(On,e=>e.errorMessage),F0=y(On,e=>{let o=Object.keys(e.signedSessions),t=o.length;return e.signedSessions[o[t-1]]}),W0=y(On,e=>{let o=Object.keys(e.signedSessions),t=o.length;return o.length>0?o[t-1]:""});n();var Zi=e=>e.toasts,jl=y(Zi,e=>e.customToasts),Ji=y(Zi,e=>e.transactionToasts);n();L();var Qi={errorMessage:Nt,successMessage:Rt,processingMessage:Dt},es=e=>e.transactionsInfo,zl=y(es,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Qi);n();n();var Je=require("@multiversx/sdk-core");k();n();n();n();function Bn(e){try{let o=atob(e),t=btoa(o),s=r.Buffer.from(e,"base64").toString(),c=r.Buffer.from(s).toString("base64"),p=e===t||t.startsWith(e),m=e===c||c.startsWith(e);if(p&&m)return!0}catch(o){return!1}return!1}function Ze(e){return Bn(e)?atob(e):e}n();n();n();n();var os=e=>{let o=e!=null?e:"";return Bn(o)?r.Buffer.from(r.Buffer.from(o,"base64").toString().trim()):r.Buffer.from(o.trim())};n();k();var Wo=({data:e,onlySetGuardian:o})=>e?o?e.startsWith("SetGuardian"):Object.values(gt).some(t=>e.startsWith(t)):!1;function Fn(e){var s,c,p;let o=u({},e);Wo({data:o.data,onlySetGuardian:!0})&&(delete o.senderUsername,delete o.receiverUsername);let t=new Je.Transaction(u(u(v(u(v(u(u({value:BigInt(o.value.valueOf()),data:os(o.data),nonce:BigInt(o.nonce.valueOf()),receiver:new Je.Address(o.receiver)},o.receiverUsername?{receiverUsername:o.receiverUsername}:{}),o.relayer?{relayer:new Je.Address(o.relayer)}:{}),{sender:new Je.Address(o.sender)}),o.senderUsername?{senderUsername:o.senderUsername}:{}),{gasLimit:BigInt((s=o.gasLimit.valueOf())!=null?s:an),gasPrice:BigInt((c=o.gasPrice.valueOf())!=null?c:sn),chainID:o.chainID.valueOf(),version:(p=o.version)!=null?p:va}),o.options?{options:o.options}:{}),o.guardian?{guardian:new Je.Address(o.guardian)}:{}));return o.relayerSignature&&(t.relayerSignature=r.Buffer.from(o.relayerSignature,"hex")),o.guardianSignature&&(t.guardianSignature=r.Buffer.from(o.guardianSignature,"hex")),o.signature&&(t.signature=r.Buffer.from(o.signature,"hex")),t}n();n();k();n();n();n();var Zl=require("@multiversx/sdk-core");n();k();n();var Ql=require("@multiversx/sdk-core");B();n();var Jl=require("@multiversx/sdk-core");n();n();n();var e_=require("@multiversx/sdk-core"),o_=x(require("bignumber.js"));C();n();n();n();var qt="blocks";n();n();n();n();z();n();n();z();n();n();n();var p_=x(require("bignumber.js"));k();n();var c_=require("@multiversx/sdk-core"),zt=x(require("bignumber.js"));k();n();var i_=x(require("bignumber.js"));n();zt.default.config({ROUNDING_MODE:zt.default.ROUND_FLOOR});n();n();n();k();n();n();n();n();k();n();k();n();var m_=require("@multiversx/sdk-core"),d_=x(require("bignumber.js"));k();n();var Xt=require("@multiversx/sdk-core"),u_=x(require("bignumber.js"));k();n();n();var l_=x(require("bignumber.js"));k();n();k();n();n();n();n();n();n();k();n();k();n();k();n();z();var f_=["reDelegateRewards","claimRewards","unBond"],T_=["wrapEgld","unwrapEgld"],h_=["unStake"],x_=["unDelegate"];n();n();n();k();n();var w_=x(require("bignumber.js"));n();n();z();n();var I_=x(require("bignumber.js"));n();n();n();n();var L_=x(require("bignumber.js"));B();n();n();var P_=require("@multiversx/sdk-core");n();n();C();n();var R_=require("@multiversx/sdk-web-wallet-provider");k();n();var C_=x(require("qs"));n();$();_o();n();_o();var Sk={search:ze()?window.location.search:"",removeParams:[]};n();n();n();H();n();z();n();n();H();n();var D_=x(require("linkifyjs"));n();k();n();var M_=x(require("bignumber.js"));n();B();n();n();C();n();C();n();n();n();z();n();z();n();var O_=x(require("bignumber.js"));k();z();n();z();n();var ss=require("react");B();n();n();z();n();n();var B_=require("@multiversx/sdk-core"),F_=x(require("bignumber.js"));z();n();B();n();n();B();var lP=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];n();var cs=require("react");B();H();n();var U_=require("react");z();var yP=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];ho();n();var Gn=e=>e.transactions,yo=y(Gn,e=>e.signedTransactions),G_=y(Gn,e=>e.signTransactionsError),ps=y(Gn,e=>e.signTransactionsCancelMessage),Hn=e=>o=>Object.entries(o).reduce((t,[s,c])=>(e(c.status)&&(t[s]=c),t),{}),ms=y(yo,Hn(go)),ds=y(yo,Hn(fo)),ls=y(yo,Hn(To)),H_=y(yo,Hn(Ot)),_s=y(Gn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:v(u({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(t=>Fn(t)))||[]})}),$_=y(yo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});n();var us=require("react");n();L();n();n();n();n();n();n();n();n();var q_=require("@multiversx/sdk-extension-provider"),j_=require("@multiversx/sdk-hw-provider"),z_=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),K_=require("@multiversx/sdk-opera-provider"),X_=require("@multiversx/sdk-passkey-provider/out"),Y_=require("@multiversx/sdk-web-wallet-provider");k();Re();n();var Qe=require("@multiversx/sdk-web-wallet-iframe-provider/out"),gs=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");C();n();var se=e=>`Unable to perform ${e}, Provider not initialized`,$n=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(se("getAccount"))}setAccount(o){throw new Error(se(`setAccount: ${o}`))}login(o){throw new Error(se(`login with options: ${o}`))}logout(o){throw new Error(se(`logout with options: ${o}`))}getAddress(){throw new Error(se("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(o,t){throw new Error(se(`sendTransaction with transactions: ${o} options: ${t}`))}signTransaction(o,t){throw new Error(se(`signTransaction with transactions: ${o} options: ${t}`))}signTransactions(o,t){throw new Error(se(`signTransactions with transactions: ${o} options: ${t}`))}signMessage(o,t){throw new Error(se(`signTransactions with ${o} and options ${t}`))}sendCustomMessage({method:o,params:t}){throw new Error(se(`sendCustomMessage with method: ${o} params: ${t}`))}sendCustomRequest(o){throw new Error(se(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Jt=new $n;var Z_=Jt;function Ae(e){Z_=e}L();n();n();n();n();n();var fs=require("@lifeomic/axios-fetch"),Qt=x(require("axios")),er=(0,fs.buildAxiosFetch)(Qt.default),or=(e,o)=>g(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let s=e.clone().json(),[c]=yield Promise.all([s]);return{data:c,status:e.status,statusText:e.statusText,headers:e.headers,config:o}});function J_(e,o,t){return g(this,null,function*(){try{let s=yield er(e,u({method:"POST",body:o?JSON.stringify(o):void 0,headers:u({"Content-Type":"application/json"},(t==null?void 0:t.headers)||{})},t));return or(s,t)}catch(s){throw console.error("Fetch Error:",s),s}})}function Q_(e,o){return g(this,null,function*(){try{let t=yield er(e,o);if(!t.ok)throw new Error(`HTTP error! Status: ${t.status}`);return or(t,o)}catch(t){throw console.error("Fetch Error:",t),t}})}function eu(e,o,t){return g(this,null,function*(){try{let s=yield er(e,u({method:"PATCH",body:o?JSON.stringify(o):void 0,headers:(t==null?void 0:t.headers)||{}},t));if(!s.ok)throw new Error(`HTTP error! Status: ${s.status}`);return or(s,t)}catch(s){throw console.error("Fetch Error:",s),s}})}var So=Qt.default.create();So.get=Q_;So.post=J_;So.patch=eu;n();n();var ou=x(require("axios"));n();var nu=x(require("swr"));n();C();n();n();n();H();n();ho();n();n();n();var tu=x(require("axios"));n();var au=x(require("axios"));po();n();k();n();var iu=x(require("axios"));n();var cu=x(require("axios"));n();n();var pu=x(require("axios"));n();var mu=x(require("axios"));n();n();L();C();n();n();n();n();H();n();D();B();n();L();n();var ys=require("@multiversx/sdk-core");B();$();n();ho();n();L();C();n();L();B();n();n();n();B();n();Po();n();n();n();n();var _u=x(require("swr"));n();n();n();n();var ws=require("react");n();n();var uu=require("react"),gu=require("@multiversx/sdk-web-wallet-provider"),fu=require("@multiversx/sdk-web-wallet-provider"),Tu=x(require("qs"));k();L();C();Tt();H();var tD=M();n();var ic=require("react"),Zn=require("@multiversx/sdk-core");n();k();Lt();n();n();n();n();n();n();n();n();n();n();po();n();var Su=x(require("axios"));n();var wu=x(require("axios"));n();var vu=x(require("axios"));n();po();n();po();n();n();n();n();var Au=require("@multiversx/sdk-opera-provider");n();var ku=require("@multiversx/sdk-extension-provider");n();Re();n();n();n();n();var js=x(qs());var bo=e=>{if(!e||!(0,js.default)(e))return null;let o=e.split(".");if(o.length!==4)return null;try{let[t,s,c,p]=o,m=JSON.parse(Ze(p)),d=Ze(t);return{ttl:Number(c),extraInfo:m,origin:d,blockHash:s}}catch(t){return console.error(`Error trying to decode ${e}:`,t),null}};n();var Js=require("@multiversx/sdk-native-auth-client");n();var Xs=x(require("axios"));n();n();n();function zs(e){return g(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}n();var Ks=(e,o,t,s=0)=>g(void 0,null,function*(){try{return yield e(...t)}catch(c){return s<o.retries?((o==null?void 0:o.delay)!=null&&(yield zs(o.delay)),yield Ks(e,o,t,s+1)):null}}),ir=(e,o={retries:5,delay:500})=>(...t)=>g(void 0,null,function*(){return yield Ks(e,o,t)});var Ju=4,Qu=3e4,Kn={current:null},eo={current:null},eg=ir((e,o,t)=>g(void 0,null,function*(){if(t){let p=Math.floor(Date.now()/1e3);return{hash:yield t(),timestamp:p}}let{data:s}=yield Xs.default.get(`${e}/${qt}?from=${Ju}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[c]=s;return c}));function Ys(e,o,t,s){return g(this,null,function*(){if(e==null)throw new Error("missing api url");let c=Date.now();if(Kn.current!=null&&c<Kn.current.timestamp*1e3+Qu&&!s)return Kn.current;if(eo.current!=null)return yield eo.current;eo.current=eg(e,o,t);try{let p=yield eo.current;if(p==null)throw eo.current=null,new Error("could not get block hash");return Kn.current={hash:p.hash,timestamp:p.timestamp},eo.current=null,p}catch(p){return eo.current=null,null}})}n();n();pn();n();function og(e){return Object.prototype.toString.call(e)==="[object String]"}var sr=e=>{var t;if(!e||!og(e))return null;let o=e.split(".");if(o.length!==3)return console.error("Invalid nativeAuthToken. You may be trying to decode a loginToken. Try using decodeLoginToken method instead"),null;try{let[s,c,p]=o,m=Ze(s),d=Ze(c),l=bo(d);if(!l)return{address:m,body:d,signature:p,blockHash:"",origin:"",ttl:0};let f=v(u({},l),{address:m,body:d,signature:p});return(t=l.extraInfo)!=null&&t.timestamp||delete f.extraInfo,f}catch(s){return null}};var cr={isExpired:!1},Zs=e=>{if(!e)return cr;let o=sr(e);if(!o)return cr;let t=lo(),{ttl:s,extraInfo:c}=o,p=c==null?void 0:c.timestamp;if(!p)return cr;let m=p+s,d=t>m,l=m-t;return{isExpired:d,expiresAt:m,secondsUntilExpires:l}};n();H();var qo={origin:M().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60},jo=e=>{var t,s,c,p,m,d;return e===!0?qo:{origin:(t=e==null?void 0:e.origin)!=null?t:qo.origin,blockHashShard:e==null?void 0:e.blockHashShard,expirySeconds:(s=e==null?void 0:e.expirySeconds)!=null?s:qo.expirySeconds,apiAddress:(c=e==null?void 0:e.apiAddress)!=null?c:qo.apiAddress,tokenExpirationToastWarningSeconds:(p=e==null?void 0:e.tokenExpirationToastWarningSeconds)!=null?p:qo.tokenExpirationToastWarningSeconds,extraInfo:(m=e==null?void 0:e.extraInfo)!=null?m:{},gatewayUrl:e==null?void 0:e.gatewayUrl,extraRequestHeaders:(d=e==null?void 0:e.extraRequestHeaders)!=null?d:{}}};var Xn=e=>{let{origin:o,apiAddress:t,expirySeconds:s,blockHashShard:c,extraInfo:p,gatewayUrl:m,extraRequestHeaders:d}=jo(e),l=new Js.NativeAuthClient({origin:o,apiUrl:t,expirySeconds:s,blockHashShard:c,gatewayUrl:m,extraRequestHeaders:d});return{getNativeAuthConfig:jo,initialize:T=>g(void 0,null,function*(){var E,P;if(!t||!o)return"";let w=()=>l.getCurrentBlockHash();try{let N=(E=T==null?void 0:T.latestBlockInfo)!=null?E:yield Ys(t,c,w,T==null?void 0:T.noCache);if(!N)return"";let{hash:V,timestamp:Q}=N,ae=l.encodeValue(JSON.stringify(u(u({},(P=T==null?void 0:T.extraInfo)!=null?P:p),Q?{timestamp:Q}:{})));return`${l.encodeValue(o)}.${V}.${s}.${ae}`}catch(N){return console.error("Error getting native auth token: ",N.toString()),""}}),getToken:({address:T,token:w,signature:E})=>l.getToken(T,w,E),getTokenExpiration:Zs}};n();var ng=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");n();var tg=require("@multiversx/sdk-passkey-provider/out");n();var mg=x(require("react")),dg=require("@fortawesome/free-solid-svg-icons");n();n();var wo=x(require("react")),rc=require("@fortawesome/free-solid-svg-icons"),ac=require("@fortawesome/react-fontawesome");n();var nc=x(require("react"));n();var zo=x(require("react"));n();_o();var Qs=()=>!ze();var ag=()=>g(void 0,null,function*(){return yield Promise.resolve().then(()=>(mr(),pr))}),ig=()=>(mr(),ee(pr)).default,dr=Qs();function oc({ssrGlobalImportCallback:e,ssrImportCallback:o,clientImportCallback:t}){let[s,c]=zo.default.useState(dr?void 0:ig()),[p,m]=zo.default.useState(dr||t==null?void 0:t()),d=()=>g(this,null,function*(){(e?e():ag()).then(l=>c(l.default)),o==null||o().then(l=>m(l.default))});return(0,zo.useEffect)(()=>{dr&&d()},[]),{globalStyles:s,styles:p}}function re(e,o){return t=>{let{globalStyles:s,styles:c}=oc({ssrGlobalImportCallback:o.ssrGlobalStyles,ssrImportCallback:o.ssrStyles,clientImportCallback:o.clientStyles});return nc.default.createElement(e,v(u({},t),{globalStyles:s!=null?s:{},styles:c!=null?c:{}}))}}var cg=({styles:e})=>wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},wo.default.createElement(ac.FontAwesomeIcon,{icon:rc.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),pg=re(cg,{ssrStyles:()=>Promise.resolve().then(()=>(_r(),lr)),clientStyles:()=>(_r(),ee(lr)).default});je();qe();n();n();L();var SM=24*60*60;k();H();n();n();n();var Tg=require("react"),hg=require("@multiversx/sdk-hw-provider");n();Ve();D();B();n();Ro();je();qe();$();n();n();n();var Xp=require("react"),Yp=require("@multiversx/sdk-core"),ih=require("@multiversx/sdk-extension-provider"),sh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),ch=require("@multiversx/sdk-passkey-provider/out"),ph=x(qp());k();n();n();var oo=()=>A(ve);n();n();n();n();n();n();n();n();var Zo=x(require("react"));var zO=(0,Zo.createContext)({}),KO=I.getState();n();var zp=x(require("react"));n();var Tr=x(require("react")),$T=x(require("axios"));n();n();Re();n();n();var XT=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),YT=require("@multiversx/sdk-webview-provider/out/WebviewProvider");D();n();D();L();B();n();n();L();n();$();n();L();B();n();n();var qT=require("@multiversx/sdk-core"),jT=x(require("bignumber.js"));k();n();var zT=x(require("bignumber.js"));k();L();C();$();H();n();var Kp=require("react"),eh=require("@multiversx/sdk-extension-provider"),oh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),nh=require("@multiversx/sdk-passkey-provider/out");Re();L();n();L();B();Ro();H();n();le();L();n();n();var dh=require("react");n();n();C();ho();n();n();var yh=require("react");n();n();var Zp=require("react");n();var lh=x(require("axios"));n();var fh=require("react");n();var xr=require("react"),gh=require("socket.io-client");L();n();var hr=class{constructor(){this.current=null;this._status="NOT_INITIALIZED";this.listeners=[]}get status(){return this._status}set status(o){this._status!==o&&(this._status=o,this.notifyListeners())}subscribe(o){return this.listeners.push(o),()=>this.unsubscribe(o)}unsubscribe(o){this.listeners=this.listeners.filter(t=>t!==o)}notifyListeners(){this.listeners.forEach(o=>o(this._status))}},Jp=new hr;n();var hh=require("react");Po();n();n();var yr=require("react");n();var xh=require("react");n();var em=require("react");n();n();var bh=require("react"),wh=require("@multiversx/sdk-extension-provider");le();D();L();C();n();function ke(){return Bo(I.getState())}be();$();$();n();n();Re();B();var Fe=e=>(Object.values(F).forEach(o=>{if(o!==(e==null?void 0:e.skipLoginMethod))switch(o){case"crossWindow":{let t=Ne.CrossWindowProvider.getInstance();t.isInitialized()&&t.dispose();break}case"iframe":{let t=Qe.IframeProvider.getInstance();t.isInitialized()&&t.dispose();break}default:break}}),null);n();var om=require("react"),ot=require("@multiversx/sdk-core");L();var Sh=(e,o)=>{var t;return o?o===!0?e:(t=o.apiAddress)!=null?t:e:null},Le=e=>{let o=A(Z),t=A(Fo),s=(0,om.useRef)(t==null?void 0:t.loginToken),c=Sh(o.apiAddress,e),p=jo(u(u({},e===!0?{}:e),c?{apiAddress:c}:{})),m=Boolean(e),d=Xn(p),{address:l}=oo(),f=G(),S=P=>{s.current=P,f(No(u(v(u({},t),{loginToken:P}),c?{nativeAuthConfig:p}:{})))},T=()=>{try{return d.initialize()}catch(P){console.error("Unable to get block. Login failed.",P);return}},w=({address:P,signature:N})=>{let V=s.current;if(!V)throw"Token not found. Call `setLoginToken` first.";if(!m){f(No({loginToken:V,signature:N}));return}let Q=d.getToken({address:P,token:V,signature:N});return f(No(u({loginToken:V,signature:N,nativeAuthToken:Q},c?{nativeAuthConfig:p}:{}))),Q};return{configuration:p,setLoginToken:S,getNativeAuthLoginToken:T,setTokenLoginInfo:w,refreshNativeAuthTokenLogin:V=>g(void 0,[V],function*({signMessageCallback:P,nativeAuthClientConfig:N}){let ae=yield Xn(N||p).initialize({noCache:Boolean(N)});if(s.current=ae,!ae)return;let Ce=new ot.Message({address:new ot.Address(l),data:r.Buffer.from(`${l}${ae}`)}),W=yield P(Ce,{});if(!(W!=null&&W.signature))throw"Message not signed";return w({address:l,signature:r.Buffer.from(W.signature).toString("hex")})})}};n();var Sr=require("react"),nm=require("@multiversx/sdk-opera-provider");le();D();L();C();be();$();H();var br=({callbackRoute:e,token:o,nativeAuth:t,onLoginRedirect:s})=>{let[c,p]=(0,Sr.useState)(""),[m,d]=(0,Sr.useState)(!1),l=t!=null,f=Le(t),S=o,T=G(),w=ke();function E(){return g(this,null,function*(){if(w)throw new Error(xe);Fe(),T(Se(X.address)),T(ne(X)),d(!0);let N=nm.OperaProvider.getInstance();try{if(!(yield N.init())){console.warn("Something went wrong trying to redirect to wallet login..");return}let{origin:Q}=M(),ae=Ke(),Ce=encodeURIComponent(`${Q}${e!=null?e:ae}`);if(l&&!S&&(S=yield f.getNativeAuthLoginToken(),!S)){console.warn("Fetching block failed. Login cancelled.");return}S&&f.setLoginToken(S);let W=u({callbackUrl:Ce},S&&{token:S});yield N.login(W),Ae(N);let{signature:no,address:to}=N.account;if(!to){d(!1),console.warn("Login cancelled.");return}no&&S&&f.setTokenLoginInfo({signature:no,address:to}),T(K({address:to,loginMethod:"opera"})),Me({callbackRoute:e,onLoginRedirect:s,options:{signature:no,address:to}})}catch(V){console.error("error loging in",V),p("error logging in"+V.message)}finally{d(!1)}})}let P=Boolean(c);return[E,{loginFailed:P,error:c,isLoading:m&&!P,isLoggedIn:w&&!P}]};n();var vh=require("react");xt();le();Re();D();L();L();C();be();H();n();var Lh=require("react");le();L();n();n();yt();n();n();var Ih=x(require("platform"));_o();n();n();n();n();n();n();n();Ve();C();n();n();var Ah=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");n();n();Ro();je();n();n();$();n();n();var kh=x(Jn());Ve();var tm,rm,am,OG=(am=(rm=(tm=q.safeWindow)==null?void 0:tm.location)==null?void 0:rm.origin)==null?void 0:am.includes("localhost");n();H();n();n();var Fh=require("react");k();n();n();k();var wr=e=>{var c;let o=A(Fo),t=(c=o==null?void 0:o.loginToken)!=null?c:e;if(!t)return;let s=bo(t);if(!!s)return s.origin.toLowerCase().startsWith(ya)};n();n();n();var Ph=x(require("bignumber.js"));n();n();n();var We=x(require("react")),lm=require("react"),_m=require("react"),Lr=x(require("classnames")),um=require("react-dom");k();n();var sm=x(require("react")),cm=x(require("classnames"));var Ch=({className:e,children:o,styles:t})=>sm.default.createElement("div",{className:(0,cm.default)(t==null?void 0:t.dappModalBody,e)},o),vr=re(Ch,{ssrStyles:()=>Promise.resolve().then(()=>(Ee(),Pe)),clientStyles:()=>(Ee(),ee(Pe)).default});n();var Ir=x(require("react")),pm=x(require("classnames"));var Nh=({visible:e,customFooter:o,className:t,footerText:s,styles:c})=>e?Ir.default.createElement("div",{className:(0,pm.default)(c==null?void 0:c.dappModalFooter,t)},o!=null?o:Ir.default.createElement("div",null,s)):null,Ar=re(Nh,{ssrStyles:()=>Promise.resolve().then(()=>(Ee(),Pe)),clientStyles:()=>(Ee(),ee(Pe)).default});n();var ko=x(require("react")),mm=require("@fortawesome/free-solid-svg-icons"),dm=require("@fortawesome/react-fontawesome"),Jo=x(require("classnames"));var Rh=({visible:e,headerText:o,customHeader:t,className:s,closeButtonClassName:c,headerTextClassName:p,onHide:m,globalStyles:d,styles:l})=>e?t?ko.default.createElement("div",{className:(0,Jo.default)(l==null?void 0:l.dappModalHeader,s)},t):ko.default.createElement("div",{className:(0,Jo.default)(l==null?void 0:l.dappModalHeader,s)},ko.default.createElement("div",{className:(0,Jo.default)(l==null?void 0:l.dappModalHeaderText,p)},o),ko.default.createElement("button",{onClick:m,className:(0,Jo.default)(l==null?void 0:l.dappModalCloseButton,d==null?void 0:d.btn,d==null?void 0:d.btnLight,c)},ko.default.createElement(dm.FontAwesomeIcon,{size:"lg",icon:mm.faTimes}))):null,kr=re(Rh,{ssrStyles:()=>Promise.resolve().then(()=>(Ee(),Pe)),clientStyles:()=>(Ee(),ee(Pe)).default});var Mh={showHeader:!0,showFooter:!1,headerText:"",footerText:""},Oh=({"data-testid":e="dappModal",children:o,className:t="dapp-modal-dialog-wrapper",closeOnEscape:s,config:c=Mh,id:p="dapp-modal",onHide:m,parentElement:d,visible:l,styles:f})=>{let[S,T]=(0,lm.useState)(!1);if((0,_m.useEffect)(()=>{T(!0)},[]),!l)return null;let{showHeader:w,showFooter:E,headerText:P,footerText:N,modalDialogClassName:V="dapp-modal-dialog",modalContentClassName:Q="dapp-modal-dialog-content",modalHeaderClassName:ae="dapp-modal-dialog-header",modalHeaderTextClassName:Ce="dapp-modal-dialog-header-text",modalCloseButtonClassName:W="dapp-modal-dialog-close-button",modalBodyClassName:no="dapp-modal-dialog-content-body",modalFooterClassName:to="dapp-modal-dialog-footer",customModalHeader:bm,customModalFooter:wm}=c,vm=Im=>{Im.key==="Escape"&&s&&(m==null||m())};return We.default.createElement(We.default.Fragment,null,S&&(0,um.createPortal)(We.default.createElement("div",{id:p,role:"dialog","aria-modal":"true",className:(0,Lr.default)(V,f==null?void 0:f.dappModal,t),"data-testid":e,onKeyDown:vm},We.default.createElement("div",{className:(0,Lr.default)(f==null?void 0:f.dappModalContent,Q)},We.default.createElement(kr,{visible:w,headerText:P,customHeader:bm,className:ae,headerTextClassName:Ce,closeButtonClassName:W,onHide:m}),We.default.createElement(vr,{className:no},o),We.default.createElement(Ar,{visible:E,customFooter:wm,footerText:N,className:to}))),d!=null?d:document==null?void 0:document.body))},Bh=re(Oh,{ssrStyles:()=>Promise.resolve().then(()=>(Ee(),Pe)),clientStyles:()=>(Ee(),ee(Pe)).default});n();n();var gm=require("react");n();n();var fm=require("react"),Uh=require("@multiversx/sdk-hw-provider");le();D();L();C();be();n();var Wh=require("react");n();n();var Pr=require("react");D();L();C();be();H();n();var Hh=require("react"),$h=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");le();D();L();C();be();$();$();n();var Vh=require("react"),qh=require("@multiversx/sdk-passkey-provider/out");le();D();L();C();be();$();$();n();n();var jh=require("react");B();n();n();n();var Kh=x(require("swr"));n();n();n();n();var zh=x(require("axios"));n();n();var Qo=x(require("react")),xm=x(require("classnames"));var Yh=({onLogin:e,text:o="Default Login Button",className:t="dapp-login-button",btnClassName:s="dapp-default-login-button",disabled:c,"data-testid":p,children:m,globalStyles:d,styles:l})=>{let f=(0,Qo.useMemo)(()=>({wrapper:(0,xm.default)(d==null?void 0:d.btn,d==null?void 0:d.btnPrimary,d==null?void 0:d.px4,d==null?void 0:d.m1,d==null?void 0:d.mx3,{[s]:s!=null},t),loginText:l==null?void 0:l.loginText,wrapperClassName:t}),[d,l]);return Qo.default.createElement("button",{"data-testid":p,disabled:c,className:f.wrapper,onClick:e},m||Qo.default.createElement("span",{className:f.loginText},o))},ym=re(Yh,{ssrStyles:()=>Promise.resolve().then(()=>(Cr(),Er)),clientStyles:()=>(Cr(),ee(Er)).default});n();n();Ve();var Sm=()=>{var e,o;return Boolean(((e=q.safeWindow)==null?void 0:e.isOpera)&&((o=q.safeWindow)==null?void 0:o.elrond))};var Zh=Sm(),Jh=({token:e,className:o="dapp-opera-login",children:t,callbackRoute:s,buttonClassName:c,nativeAuth:p,loginButtonText:m="Opera Crypto Wallet",onLoginRedirect:d,disabled:l,"data-testid":f="operaLoginButton"})=>{let S=wr(e),[T]=br({callbackRoute:s,token:e,onLoginRedirect:d,nativeAuth:p});return Zh?nt.default.createElement(ym,{onLogin:()=>{T()},className:o,btnClassName:c,text:m,disabled:l||S,"data-testid":f},t):nt.default.createElement(nt.default.Fragment,null)};0&&(module.exports={OperaWalletLoginButton});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
