// @multiversx/sdk-dapp/__commonjs/components/ProviderInitializer/index.js
"use strict";var bs=Object.create;var Do=Object.defineProperty,Ss=Object.defineProperties,ws=Object.getOwnPropertyDescriptor,vs=Object.getOwnPropertyDescriptors,As=Object.getOwnPropertyNames,an=Object.getOwnPropertySymbols,ks=Object.getPrototypeOf,ft=Object.prototype.hasOwnProperty,Ur=Object.prototype.propertyIsEnumerable;var Br=(e,o,n)=>o in e?Do(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,f=(e,o)=>{for(var n in o||(o={}))ft.call(o,n)&&Br(e,n,o[n]);if(an)for(var n of an(o))Ur.call(o,n)&&Br(e,n,o[n]);return e},w=(e,o)=>Ss(e,vs(o));var Ue=(e,o)=>{var n={};for(var r in e)ft.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&an)for(var r of an(e))o.indexOf(r)<0&&Ur.call(e,r)&&(n[r]=e[r]);return n};var x=(e,o)=>()=>(e&&(o=e(e=0)),o);var Z=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),Ee=(e,o)=>{for(var n in o)Do(e,n,{get:o[n],enumerable:!0})},Wr=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of As(o))!ft.call(e,a)&&a!==n&&Do(e,a,{get:()=>o[a],enumerable:!(r=ws(o,a))||r.enumerable});return e};var S=(e,o,n)=>(n=e!=null?bs(ks(e)):{},Wr(o||!e||!e.__esModule?Do(n,"default",{value:e,enumerable:!0}):n,e)),Ke=e=>Wr(Do({},"__esModule",{value:!0}),e);var h=(e,o,n)=>new Promise((r,a)=>{var s=l=>{try{d(n.next(l))}catch(u){a(u)}},m=l=>{try{d(n.throw(l))}catch(u){a(u)}},d=l=>l.done?r(l.value):Promise.resolve(l.value).then(s,m);d((n=n.apply(e,o)).next())});var $r=Z(cn=>{"use strict";t();cn.byteLength=Ls;cn.toByteArray=Ps;cn.fromByteArray=Rs;var Te=[],me=[],Is=typeof Uint8Array!="undefined"?Uint8Array:Array,ht="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Xe=0,Gr=ht.length;Xe<Gr;++Xe)Te[Xe]=ht[Xe],me[ht.charCodeAt(Xe)]=Xe;var Xe,Gr;me["-".charCodeAt(0)]=62;me["_".charCodeAt(0)]=63;function Vr(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Ls(e){var o=Vr(e),n=o[0],r=o[1];return(n+r)*3/4-r}function Es(e,o,n){return(o+n)*3/4-n}function Ps(e){var o,n=Vr(e),r=n[0],a=n[1],s=new Is(Es(e,r,a)),m=0,d=a>0?r-4:r,l;for(l=0;l<d;l+=4)o=me[e.charCodeAt(l)]<<18|me[e.charCodeAt(l+1)]<<12|me[e.charCodeAt(l+2)]<<6|me[e.charCodeAt(l+3)],s[m++]=o>>16&255,s[m++]=o>>8&255,s[m++]=o&255;return a===2&&(o=me[e.charCodeAt(l)]<<2|me[e.charCodeAt(l+1)]>>4,s[m++]=o&255),a===1&&(o=me[e.charCodeAt(l)]<<10|me[e.charCodeAt(l+1)]<<4|me[e.charCodeAt(l+2)]>>2,s[m++]=o>>8&255,s[m++]=o&255),s}function Cs(e){return Te[e>>18&63]+Te[e>>12&63]+Te[e>>6&63]+Te[e&63]}function Ns(e,o,n){for(var r,a=[],s=o;s<n;s+=3)r=(e[s]<<16&16711680)+(e[s+1]<<8&65280)+(e[s+2]&255),a.push(Cs(r));return a.join("")}function Rs(e){for(var o,n=e.length,r=n%3,a=[],s=16383,m=0,d=n-r;m<d;m+=s)a.push(Ns(e,m,m+s>d?d:m+s));return r===1?(o=e[n-1],a.push(Te[o>>2]+Te[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(Te[o>>10]+Te[o>>4&63]+Te[o<<2&63]+"=")),a.join("")}});var Hr=Z(xt=>{t();xt.read=function(e,o,n,r,a){var s,m,d=a*8-r-1,l=(1<<d)-1,u=l>>1,y=-7,g=n?a-1:0,b=n?-1:1,L=e[o+g];for(g+=b,s=L&(1<<-y)-1,L>>=-y,y+=d;y>0;s=s*256+e[o+g],g+=b,y-=8);for(m=s&(1<<-y)-1,s>>=-y,y+=r;y>0;m=m*256+e[o+g],g+=b,y-=8);if(s===0)s=1-u;else{if(s===l)return m?NaN:(L?-1:1)*(1/0);m=m+Math.pow(2,r),s=s-u}return(L?-1:1)*m*Math.pow(2,s-r)};xt.write=function(e,o,n,r,a,s){var m,d,l,u=s*8-a-1,y=(1<<u)-1,g=y>>1,b=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,L=r?0:s-1,E=r?1:-1,R=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(d=isNaN(o)?1:0,m=y):(m=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+g>=1?o+=b/l:o+=b*Math.pow(2,1-g),o*l>=2&&(m++,l/=2),m+g>=y?(d=0,m=y):m+g>=1?(d=(o*l-1)*Math.pow(2,a),m=m+g):(d=o*Math.pow(2,g-1)*Math.pow(2,a),m=0));a>=8;e[n+L]=d&255,L+=E,d/=256,a-=8);for(m=m<<a|d,u+=a;u>0;e[n+L]=m&255,L+=E,m/=256,u-=8);e[n+L-E]|=R*128}});var ta=Z(_o=>{"use strict";t();var yt=$r(),mo=Hr(),jr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;_o.Buffer=_;_o.SlowBuffer=Us;_o.INSPECT_MAX_BYTES=50;var pn=2147483647;_o.kMaxLength=pn;_.TYPED_ARRAY_SUPPORT=Ds();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Ds(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function Pe(e){if(e>pn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,_.prototype),o}function _(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return wt(e)}return Kr(e,o,n)}_.poolSize=8192;function Kr(e,o,n){if(typeof e=="string")return Ms(e,o);if(ArrayBuffer.isView(e))return Fs(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(be(e,ArrayBuffer)||e&&be(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(be(e,SharedArrayBuffer)||e&&be(e.buffer,SharedArrayBuffer)))return bt(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return _.from(r,o,n);var a=Bs(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return _.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}_.from=function(e,o,n){return Kr(e,o,n)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function Xr(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Os(e,o,n){return Xr(e),e<=0?Pe(e):o!==void 0?typeof n=="string"?Pe(e).fill(o,n):Pe(e).fill(o):Pe(e)}_.alloc=function(e,o,n){return Os(e,o,n)};function wt(e){return Xr(e),Pe(e<0?0:vt(e)|0)}_.allocUnsafe=function(e){return wt(e)};_.allocUnsafeSlow=function(e){return wt(e)};function Ms(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!_.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=Yr(e,o)|0,r=Pe(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function Tt(e){for(var o=e.length<0?0:vt(e.length)|0,n=Pe(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Fs(e){if(be(e,Uint8Array)){var o=new Uint8Array(e);return bt(o.buffer,o.byteOffset,o.byteLength)}return Tt(e)}function bt(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,_.prototype),r}function Bs(e){if(_.isBuffer(e)){var o=vt(e.length)|0,n=Pe(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||At(e.length)?Pe(0):Tt(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Tt(e.data)}function vt(e){if(e>=pn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+pn.toString(16)+" bytes");return e|0}function Us(e){return+e!=e&&(e=0),_.alloc(+e)}_.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==_.prototype};_.compare=function(o,n){if(be(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),be(n,Uint8Array)&&(n=_.from(n,n.offset,n.byteLength)),!_.isBuffer(o)||!_.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,s=0,m=Math.min(r,a);s<m;++s)if(o[s]!==n[s]){r=o[s],a=n[s];break}return r<a?-1:a<r?1:0};_.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return _.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=_.allocUnsafe(n),s=0;for(r=0;r<o.length;++r){var m=o[r];if(be(m,Uint8Array))s+m.length>a.length?_.from(m).copy(a,s):Uint8Array.prototype.set.call(a,m,s);else if(_.isBuffer(m))m.copy(a,s);else throw new TypeError('"list" argument must be an Array of Buffers');s+=m.length}return a};function Yr(e,o){if(_.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||be(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return St(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return na(e).length;default:if(a)return r?-1:St(e).length;o=(""+o).toLowerCase(),a=!0}}_.byteLength=Yr;function Ws(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return Ys(this,o,n);case"utf8":case"utf-8":return Jr(this,o,n);case"ascii":return Ks(this,o,n);case"latin1":case"binary":return Xs(this,o,n);case"base64":return zs(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Zs(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}_.prototype._isBuffer=!0;function Ye(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}_.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)Ye(this,n,n+1);return this};_.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)Ye(this,n,n+3),Ye(this,n+1,n+2);return this};_.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)Ye(this,n,n+7),Ye(this,n+1,n+6),Ye(this,n+2,n+5),Ye(this,n+3,n+4);return this};_.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?Jr(this,0,o):Ws.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(o){if(!_.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:_.compare(this,o)===0};_.prototype.inspect=function(){var o="",n=_o.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};jr&&(_.prototype[jr]=_.prototype.inspect);_.prototype.compare=function(o,n,r,a,s){if(be(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),!_.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),s===void 0&&(s=this.length),n<0||r>o.length||a<0||s>this.length)throw new RangeError("out of range index");if(a>=s&&n>=r)return 0;if(a>=s)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,s>>>=0,this===o)return 0;for(var m=s-a,d=r-n,l=Math.min(m,d),u=this.slice(a,s),y=o.slice(n,r),g=0;g<l;++g)if(u[g]!==y[g]){m=u[g],d=y[g];break}return m<d?-1:d<m?1:0};function Zr(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,At(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=_.from(o,r)),_.isBuffer(o))return o.length===0?-1:zr(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):zr(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function zr(e,o,n,r,a){var s=1,m=e.length,d=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;s=2,m/=2,d/=2,n/=2}function l(L,E){return s===1?L[E]:L.readUInt16BE(E*s)}var u;if(a){var y=-1;for(u=n;u<m;u++)if(l(e,u)===l(o,y===-1?0:u-y)){if(y===-1&&(y=u),u-y+1===d)return y*s}else y!==-1&&(u-=u-y),y=-1}else for(n+d>m&&(n=m-d),u=n;u>=0;u--){for(var g=!0,b=0;b<d;b++)if(l(e,u+b)!==l(o,b)){g=!1;break}if(g)return u}return-1}_.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};_.prototype.indexOf=function(o,n,r){return Zr(this,o,n,r,!0)};_.prototype.lastIndexOf=function(o,n,r){return Zr(this,o,n,r,!1)};function Gs(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var s=o.length;r>s/2&&(r=s/2);for(var m=0;m<r;++m){var d=parseInt(o.substr(m*2,2),16);if(At(d))return m;e[n+m]=d}return m}function Vs(e,o,n,r){return sn(St(o,e.length-n),e,n,r)}function $s(e,o,n,r){return sn(em(o),e,n,r)}function Hs(e,o,n,r){return sn(na(o),e,n,r)}function js(e,o,n,r){return sn(om(o,e.length-n),e,n,r)}_.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var s=this.length-n;if((r===void 0||r>s)&&(r=s),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var m=!1;;)switch(a){case"hex":return Gs(this,o,n,r);case"utf8":case"utf-8":return Vs(this,o,n,r);case"ascii":case"latin1":case"binary":return $s(this,o,n,r);case"base64":return Hs(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return js(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),m=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function zs(e,o,n){return o===0&&n===e.length?yt.fromByteArray(e):yt.fromByteArray(e.slice(o,n))}function Jr(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var s=e[a],m=null,d=s>239?4:s>223?3:s>191?2:1;if(a+d<=n){var l,u,y,g;switch(d){case 1:s<128&&(m=s);break;case 2:l=e[a+1],(l&192)===128&&(g=(s&31)<<6|l&63,g>127&&(m=g));break;case 3:l=e[a+1],u=e[a+2],(l&192)===128&&(u&192)===128&&(g=(s&15)<<12|(l&63)<<6|u&63,g>2047&&(g<55296||g>57343)&&(m=g));break;case 4:l=e[a+1],u=e[a+2],y=e[a+3],(l&192)===128&&(u&192)===128&&(y&192)===128&&(g=(s&15)<<18|(l&63)<<12|(u&63)<<6|y&63,g>65535&&g<1114112&&(m=g))}}m===null?(m=65533,d=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),a+=d}return qs(r)}var qr=4096;function qs(e){var o=e.length;if(o<=qr)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=qr));return n}function Ks(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function Xs(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function Ys(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",s=o;s<n;++s)a+=nm[e[s]];return a}function Zs(e,o,n){for(var r=e.slice(o,n),a="",s=0;s<r.length-1;s+=2)a+=String.fromCharCode(r[s]+r[s+1]*256);return a}_.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,_.prototype),a};function $(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=this[o],s=1,m=0;++m<n&&(s*=256);)a+=this[o+m]*s;return a};_.prototype.readUintBE=_.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=this[o+--n],s=1;n>0&&(s*=256);)a+=this[o+--n]*s;return a};_.prototype.readUint8=_.prototype.readUInt8=function(o,n){return o=o>>>0,n||$(o,1,this.length),this[o]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||$(o,2,this.length),this[o]|this[o+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||$(o,2,this.length),this[o]<<8|this[o+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||$(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||$(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};_.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=this[o],s=1,m=0;++m<n&&(s*=256);)a+=this[o+m]*s;return s*=128,a>=s&&(a-=Math.pow(2,8*n)),a};_.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=n,s=1,m=this[o+--a];a>0&&(s*=256);)m+=this[o+--a]*s;return s*=128,m>=s&&(m-=Math.pow(2,8*n)),m};_.prototype.readInt8=function(o,n){return o=o>>>0,n||$(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};_.prototype.readInt16LE=function(o,n){o=o>>>0,n||$(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};_.prototype.readInt16BE=function(o,n){o=o>>>0,n||$(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};_.prototype.readInt32LE=function(o,n){return o=o>>>0,n||$(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};_.prototype.readInt32BE=function(o,n){return o=o>>>0,n||$(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};_.prototype.readFloatLE=function(o,n){return o=o>>>0,n||$(o,4,this.length),mo.read(this,o,!0,23,4)};_.prototype.readFloatBE=function(o,n){return o=o>>>0,n||$(o,4,this.length),mo.read(this,o,!1,23,4)};_.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||$(o,8,this.length),mo.read(this,o,!0,52,8)};_.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||$(o,8,this.length),mo.read(this,o,!1,52,8)};function J(e,o,n,r,a,s){if(!_.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<s)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var s=Math.pow(2,8*r)-1;J(this,o,n,r,s,0)}var m=1,d=0;for(this[n]=o&255;++d<r&&(m*=256);)this[n+d]=o/m&255;return n+r};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var s=Math.pow(2,8*r)-1;J(this,o,n,r,s,0)}var m=r-1,d=1;for(this[n+m]=o&255;--m>=0&&(d*=256);)this[n+m]=o/d&255;return n+r};_.prototype.writeUint8=_.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,1,255,0),this[n]=o&255,n+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};_.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var s=Math.pow(2,8*r-1);J(this,o,n,r,s-1,-s)}var m=0,d=1,l=0;for(this[n]=o&255;++m<r&&(d*=256);)o<0&&l===0&&this[n+m-1]!==0&&(l=1),this[n+m]=(o/d>>0)-l&255;return n+r};_.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var s=Math.pow(2,8*r-1);J(this,o,n,r,s-1,-s)}var m=r-1,d=1,l=0;for(this[n+m]=o&255;--m>=0&&(d*=256);)o<0&&l===0&&this[n+m+1]!==0&&(l=1),this[n+m]=(o/d>>0)-l&255;return n+r};_.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};_.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};_.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||J(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function Qr(e,o,n,r,a,s){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function ea(e,o,n,r,a){return o=+o,n=n>>>0,a||Qr(e,o,n,4,34028234663852886e22,-34028234663852886e22),mo.write(e,o,n,r,23,4),n+4}_.prototype.writeFloatLE=function(o,n,r){return ea(this,o,n,!0,r)};_.prototype.writeFloatBE=function(o,n,r){return ea(this,o,n,!1,r)};function oa(e,o,n,r,a){return o=+o,n=n>>>0,a||Qr(e,o,n,8,17976931348623157e292,-17976931348623157e292),mo.write(e,o,n,r,52,8),n+8}_.prototype.writeDoubleLE=function(o,n,r){return oa(this,o,n,!0,r)};_.prototype.writeDoubleBE=function(o,n,r){return oa(this,o,n,!1,r)};_.prototype.copy=function(o,n,r,a){if(!_.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var s=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),s};_.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!_.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var s=o.charCodeAt(0);(a==="utf8"&&s<128||a==="latin1")&&(o=s)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var d=_.isBuffer(o)?o:_.from(o,a),l=d.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=d[m%l]}return this};var Js=/[^+/0-9A-Za-z-_]/g;function Qs(e){if(e=e.split("=")[0],e=e.trim().replace(Js,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function St(e,o){o=o||1/0;for(var n,r=e.length,a=null,s=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&s.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&s.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&s.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&s.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;s.push(n)}else if(n<2048){if((o-=2)<0)break;s.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;s.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;s.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return s}function em(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function om(e,o){for(var n,r,a,s=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,a=n%256,s.push(a),s.push(r);return s}function na(e){return yt.toByteArray(Qs(e))}function sn(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function be(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function At(e){return e!==e}var nm=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var pa=Z((wl,ca)=>{t();var G=ca.exports={},Se,we;function kt(){throw new Error("setTimeout has not been defined")}function It(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?Se=setTimeout:Se=kt}catch(e){Se=kt}try{typeof clearTimeout=="function"?we=clearTimeout:we=It}catch(e){we=It}})();function ra(e){if(Se===setTimeout)return setTimeout(e,0);if((Se===kt||!Se)&&setTimeout)return Se=setTimeout,setTimeout(e,0);try{return Se(e,0)}catch(o){try{return Se.call(null,e,0)}catch(n){return Se.call(this,e,0)}}}function tm(e){if(we===clearTimeout)return clearTimeout(e);if((we===It||!we)&&clearTimeout)return we=clearTimeout,clearTimeout(e);try{return we(e)}catch(o){try{return we.call(null,e)}catch(n){return we.call(this,e)}}}var Ce=[],lo=!1,Ze,mn=-1;function rm(){!lo||!Ze||(lo=!1,Ze.length?Ce=Ze.concat(Ce):mn=-1,Ce.length&&aa())}function aa(){if(!lo){var e=ra(rm);lo=!0;for(var o=Ce.length;o;){for(Ze=Ce,Ce=[];++mn<o;)Ze&&Ze[mn].run();mn=-1,o=Ce.length}Ze=null,lo=!1,tm(e)}}G.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];Ce.push(new ia(e,o)),Ce.length===1&&!lo&&ra(aa)};function ia(e,o){this.fun=e,this.array=o}ia.prototype.run=function(){this.fun.apply(null,this.array)};G.title="browser";G.browser=!0;G.env={};G.argv=[];G.version="";G.versions={};function Ne(){}G.on=Ne;G.addListener=Ne;G.once=Ne;G.off=Ne;G.removeListener=Ne;G.removeAllListeners=Ne;G.emit=Ne;G.prependListener=Ne;G.prependOnceListener=Ne;G.listeners=function(e){return[]};G.binding=function(e){throw new Error("process.binding is not supported")};G.cwd=function(){return"/"};G.chdir=function(e){throw new Error("process.chdir is not supported")};G.umask=function(){return 0}});var i,p,am,c,t=x(()=>{i=S(ta()),p=S(pa()),am=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object),c=am});var sa=x(()=>{"use strict";t()});var ma=x(()=>{"use strict";t()});var da=x(()=>{"use strict";t()});var Lt,_a=x(()=>{"use strict";t();Lt=(r=>(r.SetGuardian="SetGuardian",r.GuardAccount="GuardAccount",r.UnGuardAccount="UnGuardAccount",r))(Lt||{})});var la=x(()=>{"use strict";t()});var Et=x(()=>{"use strict";t()});var ua=x(()=>{"use strict";t()});var ga=x(()=>{"use strict";t()});var fa=x(()=>{"use strict";t()});var ha=x(()=>{"use strict";t()});var We,uo,Oo=x(()=>{"use strict";t();We=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),uo=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var de,xa,Ol,ya,Ml,re=x(()=>{"use strict";t();Oo();de=(b=>(b.ledger="ledger",b.walletconnect="walletconnect",b.walletconnectv2="walletconnectv2",b.wallet="wallet",b.crossWindow="crossWindow",b.iframe="iframe",b.extension="extension",b.passkey="passkey",b.metamask="metamask",b.opera="opera",b.extra="extra",b.none="",b))(de||{}),xa=(n=>(n.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",n.reloginRequest="RELOGIN_REQUEST",n))(xa||{}),Ol=f(f({},uo.WindowProviderRequestEnums),xa),ya=(o=>(o.reloginResponse="RELOGIN_RESPONSE",o))(ya||{}),Ml=f(f({},uo.WindowProviderResponseEnums),ya)});var Ta=x(()=>{"use strict";t()});var ba=x(()=>{"use strict";t()});var Q=x(()=>{"use strict";t()});var Sa=x(()=>{"use strict";t()});var wa=x(()=>{"use strict";t()});var va=x(()=>{"use strict";t()});var q=x(()=>{"use strict";t();ga();fa();ha();re();Ta();ba();Q();Sa();wa();va()});var Mo,dn,Ql,_n,eu,ln,ou,nu,im,go=x(()=>{"use strict";t();q();Mo={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:dn,egldLabel:Ql}=Mo["devnet"],{chainId:_n,egldLabel:eu}=Mo["testnet"],{chainId:ln,egldLabel:ou}=Mo["mainnet"],nu={["devnet"]:dn,["testnet"]:_n,["mainnet"]:ln},im={[dn]:"devnet",[_n]:"testnet",[ln]:"mainnet"}});var Pt=x(()=>{"use strict";t()});var ve,Aa=x(()=>{"use strict";t();ve=require("@multiversx/sdk-web-wallet-provider")});var ae,Fo=x(()=>{"use strict";t();ae=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var ka,Ia,Ct,pu,su,La=x(()=>{"use strict";t();Fo();Ct=String((Ia=(ka=ae.safeWindow)==null?void 0:ka.navigator)==null?void 0:Ia.userAgent),pu=/^((?!chrome|android).)*safari/i.test(Ct),su=/firefox/i.test(Ct)&&/windows/i.test(Ct)});var Nt,Rt,Ea,Pa,un,Ca,fo,N=x(()=>{"use strict";t();sa();ma();da();_a();la();Et();ua();go();Pt();Aa();La();Nt=5e4,Rt=1e9,Ea=1,Pa=1,un="logout",Ca="login",fo="0"});var ho,gn=x(()=>{"use strict";t();ho=()=>Date.now()/1e3});var Na=x(()=>{"use strict";t()});var Ra=x(()=>{"use strict";t()});var Da=x(()=>{"use strict";t();gn();Na();Ra()});var Dt={};Ee(Dt,{clear:()=>mm,getItem:()=>pm,localStorageKeys:()=>ee,removeItem:()=>sm,setItem:()=>cm});var ee,fn,cm,pm,sm,mm,Je=x(()=>{"use strict";t();Da();ee={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},fn=typeof localStorage!="undefined",cm=({key:e,data:o,expires:n})=>{!fn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},pm=e=>{if(!fn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:ho()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},sm=e=>{!fn||localStorage.removeItem(String(e))},mm=()=>{!fn||localStorage.clear()}});var Ot={};Ee(Ot,{clear:()=>Ba,getItem:()=>Ma,removeItem:()=>Fa,setItem:()=>Oa,storage:()=>dm});var Oa,Ma,Fa,Ba,dm,Ua=x(()=>{"use strict";t();Oa=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Ma=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Fa=e=>sessionStorage.removeItem(String(e)),Ba=()=>sessionStorage.clear(),dm={setItem:Oa,getItem:Ma,removeItem:Fa,clear:Ba}});var ie,xo=x(()=>{"use strict";t();Je();Ua();ie={session:Ot,local:Dt}});var Mt,B,he,K=x(()=>{"use strict";t();Mt=require("@reduxjs/toolkit");N();B=(0,Mt.createAction)(un),he=(0,Mt.createAction)(Ca,e=>({payload:e}))});var Bt,Wa,Ga,Qe,Ft,Va,Re,Bo,Ut,Wu,$a,Gu,Vu,yo,Ha,$u,Hu,hn,xn=x(()=>{"use strict";t();Bt=require("@multiversx/sdk-core"),Wa=require("@reduxjs/toolkit"),Ga=require("redux-persist");N();xo();Je();K();Qe={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:fo},Ft={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":Qe},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Va=(0,Wa.createSlice)({name:"accountInfoSlice",initialState:Ft,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=n?new Bt.Address(n).hex():""},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:Qe},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}},setWebsocketBatchEvent:(e,o)=>{e.websocketBatchEvent={timestamp:Date.now(),data:o.payload}}},extraReducers:e=>{e.addCase(B,()=>(ie.local.removeItem(ee.loginExpiresAt),Ft)),e.addCase(he,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new Bt.Address(r).toHex()}),e.addCase(Ga.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:a,shard:s,accounts:m,publicKey:d}=r;o.address=a,o.shard=s;let l=m&&a in m;o.accounts=l?m:Ft.accounts,o.publicKey=d})}}),{setAccount:Re,setAddress:Bo,setAccountNonce:Ut,setAccountShard:Wu,setLedgerAccount:$a,updateLedgerAccount:Gu,setWalletConnectAccount:Vu,setIsAccountLoading:yo,setAccountLoadingError:Ha,setWebsocketEvent:$u,setWebsocketBatchEvent:Hu}=Va.actions,hn=Va.reducer});function Uo(){return new Date().setHours(new Date().getHours()+24)}function Wo(e){ie.local.setItem({key:ee.loginExpiresAt,data:e,expires:e})}var Wt=x(()=>{"use strict";t();xo();Je()});var za,ja,qa,eg,Ka,og,Ge,ng,Gt,Xa,tg,Vt,rg,yn,Tn=x(()=>{"use strict";t();za=require("@reduxjs/toolkit");Wt();re();K();ja={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},qa=(0,za.createSlice)({name:"loginInfoSlice",initialState:ja,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},setWebviewLogin:(e,o)=>{e.webviewLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload},setIsWalletConnectV2Initialized:(e,o)=>{e.isWalletConnectV2Initialized=o.payload}},extraReducers:e=>{e.addCase(B,()=>ja),e.addCase(he,(o,n)=>{o.isLoginSessionInvalid=!1,o.loginMethod=n.payload.loginMethod,o.iframeLoginType=n.payload.iframeLoginType,Wo(Uo())})}}),{setLoginMethod:eg,setWalletConnectLogin:Ka,setLedgerLogin:og,setTokenLogin:Ge,setTokenLoginSignature:ng,setWalletLogin:Gt,invalidateLoginSession:Xa,setLogoutRoute:tg,setIsWalletConnectV2Initialized:Vt,setWebviewLogin:rg}=qa.actions,yn=qa.reducer});var Za,Ya,Ja,pg,sg,mg,dg,bn,Sn=x(()=>{"use strict";t();Za=require("@reduxjs/toolkit");K();Ya={},Ja=(0,Za.createSlice)({name:"modalsSlice",initialState:Ya,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(B,()=>Ya)}}),{setTxSubmittedModal:pg,setNotificationModal:sg,clearTxSubmittedModal:mg,clearNotificationModal:dg}=Ja.actions,bn=Ja.reducer});var Ve,Go=x(()=>{"use strict";t();Ve=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var H,Ae=x(()=>{"use strict";t();Go();H=()=>{if(!Ve())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:o,hash:n,origin:r,href:a,search:s}}=window;return{pathname:o,hash:n,origin:r,href:a,search:s}}});var $t,Qa=x(()=>{"use strict";t();Ae();$t=(e="")=>{let o=H(),n=e.startsWith("http")||e.startsWith("www.");return!o.origin||e.startsWith(o.origin)||n?e:`${o.origin}/${e.replace("/","")}`}});var ei=x(()=>{"use strict";t();To()});var oi,ni=x(()=>{"use strict";t();Fo();oi=e=>{if(!ae.safeWindow.location)return;if(!new URL(ae.safeWindow.location.href).protocol.startsWith("http"))return ae.safeWindow.location.reload();ae.safeWindow.location.assign(e)}});var To=x(()=>{"use strict";t();Qa();ei();Ae();ni()});var ti,ri,wn,Ht=x(()=>{"use strict";t();To();ti=!1,ri=(e=!0)=>{ti=e},wn=({timeout:e=0,url:o})=>{ti||setTimeout(()=>{if(!!window)return oi(o)},e)}});function ai({callbackRoute:e,onLoginRedirect:o,options:n}){let r=Boolean(e),a=typeof o=="function";if(r&&e!=null){if(a)return o(e,n);wn({url:e,timeout:_m})}}var _m,ii=x(()=>{"use strict";t();Ht();_m=200});function ci(e){return e[Math.floor(Math.random()*e.length)]}var pi=x(()=>{"use strict";t()});var si=x(()=>{"use strict";t();Et()});var jt=x(()=>{"use strict";t();ii();pi();si()});var mi,di,_i,zt,um,li,$g,qt,Hg,vn,An=x(()=>{"use strict";t();mi=require("@reduxjs/toolkit"),di=S(require("lodash.omit")),_i=require("redux-persist");Pt();K();jt();zt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},um={network:zt},li=(0,mi.createSlice)({name:"appConfig",initialState:um,reducers:{initializeNetworkConfig:(e,o)=>{let n=ci(o.payload.walletConnectV2RelayAddresses),r=(0,di.default)(o.payload,"walletConnectV2RelayAddresses");e.network=w(f(f({},e.network),r),{walletConnectV2RelayAddress:n})},updateNetworkConfig:(e,o)=>{e.network=f(f({},e.network),o.payload)},setCustomWalletAddress:(e,o)=>{e.network.customWalletAddress=o.payload}},extraReducers:e=>{e.addCase(B,o=>{o.network.customWalletAddress=void 0}),e.addCase(_i.REHYDRATE,(o,n)=>{var a,s;if(!((s=(a=n.payload)==null?void 0:a.network)!=null&&s.customWalletAddress))return;let{network:{customWalletAddress:r}}=n.payload;o.network.customWalletAddress=r})}}),{initializeNetworkConfig:$g,updateNetworkConfig:qt,setCustomWalletAddress:Hg}=li.actions,vn=li.reducer});var ui,Kt,gi,Yg,Zg,Jg,kn,In=x(()=>{"use strict";t();ui=require("@reduxjs/toolkit");q();K();Kt={isSigning:!1,signedSessions:{}},gi=(0,ui.createSlice)({name:"signedMessageInfoSliceState",initialState:Kt,reducers:{setSignSession:(e,o)=>{let{sessionId:n,signedSession:r,errorMessage:a}=o.payload;a&&(e.errorMessage=a),e.isSigning=r.status==="pending",e.signedSessions[n]=f(f({},e.signedSessions[n]),r)},setSignSessionState:(e,o)=>f(f({},e),o.payload),clearSignedMessageInfo:()=>Kt},extraReducers:e=>{e.addCase(B,()=>Kt)}}),{setSignSession:Yg,clearSignedMessageInfo:Zg,setSignSessionState:Jg}=gi.actions,kn=gi.reducer});var hi,xi,fi,yi,Xt,gm,af,cf,Ln,En=x(()=>{"use strict";t();hi=require("@reduxjs/toolkit"),xi=require("redux-persist");q();gn();K();fi={customToasts:[],transactionToasts:[]},yi=(0,hi.createSlice)({name:"toastsSlice",initialState:fi,reducers:{addCustomToast:(e,o)=>{let n=o.payload.toastId||`custom-toast-${e.customToasts.length+1}`,r=e.customToasts.findIndex(a=>a.toastId===n);if(r!==-1){e.customToasts[r]=w(f(f({},e.customToasts[r]),o.payload),{type:"custom",toastId:n});return}e.customToasts.push(w(f({},o.payload),{type:"custom",toastId:n}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:ho(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)}},extraReducers:e=>{e.addCase(B,()=>fi),e.addCase(xi.REHYDRATE,(o,n)=>{var a,s;let r=(s=(a=n.customToasts)==null?void 0:a.filter(m=>!("component"in m)))!=null?s:[];o.customToasts=r})}}),{addCustomToast:Xt,removeCustomToast:gm,addTransactionToast:af,removeTransactionToast:cf}=yi.actions,Ln=yi.reducer});var Ti,Zt,Jt,Qt,fm,Yt,bi,df,_f,lf,Pn,Cn=x(()=>{"use strict";t();Ti=require("@reduxjs/toolkit");K();Zt="Transaction failed",Jt="Transaction successful",Qt="Processing transaction",fm="Transaction submitted",Yt={},bi=(0,Ti.createSlice)({name:"transactionsInfo",initialState:Yt,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||Zt,successMessage:(r==null?void 0:r.successMessage)||Jt,processingMessage:(r==null?void 0:r.processingMessage)||Qt,submittedMessage:(r==null?void 0:r.submittedMessage)||fm,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>Yt},extraReducers:e=>{e.addCase(B,()=>Yt)}}),{clearTransactionsInfo:df,setTransactionsDisplayInfo:_f,clearTransactionsInfoForSessionId:lf}=bi.actions,Pn=bi.reducer});function Si(e){return e!=null&&(Im(e)||Nm(e))}function Nn(e){return e!=null&&(Lm(e)||Rm(e))}function Rn(e){return e!=null&&(Em(e)||Dm(e))}function wi(e){return e!=null&&(Pm(e)||Om(e))}function vi(e){return e!=null&&Cm(e)}function Im(e){return e!=null&&xm.includes(e)}function Lm(e){return e!=null&&ym.includes(e)}function Em(e){return e!=null&&Tm.includes(e)}function Pm(e){return e!=null&&bm.includes(e)}function Cm(e){return e!=null&&Sm.includes(e)}function Nm(e){return e!=null&&wm.includes(e)}function Rm(e){return e!=null&&vm.includes(e)}function Dm(e){return e!=null&&Am.includes(e)}function Om(e){return e!=null&&km.includes(e)}var xm,ym,Tm,bm,Sm,wm,vm,Am,km,or=x(()=>{"use strict";t();re();xm=["sent"],ym=["success"],Tm=["fail","cancelled","timedOut"],bm=["invalid"],Sm=["timedOut"],wm=["pending"],vm=["success"],Am=["fail","invalid"],km=["not executed"]});var Ai,ki,Vo,Mm,Ii,Sf,wf,vf,Af,kf,If,Lf,Ef,Pf,Cf,Nf,Dn,On=x(()=>{"use strict";t();Ai=require("@reduxjs/toolkit"),ki=require("redux-persist");re();or();K();Vo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Mm={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Ii=(0,Ai.createSlice)({name:"transactionsSlice",initialState:Vo,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,u;let{sessionId:n,transactions:r,errorMessage:a,status:s,redirectRoute:m,customTransactionInformation:d}=o.payload;e.customTransactionInformationForSessionId[n]=f(f(f({},Mm),((l=e.signedTransactions[n])==null?void 0:l.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[n]||{}),e.signedTransactions[n]={transactions:r,status:s,errorMessage:a,redirectRoute:m,customTransactionInformation:f(f({},e.customTransactionInformationForSessionId[n]),d!=null?d:{})},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=Vo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=f(f({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:s}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),s!=null&&(e.signedTransactions[n].transactions=s))},updateSignedTransactionStatus:(e,o)=>{var u,y,g,b,L,E,R,z;let{sessionId:n,status:r,errorMessage:a,transactionHash:s,serverTransaction:m,inTransit:d}=o.payload,l=(y=(u=e.signedTransactions)==null?void 0:u[n])==null?void 0:y.transactions;if(l!=null){e.signedTransactions[n].transactions=l.map(F=>F.hash===s?w(f(f({},m!=null?m:{}),F),{status:r,errorMessage:a,inTransit:d}):F);let V=(b=(g=e.signedTransactions[n])==null?void 0:g.transactions)==null?void 0:b.every(F=>Nn(F.status)),j=(E=(L=e.signedTransactions[n])==null?void 0:L.transactions)==null?void 0:E.some(F=>Rn(F.status)),ge=(z=(R=e.signedTransactions[n])==null?void 0:R.transactions)==null?void 0:z.every(F=>wi(F.status));V&&(e.signedTransactions[n].status="success"),j&&(e.signedTransactions[n].status="fail"),ge&&(e.signedTransactions[n].status="invalid")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Vo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Vo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload},updateSignedTransactionsCustomTransactionInformation:(e,o)=>{let{sessionId:n,customTransactionInformationOverrides:r}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].customTransactionInformation=f(f({},e.signedTransactions[n].customTransactionInformation),r))}},extraReducers:e=>{e.addCase(B,()=>Vo),e.addCase(ki.REHYDRATE,(o,n)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,s=Object.entries(r).reduce((d,[l,u])=>{let y=new Date(l),g=new Date;return g.setHours(g.getHours()+5),g-y>0||(d[l]=u),d},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=s)})}}),{updateSignedTransactionStatus:Sf,updateSignedTransactions:wf,setTransactionsToSign:vf,clearAllTransactionsToSign:Af,clearAllSignedTransactions:kf,clearSignedTransaction:If,clearTransactionToSign:Lf,setSignTransactionsError:Ef,setSignTransactionsCancelMessage:Pf,moveTransactionsToSignedState:Cf,updateSignedTransactionsCustomTransactionInformation:Nf}=Ii.actions,Dn=Ii.reducer});var Li,nr,Ei,Mf,Ff,Bf,Uf,Mn,Fn=x(()=>{"use strict";t();Li=require("@reduxjs/toolkit");K();nr={},Ei=(0,Li.createSlice)({name:"batchTransactionsSlice",initialState:nr,reducers:{setBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},updateBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},clearBatchTransactions:(e,o)=>{delete e[o.payload.batchId]},clearAllBatchTransactions:()=>nr},extraReducers:e=>{e.addCase(B,()=>nr)}}),{setBatchTransactions:Mf,updateBatchTransactions:Ff,clearBatchTransactions:Bf,clearAllBatchTransactions:Uf}=Ei.actions,Mn=Ei.reducer});var Ci,Pi,Ni,$f,Ri,tr=x(()=>{"use strict";t();Ci=require("@reduxjs/toolkit");K();Pi={},Ni=(0,Ci.createSlice)({name:"dappConfigSlice",initialState:Pi,reducers:{setDappConfig:(e,o)=>o.payload},extraReducers:e=>{e.addCase(B,()=>Pi)}}),{setDappConfig:$f}=Ni.actions,Ri=Ni.reducer});var ce=x(()=>{"use strict";t();xn();Tn();Sn();An();In();En();Cn();On();Fn();tr()});var ir=x(()=>{"use strict";t()});var Ji,Km,Xm,no,qn=x(()=>{"use strict";t();Ji=require("@reduxjs/toolkit");ir();xn();Fn();tr();Tn();Sn();An();In();En();Cn();On();Km={["account"]:hn,["dappConfig"]:Ri,["loginInfo"]:yn,["modals"]:bn,["networkConfig"]:vn,["signedMessageInfo"]:kn,["toasts"]:Ln,["transactionsInfo"]:Pn,["transactions"]:Dn,["batchTransactions"]:Mn},Xm=(e={})=>(0,Ji.combineReducers)(f(f({},Km),e)),no=Xm});var nc={};Ee(nc,{default:()=>dd,sessionStorageReducers:()=>cr});function Me(e,o=[]){return{key:e,version:1,storage:ec.default,blacklist:o}}var oe,Qi,ec,Ym,Ho,Zm,Jm,Qm,ed,od,nd,td,rd,ad,id,oc,cd,cr,pd,sd,md,dd,tc=x(()=>{"use strict";t();oe=require("redux-persist"),Qi=S(require("redux-persist/lib/storage")),ec=S(require("redux-persist/lib/storage/session"));qn();ce();xn();Fn();Tn();Sn();An();In();En();Cn();On();ir();Ym={persistReducersStorageType:"localStorage"},Ho={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Zm=Me(Ho["account"]),Jm=Me(Ho["loginInfo"]),Qm=Me(Ho["modals"]),ed=Me(Ho["networkConfig"]),od={2:e=>w(f({},e),{networkConfig:zt})};nd=Me("sdk-dapp-transactionsInfo"),td=Me("sdk-dapp-transactions",["transactionsToSign"]),rd=Me("sdk-dapp-batchTransactions",["batchTransactions"]),ad=Me("sdk-dapp-toasts"),id=Me("sdk-dapp-signedMessageInfo"),oc={key:"sdk-dapp-store",version:2,storage:Qi.default,whitelist:Object.keys(Ho),migrate:(0,oe.createMigrate)(od,{debug:!1})},cd=w(f({},oc),{whitelist:[]}),cr={["toasts"]:(0,oe.persistReducer)(ad,Ln),["transactions"]:(0,oe.persistReducer)(td,Dn),["transactionsInfo"]:(0,oe.persistReducer)(nd,Pn),["batchTransactions"]:(0,oe.persistReducer)(rd,Mn),["signedMessageInfo"]:(0,oe.persistReducer)(id,kn)},pd=w(f({},cr),{["account"]:(0,oe.persistReducer)(Zm,hn),["loginInfo"]:(0,oe.persistReducer)(Jm,yn),["modals"]:(0,oe.persistReducer)(Qm,bn),["networkConfig"]:(0,oe.persistReducer)(ed,vn)}),sd=Ym.persistReducersStorageType==="localStorage",md=sd?(0,oe.persistReducer)(oc,no(cr)):(0,oe.persistReducer)(cd,no(pd)),dd=md});var rc={};Ee(rc,{default:()=>_d});var _d,ac=x(()=>{"use strict";t();qn();_d=no()});var ic={};Ee(ic,{default:()=>ud});var _e,ld,ud,cc=x(()=>{"use strict";t();_e=require("redux-persist"),ld=[_e.FLUSH,_e.REHYDRATE,_e.PAUSE,_e.PERSIST,_e.PURGE,_e.REGISTER],ud=ld});var mc={};Ee(mc,{default:()=>sc});function sc(e){return(0,pc.persistStore)(e)}var pc,dc=x(()=>{"use strict";t();pc=require("redux-persist")});var op=Z((lL,ep)=>{t();var E_=typeof c=="object"&&c&&c.Object===Object&&c;ep.exports=E_});var tp=Z((uL,np)=>{t();var P_=op(),C_=typeof self=="object"&&self&&self.Object===Object&&self,N_=P_||C_||Function("return this")();np.exports=N_});var kr=Z((gL,rp)=>{t();var R_=tp(),D_=R_.Symbol;rp.exports=D_});var pp=Z((fL,cp)=>{t();var ap=kr(),ip=Object.prototype,O_=ip.hasOwnProperty,M_=ip.toString,Xo=ap?ap.toStringTag:void 0;function F_(e){var o=O_.call(e,Xo),n=e[Xo];try{e[Xo]=void 0;var r=!0}catch(s){}var a=M_.call(e);return r&&(o?e[Xo]=n:delete e[Xo]),a}cp.exports=F_});var mp=Z((hL,sp)=>{t();var B_=Object.prototype,U_=B_.toString;function W_(e){return U_.call(e)}sp.exports=W_});var up=Z((xL,lp)=>{t();var dp=kr(),G_=pp(),V_=mp(),$_="[object Null]",H_="[object Undefined]",_p=dp?dp.toStringTag:void 0;function j_(e){return e==null?e===void 0?H_:$_:_p&&_p in Object(e)?G_(e):V_(e)}lp.exports=j_});var fp=Z((yL,gp)=>{t();var z_=Array.isArray;gp.exports=z_});var xp=Z((TL,hp)=>{t();function q_(e){return e!=null&&typeof e=="object"}hp.exports=q_});var Tp=Z((bL,yp)=>{t();var K_=up(),X_=fp(),Y_=xp(),Z_="[object String]";function J_(e){return typeof e=="string"||!X_(e)&&Y_(e)&&K_(e)==Z_}yp.exports=J_});var Ip=Z((YL,kp)=>{t();function ol(e){var o=typeof e;return e!=null&&(o=="object"||o=="function")}kp.exports=ol});var Er={};Ee(Er,{css:()=>cs,default:()=>pl});var cs,pl,Pr=x(()=>{"use strict";t();cs=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(cs));pl={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var Nr={};Ee(Nr,{css:()=>ds,default:()=>dl});var ds,dl,Rr=x(()=>{"use strict";t();ds=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ds));dl={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var xl={};Ee(xl,{ProviderInitializer:()=>hl});module.exports=Ke(xl);t();t();var ue=require("react");t();t();t();var r_=S(require("axios"));t();t();ce();t();var Di=S(require("lodash.isequal")),Bn=require("reselect"),T=(0,Bn.createSelectorCreator)(Bn.defaultMemoize,Di.default);var ke=e=>e.account,$e=T(ke,e=>e.address),He=T(ke,$e,(e,o)=>o in e.accounts?e.accounts[o]:Qe),Fm=T(ke,He,(e,o)=>{let a=e,{accounts:n}=a,r=Ue(a,["accounts"]);return w(f({},r),{address:o.address,account:o})}),ah=T(He,e=>e.balance),Oi=T(He,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),ih=T(ke,e=>e.shard),Mi=T(ke,e=>e.ledgerAccount),ch=T(ke,e=>e.walletConnectAccount),ph=T(ke,e=>e.isAccountLoading),sh=T(ke,e=>e.accountLoadingError),Bm=T(ke,e=>e.websocketEvent),Um=T(ke,e=>e.websocketBatchEvent);t();var Wm=e=>e.dappConfig,Fi=T(Wm,e=>e.shouldUseWebViewProvider);t();var X=e=>e.loginInfo,Bi=T(X,e=>e.loginMethod),De=T(X,$e,(e,o)=>Boolean(o)),Ui=T(X,e=>e.walletConnectLogin),Un=T(X,e=>e.ledgerLogin),Wn=T(X,e=>e.walletLogin),gh=T(X,e=>e.isLoginSessionInvalid),bo=T(X,e=>e.tokenLogin),Gn=T(X,e=>e.logoutRoute),Gm=T(X,e=>e.isWalletConnectV2Initialized);t();var Wi=e=>e.modals,xh=T(Wi,e=>e.txSubmittedModal),yh=T(Wi,e=>e.notificationModal);t();var Oe=e=>e.networkConfig,Vn=T(Oe,e=>e.network.chainId),Sh=T(Oe,e=>e.network.roundDuration),wh=T(Oe,e=>e.network.walletConnectBridgeAddress),Gi=T(Oe,e=>e.network.walletConnectV2RelayAddress),Vi=T(Oe,e=>e.network.walletConnectV2ProjectId),$i=T(Oe,e=>e.network.walletConnectV2Options),Hi=T(Oe,e=>e.network.walletConnectDeepLink),xe=T(Oe,e=>e.network),rr=T(xe,e=>e.apiAddress),ji=T(xe,e=>e.explorerAddress),zi=T(xe,e=>{var o;return(o=e.customWalletAddress)!=null?o:e.walletAddress}),vh=T(xe,e=>e.xAliasAddress),Ah=T(xe,e=>e.egldLabel);t();var $n=e=>e.signedMessageInfo,Lh=T($n,e=>e.isSigning),Eh=T($n,e=>e.errorMessage),Ph=T($n,e=>{let o=Object.keys(e.signedSessions),n=o.length;return e.signedSessions[o[n-1]]}),Ch=T($n,e=>{let o=Object.keys(e.signedSessions),n=o.length;return o.length>0?o[n-1]:""});t();var qi=e=>e.toasts,Dh=T(qi,e=>e.customToasts),Oh=T(qi,e=>e.transactionToasts);t();ce();var Ki={errorMessage:Zt,successMessage:Jt,processingMessage:Qt},Vm=e=>e.transactionsInfo,Uh=T(Vm,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Ki);t();t();var oo=require("@multiversx/sdk-core");N();t();t();t();function $o(e){try{let o=atob(e),n=btoa(o),r=i.Buffer.from(e,"base64").toString(),a=i.Buffer.from(r).toString("base64"),s=e===n||n.startsWith(e),m=e===a||a.startsWith(e);if(s&&m)return!0}catch(o){return!1}return!1}function eo(e){return $o(e)?atob(e):e}t();t();t();t();var Xi=e=>{let o=e!=null?e:"";return $o(o)?i.Buffer.from(i.Buffer.from(o,"base64").toString().trim()):i.Buffer.from(o.trim())};t();N();var Hn=({data:e,onlySetGuardian:o})=>e?o?e.startsWith("SetGuardian"):Object.values(Lt).some(n=>e.startsWith(n)):!1;function ar(e){var r,a,s;let o=f({},e);Hn({data:o.data,onlySetGuardian:!0})&&(delete o.senderUsername,delete o.receiverUsername);let n=new oo.Transaction(f(f(w(f(w(f(f({value:BigInt(o.value.valueOf()),data:Xi(o.data),nonce:BigInt(o.nonce.valueOf()),receiver:new oo.Address(o.receiver)},o.receiverUsername?{receiverUsername:o.receiverUsername}:{}),o.relayer?{relayer:new oo.Address(o.relayer)}:{}),{sender:new oo.Address(o.sender)}),o.senderUsername?{senderUsername:o.senderUsername}:{}),{gasLimit:BigInt((r=o.gasLimit.valueOf())!=null?r:Nt),gasPrice:BigInt((a=o.gasPrice.valueOf())!=null?a:Rt),chainID:o.chainID.valueOf(),version:(s=o.version)!=null?s:Ea}),o.options?{options:o.options}:{}),o.guardian?{guardian:new oo.Address(o.guardian)}:{}));return o.relayerSignature&&(n.relayerSignature=i.Buffer.from(o.relayerSignature,"hex")),o.guardianSignature&&(n.guardianSignature=i.Buffer.from(o.guardianSignature,"hex")),o.signature&&(n.signature=i.Buffer.from(o.signature,"hex")),n}t();t();N();t();t();var pr=require("@reduxjs/toolkit"),gc=require("react-redux/lib/utils/Subscription");ce();t();var Yi=S(require("lodash.throttle"));N();ce();Wt();xo();Je();var jm=[un],jn=!1,zm=(0,Yi.default)(()=>{Wo(Uo())},5e3),Zi=e=>o=>n=>{if(jm.includes(n.type))return o(n);let r=e.getState(),a=ie.local.getItem(ee.loginExpiresAt);if(!Boolean(r==null?void 0:r.account.address))return o(n);if(a==null)return Wo(Uo());let m=Date.now();return a-m<0&&!jn?setTimeout(()=>{jn=!0,e.dispatch(Xa())},1e3):(jn&&(jn=!1),zm()),o(n)};t();t();function zn(){return typeof sessionStorage!="undefined"}var _c=zn()?(tc(),Ke(nc)).default:(ac(),Ke(rc)).default,lc=zn()?(cc(),Ke(ic)).default:[],uc=zn()?(dc(),Ke(mc)).default:e=>e;qn();var A=(0,pr.configureStore)({reducer:_c,middleware:e=>e({serializableCheck:{ignoredActions:[...lc,Ut.type,Re.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Zi)}),fc=(0,gc.createSubscription)(A),Kx=uc(A),Xx=(0,pr.configureStore)({reducer:no});t();t();var fd=require("@multiversx/sdk-core");t();N();t();var sr=require("@multiversx/sdk-core");q();t();var hc=require("@multiversx/sdk-core");function hd(e){try{return hc.Address.isValid(e)}catch(o){return!1}}function So(e){return/^\w+$/.test(e)&&hd(e)}var xd=["ESDTNFTTransfer","ESDTNFTBurn","ESDTNFTAddQuantity","ESDTNFTCreate","MultiESDTNFTTransfer","ESDTTransfer","ESDTBurn","ESDTLocalMint","ESDTLocalBurn","ESDTWipe","ESDTFreeze"];function mr(e,o,n=""){if(!So(e))return!1;if(new sr.Address(e).isContractAddress())return!0;let s=xc({receiver:e,data:n});return s?new sr.Address(s).isSmartContract()||bd(e,o,n):!1}var yd=e=>e.toLowerCase().match(/[0-9a-f]/g),Td=e=>e.length%2===0;function bd(e,o,n){let r=n==null?void 0:n.split("@");if(r==null)return!1;let[a,...s]=r,m=o!=null&&e!=null&&e===o,d=xd.includes(a),l=s.every(u=>yd(u)&&Td(u));return m&&d&&l}function xc({receiver:e,data:o}){try{if(!o)return e;let n=$o(o)?i.Buffer.from(o,"base64").toString():o,r=Sd(n),a=n.split("@");return r>-1?a[r]:e}catch(n){console.log(n);return}}function Sd(e){return e.includes("MultiESDTNFTTransfer")?1:e.includes("ESDTNFTTransfer")?4:-1}t();t();t();var wd=require("@multiversx/sdk-core"),vd=S(require("bignumber.js"));re();t();t();t();var wo="accounts";var dr="blocks";var _r="network/config";t();t();t();t();Q();t();t();Q();t();t();t();var Pd=S(require("bignumber.js"));N();t();var Ed=require("@multiversx/sdk-core"),lr=S(require("bignumber.js"));N();t();var Id=S(require("bignumber.js"));t();lr.default.config({ROUNDING_MODE:lr.default.ROUND_FLOOR});t();t();t();N();t();t();t();t();N();t();N();t();var Cd=require("@multiversx/sdk-core"),Nd=S(require("bignumber.js"));N();t();var gr=require("@multiversx/sdk-core"),Dd=S(require("bignumber.js"));N();t();t();var Rd=S(require("bignumber.js"));N();t();N();t();t();t();t();t();t();N();t();N();t();N();t();Q();var Md=["reDelegateRewards","claimRewards","unBond"],Fd=["wrapEgld","unwrapEgld"],Bd=["unStake"],Ud=["unDelegate"];t();t();t();N();t();var Vd=S(require("bignumber.js"));t();t();Q();t();var Hd=S(require("bignumber.js"));t();t();t();t();var zd=S(require("bignumber.js"));q();t();t();var qd=require("@multiversx/sdk-core");t();t();re();t();var Yd=require("@multiversx/sdk-web-wallet-provider");N();t();var Xd=S(require("qs"));t();To();var hr=e=>{let o=new URLSearchParams(e).toString(),{pathname:n,hash:r}=H(),a=o?`?${o}`:"",s=n?`${n}${a}${r}`:"./";setTimeout(()=>{window==null||window.history.replaceState({},document==null?void 0:document.title,s)})};Go();t();Go();var Kd={search:Ve()?window.location.search:"",removeParams:[]},Yn=(e,o=Kd)=>{var m;let n={},r=Ve()?window.location.search:"",a=(m=o.search)!=null?m:r;if(a){let d=a?new URLSearchParams(a):[];n=Object.fromEntries(d)}let s={};return e.forEach(d=>{s[d]=n[d],delete n[d]}),o.removeParams!=null&&Object.keys(n).forEach(d=>{var u,y;let[l]=d.split("[");(((u=o.removeParams)==null?void 0:u.includes(d))||((y=o.removeParams)==null?void 0:y.includes(l)))&&delete n[d]}),{remainingParams:s,params:n,clearNavigationHistory:()=>hr(n)}};t();t();t();Ae();t();Q();t();t();Ae();t();var Zd=S(require("linkifyjs"));t();N();t();var Jd=S(require("bignumber.js"));t();q();t();t();re();t();re();t();t();t();Q();t();Q();t();var Qd=S(require("bignumber.js"));N();Q();t();Q();t();var vc=require("react");q();t();t();Q();t();t();var e_=require("@multiversx/sdk-core"),o_=S(require("bignumber.js"));Q();t();q();t();t();q();var Jw=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var Ac=require("react");q();Ae();t();var t_=require("react");Q();var iv=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];or();t();var Zn=e=>e.transactions,qo=T(Zn,e=>e.signedTransactions),Yv=T(Zn,e=>e.signTransactionsError),Zv=T(Zn,e=>e.signTransactionsCancelMessage),Jn=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),Jv=T(qo,Jn(Si)),Qv=T(qo,Jn(Nn)),eA=T(qo,Jn(Rn)),oA=T(qo,Jn(vi)),nA=T(Zn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:w(f({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>ar(n)))||[]})}),tA=T(qo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();var a_=S(require("axios"));go();t();N();t();var i_=S(require("axios"));t();var p_=S(require("axios"));t();t();var s_=S(require("axios"));t();var m_=S(require("axios"));t();t();t();t();t();var Ic=require("@lifeomic/axios-fetch"),xr=S(require("axios")),yr=(0,Ic.buildAxiosFetch)(xr.default),Tr=(e,o)=>h(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let r=e.clone().json(),[a]=yield Promise.all([r]);return{data:a,status:e.status,statusText:e.statusText,headers:e.headers,config:o}});function d_(e,o,n){return h(this,null,function*(){try{let r=yield yr(e,f({method:"POST",body:o?JSON.stringify(o):void 0,headers:f({"Content-Type":"application/json"},(n==null?void 0:n.headers)||{})},n));return Tr(r,n)}catch(r){throw console.error("Fetch Error:",r),r}})}function __(e,o){return h(this,null,function*(){try{let n=yield yr(e,o);if(!n.ok)throw new Error(`HTTP error! Status: ${n.status}`);return Tr(n,o)}catch(n){throw console.error("Fetch Error:",n),n}})}function l_(e,o,n){return h(this,null,function*(){try{let r=yield yr(e,f({method:"PATCH",body:o?JSON.stringify(o):void 0,headers:(n==null?void 0:n.headers)||{}},n));if(!r.ok)throw new Error(`HTTP error! Status: ${r.status}`);return Tr(r,n)}catch(r){throw console.error("Fetch Error:",r),r}})}var to=xr.default.create();to.get=__;to.post=d_;to.patch=l_;var Lc=e=>to.get(e).then(o=>o.data);t();var Be=e=>{let o=e!=null?e:rr(A.getState());return o.endsWith("/")?o.slice(0,-1):o};t();var u_=S(require("axios"));t();var Ec=S(require("swr"));var vo=({apiEndpoint:e})=>{let o=Be(),n=e?`${o}/${e}`:null;return(0,Ec.default)(n,Lc)};t();t();var g_=(e,o)=>{var a;let r=`${(a=o==null?void 0:o.apiAddress)!=null?a:Be()}/${wo}/${e}?withGuardianInfo=true`;return to.get(r)},Pc=(e,o)=>h(void 0,null,function*(){if(!e)return null;try{let{data:n}=yield g_(e,o);return n}catch(n){console.error("error fetching configuration for ",e,n)}return null});t();var Cc=e=>vo({apiEndpoint:e?`${wo}/${e}?withGuardianInfo=true`:null});t();t();go();t();var x_=S(require("axios"));t();var Nc=S(require("axios"));var y_=e=>{try{return Boolean(new URL(e))}catch(o){return!1}};function Rc(){return h(this,null,function*(){var n;let e=Be();if(!y_(e))return null;let o=`${e}/${_r}`;try{let{data:r}=yield Nc.default.get(o);if(r!=null)return(n=r==null?void 0:r.data)==null?void 0:n.config}catch(r){console.error("error fetching configuration for ",o)}return null})}t();var Dc=S(require("axios"));var T_="transactions/ppu";function Oc(e){return h(this,null,function*(){let n=`${Be()}/${T_}/${e}`;try{let{data:r}=yield Dc.default.get(n);return r?{[e]:{lastBlock:r.lastBlock,fast:r.fast,faster:r.faster}}:null}catch(r){return console.error("Error fetching gas station metadata from:",n,r),null}})}t();go();t();go();t();t();t();t();var S_=S(require("axios"));t();N();t();var vp=require("react"),nt=require("@multiversx/sdk-core");t();t();t();var Mc=require("react"),Ao=require("react-redux");var w_={store:A,subscription:fc},br=(0,Mc.createContext)(w_),PI=(0,Ao.createStoreHook)(br),je=(0,Ao.createDispatchHook)(br),v=(0,Ao.createSelectorHook)(br);var Sr=()=>v(He);t();t();t();t();t();var Fc=require("@multiversx/sdk-extension-provider"),Bc=require("@multiversx/sdk-hw-provider"),Uc=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Wc=require("@multiversx/sdk-opera-provider"),Gc=require("@multiversx/sdk-passkey-provider/out"),wr=require("@multiversx/sdk-web-wallet-provider");N();Oo();t();var ro=require("@multiversx/sdk-web-wallet-iframe-provider/out"),ko=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");re();t();var Qn={},et=Io;var v_="/dapp/init",le=e=>{switch(e==null?void 0:e.constructor){case wr.WalletProvider:return"wallet";case et:return"walletconnectv2";case Bc.HWProvider:return"ledger";case Fc.ExtensionProvider:return"extension";case Gc.PasskeyProvider:return"passkey";case Uc.MetamaskProvider:return"metamask";case Wc.OperaProvider:return"opera";case We.CrossWindowProvider:return"crossWindow";case ro.IframeProvider:return"iframe";case Io:return"";default:return"extra"}},Vc=e=>new wr.WalletProvider(`${e}${v_}`),$c=e=>h(void 0,null,function*(){if(!e.isInitialized())throw new Error("Unable to get version. Provider not initialized");let o=e.hwApp,{contractData:n,version:r}=yield o.getAppConfiguration();return{version:r,dataEnabled:n===Pa}}),ye=e=>`Unable to perform ${e}, Provider not initialized`,Io=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(ye("getAccount"))}setAccount(o){throw new Error(ye(`setAccount: ${o}`))}login(o){throw new Error(ye(`login with options: ${o}`))}logout(o){throw new Error(ye(`logout with options: ${o}`))}getAddress(){throw new Error(ye("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(o,n){throw new Error(ye(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(ye(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(ye(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(ye(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(ye(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(ye(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},ao=new Io;var vr=ao,Hc=null;function pe(e){vr=e}function jc(){Hc!=null&&(vr=Hc)}function ne(){return vr||ao}var zc=()=>{let e=ne(),o=le(e);return{provider:e,providerType:o}};t();t();t();t();t();t();t();var Ko=S(require("react"));var d1=(0,Ko.createContext)({}),_1=A.getState();t();var Kc=S(require("react"));t();var Ar=S(require("react")),A_=S(require("axios"));t();t();ce();t();t();var wp=require("@multiversx/sdk-native-auth-client");t();var Jc=S(require("axios"));t();t();t();function Xc(e){return h(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}t();var Yc=(e,o,n,r=0)=>h(void 0,null,function*(){try{return yield e(...n)}catch(a){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield Xc(o.delay)),yield Yc(e,o,n,r+1)):null}}),Zc=(e,o={retries:5,delay:500})=>(...n)=>h(void 0,null,function*(){return yield Yc(e,o,n)});var k_=4,I_=3e4,ot={current:null},io={current:null},L_=Zc((e,o,n)=>h(void 0,null,function*(){if(n){let s=Math.floor(Date.now()/1e3);return{hash:yield n(),timestamp:s}}let{data:r}=yield Jc.default.get(`${e}/${dr}?from=${k_}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[a]=r;return a}));function Qc(e,o,n,r){return h(this,null,function*(){if(e==null)throw new Error("missing api url");let a=Date.now();if(ot.current!=null&&a<ot.current.timestamp*1e3+I_&&!r)return ot.current;if(io.current!=null)return yield io.current;io.current=L_(e,o,n);try{let s=yield io.current;if(s==null)throw io.current=null,new Error("could not get block hash");return ot.current={hash:s.hash,timestamp:s.timestamp},io.current=null,s}catch(s){return io.current=null,null}})}t();t();gn();t();t();var bp=S(Tp());var Lo=e=>{if(!e||!(0,bp.default)(e))return null;let o=e.split(".");if(o.length!==4)return null;try{let[n,r,a,s]=o,m=JSON.parse(eo(s)),d=eo(n);return{ttl:Number(a),extraInfo:m,origin:d,blockHash:r}}catch(n){return console.error(`Error trying to decode ${e}:`,n),null}};function Q_(e){return Object.prototype.toString.call(e)==="[object String]"}var Eo=e=>{var n;if(!e||!Q_(e))return null;let o=e.split(".");if(o.length!==3)return console.error("Invalid nativeAuthToken. You may be trying to decode a loginToken. Try using decodeLoginToken method instead"),null;try{let[r,a,s]=o,m=eo(r),d=eo(a),l=Lo(d);if(!l)return{address:m,body:d,signature:s,blockHash:"",origin:"",ttl:0};let u=w(f({},l),{address:m,body:d,signature:s});return(n=l.extraInfo)!=null&&n.timestamp||delete u.extraInfo,u}catch(r){return null}};var Ir={isExpired:!1},Sp=e=>{if(!e)return Ir;let o=Eo(e);if(!o)return Ir;let n=ho(),{ttl:r,extraInfo:a}=o,s=a==null?void 0:a.timestamp;if(!s)return Ir;let m=s+r,d=n>m,l=m-n;return{isExpired:d,expiresAt:m,secondsUntilExpires:l}};t();Ae();var Yo={origin:H().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60},Zo=e=>{var n,r,a,s,m,d;return e===!0?Yo:{origin:(n=e==null?void 0:e.origin)!=null?n:Yo.origin,blockHashShard:e==null?void 0:e.blockHashShard,expirySeconds:(r=e==null?void 0:e.expirySeconds)!=null?r:Yo.expirySeconds,apiAddress:(a=e==null?void 0:e.apiAddress)!=null?a:Yo.apiAddress,tokenExpirationToastWarningSeconds:(s=e==null?void 0:e.tokenExpirationToastWarningSeconds)!=null?s:Yo.tokenExpirationToastWarningSeconds,extraInfo:(m=e==null?void 0:e.extraInfo)!=null?m:{},gatewayUrl:e==null?void 0:e.gatewayUrl,extraRequestHeaders:(d=e==null?void 0:e.extraRequestHeaders)!=null?d:{}}};var Jo=e=>{let{origin:o,apiAddress:n,expirySeconds:r,blockHashShard:a,extraInfo:s,gatewayUrl:m,extraRequestHeaders:d}=Zo(e),l=new wp.NativeAuthClient({origin:o,apiUrl:n,expirySeconds:r,blockHashShard:a,gatewayUrl:m,extraRequestHeaders:d});return{getNativeAuthConfig:Zo,initialize:g=>h(void 0,null,function*(){var L,E;if(!n||!o)return"";let b=()=>l.getCurrentBlockHash();try{let R=(L=g==null?void 0:g.latestBlockInfo)!=null?L:yield Qc(n,a,b,g==null?void 0:g.noCache);if(!R)return"";let{hash:z,timestamp:V}=R,j=l.encodeValue(JSON.stringify(f(f({},(E=g==null?void 0:g.extraInfo)!=null?E:s),V?{timestamp:V}:{})));return`${l.encodeValue(o)}.${z}.${r}.${j}`}catch(R){return console.error("Error getting native auth token: ",R.toString()),""}}),getToken:({address:g,token:b,signature:L})=>l.getToken(g,b,L),getTokenExpiration:Sp}};var el=(e,o)=>{var n;return o?o===!0?e:(n=o.apiAddress)!=null?n:e:null},tt=e=>{let o=v(xe),n=v(bo),r=(0,vp.useRef)(n==null?void 0:n.loginToken),a=el(o.apiAddress,e),s=Zo(f(f({},e===!0?{}:e),a?{apiAddress:a}:{})),m=Boolean(e),d=Jo(s),{address:l}=Sr(),u=je(),y=E=>{r.current=E,u(Ge(f(w(f({},n),{loginToken:E}),a?{nativeAuthConfig:s}:{})))},g=()=>{try{return d.initialize()}catch(E){console.error("Unable to get block. Login failed.",E);return}},b=({address:E,signature:R})=>{let z=r.current;if(!z)throw"Token not found. Call `setLoginToken` first.";if(!m){u(Ge({loginToken:z,signature:R}));return}let V=d.getToken({address:E,token:z,signature:R});return u(Ge(f({loginToken:z,signature:R,nativeAuthToken:V},a?{nativeAuthConfig:s}:{}))),V};return{configuration:s,setLoginToken:y,getNativeAuthLoginToken:g,setTokenLoginInfo:b,refreshNativeAuthTokenLogin:z=>h(void 0,[z],function*({signMessageCallback:E,nativeAuthClientConfig:R}){let j=yield Jo(R||s).initialize({noCache:Boolean(R)});if(r.current=j,!j)return;let ge=new nt.Message({address:new nt.Address(l),data:i.Buffer.from(`${l}${j}`)}),F=yield E(ge,{});if(!(F!=null&&F.signature))throw"Message not signed";return b({address:l,signature:i.Buffer.from(F.signature).toString("hex")})})}};t();var te=require("react");t();var rt=require("react");function Ap(e,o=[]){let n=(0,rt.useRef)(!0);(0,rt.useEffect)(()=>{n.current?n.current=!1:e()},o)}K();ce();re();t();t();var Cp=S(Ip());Fo();var Lp,Ep,Pp,nl=(Pp=(Ep=(Lp=ae.safeWindow)==null?void 0:Lp.location)==null?void 0:Ep.origin)==null?void 0:Pp.includes("localhost"),Np=e=>{if(!(0,Cp.default)(e))return Boolean(e);let n=f({},e);return Boolean(nl)&&delete n.origin,Object.keys(n).length>0};t();function Lr(){return De(A.getState())}jt();t();Fo();K();q();t();function Rp(e,o){if(!e){typeof console!="undefined"&&console.warn(o);try{throw new Error(o)}catch(n){}}}function Dp(e,o){typeof e=="string"&&(e={path:e,caseSensitive:!1,end:!0});let[n,r]=tl(e.path,e.caseSensitive,e.end),a=o.match(n);if(!a)return null;let s=a[0],m=s.replace(/(.)\/+$/,"$1"),d=a.slice(1);return{params:r.reduce((u,y,g)=>{if(y==="*"){let b=d[g]||"";m=s.slice(0,s.length-b.length).replace(/(.)\/+$/,"$1")}return u[y]=rl(d[g]||"",y),u},{}),pathname:s,pathnameBase:m,pattern:e}}function tl(e,o=!1,n=!0){Rp(e==="*"||!e.endsWith("*")||e.endsWith("/*"),`Route path "${e}" will be treated as if it were "${e.replace(/\*$/,"/*")}" because the \`*\` character must always follow a \`/\` in the pattern. To get rid of this warning, please change the route path to "${e.replace(/\*$/,"/*")}".`);let r=[],a="^"+e.replace(/\/*\*?$/,"").replace(/^\/*/,"/").replace(/[\\.*+^$?{}|()[\]]/g,"\\$&").replace(/:(\w+)/g,(m,d)=>(r.push(d),"([^\\/]+)"));return e.endsWith("*")?(r.push("*"),a+=e==="*"||e==="/*"?"(.*)$":"(?:\\/(.+)|\\/*)$"):a+=n?"\\/*$":"(?:(?=[.~-]|%[0-9A-F]{2})|\\b|\\/|$)",[new RegExp(a,o?void 0:"i"),r]}function rl(e,o){try{return decodeURIComponent(e)}catch(n){return Rp(!1,`The value for the URL param "${o}" will not be decoded because the string "${e}" is a malformed URL segment. This is probably due to a bad percent encoding (${n}).`),e}}t();t();Ae();var Op=()=>{let{search:e}=H(),o=new URLSearchParams(e),n=Object.fromEntries(o),r=n==null?void 0:n.address;return So(r)?r:null};t();var Qo=(e,o)=>Pc(e,o);t();t();function Mp(){let e=ne();return le(e)}t();re();t();function en(e){return Mp()===e}function se(){let e=A.getState(),o=ne(),n=$e(e),r=De(e),a=Wn(e);if(!o)throw"provider not initialized";return en("ledger")&&r?new Promise(s=>{s(n)}):!en("")&&!en("wallet")&&!en("extra")?o.getAddress():new Promise(s=>{let m=Op();a!=null&&m&&s(m),r&&s(n),s("")})}t();function at(e){let o=A.getState(),n=Oi(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();Ae();var Fp=()=>{let{search:e}=H(),o=new URLSearchParams(e),n=Object.fromEntries(o);return n==null?void 0:n.accessToken};var it=()=>{var n;let e=Fp(),o=X(A.getState());return e!=null?e:(n=o.webviewLogin)==null?void 0:n.data};t();K();q();t();ce();var Bp=()=>h(void 0,null,function*(){try{let e=yield se();try{let o=yield Qo(e);if(o!=null){let n=w(f({},o),{nonce:at(o)});return A.dispatch(Re(n)),n}}catch(o){console.error("Failed getting account ",o)}}catch(e){console.error("Failed getting address ",e)}return null});function Up(){return h(this,null,function*(){let e=ne();if(e==null)throw"Provider not initialized";if(!e.isInitialized||e.isInitialized())return Bp();try{if(!e.init)throw"Current provider does not have init() function";return(yield e.init())?Bp():void 0}catch(o){console.error("Failed initializing provider ",o)}})}t();ce();t();var Wp=require("@multiversx/sdk-core");q();To();Ht();xo();Je();To();var Gp=({callbackUrl:e,onRedirect:o})=>{typeof o=="function"?o(e):e&&wn({url:e})},al=e=>{let o=ie.local.getItem(ee.logoutEvent),{data:n}=o?JSON.parse(o):{data:e};e===n&&(ie.local.setItem({key:ee.logoutEvent,data:e,expires:0}),ie.local.removeItem(ee.logoutEvent))},il=500;function on(a,s){return h(this,arguments,function*(e,o,n=Boolean(it()),r={shouldBroadcastLogoutAcrossTabs:!0,hasConsentPopup:!1}){var R,z,V;let m=yield se(),d=ne(),l=le(d),u=l==="wallet",y=((R=d==null?void 0:d.isInitialized)==null?void 0:R.call(d))===!0;if(n&&(d==null?void 0:d.relogin)!=null)return d.relogin();m&&r.shouldBroadcastLogoutAcrossTabs&&al(m);let g=$t(e),b=H(),{pathname:L,origin:E}=new URL(decodeURIComponent(g));if((Dp(b.pathname,L)||u&&y||((z=ae.safeWindow)==null?void 0:z.opener)&&E===((V=ae.safeWindow)==null?void 0:V.origin))&&ri(),!m)return Gp({callbackUrl:g,onRedirect:o});try{if(A.dispatch(B()),u&&y)return setTimeout(()=>{d.logout({callbackUrl:g})},il);r.hasConsentPopup&&l==="crossWindow"&&d.setShouldShowConsentPopup(!0),y&&(yield d.logout({callbackUrl:g}))}catch(j){console.error("Logging out error:",j)}finally{u||Gp({callbackUrl:g,onRedirect:o})}})}t();var Vp=({maxRetries:e})=>new Promise((o,n)=>{let r=0,a=()=>{let s=Vn(A.getState());if(s){o(s);return}if(r<e){r++,setTimeout(a,1e3);return}n(null)};a()});Ae();t();t();Oo();q();var $p=e=>(Object.values(de).forEach(o=>{if(o!==(e==null?void 0:e.skipLoginMethod))switch(o){case"crossWindow":{let n=We.CrossWindowProvider.getInstance();n.isInitialized()&&n.dispose();break}case"iframe":{let n=ro.IframeProvider.getInstance();n.isInitialized()&&n.dispose();break}default:break}}),null);var Hp=({callbackRoute:e,token:o,nativeAuth:n,onLoginRedirect:r,logoutRoute:a,customRequestMethods:s=[]})=>{var U;let m=je(),d=Np(n),l=tt(d?n:!1),u=o,[y,g]=(0,te.useState)(""),[b,L]=(0,te.useState)(""),[E,R]=(0,te.useState)(!0),[z,V]=(0,te.useState)([]),[j,ge]=(0,te.useState)(null),{provider:F}=zc(),D=v(Gi),co=v(Vi),Ie=v($i),ct=v(Hi),pt=v(Gn),st=v(Bi),P=(0,te.useRef)(F),ze=(0,te.useRef)(!1),qe=(0,te.useRef)(!1),tn=(U=a!=null?a:pt)!=null?U:"/",po=[Qn.CANCEL_ACTION,...s];o&&po.push(Qn.SIGN_LOGIN_TOKEN),d&&po.push(Qn.SIGN_NATIVE_AUTH_TOKEN);let mt=E?"":`${ct}?wallet-connect=${encodeURIComponent(b)}`,Co=Boolean(y),No=Lr(),dt=()=>{on(tn)},_t=k=>{console.log("WalletConnect Session Event: ",k)},lt=()=>h(void 0,null,function*(){var k,C,M,W;try{if(ze.current||P.current==null)return;if(!qe.current){try{yield(k=P.current)==null?void 0:k.logout()}catch(Ro){console.warn("Unable to logout")}return}let Y=yield(M=(C=P.current)==null?void 0:C.getAddress)==null?void 0:M.call(C);if(!Y){console.warn("Login cancelled.");return}let fe=yield(W=P.current)==null?void 0:W.getSignature(),Le={address:Y,loginMethod:"walletconnectv2"},so={logoutRoute:tn,loginType:"walletconnectv2",callbackRoute:e!=null?e:H().href};m(Ka(so)),fe&&l.setTokenLoginInfo({signature:fe,address:Y}),m(he(Le)),ai({callbackRoute:e,onLoginRedirect:r,options:{address:Y,signature:fe}})}catch(Y){g("User rejected connection proposal"),console.error(Y)}}),rn=()=>h(void 0,null,function*(){var C,M,W,Y,fe;if(le(P.current)==="walletconnectv2")try{((Y=(W=(M=(C=P.current)==null?void 0:C.walletConnector)==null?void 0:M.session)==null?void 0:W.getAll())!=null?Y:[]).length>0&&(yield(fe=P.current)==null?void 0:fe.logout()),P.current=ao,ge(null)}catch(Le){console.warn("Unable to logout")}}),ut=k=>h(void 0,null,function*(){var C,M,W;if(!D||!co){g("Invalid WalletConnect setup");return}if(!k||!k.topic){g("Expired connection");return}try{if((P.current?le(P.current):!1)!=="walletconnectv2"){yield O();return}if(R(!0),yield rn(),d&&!u&&(u=yield l.getNativeAuthLoginToken(),!u)){console.warn("Fetching block failed. Login cancelled.");return}u&&l.setLoginToken(u),yield O(!1);let{approval:fe}=yield(C=P.current)==null?void 0:C.connect({topic:k.topic,methods:po});try{yield(M=P.current)==null?void 0:M.login({approval:fe,token:u})}catch(Le){console.error("User rejected existing connection proposal",Le),g("User rejected existing connection proposal"),R(!0),yield O()}}catch(Y){console.error("Unable to connect to existing session",Y),g("Unable to connect to existing session")}finally{V((W=P.current)==null?void 0:W.pairings)}}),gt=k=>h(void 0,null,function*(){var C,M;try{k&&(yield(C=P.current)==null?void 0:C.logout({topic:k}))}catch(W){console.error("Unable to remove existing pairing",W),g("Unable to remove existing pairing")}finally{let W=yield(M=P.current)==null?void 0:M.getPairings();V(W)}});function O(k=!0){return h(this,null,function*(){var Ro,Mr,Fr;$p(),k&&(m(Bo(Qe.address)),m(Re(Qe)));let C=yield Vp({maxRetries:15});if(!C){console.error("Invalid chainID"),g("Invalid chainID");return}if(!co||!D){console.error("Invalid WalletConnect setup"),g("Invalid WalletConnect setup");return}let M=Lr(),W=qe.current===!1&&!M,Y=le(P.current)==="walletconnectv2",fe=((Mr=(Ro=P.current)==null?void 0:Ro.isInitialized)==null?void 0:Mr.call(Ro))&&Y;if(ze.current||W||fe)return;if(ze.current=!0,(Fr=P.current)!=null&&Fr.walletConnector){yield P.current.init(),ge(P.current),ze.current=!1,k&&(yield I());return}let Le={onClientLogin:lt,onClientLogout:dt,onClientEvent:_t},so=new et(Le,C,D,co,Ie);yield so.init(),ge(so),P.current=so,ze.current=!1,k&&(V(so.pairings),yield I())})}function I(){return h(this,null,function*(){var k,C;if(!!P.current){if(!D||!co){g("Invalid WalletConnect setup");return}try{if(!qe.current)return;let{uri:M,approval:W}=yield(k=P.current)==null?void 0:k.connect({methods:po});if(!Boolean(M))return;if(L(M),(Ie==null?void 0:Ie.logger)==="debug"&&console.log("WalletConnect uri: ",M),d&&!u&&(u=yield l.getNativeAuthLoginToken(),!u)){console.warn("Fetching block failed. Login cancelled.");return}if(u&&l.setLoginToken(u),(P.current?le(P.current):!1)!=="walletconnectv2"){R(!0),yield O();return}try{yield(C=P.current)==null?void 0:C.login({approval:W,token:u})}catch(Le){console.error("User rejected connection proposal",Le),g("User rejected connection proposal"),R(!0)}}catch(M){console.error("Unable to connect",M)}}})}return Ap(()=>{P.current=F},[F]),(0,te.useEffect)(()=>(qe.current=!0,()=>{qe.current=!1}),[]),(0,te.useEffect)(()=>{R(!b)},[b]),(0,te.useEffect)(()=>{if(!j)return;(Boolean(j.session)||st==="walletconnectv2")&&No&&pe(j)},[j,No]),[O,{error:y,loginFailed:Co,isLoading:E&&!Co,isLoggedIn:No&&!Co},{uriDeepLink:mt,walletConnectUri:b,cancelLogin:rn,connectExisting:ut,removeExistingPairing:gt,wcPairings:z}]};t();var zp=require("react");t();K();ce();q();function jp(e,o=A.dispatch){let n=Eo(e);if(n==null)return;let{signature:r,address:a}=n;r&&e&&a&&(o(Ge({loginToken:e,signature:r,nativeAuthToken:e})),o(he({address:a,loginMethod:"extra"})))}function qp(){let e=je(),o=v(Fi),n=it();(0,zp.useEffect)(()=>{!o||!n||jp(n,e)},[n,o])}K();ce();t();t();N();t();t();t();var cl=S(require("bignumber.js"));re();t();t();var Kp=require("@multiversx/sdk-opera-provider");function Xp(e){return h(this,null,function*(){try{let o=Kp.OperaProvider.getInstance().setAddress(e);if(yield o.init())return o;console.error("Could not initialise opera crypto wallet, make sure that opera crypto wallet is installed.")}catch(o){console.error("Unable to login to OperaProvider",o)}return null})}t();var Yp=require("@multiversx/sdk-extension-provider");function Zp(e){return h(this,null,function*(){try{let o=Yp.ExtensionProvider.getInstance().setAddress(e);if(yield o.init())return o;console.error("Could not initialise extension, make sure MultiversX wallet extension is installed.")}catch(o){console.error("Unable to login to ExtensionProvider",o)}return null})}t();Oo();function Jp(n){return h(this,arguments,function*({address:e,walletUrl:o}){try{let r=yield We.CrossWindowProvider.getInstance().init(),a=We.CrossWindowProvider.getInstance().setAddress(e).setWalletUrl(o);if(r)return a;console.error("Could not initialise Cross Window Wallet provider")}catch(r){console.error("Unable to login to CrossWindowWalletProvider",r)}return null})}t();function Qp(r){return h(this,arguments,function*({address:e,walletUrl:o,loginType:n}){try{let a=ro.IframeProvider.getInstance();if(a.isInitialized())return a;switch(n){case ko.IframeLoginTypes.passkey:a.setLoginType(ko.IframeLoginTypes.passkey);break;case ko.IframeLoginTypes.metamask:a.setLoginType(ko.IframeLoginTypes.metamask);break;default:break}if(a.setAddress(e).setWalletUrl(o),yield a.init())return a;console.error("Could not initialise IframeProvider")}catch(a){console.error("Unable to login to IframeProvider",a)}return null})}t();t();var es=n=>h(void 0,[n],function*({loginToken:e,extraInfoData:o}){if(e==null||Object.keys(o).length===0)return null;let r=Lo(String(e)),u=(r==null?void 0:r.extraInfo)||{},{timestamp:a}=u,s=Ue(u,["timestamp"]);if(!(r&&a!=null))return null;let d={hash:String(r==null?void 0:r.blockHash),timestamp:Number(a)};return yield Jo({extraInfo:f(f({},s),o),expirySeconds:r==null?void 0:r.ttl,origin:r==null?void 0:r.origin}).initialize({latestBlockInfo:d})});var os=s=>h(void 0,[s],function*({loginToken:e,extraInfoData:o,address:n,signature:r,loginService:a}){let m=yield es({loginToken:e,extraInfoData:o}),d=o.multisig||o.impersonate||n,l=m!=null?d:n;return m!=null&&a.setLoginToken(m),r&&a.setTokenLoginInfo({signature:r,address:n}),yield Qo(l)});t();var ns=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");function ts(e){return h(this,null,function*(){try{let o=ns.MetamaskProvider.getInstance().setAddress(e);if(yield o.init())return o;console.error("Could not initialise metamask, make suremetamask flask is installed.")}catch(o){console.error("Unable to login to MetamaskProvider",o)}return null})}t();var rs=require("@multiversx/sdk-passkey-provider/out");function as(n){return h(this,arguments,function*({address:e,extrasApiAddress:o=""}){try{let r=rs.PasskeyProvider.getInstance().setPasskeyServiceUrl(o).setAddress(e);if(yield r.init())return r;console.error("Could not initialise passkey provider")}catch(r){console.error("Unable to login to PasskeyProvider",r)}return null})}t();var fs=S(require("react")),hs=require("@fortawesome/free-solid-svg-icons");t();t();var Po=S(require("react")),_s=require("@fortawesome/free-solid-svg-icons"),ls=require("@fortawesome/react-fontawesome");t();var ss=S(require("react"));t();var nn=S(require("react"));t();Go();var is=()=>!Ve();var sl=()=>h(void 0,null,function*(){return yield Promise.resolve().then(()=>(Pr(),Er))}),ml=()=>(Pr(),Ke(Er)).default,Cr=is();function ps({ssrGlobalImportCallback:e,ssrImportCallback:o,clientImportCallback:n}){let[r,a]=nn.default.useState(Cr?void 0:ml()),[s,m]=nn.default.useState(Cr||n==null?void 0:n()),d=()=>h(this,null,function*(){(e?e():sl()).then(l=>a(l.default)),o==null||o().then(l=>m(l.default))});return(0,nn.useEffect)(()=>{Cr&&d()},[]),{globalStyles:r,styles:s}}function ms(e,o){return n=>{let{globalStyles:r,styles:a}=ps({ssrGlobalImportCallback:o.ssrGlobalStyles,ssrImportCallback:o.ssrStyles,clientImportCallback:o.clientStyles});return ss.default.createElement(e,w(f({},n),{globalStyles:r!=null?r:{},styles:a!=null?a:{}}))}}var _l=({styles:e})=>Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},Po.default.createElement(ls.FontAwesomeIcon,{icon:_s.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),us=ms(_l,{ssrStyles:()=>Promise.resolve().then(()=>(Rr(),Nr)),clientStyles:()=>(Rr(),Ke(Nr)).default});xo();Je();t();t();ce();var ll={},ul={};var gs=e=>{if(e.component!=null){let o=e,{component:n,onClose:r}=o,a=Ue(o,["component","onClose"]),s=A.dispatch(Xt(w(f({},a),{component:null,onClose:void 0})));return ll[s==null?void 0:s.payload.toastId]=n,r&&(ul[s==null?void 0:s.payload.toastId]=r),s==null?void 0:s.payload}return A.dispatch(Xt(e)).payload};var gl=3,fl=24*60*60,xs=e=>{var a;let o=()=>{let s=fl*gl,m=Math.floor(Date.now()/1e3);ie.local.setItem({key:ee.guardianBreachToastDismissTimestamp,data:m,expires:m+s})},n=(a=ie)==null?void 0:a.local.getItem(ee.guardianBreachToastDismissTimestamp);e.isGuarded&&e.activeGuardianAddress&&e.pendingGuardianAddress&&!n&&gs({toastId:"guardianUnconfirmedChangeWarning",title:"Account at risk!",component:()=>fs.default.createElement(us,null),icon:hs.faWarning,onClose:o})};t();t();var ys=require("react"),Dr=require("@multiversx/sdk-hw-provider");var Ts=()=>{let e=v(Gn),o=v(De),n=v(Un),r=ne(),[a,s]=(0,ys.useState)(),m=()=>h(void 0,null,function*(){let l=(n==null?void 0:n.index)!=null;try{if(r instanceof Dr.HWProvider&&r.isInitialized())return l&&(yield r.setAddressIndex(n.index)),r;let u=new Dr.HWProvider;return(yield u.init())?(l&&(yield u.setAddressIndex(n.index)),u):null}catch(u){return console.error("Failed to initialise Ledger Provider"),null}});function d(l){return h(this,null,function*(){let u,y=o&&!(l!=null&&l.isRelogin);try{if(u=yield m(),!u){console.warn("Could not initialise ledger app"),y&&on(e);return}let g=yield $c(u);return pe(u),s(g),u}catch(g){console.error("Could not initialise ledger app",g),y&&on(e)}return null})}return{setLedgerProvider:d,ledgerData:a}};var Or=!1;function hl(){let{loginMethod:e,iframeLoginType:o}=v(X),n=v(xe),r=v(zi),a=v(Ui),s=v(Wn),m=v($e),d=v(Mi),l=v(Un),u=v(De),y=v(Vn),g=v(bo),b=v(He),L=g==null?void 0:g.nativeAuthConfig,[E,R]=(0,ue.useState)(y),z=tt(L||!1),{data:V,isLoading:j,error:ge}=Cc(m),F=(0,ue.useRef)(!1),D=je(),{setLedgerProvider:co,ledgerData:Ie}=Ts();qp();let{callbackRoute:ct,logoutRoute:pt}=a||{callbackRoute:"",logoutRoute:""},[st]=Hp({callbackRoute:ct,logoutRoute:pt});(0,ue.useEffect)(()=>{ze()},[n]),(0,ue.useEffect)(()=>{gt()},[m,e,y]),(0,ue.useEffect)(()=>{tn()},[g==null?void 0:g.nativeAuthToken,m]),(0,ue.useEffect)(()=>{po()},[V,j]),(0,ue.useEffect)(()=>{qe()},[d,u,Ie]),(0,ue.useEffect)(()=>{u&&b.address&&xs(b)},[u,b]),(0,ue.useEffect)(()=>{P()},[b.shard,b.address,u,E,y]);function P(){return h(this,null,function*(){var M;if(!(u&&b.shard!=null))return;let I=Number(b.shard),U=yield Oc(I);if(!((M=U==null?void 0:U[I])!=null&&M.lastBlock))return;let k=!n.gasStationMetadata||!n.gasStationMetadata[I]||n.gasStationMetadata[I].lastBlock!==U[I].lastBlock,C=E!==y;C&&R(y),(k||C)&&D(qt({gasStationMetadata:U}))})}function ze(){return h(this,null,function*(){let O=n.chainId&&![dn,_n,ln].includes(n.chainId)&&!n.roundDuration;if(!!(!n.chainId||O))try{let U=yield Rc(),k={};if(U){let C=U;(n.chainId!==C.erd_chain_id||n.roundDuration!==C.erd_round_duration)&&(k.chainId=C.erd_chain_id,k.roundDuration=C.erd_round_duration)}Object.keys(k).length>0&&D(qt(k))}catch(U){console.error("Failed refreshing network config:",U)}})}function qe(){d==null&&l!=null&&Ie&&D($a({index:l.index,address:m,hasContractDataEnabled:Ie.dataEnabled,version:Ie.version}))}function tn(){return h(this,null,function*(){let{remainingParams:{impersonate:O,multisig:I}}=Yn(["impersonate"]),U=mr(m);if(!(g!=null&&g.nativeAuthToken)||O||I||U)return;let k=Eo(g==null?void 0:g.nativeAuthToken);(k==null?void 0:k.address)&&k.address!==m&&(D(Bo(k.address)),yield Up())})}function po(){return h(this,null,function*(){if(j){D(yo(!0));return}if(ge){D(Ha("Failed getting account")),console.error("Failed getting account ",ge);return}if(F.current){F.current=!1,D(yo(!1));return}V?D(Re(w(f({},V),{address:m,nonce:V.nonce.valueOf()}))):u||D(Bo("")),D(yo(!1))})}function mt(){return h(this,null,function*(){let O=Vc(r);if(pe(O),s!=null){try{let I=yield se(),{clearNavigationHistory:U,remainingParams:{signature:k,multisig:C,impersonate:M}}=Yn(["signature","loginToken","address","multisig","impersonate"]);if(!I)return pe(ao),D(Ge(null)),D(Gt(null)),U();let W=yield os({loginToken:g==null?void 0:g.loginToken,extraInfoData:{multisig:C,impersonate:M},address:I,signature:k,loginService:z});W&&(F.current=!0,D(yo(!0)),D(he({address:W.address,loginMethod:"wallet"})),D(Re(w(f({},W),{nonce:at(W)}))),D(yo(!1))),U()}catch(I){console.error("Failed authenticating wallet user ",I)}D(Gt(null))}})}function Co(){return h(this,null,function*(){let O=yield se(),I=yield Zp(O);I&&pe(I)})}function No(){return h(this,null,function*(){let O=yield se(),I=yield as({address:O,extrasApiAddress:n.extrasApiAddress});I&&pe(I)})}function dt(){return h(this,null,function*(){let O=yield se(),I=yield ts(O);I&&pe(I)})}function _t(){return h(this,null,function*(){let O=yield se(),I=yield Xp(O);I&&pe(I)})}function lt(){return h(this,null,function*(){var U;let O=yield se(),I=yield Jp({address:O,walletUrl:(U=n.customWalletAddress)!=null?U:n.walletAddress});I&&pe(I)})}function rn(){return h(this,null,function*(){let O=yield se();if(!n.metamaskSnapWalletAddress)throw new Error("Metamask snap wallet URL is not set.");let I=yield Qp({address:O,walletUrl:n.metamaskSnapWalletAddress,loginType:o});I&&pe(I)})}function ut(){return h(this,null,function*(){try{D(Vt(!0)),yield st(!1)}catch(O){console.error("Could not initialize WalletConnect")}finally{D(Vt(!1))}})}function gt(){return h(this,null,function*(){if(!(e==null||Or))switch(e){case"ledger":{Or=!0,yield co(),Or=!1;break}case"walletconnectv2":{ut();break}case"extension":{Co();break}case"passkey":{No();break}case"metamask":{dt();break}case"opera":{_t();break}case"crossWindow":{lt();break}case"iframe":rn();break;case"extra":{jc();break}case"wallet":case"":{mt();break}default:return}})}return null}0&&(module.exports={ProviderInitializer});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
