// @multiversx/sdk-dapp/__commonjs/hooks/transactions/useSignTransactions.js
"use strict";var fm=Object.create;var Io=Object.defineProperty,hm=Object.defineProperties,xm=Object.getOwnPropertyDescriptor,Tm=Object.getOwnPropertyDescriptors,ym=Object.getOwnPropertyNames,Qo=Object.getOwnPropertySymbols,bm=Object.getPrototypeOf,tt=Object.prototype.hasOwnProperty,Nr=Object.prototype.propertyIsEnumerable;var Lr=(e,o,n)=>o in e?Io(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,g=(e,o)=>{for(var n in o||(o={}))tt.call(o,n)&&Lr(e,n,o[n]);if(Qo)for(var n of Qo(o))Nr.call(o,n)&&Lr(e,n,o[n]);return e},w=(e,o)=>hm(e,Tm(o));var to=(e,o)=>{var n={};for(var r in e)tt.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&Qo)for(var r of Qo(e))o.indexOf(r)<0&&Nr.call(e,r)&&(n[r]=e[r]);return n};var f=(e,o)=>()=>(e&&(o=e(e=0)),o);var y=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),be=(e,o)=>{for(var n in o)Io(e,n,{get:o[n],enumerable:!0})},Cr=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let s of ym(o))!tt.call(e,s)&&s!==n&&Io(e,s,{get:()=>o[s],enumerable:!(r=xm(o,s))||r.enumerable});return e};var S=(e,o,n)=>(n=e!=null?fm(bm(e)):{},Cr(o||!e||!e.__esModule?Io(n,"default",{value:e,enumerable:!0}):n,e)),Oe=e=>Cr(Io({},"__esModule",{value:!0}),e);var x=(e,o,n)=>new Promise((r,s)=>{var p=l=>{try{_(n.next(l))}catch(h){s(h)}},m=l=>{try{_(n.throw(l))}catch(h){s(h)}},_=l=>l.done?r(l.value):Promise.resolve(l.value).then(p,m);_((n=n.apply(e,o)).next())});var Or=y(en=>{"use strict";t();en.byteLength=wm;en.toByteArray=Am;en.fromByteArray=Em;var me=[],ae=[],Sm=typeof Uint8Array!="undefined"?Uint8Array:Array,rt="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Me=0,Rr=rt.length;Me<Rr;++Me)me[Me]=rt[Me],ae[rt.charCodeAt(Me)]=Me;var Me,Rr;ae["-".charCodeAt(0)]=62;ae["_".charCodeAt(0)]=63;function Dr(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function wm(e){var o=Dr(e),n=o[0],r=o[1];return(n+r)*3/4-r}function vm(e,o,n){return(o+n)*3/4-n}function Am(e){var o,n=Dr(e),r=n[0],s=n[1],p=new Sm(vm(e,r,s)),m=0,_=s>0?r-4:r,l;for(l=0;l<_;l+=4)o=ae[e.charCodeAt(l)]<<18|ae[e.charCodeAt(l+1)]<<12|ae[e.charCodeAt(l+2)]<<6|ae[e.charCodeAt(l+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return s===2&&(o=ae[e.charCodeAt(l)]<<2|ae[e.charCodeAt(l+1)]>>4,p[m++]=o&255),s===1&&(o=ae[e.charCodeAt(l)]<<10|ae[e.charCodeAt(l+1)]<<4|ae[e.charCodeAt(l+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function km(e){return me[e>>18&63]+me[e>>12&63]+me[e>>6&63]+me[e&63]}function Im(e,o,n){for(var r,s=[],p=o;p<n;p+=3)r=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),s.push(km(r));return s.join("")}function Em(e){for(var o,n=e.length,r=n%3,s=[],p=16383,m=0,_=n-r;m<_;m+=p)s.push(Im(e,m,m+p>_?_:m+p));return r===1?(o=e[n-1],s.push(me[o>>2]+me[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],s.push(me[o>>10]+me[o>>4&63]+me[o<<2&63]+"=")),s.join("")}});var Mr=y(at=>{t();at.read=function(e,o,n,r,s){var p,m,_=s*8-r-1,l=(1<<_)-1,h=l>>1,u=-7,b=n?s-1:0,L=n?-1:1,k=e[o+b];for(b+=L,p=k&(1<<-u)-1,k>>=-u,u+=_;u>0;p=p*256+e[o+b],b+=L,u-=8);for(m=p&(1<<-u)-1,p>>=-u,u+=r;u>0;m=m*256+e[o+b],b+=L,u-=8);if(p===0)p=1-h;else{if(p===l)return m?NaN:(k?-1:1)*(1/0);m=m+Math.pow(2,r),p=p-h}return(k?-1:1)*m*Math.pow(2,p-r)};at.write=function(e,o,n,r,s,p){var m,_,l,h=p*8-s-1,u=(1<<h)-1,b=u>>1,L=s===23?Math.pow(2,-24)-Math.pow(2,-77):0,k=r?0:p-1,I=r?1:-1,te=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(_=isNaN(o)?1:0,m=u):(m=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+b>=1?o+=L/l:o+=L*Math.pow(2,1-b),o*l>=2&&(m++,l/=2),m+b>=u?(_=0,m=u):m+b>=1?(_=(o*l-1)*Math.pow(2,s),m=m+b):(_=o*Math.pow(2,b-1)*Math.pow(2,s),m=0));s>=8;e[n+k]=_&255,k+=I,_/=256,s-=8);for(m=m<<s|_,h+=s;h>0;e[n+k]=m&255,k+=I,m/=256,h-=8);e[n+k-I]|=te*128}});var Xr=y(ao=>{"use strict";t();var it=Or(),ro=Mr(),Fr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;ao.Buffer=d;ao.SlowBuffer=Dm;ao.INSPECT_MAX_BYTES=50;var on=2147483647;ao.kMaxLength=on;d.TYPED_ARRAY_SUPPORT=Pm();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Pm(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function Se(e){if(e>on)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,d.prototype),o}function d(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return mt(e)}return Wr(e,o,n)}d.poolSize=8192;function Wr(e,o,n){if(typeof e=="string")return Nm(e,o);if(ArrayBuffer.isView(e))return Cm(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(de(e,ArrayBuffer)||e&&de(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(de(e,SharedArrayBuffer)||e&&de(e.buffer,SharedArrayBuffer)))return st(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return d.from(r,o,n);var s=Rm(e);if(s)return s;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,o,n){return Wr(e,o,n)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function Gr(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Lm(e,o,n){return Gr(e),e<=0?Se(e):o!==void 0?typeof n=="string"?Se(e).fill(o,n):Se(e).fill(o):Se(e)}d.alloc=function(e,o,n){return Lm(e,o,n)};function mt(e){return Gr(e),Se(e<0?0:dt(e)|0)}d.allocUnsafe=function(e){return mt(e)};d.allocUnsafeSlow=function(e){return mt(e)};function Nm(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!d.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=$r(e,o)|0,r=Se(n),s=r.write(e,o);return s!==n&&(r=r.slice(0,s)),r}function ct(e){for(var o=e.length<0?0:dt(e.length)|0,n=Se(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Cm(e){if(de(e,Uint8Array)){var o=new Uint8Array(e);return st(o.buffer,o.byteOffset,o.byteLength)}return ct(e)}function st(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,d.prototype),r}function Rm(e){if(d.isBuffer(e)){var o=dt(e.length)|0,n=Se(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||_t(e.length)?Se(0):ct(e);if(e.type==="Buffer"&&Array.isArray(e.data))return ct(e.data)}function dt(e){if(e>=on)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+on.toString(16)+" bytes");return e|0}function Dm(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==d.prototype};d.compare=function(o,n){if(de(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),de(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(o)||!d.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,s=n.length,p=0,m=Math.min(r,s);p<m;++p)if(o[p]!==n[p]){r=o[p],s=n[p];break}return r<s?-1:s<r?1:0};d.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return d.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var s=d.allocUnsafe(n),p=0;for(r=0;r<o.length;++r){var m=o[r];if(de(m,Uint8Array))p+m.length>s.length?d.from(m).copy(s,p):Uint8Array.prototype.set.call(s,m,p);else if(d.isBuffer(m))m.copy(s,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return s};function $r(e,o){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||de(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var s=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return pt(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Kr(e).length;default:if(s)return r?-1:pt(e).length;o=(""+o).toLowerCase(),s=!0}}d.byteLength=$r;function Om(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return qm(this,o,n);case"utf8":case"utf-8":return Hr(this,o,n);case"ascii":return Vm(this,o,n);case"latin1":case"binary":return Hm(this,o,n);case"base64":return Gm(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return jm(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}d.prototype._isBuffer=!0;function Fe(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}d.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)Fe(this,n,n+1);return this};d.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)Fe(this,n,n+3),Fe(this,n+1,n+2);return this};d.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)Fe(this,n,n+7),Fe(this,n+1,n+6),Fe(this,n+2,n+5),Fe(this,n+3,n+4);return this};d.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?Hr(this,0,o):Om.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(o){if(!d.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:d.compare(this,o)===0};d.prototype.inspect=function(){var o="",n=ao.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};Fr&&(d.prototype[Fr]=d.prototype.inspect);d.prototype.compare=function(o,n,r,s,p){if(de(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),s===void 0&&(s=0),p===void 0&&(p=this.length),n<0||r>o.length||s<0||p>this.length)throw new RangeError("out of range index");if(s>=p&&n>=r)return 0;if(s>=p)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,s>>>=0,p>>>=0,this===o)return 0;for(var m=p-s,_=r-n,l=Math.min(m,_),h=this.slice(s,p),u=o.slice(n,r),b=0;b<l;++b)if(h[b]!==u[b]){m=h[b],_=u[b];break}return m<_?-1:_<m?1:0};function Vr(e,o,n,r,s){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,_t(n)&&(n=s?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(s)return-1;n=e.length-1}else if(n<0)if(s)n=0;else return-1;if(typeof o=="string"&&(o=d.from(o,r)),d.isBuffer(o))return o.length===0?-1:Br(e,o,n,r,s);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?s?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Br(e,[o],n,r,s);throw new TypeError("val must be string, number or Buffer")}function Br(e,o,n,r,s){var p=1,m=e.length,_=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,_/=2,n/=2}function l(k,I){return p===1?k[I]:k.readUInt16BE(I*p)}var h;if(s){var u=-1;for(h=n;h<m;h++)if(l(e,h)===l(o,u===-1?0:h-u)){if(u===-1&&(u=h),h-u+1===_)return u*p}else u!==-1&&(h-=h-u),u=-1}else for(n+_>m&&(n=m-_),h=n;h>=0;h--){for(var b=!0,L=0;L<_;L++)if(l(e,h+L)!==l(o,L)){b=!1;break}if(b)return h}return-1}d.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};d.prototype.indexOf=function(o,n,r){return Vr(this,o,n,r,!0)};d.prototype.lastIndexOf=function(o,n,r){return Vr(this,o,n,r,!1)};function Mm(e,o,n,r){n=Number(n)||0;var s=e.length-n;r?(r=Number(r),r>s&&(r=s)):r=s;var p=o.length;r>p/2&&(r=p/2);for(var m=0;m<r;++m){var _=parseInt(o.substr(m*2,2),16);if(_t(_))return m;e[n+m]=_}return m}function Fm(e,o,n,r){return nn(pt(o,e.length-n),e,n,r)}function Bm(e,o,n,r){return nn(Xm(o),e,n,r)}function Um(e,o,n,r){return nn(Kr(o),e,n,r)}function Wm(e,o,n,r){return nn(Ym(o,e.length-n),e,n,r)}d.prototype.write=function(o,n,r,s){if(n===void 0)s="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")s=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,s===void 0&&(s="utf8")):(s=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-n;if((r===void 0||r>p)&&(r=p),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");s||(s="utf8");for(var m=!1;;)switch(s){case"hex":return Mm(this,o,n,r);case"utf8":case"utf-8":return Fm(this,o,n,r);case"ascii":case"latin1":case"binary":return Bm(this,o,n,r);case"base64":return Um(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Wm(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+s);s=(""+s).toLowerCase(),m=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Gm(e,o,n){return o===0&&n===e.length?it.fromByteArray(e):it.fromByteArray(e.slice(o,n))}function Hr(e,o,n){n=Math.min(e.length,n);for(var r=[],s=o;s<n;){var p=e[s],m=null,_=p>239?4:p>223?3:p>191?2:1;if(s+_<=n){var l,h,u,b;switch(_){case 1:p<128&&(m=p);break;case 2:l=e[s+1],(l&192)===128&&(b=(p&31)<<6|l&63,b>127&&(m=b));break;case 3:l=e[s+1],h=e[s+2],(l&192)===128&&(h&192)===128&&(b=(p&15)<<12|(l&63)<<6|h&63,b>2047&&(b<55296||b>57343)&&(m=b));break;case 4:l=e[s+1],h=e[s+2],u=e[s+3],(l&192)===128&&(h&192)===128&&(u&192)===128&&(b=(p&15)<<18|(l&63)<<12|(h&63)<<6|u&63,b>65535&&b<1114112&&(m=b))}}m===null?(m=65533,_=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),s+=_}return $m(r)}var Ur=4096;function $m(e){var o=e.length;if(o<=Ur)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=Ur));return n}function Vm(e,o,n){var r="";n=Math.min(e.length,n);for(var s=o;s<n;++s)r+=String.fromCharCode(e[s]&127);return r}function Hm(e,o,n){var r="";n=Math.min(e.length,n);for(var s=o;s<n;++s)r+=String.fromCharCode(e[s]);return r}function qm(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var s="",p=o;p<n;++p)s+=Zm[e[p]];return s}function jm(e,o,n){for(var r=e.slice(o,n),s="",p=0;p<r.length-1;p+=2)s+=String.fromCharCode(r[p]+r[p+1]*256);return s}d.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var s=this.subarray(o,n);return Object.setPrototypeOf(s,d.prototype),s};function B(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var s=this[o],p=1,m=0;++m<n&&(p*=256);)s+=this[o+m]*p;return s};d.prototype.readUintBE=d.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var s=this[o+--n],p=1;n>0&&(p*=256);)s+=this[o+--n]*p;return s};d.prototype.readUint8=d.prototype.readUInt8=function(o,n){return o=o>>>0,n||B(o,1,this.length),this[o]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||B(o,2,this.length),this[o]|this[o+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||B(o,2,this.length),this[o]<<8|this[o+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||B(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||B(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};d.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var s=this[o],p=1,m=0;++m<n&&(p*=256);)s+=this[o+m]*p;return p*=128,s>=p&&(s-=Math.pow(2,8*n)),s};d.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var s=n,p=1,m=this[o+--s];s>0&&(p*=256);)m+=this[o+--s]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*n)),m};d.prototype.readInt8=function(o,n){return o=o>>>0,n||B(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};d.prototype.readInt16LE=function(o,n){o=o>>>0,n||B(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};d.prototype.readInt16BE=function(o,n){o=o>>>0,n||B(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};d.prototype.readInt32LE=function(o,n){return o=o>>>0,n||B(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};d.prototype.readInt32BE=function(o,n){return o=o>>>0,n||B(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};d.prototype.readFloatLE=function(o,n){return o=o>>>0,n||B(o,4,this.length),ro.read(this,o,!0,23,4)};d.prototype.readFloatBE=function(o,n){return o=o>>>0,n||B(o,4,this.length),ro.read(this,o,!1,23,4)};d.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||B(o,8,this.length),ro.read(this,o,!0,52,8)};d.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||B(o,8,this.length),ro.read(this,o,!1,52,8)};function K(e,o,n,r,s,p){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>s||o<p)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(o,n,r,s){if(o=+o,n=n>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;K(this,o,n,r,p,0)}var m=1,_=0;for(this[n]=o&255;++_<r&&(m*=256);)this[n+_]=o/m&255;return n+r};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(o,n,r,s){if(o=+o,n=n>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;K(this,o,n,r,p,0)}var m=r-1,_=1;for(this[n+m]=o&255;--m>=0&&(_*=256);)this[n+m]=o/_&255;return n+r};d.prototype.writeUint8=d.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,1,255,0),this[n]=o&255,n+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};d.prototype.writeIntLE=function(o,n,r,s){if(o=+o,n=n>>>0,!s){var p=Math.pow(2,8*r-1);K(this,o,n,r,p-1,-p)}var m=0,_=1,l=0;for(this[n]=o&255;++m<r&&(_*=256);)o<0&&l===0&&this[n+m-1]!==0&&(l=1),this[n+m]=(o/_>>0)-l&255;return n+r};d.prototype.writeIntBE=function(o,n,r,s){if(o=+o,n=n>>>0,!s){var p=Math.pow(2,8*r-1);K(this,o,n,r,p-1,-p)}var m=r-1,_=1,l=0;for(this[n+m]=o&255;--m>=0&&(_*=256);)o<0&&l===0&&this[n+m+1]!==0&&(l=1),this[n+m]=(o/_>>0)-l&255;return n+r};d.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};d.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};d.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function qr(e,o,n,r,s,p){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function jr(e,o,n,r,s){return o=+o,n=n>>>0,s||qr(e,o,n,4,34028234663852886e22,-34028234663852886e22),ro.write(e,o,n,r,23,4),n+4}d.prototype.writeFloatLE=function(o,n,r){return jr(this,o,n,!0,r)};d.prototype.writeFloatBE=function(o,n,r){return jr(this,o,n,!1,r)};function zr(e,o,n,r,s){return o=+o,n=n>>>0,s||qr(e,o,n,8,17976931348623157e292,-17976931348623157e292),ro.write(e,o,n,r,52,8),n+8}d.prototype.writeDoubleLE=function(o,n,r){return zr(this,o,n,!0,r)};d.prototype.writeDoubleBE=function(o,n,r){return zr(this,o,n,!1,r)};d.prototype.copy=function(o,n,r,s){if(!d.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!s&&s!==0&&(s=this.length),n>=o.length&&(n=o.length),n||(n=0),s>0&&s<r&&(s=r),s===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(s<0)throw new RangeError("sourceEnd out of bounds");s>this.length&&(s=this.length),o.length-n<s-r&&(s=o.length-n+r);var p=s-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,s):Uint8Array.prototype.set.call(o,this.subarray(r,s),n),p};d.prototype.fill=function(o,n,r,s){if(typeof o=="string"){if(typeof n=="string"?(s=n,n=0,r=this.length):typeof r=="string"&&(s=r,r=this.length),s!==void 0&&typeof s!="string")throw new TypeError("encoding must be a string");if(typeof s=="string"&&!d.isEncoding(s))throw new TypeError("Unknown encoding: "+s);if(o.length===1){var p=o.charCodeAt(0);(s==="utf8"&&p<128||s==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var _=d.isBuffer(o)?o:d.from(o,s),l=_.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=_[m%l]}return this};var zm=/[^+/0-9A-Za-z-_]/g;function Km(e){if(e=e.split("=")[0],e=e.trim().replace(zm,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function pt(e,o){o=o||1/0;for(var n,r=e.length,s=null,p=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!s){if(n>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&p.push(239,191,189);continue}s=n;continue}if(n<56320){(o-=3)>-1&&p.push(239,191,189),s=n;continue}n=(s-55296<<10|n-56320)+65536}else s&&(o-=3)>-1&&p.push(239,191,189);if(s=null,n<128){if((o-=1)<0)break;p.push(n)}else if(n<2048){if((o-=2)<0)break;p.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;p.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;p.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return p}function Xm(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function Ym(e,o){for(var n,r,s,p=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,s=n%256,p.push(s),p.push(r);return p}function Kr(e){return it.toByteArray(Km(e))}function nn(e,o,n,r){for(var s=0;s<r&&!(s+n>=o.length||s>=e.length);++s)o[s+n]=e[s];return s}function de(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function _t(e){return e!==e}var Zm=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,s=0;s<16;++s)o[r+s]=e[n]+e[s];return o}()});var ea=y((Mf,Qr)=>{t();var N=Qr.exports={},_e,le;function lt(){throw new Error("setTimeout has not been defined")}function ut(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?_e=setTimeout:_e=lt}catch(e){_e=lt}try{typeof clearTimeout=="function"?le=clearTimeout:le=ut}catch(e){le=ut}})();function Yr(e){if(_e===setTimeout)return setTimeout(e,0);if((_e===lt||!_e)&&setTimeout)return _e=setTimeout,setTimeout(e,0);try{return _e(e,0)}catch(o){try{return _e.call(null,e,0)}catch(n){return _e.call(this,e,0)}}}function Jm(e){if(le===clearTimeout)return clearTimeout(e);if((le===ut||!le)&&clearTimeout)return le=clearTimeout,clearTimeout(e);try{return le(e)}catch(o){try{return le.call(null,e)}catch(n){return le.call(this,e)}}}var we=[],io=!1,Be,tn=-1;function Qm(){!io||!Be||(io=!1,Be.length?we=Be.concat(we):tn=-1,we.length&&Zr())}function Zr(){if(!io){var e=Yr(Qm);io=!0;for(var o=we.length;o;){for(Be=we,we=[];++tn<o;)Be&&Be[tn].run();tn=-1,o=we.length}Be=null,io=!1,Jm(e)}}N.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];we.push(new Jr(e,o)),we.length===1&&!io&&Yr(Zr)};function Jr(e,o){this.fun=e,this.array=o}Jr.prototype.run=function(){this.fun.apply(null,this.array)};N.title="browser";N.browser=!0;N.env={};N.argv=[];N.version="";N.versions={};function ve(){}N.on=ve;N.addListener=ve;N.once=ve;N.off=ve;N.removeListener=ve;N.removeAllListeners=ve;N.emit=ve;N.prependListener=ve;N.prependOnceListener=ve;N.listeners=function(e){return[]};N.binding=function(e){throw new Error("process.binding is not supported")};N.cwd=function(){return"/"};N.chdir=function(e){throw new Error("process.chdir is not supported")};N.umask=function(){return 0}});var a,c,ed,i,t=f(()=>{a=S(Xr()),c=S(ea()),ed=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object),i=ed});var na=y((Bf,oa)=>{t();var od=typeof i=="object"&&i&&i.Object===Object&&i;oa.exports=od});var Eo=y((Uf,ta)=>{t();var nd=na(),td=typeof self=="object"&&self&&self.Object===Object&&self,rd=nd||td||Function("return this")();ta.exports=rd});var gt=y((Wf,ra)=>{t();var ad=Eo(),id=ad.Symbol;ra.exports=id});var sa=y((Gf,ca)=>{t();var aa=gt(),ia=Object.prototype,cd=ia.hasOwnProperty,sd=ia.toString,Po=aa?aa.toStringTag:void 0;function pd(e){var o=cd.call(e,Po),n=e[Po];try{e[Po]=void 0;var r=!0}catch(p){}var s=sd.call(e);return r&&(o?e[Po]=n:delete e[Po]),s}ca.exports=pd});var ma=y(($f,pa)=>{t();var md=Object.prototype,dd=md.toString;function _d(e){return dd.call(e)}pa.exports=_d});var ft=y((Vf,la)=>{t();var da=gt(),ld=sa(),ud=ma(),gd="[object Null]",fd="[object Undefined]",_a=da?da.toStringTag:void 0;function hd(e){return e==null?e===void 0?fd:gd:_a&&_a in Object(e)?ld(e):ud(e)}la.exports=hd});var ht=y((Hf,ua)=>{t();function xd(e){var o=typeof e;return e!=null&&(o=="object"||o=="function")}ua.exports=xd});var fa=y((qf,ga)=>{t();var Td=ft(),yd=ht(),bd="[object AsyncFunction]",Sd="[object Function]",wd="[object GeneratorFunction]",vd="[object Proxy]";function Ad(e){if(!yd(e))return!1;var o=Td(e);return o==Sd||o==wd||o==bd||o==vd}ga.exports=Ad});var xa=y((jf,ha)=>{t();var kd=Eo(),Id=kd["__core-js_shared__"];ha.exports=Id});var ba=y((zf,ya)=>{t();var xt=xa(),Ta=function(){var e=/[^.]+$/.exec(xt&&xt.keys&&xt.keys.IE_PROTO||"");return e?"Symbol(src)_1."+e:""}();function Ed(e){return!!Ta&&Ta in e}ya.exports=Ed});var wa=y((Kf,Sa)=>{t();var Pd=Function.prototype,Ld=Pd.toString;function Nd(e){if(e!=null){try{return Ld.call(e)}catch(o){}try{return e+""}catch(o){}}return""}Sa.exports=Nd});var Aa=y((Xf,va)=>{t();var Cd=fa(),Rd=ba(),Dd=ht(),Od=wa(),Md=/[\\^$.*+?()[\]{}|]/g,Fd=/^\[object .+?Constructor\]$/,Bd=Function.prototype,Ud=Object.prototype,Wd=Bd.toString,Gd=Ud.hasOwnProperty,$d=RegExp("^"+Wd.call(Gd).replace(Md,"\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g,"$1.*?")+"$");function Vd(e){if(!Dd(e)||Rd(e))return!1;var o=Cd(e)?$d:Fd;return o.test(Od(e))}va.exports=Vd});var Ia=y((Yf,ka)=>{t();function Hd(e,o){return e==null?void 0:e[o]}ka.exports=Hd});var rn=y((Zf,Ea)=>{t();var qd=Aa(),jd=Ia();function zd(e,o){var n=jd(e,o);return qd(n)?n:void 0}Ea.exports=zd});var Lo=y((Jf,Pa)=>{t();var Kd=rn(),Xd=Kd(Object,"create");Pa.exports=Xd});var Ca=y((Qf,Na)=>{t();var La=Lo();function Yd(){this.__data__=La?La(null):{},this.size=0}Na.exports=Yd});var Da=y((eh,Ra)=>{t();function Zd(e){var o=this.has(e)&&delete this.__data__[e];return this.size-=o?1:0,o}Ra.exports=Zd});var Ma=y((oh,Oa)=>{t();var Jd=Lo(),Qd="__lodash_hash_undefined__",e_=Object.prototype,o_=e_.hasOwnProperty;function n_(e){var o=this.__data__;if(Jd){var n=o[e];return n===Qd?void 0:n}return o_.call(o,e)?o[e]:void 0}Oa.exports=n_});var Ba=y((nh,Fa)=>{t();var t_=Lo(),r_=Object.prototype,a_=r_.hasOwnProperty;function i_(e){var o=this.__data__;return t_?o[e]!==void 0:a_.call(o,e)}Fa.exports=i_});var Wa=y((th,Ua)=>{t();var c_=Lo(),s_="__lodash_hash_undefined__";function p_(e,o){var n=this.__data__;return this.size+=this.has(e)?0:1,n[e]=c_&&o===void 0?s_:o,this}Ua.exports=p_});var $a=y((rh,Ga)=>{t();var m_=Ca(),d_=Da(),__=Ma(),l_=Ba(),u_=Wa();function co(e){var o=-1,n=e==null?0:e.length;for(this.clear();++o<n;){var r=e[o];this.set(r[0],r[1])}}co.prototype.clear=m_;co.prototype.delete=d_;co.prototype.get=__;co.prototype.has=l_;co.prototype.set=u_;Ga.exports=co});var Ha=y((ah,Va)=>{t();function g_(){this.__data__=[],this.size=0}Va.exports=g_});var ja=y((ih,qa)=>{t();function f_(e,o){return e===o||e!==e&&o!==o}qa.exports=f_});var No=y((ch,za)=>{t();var h_=ja();function x_(e,o){for(var n=e.length;n--;)if(h_(e[n][0],o))return n;return-1}za.exports=x_});var Xa=y((sh,Ka)=>{t();var T_=No(),y_=Array.prototype,b_=y_.splice;function S_(e){var o=this.__data__,n=T_(o,e);if(n<0)return!1;var r=o.length-1;return n==r?o.pop():b_.call(o,n,1),--this.size,!0}Ka.exports=S_});var Za=y((ph,Ya)=>{t();var w_=No();function v_(e){var o=this.__data__,n=w_(o,e);return n<0?void 0:o[n][1]}Ya.exports=v_});var Qa=y((mh,Ja)=>{t();var A_=No();function k_(e){return A_(this.__data__,e)>-1}Ja.exports=k_});var oi=y((dh,ei)=>{t();var I_=No();function E_(e,o){var n=this.__data__,r=I_(n,e);return r<0?(++this.size,n.push([e,o])):n[r][1]=o,this}ei.exports=E_});var ti=y((_h,ni)=>{t();var P_=Ha(),L_=Xa(),N_=Za(),C_=Qa(),R_=oi();function so(e){var o=-1,n=e==null?0:e.length;for(this.clear();++o<n;){var r=e[o];this.set(r[0],r[1])}}so.prototype.clear=P_;so.prototype.delete=L_;so.prototype.get=N_;so.prototype.has=C_;so.prototype.set=R_;ni.exports=so});var ai=y((lh,ri)=>{t();var D_=rn(),O_=Eo(),M_=D_(O_,"Map");ri.exports=M_});var si=y((uh,ci)=>{t();var ii=$a(),F_=ti(),B_=ai();function U_(){this.size=0,this.__data__={hash:new ii,map:new(B_||F_),string:new ii}}ci.exports=U_});var mi=y((gh,pi)=>{t();function W_(e){var o=typeof e;return o=="string"||o=="number"||o=="symbol"||o=="boolean"?e!=="__proto__":e===null}pi.exports=W_});var Co=y((fh,di)=>{t();var G_=mi();function $_(e,o){var n=e.__data__;return G_(o)?n[typeof o=="string"?"string":"hash"]:n.map}di.exports=$_});var li=y((hh,_i)=>{t();var V_=Co();function H_(e){var o=V_(this,e).delete(e);return this.size-=o?1:0,o}_i.exports=H_});var gi=y((xh,ui)=>{t();var q_=Co();function j_(e){return q_(this,e).get(e)}ui.exports=j_});var hi=y((Th,fi)=>{t();var z_=Co();function K_(e){return z_(this,e).has(e)}fi.exports=K_});var Ti=y((yh,xi)=>{t();var X_=Co();function Y_(e,o){var n=X_(this,e),r=n.size;return n.set(e,o),this.size+=n.size==r?0:1,this}xi.exports=Y_});var bi=y((bh,yi)=>{t();var Z_=si(),J_=li(),Q_=gi(),el=hi(),ol=Ti();function po(e){var o=-1,n=e==null?0:e.length;for(this.clear();++o<n;){var r=e[o];this.set(r[0],r[1])}}po.prototype.clear=Z_;po.prototype.delete=J_;po.prototype.get=Q_;po.prototype.has=el;po.prototype.set=ol;yi.exports=po});var wi=y((Sh,Si)=>{t();var nl="__lodash_hash_undefined__";function tl(e){return this.__data__.set(e,nl),this}Si.exports=tl});var Ai=y((wh,vi)=>{t();function rl(e){return this.__data__.has(e)}vi.exports=rl});var Ii=y((vh,ki)=>{t();var al=bi(),il=wi(),cl=Ai();function an(e){var o=-1,n=e==null?0:e.length;for(this.__data__=new al;++o<n;)this.add(e[o])}an.prototype.add=an.prototype.push=il;an.prototype.has=cl;ki.exports=an});var Pi=y((Ah,Ei)=>{t();function sl(e,o,n,r){for(var s=e.length,p=n+(r?1:-1);r?p--:++p<s;)if(o(e[p],p,e))return p;return-1}Ei.exports=sl});var Ni=y((kh,Li)=>{t();function pl(e){return e!==e}Li.exports=pl});var Ri=y((Ih,Ci)=>{t();function ml(e,o,n){for(var r=n-1,s=e.length;++r<s;)if(e[r]===o)return r;return-1}Ci.exports=ml});var Oi=y((Eh,Di)=>{t();var dl=Pi(),_l=Ni(),ll=Ri();function ul(e,o,n){return o===o?ll(e,o,n):dl(e,_l,n)}Di.exports=ul});var Fi=y((Ph,Mi)=>{t();var gl=Oi();function fl(e,o){var n=e==null?0:e.length;return!!n&&gl(e,o,0)>-1}Mi.exports=fl});var Ui=y((Lh,Bi)=>{t();function hl(e,o,n){for(var r=-1,s=e==null?0:e.length;++r<s;)if(n(o,e[r]))return!0;return!1}Bi.exports=hl});var Gi=y((Nh,Wi)=>{t();function xl(e,o){return e.has(o)}Wi.exports=xl});var Vi=y((Ch,$i)=>{t();var Tl=rn(),yl=Eo(),bl=Tl(yl,"Set");$i.exports=bl});var qi=y((Rh,Hi)=>{t();function Sl(){}Hi.exports=Sl});var Tt=y((Dh,ji)=>{t();function wl(e){var o=-1,n=Array(e.size);return e.forEach(function(r){n[++o]=r}),n}ji.exports=wl});var Ki=y((Oh,zi)=>{t();var yt=Vi(),vl=qi(),Al=Tt(),kl=1/0,Il=yt&&1/Al(new yt([,-0]))[1]==kl?function(e){return new yt(e)}:vl;zi.exports=Il});var Yi=y((Mh,Xi)=>{t();var El=Ii(),Pl=Fi(),Ll=Ui(),Nl=Gi(),Cl=Ki(),Rl=Tt(),Dl=200;function Ol(e,o,n){var r=-1,s=Pl,p=e.length,m=!0,_=[],l=_;if(n)m=!1,s=Ll;else if(p>=Dl){var h=o?null:Cl(e);if(h)return Rl(h);m=!1,s=Nl,l=new El}else l=o?[]:_;e:for(;++r<p;){var u=e[r],b=o?o(u):u;if(u=n||u!==0?u:0,m&&b===b){for(var L=l.length;L--;)if(l[L]===b)continue e;o&&l.push(b),_.push(u)}else s(l,b,n)||(l!==_&&l.push(b),_.push(u))}return _}Xi.exports=Ol});var Ji=y((Fh,Zi)=>{t();var Ml=Yi();function Fl(e){return e&&e.length?Ml(e):[]}Zi.exports=Fl});var oc=f(()=>{"use strict";t()});var nc,We,tc,At,rc,kt,ac=f(()=>{"use strict";t();nc="error when signing",We="Transaction canceled",tc="error signing transaction",At="provider not initialized",rc="You need a signer/valid signer to send a transaction,use either WalletProvider, LedgerProvider or WalletConnect",kt="You cannot sign transactions from a different account"});var ic=f(()=>{"use strict";t()});var It,cc=f(()=>{"use strict";t();It=(r=>(r.SetGuardian="SetGuardian",r.GuardAccount="GuardAccount",r.UnGuardAccount="UnGuardAccount",r))(It||{})});var sc=f(()=>{"use strict";t()});var Et=f(()=>{"use strict";t()});var pc=f(()=>{"use strict";t()});var Pt,Lt=f(()=>{"use strict";t();Pt=(s=>(s.signature="signature",s.sessionId="sessionId",s.status="status",s.address="address",s))(Pt||{})});var mc=f(()=>{"use strict";t()});var dc=f(()=>{"use strict";t()});var Q,_o,lo=f(()=>{"use strict";t();Q=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),_o=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var _c,ex,lc,ox,ee=f(()=>{"use strict";t();lo();_c=(n=>(n.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",n.reloginRequest="RELOGIN_REQUEST",n))(_c||{}),ex=g(g({},_o.WindowProviderRequestEnums),_c),lc=(o=>(o.reloginResponse="RELOGIN_RESPONSE",o))(lc||{}),ox=g(g({},_o.WindowProviderResponseEnums),lc)});var uc=f(()=>{"use strict";t()});var gc=f(()=>{"use strict";t()});var X=f(()=>{"use strict";t()});var fc=f(()=>{"use strict";t()});var hc=f(()=>{"use strict";t()});var xc=f(()=>{"use strict";t()});var G=f(()=>{"use strict";t();Lt();mc();dc();ee();uc();gc();X();fc();hc();xc()});var Ge,Tc,xx,yc,Tx,bc,yx,bx,Sc,Ro=f(()=>{"use strict";t();G();Ge={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:Tc,egldLabel:xx}=Ge["devnet"],{chainId:yc,egldLabel:Tx}=Ge["testnet"],{chainId:bc,egldLabel:yx}=Ge["mainnet"],bx={["devnet"]:Tc,["testnet"]:yc,["mainnet"]:bc},Sc={[Tc]:"devnet",[yc]:"testnet",[bc]:"mainnet"}});var Nt=f(()=>{"use strict";t()});var ue,wc=f(()=>{"use strict";t();ue=require("@multiversx/sdk-web-wallet-provider")});var Do,cn=f(()=>{"use strict";t();Do=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var vc,Ac,Ct,Ix,Ex,kc=f(()=>{"use strict";t();cn();Ct=String((Ac=(vc=Do.safeWindow)==null?void 0:vc.navigator)==null?void 0:Ac.userAgent),Ix=/^((?!chrome|android).)*safari/i.test(Ct),Ex=/firefox/i.test(Ct)&&/windows/i.test(Ct)});var sn,pn,Ic,Ec,ge,mn,dn,Pc,uo,A=f(()=>{"use strict";t();oc();ac();ic();cc();sc();Et();pc();Ro();Nt();wc();kc();sn=5e4,pn=1e9,Ic=1,Ec="/dapp/init",ge="signSession",mn="sdk-dapp-version",dn="logout",Pc="login",uo="0"});var Oo,_n=f(()=>{"use strict";t();Oo=()=>Date.now()/1e3});var Lc=f(()=>{"use strict";t()});var Nc=f(()=>{"use strict";t()});var Cc=f(()=>{"use strict";t();_n();Lc();Nc()});var Rt={};be(Rt,{clear:()=>Hl,getItem:()=>$l,localStorageKeys:()=>Ae,removeItem:()=>Vl,setItem:()=>Gl});var Ae,ln,Gl,$l,Vl,Hl,$e=f(()=>{"use strict";t();Cc();Ae={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},ln=typeof localStorage!="undefined",Gl=({key:e,data:o,expires:n})=>{!ln||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},$l=e=>{if(!ln)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Oo()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},Vl=e=>{!ln||localStorage.removeItem(String(e))},Hl=()=>{!ln||localStorage.clear()}});var Dt={};be(Dt,{clear:()=>Mc,getItem:()=>Dc,removeItem:()=>Oc,setItem:()=>Rc,storage:()=>ql});var Rc,Dc,Oc,Mc,ql,Fc=f(()=>{"use strict";t();Rc=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Dc=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Oc=e=>sessionStorage.removeItem(String(e)),Mc=()=>sessionStorage.clear(),ql={setItem:Rc,getItem:Dc,removeItem:Oc,clear:Mc}});var Ce,go=f(()=>{"use strict";t();$e();Fc();Ce={session:Dt,local:Rt}});var Ot,P,Ve,q=f(()=>{"use strict";t();Ot=require("@reduxjs/toolkit");A();P=(0,Ot.createAction)(dn),Ve=(0,Ot.createAction)(Pc,e=>({payload:e}))});var Ft,Bc,Uc,un,Mt,Wc,Bt,aT,Ut,iT,cT,sT,pT,mT,dT,_T,lT,gn,fn=f(()=>{"use strict";t();Ft=require("@multiversx/sdk-core"),Bc=require("@reduxjs/toolkit"),Uc=require("redux-persist");A();go();$e();q();un={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:uo},Mt={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":un},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Wc=(0,Bc.createSlice)({name:"accountInfoSlice",initialState:Mt,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=n?new Ft.Address(n).hex():""},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:un},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}},setWebsocketBatchEvent:(e,o)=>{e.websocketBatchEvent={timestamp:Date.now(),data:o.payload}}},extraReducers:e=>{e.addCase(P,()=>(Ce.local.removeItem(Ae.loginExpiresAt),Mt)),e.addCase(Ve,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new Ft.Address(r).toHex()}),e.addCase(Uc.REHYDRATE,(o,n)=>{var h;if(!((h=n.payload)!=null&&h.account))return;let{account:r}=n.payload,{address:s,shard:p,accounts:m,publicKey:_}=r;o.address=s,o.shard=p;let l=m&&s in m;o.accounts=l?m:Mt.accounts,o.publicKey=_})}}),{setAccount:Bt,setAddress:aT,setAccountNonce:Ut,setAccountShard:iT,setLedgerAccount:cT,updateLedgerAccount:sT,setWalletConnectAccount:pT,setIsAccountLoading:mT,setAccountLoadingError:dT,setWebsocketEvent:_T,setWebsocketBatchEvent:lT}=Wc.actions,gn=Wc.reducer});function Mo(){return new Date().setHours(new Date().getHours()+24)}function Fo(e){Ce.local.setItem({key:Ae.loginExpiresAt,data:e,expires:e})}var Wt=f(()=>{"use strict";t();go();$e()});var $c,Gc,Vc,wT,vT,AT,Hc,kT,IT,qc,ET,PT,LT,hn,xn=f(()=>{"use strict";t();$c=require("@reduxjs/toolkit");Wt();ee();q();Gc={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},Vc=(0,$c.createSlice)({name:"loginInfoSlice",initialState:Gc,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},setWebviewLogin:(e,o)=>{e.webviewLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload},setIsWalletConnectV2Initialized:(e,o)=>{e.isWalletConnectV2Initialized=o.payload}},extraReducers:e=>{e.addCase(P,()=>Gc),e.addCase(Ve,(o,n)=>{o.isLoginSessionInvalid=!1,o.loginMethod=n.payload.loginMethod,o.iframeLoginType=n.payload.iframeLoginType,Fo(Mo())})}}),{setLoginMethod:wT,setWalletConnectLogin:vT,setLedgerLogin:AT,setTokenLogin:Hc,setTokenLoginSignature:kT,setWalletLogin:IT,invalidateLoginSession:qc,setLogoutRoute:ET,setIsWalletConnectV2Initialized:PT,setWebviewLogin:LT}=Vc.actions,hn=Vc.reducer});var zc,jc,Kc,DT,jl,OT,MT,Tn,yn=f(()=>{"use strict";t();zc=require("@reduxjs/toolkit");q();jc={},Kc=(0,zc.createSlice)({name:"modalsSlice",initialState:jc,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(P,()=>jc)}}),{setTxSubmittedModal:DT,setNotificationModal:jl,clearTxSubmittedModal:OT,clearNotificationModal:MT}=Kc.actions,Tn=Kc.reducer});var ke,Bo=f(()=>{"use strict";t();ke=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var C,ie=f(()=>{"use strict";t();Bo();C=()=>{if(!ke())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:o,hash:n,origin:r,href:s,search:p}}=window;return{pathname:o,hash:n,origin:r,href:s,search:p}}});var Xc=f(()=>{"use strict";t();ie()});var Gt,Yc=f(()=>{"use strict";t();Re();Gt=()=>{let{pathname:e,search:o,hash:n}=C();return`${e!=null?e:""}${o!=null?o:""}${n!=null?n:""}`}});var Zc=f(()=>{"use strict";t();cn()});var Re=f(()=>{"use strict";t();Xc();Yc();ie();Zc()});var $t=f(()=>{"use strict";t();Re()});var Jc=f(()=>{"use strict";t();$t()});function Qc(e){return e[Math.floor(Math.random()*e.length)]}var es=f(()=>{"use strict";t()});var os=f(()=>{"use strict";t();Et()});var ns=f(()=>{"use strict";t();Jc();es();os()});var ts,rs,as,Vt,Xl,is,gy,fy,hy,bn,Sn=f(()=>{"use strict";t();ts=require("@reduxjs/toolkit"),rs=S(require("lodash.omit")),as=require("redux-persist");Nt();q();ns();Vt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},Xl={network:Vt},is=(0,ts.createSlice)({name:"appConfig",initialState:Xl,reducers:{initializeNetworkConfig:(e,o)=>{let n=Qc(o.payload.walletConnectV2RelayAddresses),r=(0,rs.default)(o.payload,"walletConnectV2RelayAddresses");e.network=w(g(g({},e.network),r),{walletConnectV2RelayAddress:n})},updateNetworkConfig:(e,o)=>{e.network=g(g({},e.network),o.payload)},setCustomWalletAddress:(e,o)=>{e.network.customWalletAddress=o.payload}},extraReducers:e=>{e.addCase(P,o=>{o.network.customWalletAddress=void 0}),e.addCase(as.REHYDRATE,(o,n)=>{var s,p;if(!((p=(s=n.payload)==null?void 0:s.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:r}}=n.payload;o.network.customWalletAddress=r})}}),{initializeNetworkConfig:gy,updateNetworkConfig:fy,setCustomWalletAddress:hy}=is.actions,bn=is.reducer});var cs,Ht,ss,wy,vy,Ay,wn,vn=f(()=>{"use strict";t();cs=require("@reduxjs/toolkit");G();q();Ht={isSigning:!1,signedSessions:{}},ss=(0,cs.createSlice)({name:"signedMessageInfoSliceState",initialState:Ht,reducers:{setSignSession:(e,o)=>{let{sessionId:n,signedSession:r,errorMessage:s}=o.payload;s&&(e.errorMessage=s),e.isSigning=r.status==="pending",e.signedSessions[n]=g(g({},e.signedSessions[n]),r)},setSignSessionState:(e,o)=>g(g({},e),o.payload),clearSignedMessageInfo:()=>Ht},extraReducers:e=>{e.addCase(P,()=>Ht)}}),{setSignSession:wy,clearSignedMessageInfo:vy,setSignSessionState:Ay}=ss.actions,wn=ss.reducer});var ms,ds,ps,_s,Yl,Zl,Cy,Jl,An,kn=f(()=>{"use strict";t();ms=require("@reduxjs/toolkit"),ds=require("redux-persist");G();_n();q();ps={customToasts:[],transactionToasts:[]},_s=(0,ms.createSlice)({name:"toastsSlice",initialState:ps,reducers:{addCustomToast:(e,o)=>{let n=o.payload.toastId||`custom-toast-${e.customToasts.length+1}`,r=e.customToasts.findIndex(s=>s.toastId===n);if(r!==-1){e.customToasts[r]=w(g(g({},e.customToasts[r]),o.payload),{type:"custom",toastId:n});return}e.customToasts.push(w(g({},o.payload),{type:"custom",toastId:n}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Oo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)}},extraReducers:e=>{e.addCase(P,()=>ps),e.addCase(ds.REHYDRATE,(o,n)=>{var s,p;let r=(p=(s=n.customToasts)==null?void 0:s.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=r})}}),{addCustomToast:Yl,removeCustomToast:Zl,addTransactionToast:Cy,removeTransactionToast:Jl}=_s.actions,An=_s.reducer});var ls,jt,zt,Kt,Ql,qt,us,My,eu,In,En,Pn=f(()=>{"use strict";t();ls=require("@reduxjs/toolkit");q();jt="Transaction failed",zt="Transaction successful",Kt="Processing transaction",Ql="Transaction submitted",qt={},us=(0,ls.createSlice)({name:"transactionsInfo",initialState:qt,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||jt,successMessage:(r==null?void 0:r.successMessage)||zt,processingMessage:(r==null?void 0:r.processingMessage)||Kt,submittedMessage:(r==null?void 0:r.submittedMessage)||Ql,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>qt},extraReducers:e=>{e.addCase(P,()=>qt)}}),{clearTransactionsInfo:My,setTransactionsDisplayInfo:eu,clearTransactionsInfoForSessionId:In}=us.actions,En=us.reducer});function gs(e){return e!=null&&(mu(e)||gu(e))}function Ln(e){return e!=null&&(du(e)||fu(e))}function Nn(e){return e!=null&&(_u(e)||hu(e))}function fs(e){return e!=null&&(lu(e)||xu(e))}function hs(e){return e!=null&&uu(e)}function mu(e){return e!=null&&ou.includes(e)}function du(e){return e!=null&&nu.includes(e)}function _u(e){return e!=null&&tu.includes(e)}function lu(e){return e!=null&&ru.includes(e)}function uu(e){return e!=null&&au.includes(e)}function gu(e){return e!=null&&iu.includes(e)}function fu(e){return e!=null&&cu.includes(e)}function hu(e){return e!=null&&su.includes(e)}function xu(e){return e!=null&&pu.includes(e)}var ou,nu,tu,ru,au,iu,cu,su,pu,Zt=f(()=>{"use strict";t();ee();ou=["sent"],nu=["success"],tu=["fail","cancelled","timedOut"],ru=["invalid"],au=["timedOut"],iu=["pending"],cu=["success"],su=["fail","invalid"],pu=["not executed"]});var xs,Ts,Uo,Tu,ys,qy,jy,yu,fo,bu,Su,zy,Ky,qe,je,Xy,Cn,Rn=f(()=>{"use strict";t();xs=require("@reduxjs/toolkit"),Ts=require("redux-persist");ee();Zt();q();Uo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Tu={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},ys=(0,xs.createSlice)({name:"transactionsSlice",initialState:Uo,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,h;let{sessionId:n,transactions:r,errorMessage:s,status:p,redirectRoute:m,customTransactionInformation:_}=o.payload;e.customTransactionInformationForSessionId[n]=g(g(g({},Tu),((l=e.signedTransactions[n])==null?void 0:l.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[n]||{}),e.signedTransactions[n]={transactions:r,status:p,errorMessage:s,redirectRoute:m,customTransactionInformation:g(g({},e.customTransactionInformationForSessionId[n]),_!=null?_:{})},((h=e==null?void 0:e.transactionsToSign)==null?void 0:h.sessionId)===n&&(e.transactionsToSign=Uo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=g(g({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:s,transactions:p}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,s!=null&&(e.signedTransactions[n].errorMessage=s),p!=null&&(e.signedTransactions[n].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var h,u,b,L,k,I,te,W;let{sessionId:n,status:r,errorMessage:s,transactionHash:p,serverTransaction:m,inTransit:_}=o.payload,l=(u=(h=e.signedTransactions)==null?void 0:h[n])==null?void 0:u.transactions;if(l!=null){e.signedTransactions[n].transactions=l.map(D=>D.hash===p?w(g(g({},m!=null?m:{}),D),{status:r,errorMessage:s,inTransit:_}):D);let $=(L=(b=e.signedTransactions[n])==null?void 0:b.transactions)==null?void 0:L.every(D=>Ln(D.status)),re=(I=(k=e.signedTransactions[n])==null?void 0:k.transactions)==null?void 0:I.some(D=>Nn(D.status)),xe=(W=(te=e.signedTransactions[n])==null?void 0:te.transactions)==null?void 0:W.every(D=>fs(D.status));$&&(e.signedTransactions[n].status="success"),re&&(e.signedTransactions[n].status="fail"),xe&&(e.signedTransactions[n].status="invalid")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Uo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Uo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload},updateSignedTransactionsCustomTransactionInformation:(e,o)=>{let{sessionId:n,customTransactionInformationOverrides:r}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].customTransactionInformation=g(g({},e.signedTransactions[n].customTransactionInformation),r))}},extraReducers:e=>{e.addCase(P,()=>Uo),e.addCase(Ts.REHYDRATE,(o,n)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:s}=n.payload.transactions,p=Object.entries(r).reduce((_,[l,h])=>{let u=new Date(l),b=new Date;return b.setHours(b.getHours()+5),b-u>0||(_[l]=h),_},{});s!=null&&(o.customTransactionInformationForSessionId=s),r!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:qy,updateSignedTransactions:jy,setTransactionsToSign:yu,clearAllTransactionsToSign:fo,clearAllSignedTransactions:bu,clearSignedTransaction:Su,clearTransactionToSign:zy,setSignTransactionsError:Ky,setSignTransactionsCancelMessage:qe,moveTransactionsToSignedState:je,updateSignedTransactionsCustomTransactionInformation:Xy}=ys.actions,Cn=ys.reducer});var bs,Jt,Ss,Qy,eb,wu,ob,Dn,On=f(()=>{"use strict";t();bs=require("@reduxjs/toolkit");q();Jt={},Ss=(0,bs.createSlice)({name:"batchTransactionsSlice",initialState:Jt,reducers:{setBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},updateBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},clearBatchTransactions:(e,o)=>{delete e[o.payload.batchId]},clearAllBatchTransactions:()=>Jt},extraReducers:e=>{e.addCase(P,()=>Jt)}}),{setBatchTransactions:Qy,updateBatchTransactions:eb,clearBatchTransactions:wu,clearAllBatchTransactions:ob}=Ss.actions,Dn=Ss.reducer});var vs,ws,As,ab,ks,Qt=f(()=>{"use strict";t();vs=require("@reduxjs/toolkit");q();ws={},As=(0,vs.createSlice)({name:"dappConfigSlice",initialState:ws,reducers:{setDappConfig:(e,o)=>o.payload},extraReducers:e=>{e.addCase(P,()=>ws)}}),{setDappConfig:ab}=As.actions,ks=As.reducer});var j=f(()=>{"use strict";t();fn();xn();yn();Sn();vn();kn();Pn();Rn();On();Qt()});var nr=f(()=>{"use strict";t()});var Us,Bu,Uu,Je,$n=f(()=>{"use strict";t();Us=require("@reduxjs/toolkit");nr();fn();On();Qt();xn();yn();Sn();vn();kn();Pn();Rn();Bu={["account"]:gn,["dappConfig"]:ks,["loginInfo"]:hn,["modals"]:Tn,["networkConfig"]:bn,["signedMessageInfo"]:wn,["toasts"]:An,["transactionsInfo"]:En,["transactions"]:Cn,["batchTransactions"]:Dn},Uu=(e={})=>(0,Us.combineReducers)(g(g({},Bu),e)),Je=Uu});var Vs={};be(Vs,{default:()=>og,sessionStorageReducers:()=>tr});function Ee(e,o=[]){return{key:e,version:1,storage:Gs.default,blacklist:o}}var Y,Ws,Gs,Wu,Go,Gu,$u,Vu,Hu,qu,ju,zu,Ku,Xu,Yu,$s,Zu,tr,Ju,Qu,eg,og,Hs=f(()=>{"use strict";t();Y=require("redux-persist"),Ws=S(require("redux-persist/lib/storage")),Gs=S(require("redux-persist/lib/storage/session"));$n();j();fn();On();xn();yn();Sn();vn();kn();Pn();Rn();nr();Wu={persistReducersStorageType:"localStorage"},Go={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Gu=Ee(Go["account"]),$u=Ee(Go["loginInfo"]),Vu=Ee(Go["modals"]),Hu=Ee(Go["networkConfig"]),qu={2:e=>w(g({},e),{networkConfig:Vt})};ju=Ee("sdk-dapp-transactionsInfo"),zu=Ee("sdk-dapp-transactions",["transactionsToSign"]),Ku=Ee("sdk-dapp-batchTransactions",["batchTransactions"]),Xu=Ee("sdk-dapp-toasts"),Yu=Ee("sdk-dapp-signedMessageInfo"),$s={key:"sdk-dapp-store",version:2,storage:Ws.default,whitelist:Object.keys(Go),migrate:(0,Y.createMigrate)(qu,{debug:!1})},Zu=w(g({},$s),{whitelist:[]}),tr={["toasts"]:(0,Y.persistReducer)(Xu,An),["transactions"]:(0,Y.persistReducer)(zu,Cn),["transactionsInfo"]:(0,Y.persistReducer)(ju,En),["batchTransactions"]:(0,Y.persistReducer)(Ku,Dn),["signedMessageInfo"]:(0,Y.persistReducer)(Yu,wn)},Ju=w(g({},tr),{["account"]:(0,Y.persistReducer)(Gu,gn),["loginInfo"]:(0,Y.persistReducer)($u,hn),["modals"]:(0,Y.persistReducer)(Vu,Tn),["networkConfig"]:(0,Y.persistReducer)(Hu,bn)}),Qu=Wu.persistReducersStorageType==="localStorage",eg=Qu?(0,Y.persistReducer)($s,Je(tr)):(0,Y.persistReducer)(Zu,Je(Ju)),og=eg});var qs={};be(qs,{default:()=>ng});var ng,js=f(()=>{"use strict";t();$n();ng=Je()});var zs={};be(zs,{default:()=>rg});var ce,tg,rg,Ks=f(()=>{"use strict";t();ce=require("redux-persist"),tg=[ce.FLUSH,ce.REHYDRATE,ce.PAUSE,ce.PERSIST,ce.PURGE,ce.REGISTER],rg=tg});var Zs={};be(Zs,{default:()=>Ys});function Ys(e){return(0,Xs.persistStore)(e)}var Xs,Js=f(()=>{"use strict";t();Xs=require("redux-persist")});var Mp=y((VL,Op)=>{t();var zg=Array.isArray;Op.exports=zg});var Bp=y((HL,Fp)=>{t();function Kg(e){return e!=null&&typeof e=="object"}Fp.exports=Kg});var Wp=y((qL,Up)=>{t();var Xg=ft(),Yg=Mp(),Zg=Bp(),Jg="[object String]";function Qg(e){return typeof e=="string"||!Yg(e)&&Zg(e)&&Xg(e)==Jg}Up.exports=Qg});var xr={};be(xr,{css:()=>tm,default:()=>Tf});var tm,Tf,Tr=f(()=>{"use strict";t();tm=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(tm));Tf={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var br={};be(br,{css:()=>cm,default:()=>Sf});var cm,Sf,Sr=f(()=>{"use strict";t();cm=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(cm));Sf={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var Nf={};be(Nf,{useSignTransactions:()=>Lf});module.exports=Oe(Nf);t();var Yo=require("react"),ot=require("@multiversx/sdk-core"),kr=require("@multiversx/sdk-extension-provider"),Ir=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Er=require("@multiversx/sdk-passkey-provider/out"),um=S(Ji());t();t();var mo="accounts";var bt="blocks";t();t();t();var Qi=require("@lifeomic/axios-fetch"),St=S(require("axios")),wt=(0,Qi.buildAxiosFetch)(St.default),vt=(e,o)=>x(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let r=e.clone().json(),[s]=yield Promise.all([r]);return{data:s,status:e.status,statusText:e.statusText,headers:e.headers,config:o}});function Bl(e,o,n){return x(this,null,function*(){try{let r=yield wt(e,g({method:"POST",body:o?JSON.stringify(o):void 0,headers:g({"Content-Type":"application/json"},(n==null?void 0:n.headers)||{})},n));return vt(r,n)}catch(r){throw console.error("Fetch Error:",r),r}})}function Ul(e,o){return x(this,null,function*(){try{let n=yield wt(e,o);if(!n.ok)throw new Error(`HTTP error! Status: ${n.status}`);return vt(n,o)}catch(n){throw console.error("Fetch Error:",n),n}})}function Wl(e,o,n){return x(this,null,function*(){try{let r=yield wt(e,g({method:"PATCH",body:o?JSON.stringify(o):void 0,headers:(n==null?void 0:n.headers)||{}},n));if(!r.ok)throw new Error(`HTTP error! Status: ${r.status}`);return vt(r,n)}catch(r){throw console.error("Fetch Error:",r),r}})}var Ue=St.default.create();Ue.get=Ul;Ue.post=Bl;Ue.patch=Wl;var ec=e=>Ue.get(e).then(o=>o.data);t();t();t();j();t();var Is=S(require("lodash.isequal")),Mn=require("reselect"),T=(0,Mn.createSelectorCreator)(Mn.defaultMemoize,Is.default);var fe=e=>e.account,ze=T(fe,e=>e.address),Ke=T(fe,ze,(e,o)=>o in e.accounts?e.accounts[o]:un),vu=T(fe,Ke,(e,o)=>{let s=e,{accounts:n}=s,r=to(s,["accounts"]);return w(g({},r),{address:o.address,account:o})}),bb=T(Ke,e=>e.balance),Es=T(Ke,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Sb=T(fe,e=>e.shard),wb=T(fe,e=>e.ledgerAccount),vb=T(fe,e=>e.walletConnectAccount),Ab=T(fe,e=>e.isAccountLoading),kb=T(fe,e=>e.accountLoadingError),Au=T(fe,e=>e.websocketEvent),ku=T(fe,e=>e.websocketBatchEvent);t();var er=e=>e.dappConfig,Pb=T(er,e=>e.shouldUseWebViewProvider);t();var oe=e=>e.loginInfo,Rb=T(oe,e=>e.loginMethod),or=T(oe,ze,(e,o)=>Boolean(o)),Db=T(oe,e=>e.walletConnectLogin),Iu=T(oe,e=>e.ledgerLogin),Eu=T(oe,e=>e.walletLogin),Ob=T(oe,e=>e.isLoginSessionInvalid),Mb=T(oe,e=>e.tokenLogin),Pu=T(oe,e=>e.logoutRoute),Lu=T(oe,e=>e.isWalletConnectV2Initialized);t();var Ps=e=>e.modals,Ub=T(Ps,e=>e.txSubmittedModal),Wb=T(Ps,e=>e.notificationModal);t();var Ie=e=>e.networkConfig,Ls=T(Ie,e=>e.network.chainId),Vb=T(Ie,e=>e.network.roundDuration),Hb=T(Ie,e=>e.network.walletConnectBridgeAddress),qb=T(Ie,e=>e.network.walletConnectV2RelayAddress),jb=T(Ie,e=>e.network.walletConnectV2ProjectId),zb=T(Ie,e=>e.network.walletConnectV2Options),Kb=T(Ie,e=>e.network.walletConnectDeepLink),Xe=T(Ie,e=>e.network),Ns=T(Xe,e=>e.apiAddress),Cs=T(Xe,e=>e.explorerAddress),Rs=T(Xe,e=>{var o;return(o=e.customWalletAddress)!=null?o:e.walletAddress}),Xb=T(Xe,e=>e.xAliasAddress),Yb=T(Xe,e=>e.egldLabel);t();var Fn=e=>e.signedMessageInfo,Qb=T(Fn,e=>e.isSigning),e0=T(Fn,e=>e.errorMessage),o0=T(Fn,e=>{let o=Object.keys(e.signedSessions),n=o.length;return e.signedSessions[o[n-1]]}),n0=T(Fn,e=>{let o=Object.keys(e.signedSessions),n=o.length;return o.length>0?o[n-1]:""});t();var Ds=e=>e.toasts,a0=T(Ds,e=>e.customToasts),i0=T(Ds,e=>e.transactionToasts);t();j();var Os={errorMessage:jt,successMessage:zt,processingMessage:Kt},Nu=e=>e.transactionsInfo,m0=T(Nu,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Os);t();t();var Ze=require("@multiversx/sdk-core");A();t();t();t();function Bn(e){try{let o=atob(e),n=btoa(o),r=a.Buffer.from(e,"base64").toString(),s=a.Buffer.from(r).toString("base64"),p=e===n||n.startsWith(e),m=e===s||s.startsWith(e);if(p&&m)return!0}catch(o){return!1}return!1}function Ye(e){return Bn(e)?atob(e):e}t();t();t();t();var Ms=e=>{let o=e!=null?e:"";return Bn(o)?a.Buffer.from(a.Buffer.from(o,"base64").toString().trim()):a.Buffer.from(o.trim())};t();A();var ho=({data:e,onlySetGuardian:o})=>e?o?e.startsWith("SetGuardian"):Object.values(It).some(n=>e.startsWith(n)):!1;function Wo(e){var r,s,p;let o=g({},e);ho({data:o.data,onlySetGuardian:!0})&&(delete o.senderUsername,delete o.receiverUsername);let n=new Ze.Transaction(g(g(w(g(w(g(g({value:BigInt(o.value.valueOf()),data:Ms(o.data),nonce:BigInt(o.nonce.valueOf()),receiver:new Ze.Address(o.receiver)},o.receiverUsername?{receiverUsername:o.receiverUsername}:{}),o.relayer?{relayer:new Ze.Address(o.relayer)}:{}),{sender:new Ze.Address(o.sender)}),o.senderUsername?{senderUsername:o.senderUsername}:{}),{gasLimit:BigInt((r=o.gasLimit.valueOf())!=null?r:sn),gasPrice:BigInt((s=o.gasPrice.valueOf())!=null?s:pn),chainID:o.chainID.valueOf(),version:(p=o.version)!=null?p:Ic}),o.options?{options:o.options}:{}),o.guardian?{guardian:new Ze.Address(o.guardian)}:{}));return o.relayerSignature&&(n.relayerSignature=a.Buffer.from(o.relayerSignature,"hex")),o.guardianSignature&&(n.guardianSignature=a.Buffer.from(o.guardianSignature,"hex")),o.signature&&(n.signature=a.Buffer.from(o.signature,"hex")),n}t();t();A();function Ru(e,o){let n=new URLSearchParams(e),r=Object.fromEntries(n);return{nextUrlParams:new URLSearchParams(g(g({},r),o)).toString(),params:r}}var Du="4.1.8";function Un({callbackUrl:e,urlParams:o={}}){let n=e;if(Object.entries(o).length>0)try{let{search:r,origin:s,pathname:p,hash:m}=new URL(e),_=w(g({},o),{[mn]:Du}),{nextUrlParams:l}=Ru(r,_);n=`${s}${p}?${l}${m}`}catch(r){return console.error("Unable to construct URL from: ",e,r),n}return n}t();t();var rr=require("@reduxjs/toolkit"),np=require("react-redux/lib/utils/Subscription");j();t();var Fs=S(require("lodash.throttle"));A();j();Wt();go();$e();var Ou=[dn],Wn=!1,Mu=(0,Fs.default)(()=>{Fo(Mo())},5e3),Bs=e=>o=>n=>{if(Ou.includes(n.type))return o(n);let r=e.getState(),s=Ce.local.getItem(Ae.loginExpiresAt);if(!Boolean(r==null?void 0:r.account.address))return o(n);if(s==null)return Fo(Mo());let m=Date.now();return s-m<0&&!Wn?setTimeout(()=>{Wn=!0,e.dispatch(qc())},1e3):(Wn&&(Wn=!1),Mu()),o(n)};t();t();function Gn(){return typeof sessionStorage!="undefined"}var Qs=Gn()?(Hs(),Oe(Vs)).default:(js(),Oe(qs)).default,ep=Gn()?(Ks(),Oe(zs)).default:[],op=Gn()?(Js(),Oe(Zs)).default:e=>e;$n();var v=(0,rr.configureStore)({reducer:Qs,middleware:e=>e({serializableCheck:{ignoredActions:[...ep,Ut.type,Bt.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Bs)}),tp=(0,np.createSubscription)(v),TS=op(v),yS=(0,rr.configureStore)({reducer:Je});t();t();var ig=require("@multiversx/sdk-core");t();A();t();var sg=require("@multiversx/sdk-core");G();t();var cg=require("@multiversx/sdk-core");t();t();t();var pg=require("@multiversx/sdk-core"),mg=S(require("bignumber.js"));ee();t();t();t();t();t();t();X();t();t();X();t();t();t();var fg=S(require("bignumber.js"));A();t();var gg=require("@multiversx/sdk-core"),ir=S(require("bignumber.js"));A();t();var lg=S(require("bignumber.js"));t();ir.default.config({ROUNDING_MODE:ir.default.ROUND_FLOOR});t();t();t();A();t();t();t();t();A();t();A();t();var hg=require("@multiversx/sdk-core"),xg=S(require("bignumber.js"));A();t();var sr=require("@multiversx/sdk-core"),bg=S(require("bignumber.js"));A();t();t();var Tg=S(require("bignumber.js"));A();t();A();t();t();t();t();t();t();A();t();A();t();A();t();X();var wg=["reDelegateRewards","claimRewards","unBond"],vg=["wrapEgld","unwrapEgld"],Ag=["unStake"],kg=["unDelegate"];t();t();t();A();t();var Pg=S(require("bignumber.js"));t();t();X();t();var Ng=S(require("bignumber.js"));t();t();t();t();var Rg=S(require("bignumber.js"));G();t();t();var sp=require("@multiversx/sdk-core");t();t();ee();function Vo(e){let n=Object.getPrototypeOf(e).toPlainObject!=null?e:Wo(e),r=new sp.TransactionComputer,s=w(g({},n.toPlainObject()),{hash:r.computeTransactionHash(n),senderUsername:n.senderUsername,receiverUsername:n.receiverUsername,status:"pending"});return ho({data:s.data,onlySetGuardian:!0})&&(delete s.senderUsername,delete s.receiverUsername),s}t();var dp=require("@multiversx/sdk-web-wallet-provider");A();t();var mp=S(require("qs"));t();Re();var Hn=e=>{let o=new URLSearchParams(e).toString(),{pathname:n,hash:r}=C(),s=o?`?${o}`:"",p=n?`${n}${s}${r}`:"./";setTimeout(()=>{window==null||window.history.replaceState({},document==null?void 0:document.title,p)})};Bo();t();Bo();var Dg={search:ke()?window.location.search:"",removeParams:[]},pp=(e,o=Dg)=>{var m;let n={},r=ke()?window.location.search:"",s=(m=o.search)!=null?m:r;if(s){let _=s?new URLSearchParams(s):[];n=Object.fromEntries(_)}let p={};return e.forEach(_=>{p[_]=n[_],delete n[_]}),o.removeParams!=null&&Object.keys(n).forEach(_=>{var h,u;let[l]=_.split("[");(((h=o.removeParams)==null?void 0:h.includes(_))||((u=o.removeParams)==null?void 0:u.includes(l)))&&delete n[_]}),{remainingParams:p,params:n,clearNavigationHistory:()=>Hn(n)}};var qn=({removeParams:e,search:o})=>{let n=ke()?window.location.search:"",r=o!=null?o:n;if(!r)return{};let s=mp.default.parse(r.replace("?","")),p=Object.keys(s).filter(_=>!e.includes(_)),{remainingParams:m}=pp(p,{search:o,removeParams:e});return Hn(m),m};var _p=({transaction:e,search:o})=>qn({removeParams:[...Object.keys(e),dp.WALLET_PROVIDER_CALLBACK_PARAM,ge,mn],search:o});t();t();t();ie();t();X();t();t();ie();t();var Og=S(require("linkifyjs"));t();A();t();var Mg=S(require("bignumber.js"));t();G();t();t();ee();t();ee();t();t();t();X();t();X();t();var Fg=S(require("bignumber.js"));A();X();t();X();t();var up=require("react");G();t();t();X();t();t();var Bg=require("@multiversx/sdk-core"),Ug=S(require("bignumber.js"));X();t();G();t();t();G();var S1=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var gp=require("react");G();ie();t();var Gg=require("react");X();var L1=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];Zt();t();var jn=e=>e.transactions,qo=T(jn,e=>e.signedTransactions),yE=T(jn,e=>e.signTransactionsError),zn=T(jn,e=>e.signTransactionsCancelMessage),Kn=e=>o=>Object.entries(o).reduce((n,[r,s])=>(e(s.status)&&(n[r]=s),n),{}),bE=T(qo,Kn(gs)),SE=T(qo,Kn(Ln)),wE=T(qo,Kn(Nn)),vE=T(qo,Kn(hs)),Xn=T(jn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:w(g({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>Wo(n)))||[]})}),AE=T(qo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});var xo=e=>{let o=e!=null?e:Ns(v.getState());return o.endsWith("/")?o.slice(0,-1):o};t();var $g=S(require("axios"));t();var fp=S(require("swr"));var Yn=({apiEndpoint:e})=>{let o=xo(),n=e?`${o}/${e}`:null;return(0,fp.default)(n,ec)};var Zn=e=>Yn({apiEndpoint:e?`${mo}/${e}?withGuardianInfo=true`:null});A();t();t();t();var hp=require("react"),To=require("react-redux");var Vg={store:v,subscription:tp},mr=(0,hp.createContext)(Vg),rP=(0,To.createStoreHook)(mr),yo=(0,To.createDispatchHook)(mr),R=(0,To.createSelectorHook)(mr);var jo=()=>R(Ke);t();t();t();t();t();var yp=require("@multiversx/sdk-extension-provider"),bp=require("@multiversx/sdk-hw-provider"),Sp=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),wp=require("@multiversx/sdk-opera-provider"),vp=require("@multiversx/sdk-passkey-provider/out"),dr=require("@multiversx/sdk-web-wallet-provider");A();lo();t();var he=require("@multiversx/sdk-web-wallet-iframe-provider/out"),xp=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");ee();t();var Tp=bo;var Hg="/dapp/init",Qe=e=>{switch(e==null?void 0:e.constructor){case dr.WalletProvider:return"wallet";case Tp:return"walletconnectv2";case bp.HWProvider:return"ledger";case yp.ExtensionProvider:return"extension";case vp.PasskeyProvider:return"passkey";case Sp.MetamaskProvider:return"metamask";case wp.OperaProvider:return"opera";case Q.CrossWindowProvider:return"crossWindow";case he.IframeProvider:return"iframe";case bo:return"";default:return"extra"}},Ap=e=>new dr.WalletProvider(`${e}${Hg}`);var pe=e=>`Unable to perform ${e}, Provider not initialized`,bo=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(pe("getAccount"))}setAccount(o){throw new Error(pe(`setAccount: ${o}`))}login(o){throw new Error(pe(`login with options: ${o}`))}logout(o){throw new Error(pe(`logout with options: ${o}`))}getAddress(){throw new Error(pe("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(o,n){throw new Error(pe(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(pe(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(pe(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(pe(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(pe(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(pe(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},_r=new bo;var Ip=_r,kp=null;function Ep(){kp!=null&&(Ip=kp)}function Le(){return Ip||_r}var zo=()=>{let e=Le(),o=Qe(e);return{provider:e,providerType:o}};t();t();t();t();t();t();t();var Ko=S(require("react"));var WP=(0,Ko.createContext)({}),GP=v.getState();t();var Lp=S(require("react"));t();var lr=S(require("react")),qg=S(require("axios"));t();t();t();var Np=require("react"),Cp=require("@multiversx/sdk-web-wallet-provider"),Rp=require("@multiversx/sdk-web-wallet-provider"),Dp=S(require("qs"));A();j();ee();Lt();ie();var jg=C();function Jn(e){let o=R(Xe),n=yo(),{shouldFetchWalletUrlOnLoad:r}=R(er),{search:s}=r?jg:C();(0,Np.useEffect)(()=>{if(s!=null){let p=Dp.default.parse(s.replace("?",""));if(p&&ge in p){let m=String(p[ge]),_=new Cp.WalletProvider(`${o.walletAddress}${Ec}`).getTransactionsFromWalletUrl(s);if(p.status==="cancelled"){n(je({sessionId:m,status:"cancelled"})),e(),qn({removeParams:[...Object.keys(Pt),Rp.WALLET_PROVIDER_CALLBACK_PARAM,ge,"address"],search:s}),n(qe(We));return}if(_.length>0){n(je({sessionId:m,status:"signed",transactions:_.map(h=>Vo(h))}));let[l]=_;_p({transaction:l,search:s})}}}},[s])}lo();t();t();var qp=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),jp=require("@multiversx/sdk-webview-provider/out/WebviewProvider");q();t();q();j();G();t();t();var Gp=S(Wp());var ur=e=>{if(!e||!(0,Gp.default)(e))return null;let o=e.split(".");if(o.length!==4)return null;try{let[n,r,s,p]=o,m=JSON.parse(Ye(p)),_=Ye(n);return{ttl:Number(s),extraInfo:m,origin:_,blockHash:r}}catch(n){return console.error(`Error trying to decode ${e}:`,n),null}};function ef(e){return Object.prototype.toString.call(e)==="[object String]"}var gr=e=>{var n;if(!e||!ef(e))return null;let o=e.split(".");if(o.length!==3)return console.error("Invalid nativeAuthToken. You may be trying to decode a loginToken. Try using decodeLoginToken method instead"),null;try{let[r,s,p]=o,m=Ye(r),_=Ye(s),l=ur(_);if(!l)return{address:m,body:_,signature:p,blockHash:"",origin:"",ttl:0};let h=w(g({},l),{address:m,body:_,signature:p});return(n=l.extraInfo)!=null&&n.timestamp||delete h.extraInfo,h}catch(r){return null}};function $p(e,o=v.dispatch){let n=gr(e);if(n==null)return;let{signature:r,address:s}=n;r&&e&&s&&(o(Hc({loginToken:e,signature:r,nativeAuthToken:e})),o(Ve({address:s,loginMethod:"extra"})))}t();t();t();A();j();function Vp(){v.dispatch(fo())}t();Re();t();j();G();t();t();var tf=require("@multiversx/sdk-core"),rf=S(require("bignumber.js"));A();t();t();var of=(e,o)=>{var s;let r=`${(s=o==null?void 0:o.apiAddress)!=null?s:xo()}/${mo}/${e}?withGuardianInfo=true`;return Ue.get(r)},Hp=(e,o)=>x(void 0,null,function*(){if(!e)return null;try{let{data:n}=yield of(e,o);return n}catch(n){console.error("error fetching configuration for ",e,n)}return null});var So=(e,o)=>Hp(e,o);t();function Qn(e){let o=v.getState(),n=Es(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();var fr=({accountNonce:e,transactionNonce:o})=>o&&o>e?o:e;t();var af=S(require("bignumber.js"));A();var ne=class{constructor(){this.init=()=>x(this,null,function*(){return yield this._provider.init()});this.login=()=>x(this,null,function*(){var n;let o=yield this._provider.login();return{address:(n=o==null?void 0:o.address)!=null?n:""}});this.logout=()=>x(this,null,function*(){return v.dispatch(P()),yield this._provider.logout()});this.relogin=()=>x(this,null,function*(){let o=yield this._provider.relogin();return o?($p(o),Ep(),o):(console.error("Unable to re-login. Missing accessToken."),null)});this.signTransactions=o=>x(this,null,function*(){let n=yield this._provider.signTransactions(o);return n||(Vp(),this._provider.cancelAction(),null)});this.signTransaction=o=>x(this,null,function*(){return yield this._provider.signTransaction(o)});this.signMessage=o=>x(this,null,function*(){return yield this._provider.signMessage(o)});this.cancelAction=()=>x(this,null,function*(){return yield this._provider.cancelAction()});this.isInitialized=()=>this._provider.isInitialized();this.isConnected=()=>this._provider.isConnected();this.sendCustomRequest=o=>x(this,null,function*(){this._provider.sendPostMessage({type:o.request.method,payload:o.request.params})});this.getAddress=(0,qp.providerNotInitializedError)("getAddress");this._provider=jp.WebviewProvider.getInstance({resetStateCallback:()=>v.dispatch(P())})}static getInstance(){return ne._instance||(ne._instance=new ne),ne._instance}getAccount(){var n;let o=this._provider.getAccount();return{address:(n=o==null?void 0:o.address)!=null?n:""}}setAccount(o){this._provider.setAccount(o)}};j();ee();Re();ie();t();t();var zp=(e,o)=>e?Array.isArray(o)?o.some(n=>e.address===n||e.activeGuardianAddress===n):e.address===o||e.activeGuardianAddress===o:!0;t();function Kp(e){let o=Xn(v.getState()),n=Object.keys(e).length===(o==null?void 0:o.transactions.length);return e&&(e&&n)}t();t();var Xp=({latestNonce:e,transactions:o})=>o.length===0?o:o.map((n,r)=>{let s=e+r,p=n.nonce,m=fr({accountNonce:s,transactionNonce:Number(p)});return n.nonce=BigInt(m),n});var et=()=>{let e=R(ze),{data:o}=Zn(e);return n=>x(void 0,null,function*(){let r=o!=null?o:yield So(e),s=Qn(r);return Xp({latestNonce:s,transactions:n})})};t();var Yp=({transactions:e,isGuarded:o})=>o?e.length===0?!1:e.every(n=>Boolean(n.getGuardianSignature().toString("hex"))):!0;t();t();t();Ro();t();var pf=S(require("axios"));t();var df=S(require("axios"));t();var _f=S(require("axios"));t();Ro();t();Ro();function hr(e){return Sc[e]}t();t();t();t();var lf=require("@multiversx/sdk-opera-provider");t();var uf=require("@multiversx/sdk-extension-provider");t();lo();function Zp(n){return x(this,arguments,function*({address:e,walletUrl:o}){try{let r=yield Q.CrossWindowProvider.getInstance().init(),s=Q.CrossWindowProvider.getInstance().setAddress(e).setWalletUrl(o);if(r)return s;console.error("Could not initialise Cross Window Wallet provider")}catch(r){console.error("Unable to login to CrossWindowWalletProvider",r)}return null})}t();t();t();t();var ff=require("@multiversx/sdk-native-auth-client");t();var om=S(require("axios"));t();t();t();function Jp(e){return x(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}t();var Qp=(e,o,n,r=0)=>x(void 0,null,function*(){try{return yield e(...n)}catch(s){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield Jp(o.delay)),yield Qp(e,o,n,r+1)):null}}),em=(e,o={retries:5,delay:500})=>(...n)=>x(void 0,null,function*(){return yield Qp(e,o,n)});var gf=4;var b2=em((e,o,n)=>x(void 0,null,function*(){if(n){let p=Math.floor(Date.now()/1e3);return{hash:yield n(),timestamp:p}}let{data:r}=yield om.default.get(`${e}/${bt}?from=${gf}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[s]=r;return s}));t();t();_n();t();ie();var I2={origin:C().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};t();var hf=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");t();var xf=require("@multiversx/sdk-passkey-provider/out");t();var Af=S(require("react")),kf=require("@fortawesome/free-solid-svg-icons");t();t();var wo=S(require("react")),sm=require("@fortawesome/free-solid-svg-icons"),pm=require("@fortawesome/react-fontawesome");t();var am=S(require("react"));t();var Xo=S(require("react"));t();Bo();var nm=()=>!ke();var yf=()=>x(void 0,null,function*(){return yield Promise.resolve().then(()=>(Tr(),xr))}),bf=()=>(Tr(),Oe(xr)).default,yr=nm();function rm({ssrGlobalImportCallback:e,ssrImportCallback:o,clientImportCallback:n}){let[r,s]=Xo.default.useState(yr?void 0:bf()),[p,m]=Xo.default.useState(yr||n==null?void 0:n()),_=()=>x(this,null,function*(){(e?e():yf()).then(l=>s(l.default)),o==null||o().then(l=>m(l.default))});return(0,Xo.useEffect)(()=>{yr&&_()},[]),{globalStyles:r,styles:p}}function im(e,o){return n=>{let{globalStyles:r,styles:s}=rm({ssrGlobalImportCallback:o.ssrGlobalStyles,ssrImportCallback:o.ssrStyles,clientImportCallback:o.clientStyles});return am.default.createElement(e,w(g({},n),{globalStyles:r!=null?r:{},styles:s!=null?s:{}}))}}var wf=({styles:e})=>wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},wo.default.createElement(pm.FontAwesomeIcon,{icon:sm.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),vf=im(wf,{ssrStyles:()=>Promise.resolve().then(()=>(Sr(),br)),clientStyles:()=>(Sr(),Oe(br)).default});go();$e();t();t();j();var h3=24*60*60;A();ie();var mm=({transactions:e,hasGuardianScreen:o,callbackRoute:n,sessionId:r,walletAddress:s,isGuarded:p})=>{let m=Yp({isGuarded:p,transactions:e}),_=e[0].getChainID().valueOf(),l=e[0].getSender().bech32().toString(),h=hr(_),u=s!=null?s:Ge[h].walletAddress;return{needs2FaSigning:p?!o&&!m&&r:!1,sendTransactionsToGuardian:()=>{let I=Ap(u),te={[ge]:String(r)},{origin:W}=C(),$=window!=null&&window.location?`${W}${n}`:`${n}`,re=Un({callbackUrl:$,urlParams:te});I.guardTransactions(e,{callbackUrl:encodeURIComponent(re)})},guardTransactions:()=>x(void 0,null,function*(){let I=yield Zp({address:l,walletUrl:u});return I==null||I.setShouldShowConsentPopup(!0),yield I==null?void 0:I.guardTransactions(e)})}};t();t();t();var Ef=require("react"),Pf=require("@multiversx/sdk-hw-provider");t();t();ie();t();t();t();ee();t();t();t();ie();t();q();G();t();j();t();j();t();var _m=require("@multiversx/sdk-core");G();Re();t();cn();q();G();t();$t();go();$e();Re();t();var vo=require("react"),wr=require("@multiversx/sdk-extension-provider"),vr=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Ar=require("@multiversx/sdk-passkey-provider/out");lo();j();var lm=()=>{let e=yo(),{provider:o}=zo(),{nonce:n}=jo(),[r,s]=(0,vo.useState)(null),[p,m]=(0,vo.useState)(),[_,l]=(0,vo.useState)([]),h=et(),u=R(Xn),b=R(zn),L=()=>x(void 0,null,function*(){var re,xe;let W=Boolean(u==null?void 0:u.transactions),$=(re=u==null?void 0:u.transactions)!=null?re:[];if(W){let D=(xe=u==null?void 0:u.customTransactionInformation)!=null&&xe.skipUpdateNonces?$:yield h($);l(D)}m(W)});(0,vo.useEffect)(()=>{L()},[u,n]);let k=()=>{s(null),e(qe(null))},I=W=>{k(),te(W)};Jn(I);function te(W){var eo,E,O,V,M,F,H,Z,J,Te,z,ye;let $=o instanceof wr.ExtensionProvider,re=o instanceof Ar.PasskeyProvider,xe=o instanceof Q.CrossWindowProvider,D=o instanceof he.IframeProvider,Zo=o instanceof vr.MetamaskProvider,nt=o instanceof ne;e(fo()),e(In(W)),!(!$&&!xe&&!D&&!re&&!Zo)&&(k(),$&&((E=(eo=wr.ExtensionProvider.getInstance())==null?void 0:eo.cancelAction)==null||E.call(eo)),re&&((V=(O=Ar.PasskeyProvider.getInstance())==null?void 0:O.cancelAction)==null||V.call(O)),Zo&&((F=(M=vr.MetamaskProvider.getInstance())==null?void 0:M.cancelAction)==null||F.call(M)),xe&&((Z=(H=Q.CrossWindowProvider.getInstance())==null?void 0:H.cancelAction)==null||Z.call(H)),D&&((Te=(J=he.IframeProvider.getInstance())==null?void 0:J.cancelAction)==null||Te.call(J)),nt&&((ye=(z=ne.getInstance())==null?void 0:z.cancelAction)==null||ye.call(z)))}return{error:r,canceledTransactionsMessage:b,clearTransactionStatusMessage:k,onAbort:I,setError:s,hasTransactions:p,transactionsToSign:u&&w(g({},u),{transactions:_})}};var Lf=()=>{let e=yo(),o=(0,Yo.useRef)("/"),{provider:n}=zo(),r=R(Rs),s=Qe(n),p=(0,Yo.useRef)(!1),m=et(),{isGuarded:_,address:l}=jo(),h=R(zn),{transactionsToSign:u,error:b,setError:L,hasTransactions:k,onAbort:I,clearTransactionStatusMessage:te}=lm();Jn(I);let W=(0,um.default)(u==null?void 0:u.transactions.map(E=>E.sender.toBech32()).filter(E=>E)),$=W==null?void 0:W[0],{data:re}=Zn(!$||$===l?null:$),xe=E=>{var J,Te,z,ye,oo,Ao,De,U,se,ko,no,Jo;let O=n instanceof kr.ExtensionProvider,V=n instanceof Er.PasskeyProvider,M=n instanceof Q.CrossWindowProvider,F=n instanceof he.IframeProvider,H=n instanceof Ir.MetamaskProvider,Z=n instanceof ne;e(fo()),e(In(E)),p.current=!1,!(!O&&!M&&!F&&!V&&!H)&&(te(),O&&((Te=(J=kr.ExtensionProvider.getInstance())==null?void 0:J.cancelAction)==null||Te.call(J)),V&&((ye=(z=Er.PasskeyProvider.getInstance())==null?void 0:z.cancelAction)==null||ye.call(z)),H&&((Ao=(oo=Ir.MetamaskProvider.getInstance())==null?void 0:oo.cancelAction)==null||Ao.call(oo)),M&&((U=(De=Q.CrossWindowProvider.getInstance())==null?void 0:De.cancelAction)==null||U.call(De)),F&&((ko=(se=he.IframeProvider.getInstance())==null?void 0:se.cancelAction)==null||ko.call(se)),Z&&((Jo=(no=ne.getInstance())==null?void 0:no.cancelAction)==null||Jo.call(no)))},D=(E,O)=>{s==="walletconnectv2"&&(E=We);let M=E.includes(We);if(xe(O),M){e(qe(We));return}L(E)},Zo=(E,O,V="")=>{let M={[ge]:O},F=V;if(window!=null&&window.location){let{search:Z,origin:J}=C(),Te=new URLSearchParams(Z);F=`${J}${V}`,Te.forEach((z,ye)=>{M[ye]=z})}let H=Un({callbackUrl:F,urlParams:M});n.signTransactions(E,{callbackUrl:encodeURIComponent(H)})},nt=()=>x(void 0,null,function*(){var Ao,De;if(p.current||!u)return;let{sessionId:E,transactions:O,callbackRoute:V,customTransactionInformation:M}=u,{redirectAfterSign:F}=M,H=Gt(),Z=V||H,J=H.includes(Z),Te=F&&!J;try{if(!(yield(Ao=n==null?void 0:n.init)==null?void 0:Ao.call(n)))return}catch(U){let se=(U==null?void 0:U.message)||U||At;console.error(se),D(At,E);return}let z=!M.skipGuardian,ye=M.hasConsentPopup,oo=n instanceof Q.CrossWindowProvider;try{p.current=!0,oo&&ye&&n.setShouldShowConsentPopup(!0);let U=(De=yield n.signTransactions(_&&z?O==null?void 0:O.map(Ne=>(Ne.version=ot.TransactionVersion.withTxOptions().valueOf(),Ne.options=ot.TransactionOptions.withOptions({guarded:!0}).valueOf(),Ne)):O))!=null?De:[];if(p.current=!1,!Kp(U))return;let ko=Object.values(U).map(Ne=>Vo(Ne)),{needs2FaSigning:no,guardTransactions:Jo}=mm({transactions:U,sessionId:E,callbackRoute:V,isGuarded:_&&z,walletAddress:r});if(no){let Ne=yield Jo();ko=Ne?Ne.map(gm=>Vo(gm)):[]}let Pr={sessionId:E,transactions:ko,status:"signed"};Te&&(Pr.redirectRoute=Z),e(je(Pr))}catch(U){p.current=!1;let se=(U==null?void 0:U.message)||U||tc;console.error(se),e(je({sessionId:E,status:"cancelled"})),D(se.includes("cancel")?We:se,E)}}),eo=()=>x(void 0,null,function*(){if(!u)return;te();let{sessionId:E,transactions:O,callbackRoute:V,customTransactionInformation:M}=u;if(!n){console.error(rc);return}if(W.length>1)throw new Error("Multiple senders are not allowed");if($&&$!==l&&!zp(re,l))return console.error(kt),D(kt);o.current=V||C().pathname;try{let F=s==="wallet",H=M.skipUpdateNonces?O:yield m(O);if(F)return Zo(H,E,V);nt()}catch(F){let Z=(F==null?void 0:F.message)||nc;console.error(Z),D(Z,E),e(je({sessionId:E,status:"cancelled"})),console.error(Z,F)}});return(0,Yo.useEffect)(()=>{k?eo():p.current=!1},[u,k,re]),{error:b,canceledTransactionsMessage:h,onAbort:I,hasTransactions:k,callbackRoute:o.current,sessionId:u==null?void 0:u.sessionId,transactions:u==null?void 0:u.transactions}};0&&(module.exports={useSignTransactions});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
