// @multiversx/sdk-dapp/__commonjs/hooks/transactions/useSignTransactionsWithLedger.js
"use strict";var bp=Object.create;var Ro=Object.defineProperty,Sp=Object.defineProperties,wp=Object.getOwnPropertyDescriptor,vp=Object.getOwnPropertyDescriptors,Ap=Object.getOwnPropertyNames,on=Object.getOwnPropertySymbols,kp=Object.getPrototypeOf,tt=Object.prototype.hasOwnProperty,Or=Object.prototype.propertyIsEnumerable;var Mr=(e,o,n)=>o in e?Ro(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,g=(e,o)=>{for(var n in o||(o={}))tt.call(o,n)&&Mr(e,n,o[n]);if(on)for(var n of on(o))Or.call(o,n)&&Mr(e,n,o[n]);return e},S=(e,o)=>Sp(e,vp(o));var lo=(e,o)=>{var n={};for(var r in e)tt.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&on)for(var r of on(e))o.indexOf(r)<0&&Or.call(e,r)&&(n[r]=e[r]);return n};var T=(e,o)=>()=>(e&&(o=e(e=0)),o);var Y=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),Ie=(e,o)=>{for(var n in o)Ro(e,n,{get:o[n],enumerable:!0})},Fr=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of Ap(o))!tt.call(e,a)&&a!==n&&Ro(e,a,{get:()=>o[a],enumerable:!(r=wp(o,a))||r.enumerable});return e};var w=(e,o,n)=>(n=e!=null?bp(kp(e)):{},Fr(o||!e||!e.__esModule?Ro(n,"default",{value:e,enumerable:!0}):n,e)),je=e=>Fr(Ro({},"__esModule",{value:!0}),e);var x=(e,o,n)=>new Promise((r,a)=>{var p=_=>{try{d(n.next(_))}catch(u){a(u)}},m=_=>{try{d(n.throw(_))}catch(u){a(u)}},d=_=>_.done?r(_.value):Promise.resolve(_.value).then(p,m);d((n=n.apply(e,o)).next())});var Wr=Y(nn=>{"use strict";t();nn.byteLength=Ep;nn.toByteArray=Lp;nn.fromByteArray=Rp;var ue=[],ee=[],Ip=typeof Uint8Array!="undefined"?Uint8Array:Array,rt="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ze=0,Br=rt.length;ze<Br;++ze)ue[ze]=rt[ze],ee[rt.charCodeAt(ze)]=ze;var ze,Br;ee["-".charCodeAt(0)]=62;ee["_".charCodeAt(0)]=63;function Ur(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Ep(e){var o=Ur(e),n=o[0],r=o[1];return(n+r)*3/4-r}function Pp(e,o,n){return(o+n)*3/4-n}function Lp(e){var o,n=Ur(e),r=n[0],a=n[1],p=new Ip(Pp(e,r,a)),m=0,d=a>0?r-4:r,_;for(_=0;_<d;_+=4)o=ee[e.charCodeAt(_)]<<18|ee[e.charCodeAt(_+1)]<<12|ee[e.charCodeAt(_+2)]<<6|ee[e.charCodeAt(_+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return a===2&&(o=ee[e.charCodeAt(_)]<<2|ee[e.charCodeAt(_+1)]>>4,p[m++]=o&255),a===1&&(o=ee[e.charCodeAt(_)]<<10|ee[e.charCodeAt(_+1)]<<4|ee[e.charCodeAt(_+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function Np(e){return ue[e>>18&63]+ue[e>>12&63]+ue[e>>6&63]+ue[e&63]}function Cp(e,o,n){for(var r,a=[],p=o;p<n;p+=3)r=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),a.push(Np(r));return a.join("")}function Rp(e){for(var o,n=e.length,r=n%3,a=[],p=16383,m=0,d=n-r;m<d;m+=p)a.push(Cp(e,m,m+p>d?d:m+p));return r===1?(o=e[n-1],a.push(ue[o>>2]+ue[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(ue[o>>10]+ue[o>>4&63]+ue[o<<2&63]+"=")),a.join("")}});var Gr=Y(at=>{t();at.read=function(e,o,n,r,a){var p,m,d=a*8-r-1,_=(1<<d)-1,u=_>>1,f=-7,h=n?a-1:0,b=n?-1:1,v=e[o+h];for(h+=b,p=v&(1<<-f)-1,v>>=-f,f+=d;f>0;p=p*256+e[o+h],h+=b,f-=8);for(m=p&(1<<-f)-1,p>>=-f,f+=r;f>0;m=m*256+e[o+h],h+=b,f-=8);if(p===0)p=1-u;else{if(p===_)return m?NaN:(v?-1:1)*(1/0);m=m+Math.pow(2,r),p=p-u}return(v?-1:1)*m*Math.pow(2,p-r)};at.write=function(e,o,n,r,a,p){var m,d,_,u=p*8-a-1,f=(1<<u)-1,h=f>>1,b=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,v=r?0:p-1,k=r?1:-1,F=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(d=isNaN(o)?1:0,m=f):(m=Math.floor(Math.log(o)/Math.LN2),o*(_=Math.pow(2,-m))<1&&(m--,_*=2),m+h>=1?o+=b/_:o+=b*Math.pow(2,1-h),o*_>=2&&(m++,_/=2),m+h>=f?(d=0,m=f):m+h>=1?(d=(o*_-1)*Math.pow(2,a),m=m+h):(d=o*Math.pow(2,h-1)*Math.pow(2,a),m=0));a>=8;e[n+v]=d&255,v+=k,d/=256,a-=8);for(m=m<<a|d,u+=a;u>0;e[n+v]=m&255,v+=k,m/=256,u-=8);e[n+v-k]|=F*128}});var ea=Y(go=>{"use strict";t();var it=Wr(),uo=Gr(),$r=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;go.Buffer=l;go.SlowBuffer=Up;go.INSPECT_MAX_BYTES=50;var tn=2147483647;go.kMaxLength=tn;l.TYPED_ARRAY_SUPPORT=Dp();!l.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Dp(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(l.prototype,"parent",{enumerable:!0,get:function(){if(!!l.isBuffer(this))return this.buffer}});Object.defineProperty(l.prototype,"offset",{enumerable:!0,get:function(){if(!!l.isBuffer(this))return this.byteOffset}});function Ee(e){if(e>tn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,l.prototype),o}function l(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return mt(e)}return jr(e,o,n)}l.poolSize=8192;function jr(e,o,n){if(typeof e=="string")return Op(e,o);if(ArrayBuffer.isView(e))return Fp(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(ge(e,ArrayBuffer)||e&&ge(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(ge(e,SharedArrayBuffer)||e&&ge(e.buffer,SharedArrayBuffer)))return st(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return l.from(r,o,n);var a=Bp(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return l.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}l.from=function(e,o,n){return jr(e,o,n)};Object.setPrototypeOf(l.prototype,Uint8Array.prototype);Object.setPrototypeOf(l,Uint8Array);function zr(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Mp(e,o,n){return zr(e),e<=0?Ee(e):o!==void 0?typeof n=="string"?Ee(e).fill(o,n):Ee(e).fill(o):Ee(e)}l.alloc=function(e,o,n){return Mp(e,o,n)};function mt(e){return zr(e),Ee(e<0?0:dt(e)|0)}l.allocUnsafe=function(e){return mt(e)};l.allocUnsafeSlow=function(e){return mt(e)};function Op(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!l.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=qr(e,o)|0,r=Ee(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function ct(e){for(var o=e.length<0?0:dt(e.length)|0,n=Ee(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Fp(e){if(ge(e,Uint8Array)){var o=new Uint8Array(e);return st(o.buffer,o.byteOffset,o.byteLength)}return ct(e)}function st(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,l.prototype),r}function Bp(e){if(l.isBuffer(e)){var o=dt(e.length)|0,n=Ee(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||_t(e.length)?Ee(0):ct(e);if(e.type==="Buffer"&&Array.isArray(e.data))return ct(e.data)}function dt(e){if(e>=tn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+tn.toString(16)+" bytes");return e|0}function Up(e){return+e!=e&&(e=0),l.alloc(+e)}l.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==l.prototype};l.compare=function(o,n){if(ge(o,Uint8Array)&&(o=l.from(o,o.offset,o.byteLength)),ge(n,Uint8Array)&&(n=l.from(n,n.offset,n.byteLength)),!l.isBuffer(o)||!l.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,p=0,m=Math.min(r,a);p<m;++p)if(o[p]!==n[p]){r=o[p],a=n[p];break}return r<a?-1:a<r?1:0};l.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};l.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return l.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=l.allocUnsafe(n),p=0;for(r=0;r<o.length;++r){var m=o[r];if(ge(m,Uint8Array))p+m.length>a.length?l.from(m).copy(a,p):Uint8Array.prototype.set.call(a,m,p);else if(l.isBuffer(m))m.copy(a,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return a};function qr(e,o){if(l.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||ge(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return pt(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Qr(e).length;default:if(a)return r?-1:pt(e).length;o=(""+o).toLowerCase(),a=!0}}l.byteLength=qr;function Wp(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return Yp(this,o,n);case"utf8":case"utf-8":return Xr(this,o,n);case"ascii":return Kp(this,o,n);case"latin1":case"binary":return Xp(this,o,n);case"base64":return zp(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Zp(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}l.prototype._isBuffer=!0;function qe(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}l.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)qe(this,n,n+1);return this};l.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)qe(this,n,n+3),qe(this,n+1,n+2);return this};l.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)qe(this,n,n+7),qe(this,n+1,n+6),qe(this,n+2,n+5),qe(this,n+3,n+4);return this};l.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?Xr(this,0,o):Wp.apply(this,arguments)};l.prototype.toLocaleString=l.prototype.toString;l.prototype.equals=function(o){if(!l.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:l.compare(this,o)===0};l.prototype.inspect=function(){var o="",n=go.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};$r&&(l.prototype[$r]=l.prototype.inspect);l.prototype.compare=function(o,n,r,a,p){if(ge(o,Uint8Array)&&(o=l.from(o,o.offset,o.byteLength)),!l.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),p===void 0&&(p=this.length),n<0||r>o.length||a<0||p>this.length)throw new RangeError("out of range index");if(a>=p&&n>=r)return 0;if(a>=p)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,p>>>=0,this===o)return 0;for(var m=p-a,d=r-n,_=Math.min(m,d),u=this.slice(a,p),f=o.slice(n,r),h=0;h<_;++h)if(u[h]!==f[h]){m=u[h],d=f[h];break}return m<d?-1:d<m?1:0};function Kr(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,_t(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=l.from(o,r)),l.isBuffer(o))return o.length===0?-1:Vr(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Vr(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function Vr(e,o,n,r,a){var p=1,m=e.length,d=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,d/=2,n/=2}function _(v,k){return p===1?v[k]:v.readUInt16BE(k*p)}var u;if(a){var f=-1;for(u=n;u<m;u++)if(_(e,u)===_(o,f===-1?0:u-f)){if(f===-1&&(f=u),u-f+1===d)return f*p}else f!==-1&&(u-=u-f),f=-1}else for(n+d>m&&(n=m-d),u=n;u>=0;u--){for(var h=!0,b=0;b<d;b++)if(_(e,u+b)!==_(o,b)){h=!1;break}if(h)return u}return-1}l.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};l.prototype.indexOf=function(o,n,r){return Kr(this,o,n,r,!0)};l.prototype.lastIndexOf=function(o,n,r){return Kr(this,o,n,r,!1)};function Gp(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var p=o.length;r>p/2&&(r=p/2);for(var m=0;m<r;++m){var d=parseInt(o.substr(m*2,2),16);if(_t(d))return m;e[n+m]=d}return m}function $p(e,o,n,r){return rn(pt(o,e.length-n),e,n,r)}function Vp(e,o,n,r){return rn(em(o),e,n,r)}function Hp(e,o,n,r){return rn(Qr(o),e,n,r)}function jp(e,o,n,r){return rn(om(o,e.length-n),e,n,r)}l.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-n;if((r===void 0||r>p)&&(r=p),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var m=!1;;)switch(a){case"hex":return Gp(this,o,n,r);case"utf8":case"utf-8":return $p(this,o,n,r);case"ascii":case"latin1":case"binary":return Vp(this,o,n,r);case"base64":return Hp(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return jp(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),m=!0}};l.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function zp(e,o,n){return o===0&&n===e.length?it.fromByteArray(e):it.fromByteArray(e.slice(o,n))}function Xr(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var p=e[a],m=null,d=p>239?4:p>223?3:p>191?2:1;if(a+d<=n){var _,u,f,h;switch(d){case 1:p<128&&(m=p);break;case 2:_=e[a+1],(_&192)===128&&(h=(p&31)<<6|_&63,h>127&&(m=h));break;case 3:_=e[a+1],u=e[a+2],(_&192)===128&&(u&192)===128&&(h=(p&15)<<12|(_&63)<<6|u&63,h>2047&&(h<55296||h>57343)&&(m=h));break;case 4:_=e[a+1],u=e[a+2],f=e[a+3],(_&192)===128&&(u&192)===128&&(f&192)===128&&(h=(p&15)<<18|(_&63)<<12|(u&63)<<6|f&63,h>65535&&h<1114112&&(m=h))}}m===null?(m=65533,d=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),a+=d}return qp(r)}var Hr=4096;function qp(e){var o=e.length;if(o<=Hr)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=Hr));return n}function Kp(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function Xp(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function Yp(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",p=o;p<n;++p)a+=nm[e[p]];return a}function Zp(e,o,n){for(var r=e.slice(o,n),a="",p=0;p<r.length-1;p+=2)a+=String.fromCharCode(r[p]+r[p+1]*256);return a}l.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,l.prototype),a};function B(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}l.prototype.readUintLE=l.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var a=this[o],p=1,m=0;++m<n&&(p*=256);)a+=this[o+m]*p;return a};l.prototype.readUintBE=l.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var a=this[o+--n],p=1;n>0&&(p*=256);)a+=this[o+--n]*p;return a};l.prototype.readUint8=l.prototype.readUInt8=function(o,n){return o=o>>>0,n||B(o,1,this.length),this[o]};l.prototype.readUint16LE=l.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||B(o,2,this.length),this[o]|this[o+1]<<8};l.prototype.readUint16BE=l.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||B(o,2,this.length),this[o]<<8|this[o+1]};l.prototype.readUint32LE=l.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||B(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};l.prototype.readUint32BE=l.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||B(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};l.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var a=this[o],p=1,m=0;++m<n&&(p*=256);)a+=this[o+m]*p;return p*=128,a>=p&&(a-=Math.pow(2,8*n)),a};l.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||B(o,n,this.length);for(var a=n,p=1,m=this[o+--a];a>0&&(p*=256);)m+=this[o+--a]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*n)),m};l.prototype.readInt8=function(o,n){return o=o>>>0,n||B(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};l.prototype.readInt16LE=function(o,n){o=o>>>0,n||B(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};l.prototype.readInt16BE=function(o,n){o=o>>>0,n||B(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};l.prototype.readInt32LE=function(o,n){return o=o>>>0,n||B(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};l.prototype.readInt32BE=function(o,n){return o=o>>>0,n||B(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};l.prototype.readFloatLE=function(o,n){return o=o>>>0,n||B(o,4,this.length),uo.read(this,o,!0,23,4)};l.prototype.readFloatBE=function(o,n){return o=o>>>0,n||B(o,4,this.length),uo.read(this,o,!1,23,4)};l.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||B(o,8,this.length),uo.read(this,o,!0,52,8)};l.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||B(o,8,this.length),uo.read(this,o,!1,52,8)};function z(e,o,n,r,a,p){if(!l.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<p)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}l.prototype.writeUintLE=l.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var p=Math.pow(2,8*r)-1;z(this,o,n,r,p,0)}var m=1,d=0;for(this[n]=o&255;++d<r&&(m*=256);)this[n+d]=o/m&255;return n+r};l.prototype.writeUintBE=l.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var p=Math.pow(2,8*r)-1;z(this,o,n,r,p,0)}var m=r-1,d=1;for(this[n+m]=o&255;--m>=0&&(d*=256);)this[n+m]=o/d&255;return n+r};l.prototype.writeUint8=l.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,1,255,0),this[n]=o&255,n+1};l.prototype.writeUint16LE=l.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};l.prototype.writeUint16BE=l.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};l.prototype.writeUint32LE=l.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};l.prototype.writeUint32BE=l.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};l.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var p=Math.pow(2,8*r-1);z(this,o,n,r,p-1,-p)}var m=0,d=1,_=0;for(this[n]=o&255;++m<r&&(d*=256);)o<0&&_===0&&this[n+m-1]!==0&&(_=1),this[n+m]=(o/d>>0)-_&255;return n+r};l.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var p=Math.pow(2,8*r-1);z(this,o,n,r,p-1,-p)}var m=r-1,d=1,_=0;for(this[n+m]=o&255;--m>=0&&(d*=256);)o<0&&_===0&&this[n+m+1]!==0&&(_=1),this[n+m]=(o/d>>0)-_&255;return n+r};l.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};l.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};l.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};l.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};l.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||z(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function Yr(e,o,n,r,a,p){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function Zr(e,o,n,r,a){return o=+o,n=n>>>0,a||Yr(e,o,n,4,34028234663852886e22,-34028234663852886e22),uo.write(e,o,n,r,23,4),n+4}l.prototype.writeFloatLE=function(o,n,r){return Zr(this,o,n,!0,r)};l.prototype.writeFloatBE=function(o,n,r){return Zr(this,o,n,!1,r)};function Jr(e,o,n,r,a){return o=+o,n=n>>>0,a||Yr(e,o,n,8,17976931348623157e292,-17976931348623157e292),uo.write(e,o,n,r,52,8),n+8}l.prototype.writeDoubleLE=function(o,n,r){return Jr(this,o,n,!0,r)};l.prototype.writeDoubleBE=function(o,n,r){return Jr(this,o,n,!1,r)};l.prototype.copy=function(o,n,r,a){if(!l.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var p=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),p};l.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!l.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var p=o.charCodeAt(0);(a==="utf8"&&p<128||a==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var d=l.isBuffer(o)?o:l.from(o,a),_=d.length;if(_===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=d[m%_]}return this};var Jp=/[^+/0-9A-Za-z-_]/g;function Qp(e){if(e=e.split("=")[0],e=e.trim().replace(Jp,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function pt(e,o){o=o||1/0;for(var n,r=e.length,a=null,p=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&p.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&p.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&p.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;p.push(n)}else if(n<2048){if((o-=2)<0)break;p.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;p.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;p.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return p}function em(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function om(e,o){for(var n,r,a,p=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,a=n%256,p.push(a),p.push(r);return p}function Qr(e){return it.toByteArray(Qp(e))}function rn(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function ge(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function _t(e){return e!==e}var nm=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var aa=Y((hl,ra)=>{t();var O=ra.exports={},fe,he;function lt(){throw new Error("setTimeout has not been defined")}function ut(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?fe=setTimeout:fe=lt}catch(e){fe=lt}try{typeof clearTimeout=="function"?he=clearTimeout:he=ut}catch(e){he=ut}})();function oa(e){if(fe===setTimeout)return setTimeout(e,0);if((fe===lt||!fe)&&setTimeout)return fe=setTimeout,setTimeout(e,0);try{return fe(e,0)}catch(o){try{return fe.call(null,e,0)}catch(n){return fe.call(this,e,0)}}}function tm(e){if(he===clearTimeout)return clearTimeout(e);if((he===ut||!he)&&clearTimeout)return he=clearTimeout,clearTimeout(e);try{return he(e)}catch(o){try{return he.call(null,e)}catch(n){return he.call(this,e)}}}var Pe=[],fo=!1,Ke,an=-1;function rm(){!fo||!Ke||(fo=!1,Ke.length?Pe=Ke.concat(Pe):an=-1,Pe.length&&na())}function na(){if(!fo){var e=oa(rm);fo=!0;for(var o=Pe.length;o;){for(Ke=Pe,Pe=[];++an<o;)Ke&&Ke[an].run();an=-1,o=Pe.length}Ke=null,fo=!1,tm(e)}}O.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];Pe.push(new ta(e,o)),Pe.length===1&&!fo&&oa(na)};function ta(e,o){this.fun=e,this.array=o}ta.prototype.run=function(){this.fun.apply(null,this.array)};O.title="browser";O.browser=!0;O.env={};O.argv=[];O.version="";O.versions={};function Le(){}O.on=Le;O.addListener=Le;O.once=Le;O.off=Le;O.removeListener=Le;O.removeAllListeners=Le;O.emit=Le;O.prependListener=Le;O.prependOnceListener=Le;O.listeners=function(e){return[]};O.binding=function(e){throw new Error("process.binding is not supported")};O.cwd=function(){return"/"};O.chdir=function(e){throw new Error("process.chdir is not supported")};O.umask=function(){return 0}});var i,s,am,c,t=T(()=>{i=w(ea()),s=w(aa()),am=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object),c=am});var ia=T(()=>{"use strict";t()});var cn,ca,gt=T(()=>{"use strict";t();cn="Transaction canceled",ca="You cannot sign transactions from a different account"});var sa=T(()=>{"use strict";t()});var ft,pa=T(()=>{"use strict";t();ft=(r=>(r.SetGuardian="SetGuardian",r.GuardAccount="GuardAccount",r.UnGuardAccount="UnGuardAccount",r))(ft||{})});var ma=T(()=>{"use strict";t()});var ht,xt=T(()=>{"use strict";t();ht={36864:{code:"codeSuccess",message:"Success"},27013:{code:"ERR_USER_DENIED",message:"Rejected by user"},27904:{code:"ERR_UNKNOWN_INSTRUCTION",message:"Unknown instruction"},28160:{code:"ERR_WRONG_CLA",message:"Wrong CLA"},28161:{code:"ERR_INVALID_ARGUMENTS",message:"Invalid arguments"},28162:{code:"ERR_INVALID_MESSAGE",message:"Invalid message"},28163:{code:"ERR_INVALID_P1",message:"Invalid P1"},28164:{code:"ERR_MESSAGE_TOO_LONG",message:"Message too long"},28165:{code:"ERR_RECEIVER_TOO_LONG",message:"Receiver too long"},28166:{code:"ERR_AMOUNT_TOO_LONG",message:"Amount too long"},28167:{code:"ERR_CONTRACT_DATA_DISABLED",message:"Contract data disabled in app options"},28168:{code:"ERR_MESSAGE_INCOMPLETE",message:"Message incomplete"},28176:{code:"ERR_SIGNATURE_FAILED",message:"Signature failed"},28169:{code:"ERR_WRONG_TX_VERSION",message:"Wrong TX version"},28170:{code:"ERR_NONCE_TOO_LONG",message:"Nonce too long"},28171:{code:"ERR_INVALID_AMOUNT",message:"Invalid amount"},28172:{code:"ERR_INVALID_FEE",message:"Invalid fee"},28173:{code:"ERR_PRETTY_FAILED",message:"Pretty failed"},28174:{code:"ERR_DATA_TOO_LONG",message:"Data too long"},28175:{code:"ERR_WRONG_TX_OPTIONS",message:"Invalid transaction options"},28177:{code:"ERR_SIGN_TX_DEPRECATED",message:"Regular transaction signing is deprecated in this version. Use hash signing."}}});var da=T(()=>{"use strict";t()});var Tt,yt=T(()=>{"use strict";t();Tt=(a=>(a.signature="signature",a.sessionId="sessionId",a.status="status",a.address="address",a))(Tt||{})});var _a=T(()=>{"use strict";t()});var la=T(()=>{"use strict";t()});var Ne,ho,Do=T(()=>{"use strict";t();Ne=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),ho=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var ua,Pl,ga,Ll,oe=T(()=>{"use strict";t();Do();ua=(n=>(n.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",n.reloginRequest="RELOGIN_REQUEST",n))(ua||{}),Pl=g(g({},ho.WindowProviderRequestEnums),ua),ga=(o=>(o.reloginResponse="RELOGIN_RESPONSE",o))(ga||{}),Ll=g(g({},ho.WindowProviderResponseEnums),ga)});var fa=T(()=>{"use strict";t()});var ha=T(()=>{"use strict";t()});var q=T(()=>{"use strict";t()});var xa=T(()=>{"use strict";t()});var Ta=T(()=>{"use strict";t()});var ya=T(()=>{"use strict";t()});var G=T(()=>{"use strict";t();yt();_a();la();oe();fa();ha();q();xa();Ta();ya()});var Xe,ba,ql,Sa,Kl,wa,Xl,Yl,va,bt,Mo=T(()=>{"use strict";t();G();Xe={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:ba,egldLabel:ql}=Xe["devnet"],{chainId:Sa,egldLabel:Kl}=Xe["testnet"],{chainId:wa,egldLabel:Xl}=Xe["mainnet"],Yl={["devnet"]:ba,["testnet"]:Sa,["mainnet"]:wa},va={[ba]:"devnet",[Sa]:"testnet",[wa]:"mainnet"},bt=0});var St=T(()=>{"use strict";t()});var xe,Aa=T(()=>{"use strict";t();xe=require("@multiversx/sdk-web-wallet-provider")});var pe,sn=T(()=>{"use strict";t();pe=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var ka,Ia,wt,nu,tu,Ea=T(()=>{"use strict";t();sn();wt=String((Ia=(ka=pe.safeWindow)==null?void 0:ka.navigator)==null?void 0:Ia.userAgent),nu=/^((?!chrome|android).)*safari/i.test(wt),tu=/firefox/i.test(wt)&&/windows/i.test(wt)});var pn,mn,Pa,La,Na,We,dn,_n,Ca,xo,P=T(()=>{"use strict";t();ia();gt();sa();pa();ma();xt();da();Mo();St();Aa();Ea();pn=5e4,mn=1e9,Pa=1,La=1,Na="/dapp/init",We="signSession",dn="sdk-dapp-version",_n="logout",Ca="login",xo="0"});var Oo,ln=T(()=>{"use strict";t();Oo=()=>Date.now()/1e3});var Ra=T(()=>{"use strict";t()});var Da=T(()=>{"use strict";t()});var Ma=T(()=>{"use strict";t();ln();Ra();Da()});var vt={};Ie(vt,{clear:()=>pm,getItem:()=>cm,localStorageKeys:()=>ne,removeItem:()=>sm,setItem:()=>im});var ne,un,im,cm,sm,pm,Ye=T(()=>{"use strict";t();Ma();ne={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},un=typeof localStorage!="undefined",im=({key:e,data:o,expires:n})=>{!un||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},cm=e=>{if(!un)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Oo()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},sm=e=>{!un||localStorage.removeItem(String(e))},pm=()=>{!un||localStorage.clear()}});var At={};Ie(At,{clear:()=>Ua,getItem:()=>Fa,removeItem:()=>Ba,setItem:()=>Oa,storage:()=>mm});var Oa,Fa,Ba,Ua,mm,Wa=T(()=>{"use strict";t();Oa=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Fa=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Ba=e=>sessionStorage.removeItem(String(e)),Ua=()=>sessionStorage.clear(),mm={setItem:Oa,getItem:Fa,removeItem:Ba,clear:Ua}});var me,To=T(()=>{"use strict";t();Ye();Wa();me={session:At,local:vt}});var kt,L,Ze,j=T(()=>{"use strict";t();kt=require("@reduxjs/toolkit");P();L=(0,kt.createAction)(_n),Ze=(0,kt.createAction)(Ca,e=>({payload:e}))});var Et,Ga,$a,gn,It,Va,Pt,Du,Lt,Mu,Ou,Fu,Bu,Uu,Wu,Gu,$u,fn,hn=T(()=>{"use strict";t();Et=require("@multiversx/sdk-core"),Ga=require("@reduxjs/toolkit"),$a=require("redux-persist");P();To();Ye();j();gn={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:xo},It={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":gn},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Va=(0,Ga.createSlice)({name:"accountInfoSlice",initialState:It,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=n?new Et.Address(n).hex():""},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:gn},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}},setWebsocketBatchEvent:(e,o)=>{e.websocketBatchEvent={timestamp:Date.now(),data:o.payload}}},extraReducers:e=>{e.addCase(L,()=>(me.local.removeItem(ne.loginExpiresAt),It)),e.addCase(Ze,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new Et.Address(r).toHex()}),e.addCase($a.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:a,shard:p,accounts:m,publicKey:d}=r;o.address=a,o.shard=p;let _=m&&a in m;o.accounts=_?m:It.accounts,o.publicKey=d})}}),{setAccount:Pt,setAddress:Du,setAccountNonce:Lt,setAccountShard:Mu,setLedgerAccount:Ou,updateLedgerAccount:Fu,setWalletConnectAccount:Bu,setIsAccountLoading:Uu,setAccountLoadingError:Wu,setWebsocketEvent:Gu,setWebsocketBatchEvent:$u}=Va.actions,fn=Va.reducer});function Fo(){return new Date().setHours(new Date().getHours()+24)}function Bo(e){me.local.setItem({key:ne.loginExpiresAt,data:e,expires:e})}var Nt=T(()=>{"use strict";t();To();Ye()});var ja,Ha,za,Ju,Qu,eg,qa,og,ng,Ka,tg,rg,ag,xn,Tn=T(()=>{"use strict";t();ja=require("@reduxjs/toolkit");Nt();oe();j();Ha={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},za=(0,ja.createSlice)({name:"loginInfoSlice",initialState:Ha,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},setWebviewLogin:(e,o)=>{e.webviewLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload},setIsWalletConnectV2Initialized:(e,o)=>{e.isWalletConnectV2Initialized=o.payload}},extraReducers:e=>{e.addCase(L,()=>Ha),e.addCase(Ze,(o,n)=>{o.isLoginSessionInvalid=!1,o.loginMethod=n.payload.loginMethod,o.iframeLoginType=n.payload.iframeLoginType,Bo(Fo())})}}),{setLoginMethod:Ju,setWalletConnectLogin:Qu,setLedgerLogin:eg,setTokenLogin:qa,setTokenLoginSignature:og,setWalletLogin:ng,invalidateLoginSession:Ka,setLogoutRoute:tg,setIsWalletConnectV2Initialized:rg,setWebviewLogin:ag}=za.actions,xn=za.reducer});var Ya,Xa,Za,pg,dm,mg,dg,yn,bn=T(()=>{"use strict";t();Ya=require("@reduxjs/toolkit");j();Xa={},Za=(0,Ya.createSlice)({name:"modalsSlice",initialState:Xa,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(L,()=>Xa)}}),{setTxSubmittedModal:pg,setNotificationModal:dm,clearTxSubmittedModal:mg,clearNotificationModal:dg}=Za.actions,yn=Za.reducer});var Ce,Uo=T(()=>{"use strict";t();Ce=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var R,te=T(()=>{"use strict";t();Uo();R=()=>{if(!Ce())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:o,hash:n,origin:r,href:a,search:p}}=window;return{pathname:o,hash:n,origin:r,href:a,search:p}}});var Ct,Ja=T(()=>{"use strict";t();te();Ct=(e="")=>{let o=R(),n=e.startsWith("http")||e.startsWith("www.");return!o.origin||e.startsWith(o.origin)||n?e:`${o.origin}/${e.replace("/","")}`}});var Qa=T(()=>{"use strict";t();Qe()});var ei,oi=T(()=>{"use strict";t();sn();ei=e=>{if(!pe.safeWindow.location)return;if(!new URL(pe.safeWindow.location.href).protocol.startsWith("http"))return pe.safeWindow.location.reload();pe.safeWindow.location.assign(e)}});var Qe=T(()=>{"use strict";t();Ja();Qa();te();oi()});var ni,ti,Wo,Sn=T(()=>{"use strict";t();Qe();ni=!1,ti=(e=!0)=>{ni=e},Wo=({timeout:e=0,url:o})=>{ni||setTimeout(()=>{if(!!window)return ei(o)},e)}});var ri=T(()=>{"use strict";t();Sn()});function ai(e){return e[Math.floor(Math.random()*e.length)]}var ii=T(()=>{"use strict";t()});function si(e){let o=null;if((e==null?void 0:e.statusCode)in ht){let n=e==null?void 0:e.statusCode,{message:r}=ht[n];o=um.includes(n)?ci:r}return{errorMessage:o,defaultErrorMessage:ci}}var ci,_m,lm,um,Rt=T(()=>{"use strict";t();xt();ci="Check if the MultiversX app is open on Ledger",_m=28161,lm=28160,um=[_m,lm]});var pi=T(()=>{"use strict";t();ri();ii();Rt()});var mi,di,_i,Dt,fm,li,$g,Vg,Hg,wn,vn=T(()=>{"use strict";t();mi=require("@reduxjs/toolkit"),di=w(require("lodash.omit")),_i=require("redux-persist");St();j();pi();Dt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},fm={network:Dt},li=(0,mi.createSlice)({name:"appConfig",initialState:fm,reducers:{initializeNetworkConfig:(e,o)=>{let n=ai(o.payload.walletConnectV2RelayAddresses),r=(0,di.default)(o.payload,"walletConnectV2RelayAddresses");e.network=S(g(g({},e.network),r),{walletConnectV2RelayAddress:n})},updateNetworkConfig:(e,o)=>{e.network=g(g({},e.network),o.payload)},setCustomWalletAddress:(e,o)=>{e.network.customWalletAddress=o.payload}},extraReducers:e=>{e.addCase(L,o=>{o.network.customWalletAddress=void 0}),e.addCase(_i.REHYDRATE,(o,n)=>{var a,p;if(!((p=(a=n.payload)==null?void 0:a.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:r}}=n.payload;o.network.customWalletAddress=r})}}),{initializeNetworkConfig:$g,updateNetworkConfig:Vg,setCustomWalletAddress:Hg}=li.actions,wn=li.reducer});var ui,Mt,gi,Yg,Zg,Jg,An,kn=T(()=>{"use strict";t();ui=require("@reduxjs/toolkit");G();j();Mt={isSigning:!1,signedSessions:{}},gi=(0,ui.createSlice)({name:"signedMessageInfoSliceState",initialState:Mt,reducers:{setSignSession:(e,o)=>{let{sessionId:n,signedSession:r,errorMessage:a}=o.payload;a&&(e.errorMessage=a),e.isSigning=r.status==="pending",e.signedSessions[n]=g(g({},e.signedSessions[n]),r)},setSignSessionState:(e,o)=>g(g({},e),o.payload),clearSignedMessageInfo:()=>Mt},extraReducers:e=>{e.addCase(L,()=>Mt)}}),{setSignSession:Yg,clearSignedMessageInfo:Zg,setSignSessionState:Jg}=gi.actions,An=gi.reducer});var hi,xi,fi,Ti,hm,xm,af,Tm,In,En=T(()=>{"use strict";t();hi=require("@reduxjs/toolkit"),xi=require("redux-persist");G();ln();j();fi={customToasts:[],transactionToasts:[]},Ti=(0,hi.createSlice)({name:"toastsSlice",initialState:fi,reducers:{addCustomToast:(e,o)=>{let n=o.payload.toastId||`custom-toast-${e.customToasts.length+1}`,r=e.customToasts.findIndex(a=>a.toastId===n);if(r!==-1){e.customToasts[r]=S(g(g({},e.customToasts[r]),o.payload),{type:"custom",toastId:n});return}e.customToasts.push(S(g({},o.payload),{type:"custom",toastId:n}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Oo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)}},extraReducers:e=>{e.addCase(L,()=>fi),e.addCase(xi.REHYDRATE,(o,n)=>{var a,p;let r=(p=(a=n.customToasts)==null?void 0:a.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=r})}}),{addCustomToast:hm,removeCustomToast:xm,addTransactionToast:af,removeTransactionToast:Tm}=Ti.actions,In=Ti.reducer});var yi,Ft,Bt,Ut,ym,Ot,bi,mf,bm,Pn,Ln,Nn=T(()=>{"use strict";t();yi=require("@reduxjs/toolkit");j();Ft="Transaction failed",Bt="Transaction successful",Ut="Processing transaction",ym="Transaction submitted",Ot={},bi=(0,yi.createSlice)({name:"transactionsInfo",initialState:Ot,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||Ft,successMessage:(r==null?void 0:r.successMessage)||Bt,processingMessage:(r==null?void 0:r.processingMessage)||Ut,submittedMessage:(r==null?void 0:r.submittedMessage)||ym,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>Ot},extraReducers:e=>{e.addCase(L,()=>Ot)}}),{clearTransactionsInfo:mf,setTransactionsDisplayInfo:bm,clearTransactionsInfoForSessionId:Pn}=bi.actions,Ln=bi.reducer});function Si(e){return e!=null&&(Nm(e)||Om(e))}function Cn(e){return e!=null&&(Cm(e)||Fm(e))}function Rn(e){return e!=null&&(Rm(e)||Bm(e))}function wi(e){return e!=null&&(Dm(e)||Um(e))}function vi(e){return e!=null&&Mm(e)}function Nm(e){return e!=null&&Sm.includes(e)}function Cm(e){return e!=null&&wm.includes(e)}function Rm(e){return e!=null&&vm.includes(e)}function Dm(e){return e!=null&&Am.includes(e)}function Mm(e){return e!=null&&km.includes(e)}function Om(e){return e!=null&&Im.includes(e)}function Fm(e){return e!=null&&Em.includes(e)}function Bm(e){return e!=null&&Pm.includes(e)}function Um(e){return e!=null&&Lm.includes(e)}var Sm,wm,vm,Am,km,Im,Em,Pm,Lm,$t=T(()=>{"use strict";t();oe();Sm=["sent"],wm=["success"],vm=["fail","cancelled","timedOut"],Am=["invalid"],km=["timedOut"],Im=["pending"],Em=["success"],Pm=["fail","invalid"],Lm=["not executed"]});var Ai,ki,Go,Wm,Ii,Tf,yf,Gm,yo,$m,Vm,bf,Ei,eo,bo,Sf,Dn,Mn=T(()=>{"use strict";t();Ai=require("@reduxjs/toolkit"),ki=require("redux-persist");oe();$t();j();Go={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Wm={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Ii=(0,Ai.createSlice)({name:"transactionsSlice",initialState:Go,reducers:{moveTransactionsToSignedState:(e,o)=>{var _,u;let{sessionId:n,transactions:r,errorMessage:a,status:p,redirectRoute:m,customTransactionInformation:d}=o.payload;e.customTransactionInformationForSessionId[n]=g(g(g({},Wm),((_=e.signedTransactions[n])==null?void 0:_.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[n]||{}),e.signedTransactions[n]={transactions:r,status:p,errorMessage:a,redirectRoute:m,customTransactionInformation:g(g({},e.customTransactionInformationForSessionId[n]),d!=null?d:{})},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=Go.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=g(g({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:p}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),p!=null&&(e.signedTransactions[n].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var u,f,h,b,v,k,F,A;let{sessionId:n,status:r,errorMessage:a,transactionHash:p,serverTransaction:m,inTransit:d}=o.payload,_=(f=(u=e.signedTransactions)==null?void 0:u[n])==null?void 0:f.transactions;if(_!=null){e.signedTransactions[n].transactions=_.map(C=>C.hash===p?S(g(g({},m!=null?m:{}),C),{status:r,errorMessage:a,inTransit:d}):C);let U=(b=(h=e.signedTransactions[n])==null?void 0:h.transactions)==null?void 0:b.every(C=>Cn(C.status)),D=(k=(v=e.signedTransactions[n])==null?void 0:v.transactions)==null?void 0:k.some(C=>Rn(C.status)),W=(A=(F=e.signedTransactions[n])==null?void 0:F.transactions)==null?void 0:A.every(C=>wi(C.status));U&&(e.signedTransactions[n].status="success"),D&&(e.signedTransactions[n].status="fail"),W&&(e.signedTransactions[n].status="invalid")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Go.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Go.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload},updateSignedTransactionsCustomTransactionInformation:(e,o)=>{let{sessionId:n,customTransactionInformationOverrides:r}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].customTransactionInformation=g(g({},e.signedTransactions[n].customTransactionInformation),r))}},extraReducers:e=>{e.addCase(L,()=>Go),e.addCase(ki.REHYDRATE,(o,n)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,p=Object.entries(r).reduce((d,[_,u])=>{let f=new Date(_),h=new Date;return h.setHours(h.getHours()+5),h-f>0||(d[_]=u),d},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:Tf,updateSignedTransactions:yf,setTransactionsToSign:Gm,clearAllTransactionsToSign:yo,clearAllSignedTransactions:$m,clearSignedTransaction:Vm,clearTransactionToSign:bf,setSignTransactionsError:Ei,setSignTransactionsCancelMessage:eo,moveTransactionsToSignedState:bo,updateSignedTransactionsCustomTransactionInformation:Sf}=Ii.actions,Dn=Ii.reducer});var Pi,Vt,Li,kf,If,Hm,Ef,On,Fn=T(()=>{"use strict";t();Pi=require("@reduxjs/toolkit");j();Vt={},Li=(0,Pi.createSlice)({name:"batchTransactionsSlice",initialState:Vt,reducers:{setBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},updateBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},clearBatchTransactions:(e,o)=>{delete e[o.payload.batchId]},clearAllBatchTransactions:()=>Vt},extraReducers:e=>{e.addCase(L,()=>Vt)}}),{setBatchTransactions:kf,updateBatchTransactions:If,clearBatchTransactions:Hm,clearAllBatchTransactions:Ef}=Li.actions,On=Li.reducer});var Ci,Ni,Ri,Cf,Di,Ht=T(()=>{"use strict";t();Ci=require("@reduxjs/toolkit");j();Ni={},Ri=(0,Ci.createSlice)({name:"dappConfigSlice",initialState:Ni,reducers:{setDappConfig:(e,o)=>o.payload},extraReducers:e=>{e.addCase(L,()=>Ni)}}),{setDappConfig:Cf}=Ri.actions,Di=Ri.reducer});var V=T(()=>{"use strict";t();hn();Tn();bn();vn();kn();En();Nn();Mn();Fn();Ht()});var qt=T(()=>{"use strict";t()});var Qi,ed,od,ao,Vn=T(()=>{"use strict";t();Qi=require("@reduxjs/toolkit");qt();hn();Fn();Ht();Tn();bn();vn();kn();En();Nn();Mn();ed={["account"]:fn,["dappConfig"]:Di,["loginInfo"]:xn,["modals"]:yn,["networkConfig"]:wn,["signedMessageInfo"]:An,["toasts"]:In,["transactionsInfo"]:Ln,["transactions"]:Dn,["batchTransactions"]:On},od=(e={})=>(0,Qi.combineReducers)(g(g({},ed),e)),ao=od});var tc={};Ie(tc,{default:()=>hd,sessionStorageReducers:()=>Kt});function Re(e,o=[]){return{key:e,version:1,storage:oc.default,blacklist:o}}var X,ec,oc,nd,jo,td,rd,ad,id,cd,sd,pd,md,dd,_d,nc,ld,Kt,ud,gd,fd,hd,rc=T(()=>{"use strict";t();X=require("redux-persist"),ec=w(require("redux-persist/lib/storage")),oc=w(require("redux-persist/lib/storage/session"));Vn();V();hn();Fn();Tn();bn();vn();kn();En();Nn();Mn();qt();nd={persistReducersStorageType:"localStorage"},jo={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},td=Re(jo["account"]),rd=Re(jo["loginInfo"]),ad=Re(jo["modals"]),id=Re(jo["networkConfig"]),cd={2:e=>S(g({},e),{networkConfig:Dt})};sd=Re("sdk-dapp-transactionsInfo"),pd=Re("sdk-dapp-transactions",["transactionsToSign"]),md=Re("sdk-dapp-batchTransactions",["batchTransactions"]),dd=Re("sdk-dapp-toasts"),_d=Re("sdk-dapp-signedMessageInfo"),nc={key:"sdk-dapp-store",version:2,storage:ec.default,whitelist:Object.keys(jo),migrate:(0,X.createMigrate)(cd,{debug:!1})},ld=S(g({},nc),{whitelist:[]}),Kt={["toasts"]:(0,X.persistReducer)(dd,In),["transactions"]:(0,X.persistReducer)(pd,Dn),["transactionsInfo"]:(0,X.persistReducer)(sd,Ln),["batchTransactions"]:(0,X.persistReducer)(md,On),["signedMessageInfo"]:(0,X.persistReducer)(_d,An)},ud=S(g({},Kt),{["account"]:(0,X.persistReducer)(td,fn),["loginInfo"]:(0,X.persistReducer)(rd,xn),["modals"]:(0,X.persistReducer)(ad,yn),["networkConfig"]:(0,X.persistReducer)(id,wn)}),gd=nd.persistReducersStorageType==="localStorage",fd=gd?(0,X.persistReducer)(nc,ao(Kt)):(0,X.persistReducer)(ld,ao(ud)),hd=fd});var ac={};Ie(ac,{default:()=>xd});var xd,ic=T(()=>{"use strict";t();Vn();xd=ao()});var cc={};Ie(cc,{default:()=>yd});var re,Td,yd,sc=T(()=>{"use strict";t();re=require("redux-persist"),Td=[re.FLUSH,re.REHYDRATE,re.PAUSE,re.PERSIST,re.PURGE,re.REGISTER],yd=Td});var dc={};Ie(dc,{default:()=>mc});function mc(e){return(0,pc.persistStore)(e)}var pc,_c=T(()=>{"use strict";t();pc=require("redux-persist")});var ds=Y((gI,ms)=>{t();var g_=typeof c=="object"&&c&&c.Object===Object&&c;ms.exports=g_});var ls=Y((fI,_s)=>{t();var f_=ds(),h_=typeof self=="object"&&self&&self.Object===Object&&self,x_=f_||h_||Function("return this")();_s.exports=x_});var ur=Y((hI,us)=>{t();var T_=ls(),y_=T_.Symbol;us.exports=y_});var xs=Y((xI,hs)=>{t();var gs=ur(),fs=Object.prototype,b_=fs.hasOwnProperty,S_=fs.toString,Jo=gs?gs.toStringTag:void 0;function w_(e){var o=b_.call(e,Jo),n=e[Jo];try{e[Jo]=void 0;var r=!0}catch(p){}var a=S_.call(e);return r&&(o?e[Jo]=n:delete e[Jo]),a}hs.exports=w_});var ys=Y((TI,Ts)=>{t();var v_=Object.prototype,A_=v_.toString;function k_(e){return A_.call(e)}Ts.exports=k_});var vs=Y((yI,ws)=>{t();var bs=ur(),I_=xs(),E_=ys(),P_="[object Null]",L_="[object Undefined]",Ss=bs?bs.toStringTag:void 0;function N_(e){return e==null?e===void 0?L_:P_:Ss&&Ss in Object(e)?I_(e):E_(e)}ws.exports=N_});var ks=Y((bI,As)=>{t();var C_=Array.isArray;As.exports=C_});var Es=Y((SI,Is)=>{t();function R_(e){return e!=null&&typeof e=="object"}Is.exports=R_});var Ls=Y((wI,Ps)=>{t();var D_=vs(),M_=ks(),O_=Es(),F_="[object String]";function B_(e){return typeof e=="string"||!M_(e)&&O_(e)&&D_(e)==F_}Ps.exports=B_});var hr={};Ie(hr,{css:()=>Fs,default:()=>H_});var Fs,H_,xr=T(()=>{"use strict";t();Fs=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Fs));H_={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var yr={};Ie(yr,{css:()=>Gs,default:()=>q_});var Gs,q_,br=T(()=>{"use strict";t();Gs=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Gs));q_={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var pl={};Ie(pl,{useSignTransactionsWithLedger:()=>sl});module.exports=je(pl);t();t();t();var Mc=w(require("axios"));t();t();V();t();var Mi=w(require("lodash.isequal")),Bn=require("reselect"),y=(0,Bn.createSelectorCreator)(Bn.defaultMemoize,Mi.default);var Te=e=>e.account,Ge=y(Te,e=>e.address),oo=y(Te,Ge,(e,o)=>o in e.accounts?e.accounts[o]:gn),Oi=y(Te,oo,(e,o)=>{let a=e,{accounts:n}=a,r=lo(a,["accounts"]);return S(g({},r),{address:o.address,account:o})}),Kf=y(oo,e=>e.balance),Fi=y(oo,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Xf=y(Te,e=>e.shard),Yf=y(Te,e=>e.ledgerAccount),Zf=y(Te,e=>e.walletConnectAccount),Jf=y(Te,e=>e.isAccountLoading),Qf=y(Te,e=>e.accountLoadingError),jm=y(Te,e=>e.websocketEvent),zm=y(Te,e=>e.websocketBatchEvent);t();var jt=e=>e.dappConfig,nh=y(jt,e=>e.shouldUseWebViewProvider);t();var K=e=>e.loginInfo,ih=y(K,e=>e.loginMethod),$o=y(K,Ge,(e,o)=>Boolean(o)),ch=y(K,e=>e.walletConnectLogin),Bi=y(K,e=>e.ledgerLogin),Ui=y(K,e=>e.walletLogin),sh=y(K,e=>e.isLoginSessionInvalid),ph=y(K,e=>e.tokenLogin),Wi=y(K,e=>e.logoutRoute),qm=y(K,e=>e.isWalletConnectV2Initialized);t();var Gi=e=>e.modals,_h=y(Gi,e=>e.txSubmittedModal),lh=y(Gi,e=>e.notificationModal);t();var ye=e=>e.networkConfig,$i=y(ye,e=>e.network.chainId),fh=y(ye,e=>e.network.roundDuration),hh=y(ye,e=>e.network.walletConnectBridgeAddress),xh=y(ye,e=>e.network.walletConnectV2RelayAddress),Th=y(ye,e=>e.network.walletConnectV2ProjectId),yh=y(ye,e=>e.network.walletConnectV2Options),bh=y(ye,e=>e.network.walletConnectDeepLink),no=y(ye,e=>e.network),Vi=y(no,e=>e.apiAddress),Hi=y(no,e=>e.explorerAddress),ji=y(no,e=>{var o;return(o=e.customWalletAddress)!=null?o:e.walletAddress}),Sh=y(no,e=>e.xAliasAddress),zi=y(no,e=>e.egldLabel);t();var Un=e=>e.signedMessageInfo,Ah=y(Un,e=>e.isSigning),kh=y(Un,e=>e.errorMessage),Ih=y(Un,e=>{let o=Object.keys(e.signedSessions),n=o.length;return e.signedSessions[o[n-1]]}),Eh=y(Un,e=>{let o=Object.keys(e.signedSessions),n=o.length;return o.length>0?o[n-1]:""});t();var qi=e=>e.toasts,Nh=y(qi,e=>e.customToasts),Ch=y(qi,e=>e.transactionToasts);t();V();var Ki={errorMessage:Ft,successMessage:Bt,processingMessage:Ut},Km=e=>e.transactionsInfo,Oh=y(Km,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Ki);t();t();var ro=require("@multiversx/sdk-core");P();t();t();t();function Wn(e){try{let o=atob(e),n=btoa(o),r=i.Buffer.from(e,"base64").toString(),a=i.Buffer.from(r).toString("base64"),p=e===n||n.startsWith(e),m=e===a||a.startsWith(e);if(p&&m)return!0}catch(o){return!1}return!1}function to(e){return Wn(e)?atob(e):e}t();t();function zt(e){for(let o=0;o<e.length;o++)if(e.charCodeAt(o)>127)return!1;return!0}function Vo(e){let o=e;try{let n=i.Buffer.from(e,"hex").toString();zt(n)&&n.length>1&&(o=n)}catch(n){}return o}t();t();var Xi=e=>{let o=e!=null?e:"";return Wn(o)?i.Buffer.from(i.Buffer.from(o,"base64").toString().trim()):i.Buffer.from(o.trim())};t();P();var So=({data:e,onlySetGuardian:o})=>e?o?e.startsWith("SetGuardian"):Object.values(ft).some(n=>e.startsWith(n)):!1;function Ho(e){var r,a,p;let o=g({},e);So({data:o.data,onlySetGuardian:!0})&&(delete o.senderUsername,delete o.receiverUsername);let n=new ro.Transaction(g(g(S(g(S(g(g({value:BigInt(o.value.valueOf()),data:Xi(o.data),nonce:BigInt(o.nonce.valueOf()),receiver:new ro.Address(o.receiver)},o.receiverUsername?{receiverUsername:o.receiverUsername}:{}),o.relayer?{relayer:new ro.Address(o.relayer)}:{}),{sender:new ro.Address(o.sender)}),o.senderUsername?{senderUsername:o.senderUsername}:{}),{gasLimit:BigInt((r=o.gasLimit.valueOf())!=null?r:pn),gasPrice:BigInt((a=o.gasPrice.valueOf())!=null?a:mn),chainID:o.chainID.valueOf(),version:(p=o.version)!=null?p:Pa}),o.options?{options:o.options}:{}),o.guardian?{guardian:new ro.Address(o.guardian)}:{}));return o.relayerSignature&&(n.relayerSignature=i.Buffer.from(o.relayerSignature,"hex")),o.guardianSignature&&(n.guardianSignature=i.Buffer.from(o.guardianSignature,"hex")),o.signature&&(n.signature=i.Buffer.from(o.signature,"hex")),n}t();t();P();function Xm(e,o){let n=new URLSearchParams(e),r=Object.fromEntries(n);return{nextUrlParams:new URLSearchParams(g(g({},r),o)).toString(),params:r}}var Ym="4.1.8";function Yi({callbackUrl:e,urlParams:o={}}){let n=e;if(Object.entries(o).length>0)try{let{search:r,origin:a,pathname:p,hash:m}=new URL(e),d=S(g({},o),{[dn]:Ym}),{nextUrlParams:_}=Xm(r,d);n=`${a}${p}?${_}${m}`}catch(r){return console.error("Unable to construct URL from: ",e,r),n}return n}t();t();var Xt=require("@reduxjs/toolkit"),fc=require("react-redux/lib/utils/Subscription");V();t();var Zi=w(require("lodash.throttle"));P();V();Nt();To();Ye();var Zm=[_n],Gn=!1,Jm=(0,Zi.default)(()=>{Bo(Fo())},5e3),Ji=e=>o=>n=>{if(Zm.includes(n.type))return o(n);let r=e.getState(),a=me.local.getItem(ne.loginExpiresAt);if(!Boolean(r==null?void 0:r.account.address))return o(n);if(a==null)return Bo(Fo());let m=Date.now();return a-m<0&&!Gn?setTimeout(()=>{Gn=!0,e.dispatch(Ka())},1e3):(Gn&&(Gn=!1),Jm()),o(n)};t();t();function $n(){return typeof sessionStorage!="undefined"}var lc=$n()?(rc(),je(tc)).default:(ic(),je(ac)).default,uc=$n()?(sc(),je(cc)).default:[],gc=$n()?(_c(),je(dc)).default:e=>e;Vn();var I=(0,Xt.configureStore)({reducer:lc,middleware:e=>e({serializableCheck:{ignoredActions:[...uc,Lt.type,Pt.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Ji)}),hc=(0,fc.createSubscription)(I),jx=gc(I),zx=(0,Xt.configureStore)({reducer:ao});t();t();var Sd=require("@multiversx/sdk-core");t();P();t();var vd=require("@multiversx/sdk-core");G();t();var xc=require("@multiversx/sdk-core");function wd(e){try{return xc.Address.isValid(e)}catch(o){return!1}}function wo(e){return/^\w+$/.test(e)&&wd(e)}t();t();t();var Zt=require("@multiversx/sdk-core"),Hn=w(require("bignumber.js"));oe();var Yt={tokenId:"",amount:""},Tc=e=>e.split("@").map((p,m)=>[2,3].includes(m)?p:Vo(p));function Jt(e){if(!e)return Yt;let o=e.startsWith("ESDTTransfer"),n=e.startsWith("ESDTNFTTransfer")&&e.includes("@"),r=e.startsWith("ESDTNFTBurn")&&e.includes("@");if(o){let[,a,p]=e.split("@");try{let m=i.Buffer.from(a,"hex").toString("ascii");if(!m)return Yt;let d=new Hn.default("0x"+p.replace("0x","")).toString(10);return{tokenId:m,amount:d}}catch(m){console.error("Error getting token from transaction data",m)}}if(n)try{let[,a,p,m,d]=Tc(e);if([a,p,m,d].every(_=>Boolean(_))&&wo(new Zt.Address(d).bech32()))return{tokenId:`${a}-${p}`,amount:new Hn.default(m,16).toString(10),collection:a,nonce:p,receiver:new Zt.Address(d).bech32()}}catch(a){}if(r)try{let[,a,p,m]=Tc(e);if([a,p,m].every(d=>Boolean(d)))return{tokenId:`${a}-${p}`,amount:new Hn.default(m,16).toString(10),collection:a,nonce:p}}catch(a){}return Yt}t();t();t();var io="accounts";var Qt="blocks";t();t();t();t();q();t();t();q();t();t();t();var Ld=w(require("bignumber.js"));P();t();var Pd=require("@multiversx/sdk-core"),er=w(require("bignumber.js"));P();t();var Id=w(require("bignumber.js"));t();er.default.config({ROUNDING_MODE:er.default.ROUND_FLOOR});t();t();t();P();t();t();t();t();P();t();P();t();var Nd=require("@multiversx/sdk-core"),Cd=w(require("bignumber.js"));P();t();var nr=require("@multiversx/sdk-core"),Md=w(require("bignumber.js"));P();t();t();var Rd=w(require("bignumber.js"));P();t();P();t();t();t();t();t();t();P();t();P();t();P();t();q();var Fd=["reDelegateRewards","claimRewards","unBond"],Bd=["wrapEgld","unwrapEgld"],Ud=["unStake"],Wd=["unDelegate"];t();t();t();P();t();var Vd=w(require("bignumber.js"));t();t();q();t();var jd=w(require("bignumber.js"));t();t();t();function wc({tokenId:e,erdLabel:o}){return Boolean(e&&e!==o)}t();var rr=w(require("bignumber.js"));G();t();var vc=(e,o)=>{let n=[],r=e.indexOf(o,0);for(;r>=0;)n.push(r),r=e.indexOf(o,r+1);return n};function Ac(e){let o=[],n=0;try{if((e==null?void 0:e.startsWith("MultiESDTNFTTransfer"))&&(e==null?void 0:e.includes("@"))){let[,r,a,...p]=e==null?void 0:e.split("@");if(r){let m=new rr.default(a,16).toNumber();if(m>=Number.MAX_SAFE_INTEGER)return[];let d=0;for(let h=0;h<m;h++){let b={type:"nftTransaction",data:"",receiver:r};for(let v=0;v<3;v++){switch(v){case 0:b.token=Vo(p[d]),b.data=p[d];break;case 1:{let k=p[d]&&p[d].length?p[d]:"";k&&k!=="00"?b.nonce=k:b.type="esdtTransaction",b.data=`${b.data}@${p[d]}`;break}case 2:b.amount=new rr.default(p[d],16).toString(10),b.data=`${b.data}@${p[d]}`;break;default:break}n=d+1,d++}o[h]=b}let _=o.length!==m,u=o.some(h=>vc(h.data,"@").length!==2),f=o.some(h=>h.data.startsWith("@"));if(_||u||f)return[];if(p[n]){let h=p[n];for(let b=n+1;b<p.length;b++)h+="@"+p[b];o[m]={type:"scCall",data:h,receiver:r}}}}}catch(r){return console.error("failed parsing tx",r),o}return o}t();var kc=require("@multiversx/sdk-core");t();t();oe();function qo(e){let n=Object.getPrototypeOf(e).toPlainObject!=null?e:Ho(e),r=new kc.TransactionComputer,a=S(g({},n.toPlainObject()),{hash:r.computeTransactionHash(n),senderUsername:n.senderUsername,receiverUsername:n.receiverUsername,status:"pending"});return So({data:a.data,onlySetGuardian:!0})&&(delete a.senderUsername,delete a.receiverUsername),a}t();var Pc=require("@multiversx/sdk-web-wallet-provider");P();t();var Ec=w(require("qs"));t();Qe();var zn=e=>{let o=new URLSearchParams(e).toString(),{pathname:n,hash:r}=R(),a=o?`?${o}`:"",p=n?`${n}${a}${r}`:"./";setTimeout(()=>{window==null||window.history.replaceState({},document==null?void 0:document.title,p)})};Uo();t();Uo();var qd={search:Ce()?window.location.search:"",removeParams:[]},Ic=(e,o=qd)=>{var m;let n={},r=Ce()?window.location.search:"",a=(m=o.search)!=null?m:r;if(a){let d=a?new URLSearchParams(a):[];n=Object.fromEntries(d)}let p={};return e.forEach(d=>{p[d]=n[d],delete n[d]}),o.removeParams!=null&&Object.keys(n).forEach(d=>{var u,f;let[_]=d.split("[");(((u=o.removeParams)==null?void 0:u.includes(d))||((f=o.removeParams)==null?void 0:f.includes(_)))&&delete n[d]}),{remainingParams:p,params:n,clearNavigationHistory:()=>zn(n)}};var qn=({removeParams:e,search:o})=>{let n=Ce()?window.location.search:"",r=o!=null?o:n;if(!r)return{};let a=Ec.default.parse(r.replace("?","")),p=Object.keys(a).filter(d=>!e.includes(d)),{remainingParams:m}=Ic(p,{search:o,removeParams:e});return zn(m),m};var Lc=({transaction:e,search:o})=>qn({removeParams:[...Object.keys(e),Pc.WALLET_PROVIDER_CALLBACK_PARAM,We,dn],search:o});t();t();t();te();t();q();t();t();te();t();var Kd=w(require("linkifyjs"));t();P();t();var Xd=w(require("bignumber.js"));t();G();t();t();oe();t();oe();t();t();t();q();t();q();t();var Yd=w(require("bignumber.js"));P();q();t();q();t();var Cc=require("react");G();t();t();q();t();t();var Zd=require("@multiversx/sdk-core"),Jd=w(require("bignumber.js"));q();t();G();t();t();G();var qw=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var Rc=require("react");G();te();t();var e_=require("react");q();var ov=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];$t();t();var Kn=e=>e.transactions,Xo=y(Kn,e=>e.signedTransactions),jv=y(Kn,e=>e.signTransactionsError),Dc=y(Kn,e=>e.signTransactionsCancelMessage),Xn=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),zv=y(Xo,Xn(Si)),qv=y(Xo,Xn(Cn)),Kv=y(Xo,Xn(Rn)),Xv=y(Xo,Xn(vi)),Yn=y(Kn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:S(g({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>Ho(n)))||[]})}),Yv=y(Xo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});function Oc(e){return x(this,null,function*(){let{network:{apiAddress:o,apiTimeout:n}}=ye(I.getState()),{data:r}=yield Mc.default.get(`${o}/${io}/${e}`,{timeout:Number(n)});return r})}t();t();var Fc=require("react"),vo=require("react-redux");var o_={store:I,subscription:hc},ar=(0,Fc.createContext)(o_),uA=(0,vo.createStoreHook)(ar),$e=(0,vo.createDispatchHook)(ar),N=(0,vo.createSelectorHook)(ar);var Bc=()=>N(Oi);t();t();t();t();var Gc=require("@multiversx/sdk-extension-provider"),$c=require("@multiversx/sdk-hw-provider"),Vc=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Hc=require("@multiversx/sdk-opera-provider"),jc=require("@multiversx/sdk-passkey-provider/out"),ir=require("@multiversx/sdk-web-wallet-provider");P();Do();t();var co=require("@multiversx/sdk-web-wallet-iframe-provider/out"),Uc=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");oe();t();var Wc=Ao;var n_="/dapp/init",so=e=>{switch(e==null?void 0:e.constructor){case ir.WalletProvider:return"wallet";case Wc:return"walletconnectv2";case $c.HWProvider:return"ledger";case Gc.ExtensionProvider:return"extension";case jc.PasskeyProvider:return"passkey";case Vc.MetamaskProvider:return"metamask";case Hc.OperaProvider:return"opera";case Ne.CrossWindowProvider:return"crossWindow";case co.IframeProvider:return"iframe";case Ao:return"";default:return"extra"}},zc=e=>new ir.WalletProvider(`${e}${n_}`),qc=e=>x(void 0,null,function*(){if(!e.isInitialized())throw new Error("Unable to get version. Provider not initialized");let o=e.hwApp,{contractData:n,version:r}=yield o.getAppConfiguration();return{version:r,dataEnabled:n===La}}),de=e=>`Unable to perform ${e}, Provider not initialized`,Ao=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(de("getAccount"))}setAccount(o){throw new Error(de(`setAccount: ${o}`))}login(o){throw new Error(de(`login with options: ${o}`))}logout(o){throw new Error(de(`logout with options: ${o}`))}getAddress(){throw new Error(de("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(o,n){throw new Error(de(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(de(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(de(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(de(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(de(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(de(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},cr=new Ao;var sr=cr,Kc=null;function Xc(e){sr=e}function Yc(){Kc!=null&&(sr=Kc)}function Z(){return sr||cr}var ko=()=>{let e=Z(),o=so(e);return{provider:e,providerType:o}};t();var be=require("react"),Me=require("@multiversx/sdk-core");t();t();t();t();var Zc=require("@lifeomic/axios-fetch"),pr=w(require("axios")),mr=(0,Zc.buildAxiosFetch)(pr.default),dr=(e,o)=>x(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let r=e.clone().json(),[a]=yield Promise.all([r]);return{data:a,status:e.status,statusText:e.statusText,headers:e.headers,config:o}});function t_(e,o,n){return x(this,null,function*(){try{let r=yield mr(e,g({method:"POST",body:o?JSON.stringify(o):void 0,headers:g({"Content-Type":"application/json"},(n==null?void 0:n.headers)||{})},n));return dr(r,n)}catch(r){throw console.error("Fetch Error:",r),r}})}function r_(e,o){return x(this,null,function*(){try{let n=yield mr(e,o);if(!n.ok)throw new Error(`HTTP error! Status: ${n.status}`);return dr(n,o)}catch(n){throw console.error("Fetch Error:",n),n}})}function a_(e,o,n){return x(this,null,function*(){try{let r=yield mr(e,g({method:"PATCH",body:o?JSON.stringify(o):void 0,headers:(n==null?void 0:n.headers)||{}},n));if(!r.ok)throw new Error(`HTTP error! Status: ${r.status}`);return dr(r,n)}catch(r){throw console.error("Fetch Error:",r),r}})}var po=pr.default.create();po.get=r_;po.post=t_;po.patch=a_;var Jc=e=>po.get(e).then(o=>o.data);t();var Io=e=>{let o=e!=null?e:Vi(I.getState());return o.endsWith("/")?o.slice(0,-1):o};t();var Qc=w(require("swr"));var Zn=({apiEndpoint:e})=>{let o=Io(),n=e?`${o}/${e}`:null;return(0,Qc.default)(n,Jc)};var Jn=e=>Zn({apiEndpoint:e?`${io}/${e}?withGuardianInfo=true`:null});P();t();var Yo=require("react");t();function es({parsedTransactionsByDataField:e,data:o,txInfo:n}){!e||(e[o]=n)}function os({transactions:e}){let o={},n=[];return!e||e.length===0?{allTransactions:n,parsedTransactionsByDataField:o}:(e.forEach((r,a)=>{let p=r.getData().toString(),m=Ac(p);if(m.length>0)m.forEach((d,_)=>{let u=_===m.length-1,f={transaction:r,multiTxData:d.data,transactionIndex:_,needsSigning:u};es({parsedTransactionsByDataField:o,data:d.data,txInfo:{tokenId:d.token?d.token:"",amount:d.amount?d.amount:"",type:d.type,nonce:d.nonce?d.nonce:"",multiTxData:d.data,receiver:d.receiver}}),n.push(f)});else{let d=r.getData().toString(),{tokenId:_,amount:u}=Jt(d);_&&es({parsedTransactionsByDataField:o,data:d,txInfo:{tokenId:_,amount:u,receiver:r.getReceiver().bech32()}}),n.push({transaction:r,transactionIndex:a,multiTxData:d,needsSigning:!0})}}),{allTransactions:n,parsedTransactionsByDataField:o})}var ns={tokenId:"",amount:"",type:"",multiTxData:"",receiver:""};function ts({transactions:e}){let[o,n]=(0,Yo.useState)({}),[r,a]=(0,Yo.useState)([]);function p(d,_){return o==null?ns:d in o?o[d]:_!=null&&String(_)in o?o[_]:ns}function m(){let{allTransactions:d,parsedTransactionsByDataField:_}=os({transactions:e});a(d),n(_)}return(0,Yo.useEffect)(()=>{m()},[e==null?void 0:e.length]),{parsedTransactionsByDataField:o,getTxInfoByDataField:p,allTransactions:r}}Rt();t();t();var rs=(e,o)=>e?Array.isArray(o)?o.some(n=>e.address===n||e.activeGuardianAddress===n):e.address===o||e.activeGuardianAddress===o:!0;t();function as(e){let o=Yn(I.getState()),n=Object.keys(e).length===(o==null?void 0:o.transactions.length);return e&&(e&&n)}t();t();t();var i_=(e,o)=>{var a;let r=`${(a=o==null?void 0:o.apiAddress)!=null?a:Io()}/${io}/${e}?withGuardianInfo=true`;return po.get(r)},is=(e,o)=>x(void 0,null,function*(){if(!e)return null;try{let{data:n}=yield i_(e,o);return n}catch(n){console.error("error fetching configuration for ",e,n)}return null});var Eo=(e,o)=>is(e,o);t();function Qn(e){let o=I.getState(),n=Fi(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();var _r=({accountNonce:e,transactionNonce:o})=>o&&o>e?o:e;var cs=({latestNonce:e,transactions:o})=>o.length===0?o:o.map((n,r)=>{let a=e+r,p=n.nonce,m=_r({accountNonce:a,transactionNonce:Number(p)});return n.nonce=BigInt(m),n});var ss=()=>{let e=N(Ge),{data:o}=Jn(e);return n=>x(void 0,null,function*(){let r=o!=null?o:yield Eo(e),a=Qn(r);return cs({latestNonce:a,transactions:n})})};t();var Zo=({transactions:e,isGuarded:o})=>o?e.length===0?!1:e.every(n=>Boolean(n.getGuardianSignature().toString("hex"))):!0;t();t();t();Mo();t();var p_=w(require("axios"));t();var d_=w(require("axios"));t();var __=w(require("axios"));t();Mo();t();Mo();function lr(e){return va[e]}t();t();t();t();var l_=require("@multiversx/sdk-opera-provider");t();var u_=require("@multiversx/sdk-extension-provider");t();Do();function ps(n){return x(this,arguments,function*({address:e,walletUrl:o}){try{let r=yield Ne.CrossWindowProvider.getInstance().init(),a=Ne.CrossWindowProvider.getInstance().setAddress(e).setWalletUrl(o);if(r)return a;console.error("Could not initialise Cross Window Wallet provider")}catch(r){console.error("Unable to login to CrossWindowWalletProvider",r)}return null})}t();t();t();t();var Ns=w(Ls());var gr=e=>{if(!e||!(0,Ns.default)(e))return null;let o=e.split(".");if(o.length!==4)return null;try{let[n,r,a,p]=o,m=JSON.parse(to(p)),d=to(n);return{ttl:Number(a),extraInfo:m,origin:d,blockHash:r}}catch(n){return console.error(`Error trying to decode ${e}:`,n),null}};t();var G_=require("@multiversx/sdk-native-auth-client");t();var Ms=w(require("axios"));t();t();t();function Cs(e){return x(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}t();var Rs=(e,o,n,r=0)=>x(void 0,null,function*(){try{return yield e(...n)}catch(a){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield Cs(o.delay)),yield Rs(e,o,n,r+1)):null}}),Ds=(e,o={retries:5,delay:500})=>(...n)=>x(void 0,null,function*(){return yield Rs(e,o,n)});var U_=4;var MI=Ds((e,o,n)=>x(void 0,null,function*(){if(n){let p=Math.floor(Date.now()/1e3);return{hash:yield n(),timestamp:p}}let{data:r}=yield Ms.default.get(`${e}/${Qt}?from=${U_}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[a]=r;return a}));t();t();ln();t();function W_(e){return Object.prototype.toString.call(e)==="[object String]"}var fr=e=>{var n;if(!e||!W_(e))return null;let o=e.split(".");if(o.length!==3)return console.error("Invalid nativeAuthToken. You may be trying to decode a loginToken. Try using decodeLoginToken method instead"),null;try{let[r,a,p]=o,m=to(r),d=to(a),_=gr(d);if(!_)return{address:m,body:d,signature:p,blockHash:"",origin:"",ttl:0};let u=S(g({},_),{address:m,body:d,signature:p});return(n=_.extraInfo)!=null&&n.timestamp||delete u.extraInfo,u}catch(r){return null}};t();te();var jI={origin:R().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};t();var $_=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");t();var V_=require("@multiversx/sdk-passkey-provider/out");t();var Y_=w(require("react")),Z_=require("@fortawesome/free-solid-svg-icons");t();t();var Po=w(require("react")),$s=require("@fortawesome/free-solid-svg-icons"),Vs=require("@fortawesome/react-fontawesome");t();var Us=w(require("react"));t();var Qo=w(require("react"));t();Uo();var Os=()=>!Ce();var j_=()=>x(void 0,null,function*(){return yield Promise.resolve().then(()=>(xr(),hr))}),z_=()=>(xr(),je(hr)).default,Tr=Os();function Bs({ssrGlobalImportCallback:e,ssrImportCallback:o,clientImportCallback:n}){let[r,a]=Qo.default.useState(Tr?void 0:z_()),[p,m]=Qo.default.useState(Tr||n==null?void 0:n()),d=()=>x(this,null,function*(){(e?e():j_()).then(_=>a(_.default)),o==null||o().then(_=>m(_.default))});return(0,Qo.useEffect)(()=>{Tr&&d()},[]),{globalStyles:r,styles:p}}function Ws(e,o){return n=>{let{globalStyles:r,styles:a}=Bs({ssrGlobalImportCallback:o.ssrGlobalStyles,ssrImportCallback:o.ssrStyles,clientImportCallback:o.clientStyles});return Us.default.createElement(e,S(g({},n),{globalStyles:r!=null?r:{},styles:a!=null?a:{}}))}}var K_=({styles:e})=>Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},Po.default.createElement(Vs.FontAwesomeIcon,{icon:$s.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),Po.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),X_=Ws(K_,{ssrStyles:()=>Promise.resolve().then(()=>(br(),yr)),clientStyles:()=>(br(),je(yr)).default});To();Ye();t();t();V();var M1=24*60*60;P();te();var Hs=({transactions:e,hasGuardianScreen:o,callbackRoute:n,sessionId:r,walletAddress:a,isGuarded:p})=>{let m=Zo({isGuarded:p,transactions:e}),d=e[0].getChainID().valueOf(),_=e[0].getSender().bech32().toString(),u=lr(d),f=a!=null?a:Xe[u].walletAddress;return{needs2FaSigning:p?!o&&!m&&r:!1,sendTransactionsToGuardian:()=>{let k=zc(f),F={[We]:String(r)},{origin:A}=R(),U=window!=null&&window.location?`${A}${n}`:`${n}`,D=Yi({callbackUrl:U,urlParams:F});k.guardTransactions(e,{callbackUrl:encodeURIComponent(D)})},guardTransactions:()=>x(void 0,null,function*(){let k=yield ps({address:_,walletUrl:f});return k==null||k.setShouldShowConsentPopup(!0),yield k==null?void 0:k.guardTransactions(e)})}};t();t();t();var Qs=require("react"),wr=require("@multiversx/sdk-hw-provider");t();t();te();var js=()=>{let{search:e}=R(),o=new URLSearchParams(e),n=Object.fromEntries(o),r=n==null?void 0:n.address;return wo(r)?r:null};t();t();function zs(){let e=Z();return so(e)}t();oe();t();function mo(e){return zs()===e}function et(){let e=I.getState(),o=Z(),n=Ge(e),r=$o(e),a=Ui(e);if(!o)throw"provider not initialized";return mo("ledger")&&r?new Promise(p=>{p(n)}):!mo("")&&!mo("wallet")&&!mo("extra")?o.getAddress():new Promise(p=>{let m=js();a!=null&&m&&p(m),r&&p(n),p("")})}t();t();te();var qs=()=>{let{search:e}=R(),o=new URLSearchParams(e),n=Object.fromEntries(o);return n==null?void 0:n.accessToken};var Ks=()=>{var n;let e=qs(),o=K(I.getState());return e!=null?e:(n=o.webviewLogin)==null?void 0:n.data};t();j();G();t();V();t();V();t();var Xs=require("@multiversx/sdk-core");G();Qe();t();sn();j();G();t();function Ys(e,o){if(!e){typeof console!="undefined"&&console.warn(o);try{throw new Error(o)}catch(n){}}}function Zs(e,o){typeof e=="string"&&(e={path:e,caseSensitive:!1,end:!0});let[n,r]=J_(e.path,e.caseSensitive,e.end),a=o.match(n);if(!a)return null;let p=a[0],m=p.replace(/(.)\/+$/,"$1"),d=a.slice(1);return{params:r.reduce((u,f,h)=>{if(f==="*"){let b=d[h]||"";m=p.slice(0,p.length-b.length).replace(/(.)\/+$/,"$1")}return u[f]=Q_(d[h]||"",f),u},{}),pathname:p,pathnameBase:m,pattern:e}}function J_(e,o=!1,n=!0){Ys(e==="*"||!e.endsWith("*")||e.endsWith("/*"),`Route path "${e}" will be treated as if it were "${e.replace(/\*$/,"/*")}" because the \`*\` character must always follow a \`/\` in the pattern. To get rid of this warning, please change the route path to "${e.replace(/\*$/,"/*")}".`);let r=[],a="^"+e.replace(/\/*\*?$/,"").replace(/^\/*/,"/").replace(/[\\.*+^$?{}|()[\]]/g,"\\$&").replace(/:(\w+)/g,(m,d)=>(r.push(d),"([^\\/]+)"));return e.endsWith("*")?(r.push("*"),a+=e==="*"||e==="/*"?"(.*)$":"(?:\\/(.+)|\\/*)$"):a+=n?"\\/*$":"(?:(?=[.~-]|%[0-9A-F]{2})|\\b|\\/|$)",[new RegExp(a,o?void 0:"i"),r]}function Q_(e,o){try{return decodeURIComponent(e)}catch(n){return Ys(!1,`The value for the URL param "${o}" will not be decoded because the string "${e}" is a malformed URL segment. This is probably due to a bad percent encoding (${n}).`),e}}Sn();To();Ye();Qe();var Js=({callbackUrl:e,onRedirect:o})=>{typeof o=="function"?o(e):e&&Wo({url:e})},el=e=>{let o=me.local.getItem(ne.logoutEvent),{data:n}=o?JSON.parse(o):{data:e};e===n&&(me.local.setItem({key:ne.logoutEvent,data:e,expires:0}),me.local.removeItem(ne.logoutEvent))},ol=500;function Sr(a,p){return x(this,arguments,function*(e,o,n=Boolean(Ks()),r={shouldBroadcastLogoutAcrossTabs:!0,hasConsentPopup:!1}){var F,A,U;let m=yield et(),d=Z(),_=so(d),u=_==="wallet",f=((F=d==null?void 0:d.isInitialized)==null?void 0:F.call(d))===!0;if(n&&(d==null?void 0:d.relogin)!=null)return d.relogin();m&&r.shouldBroadcastLogoutAcrossTabs&&el(m);let h=Ct(e),b=R(),{pathname:v,origin:k}=new URL(decodeURIComponent(h));if((Zs(b.pathname,v)||u&&f||((A=pe.safeWindow)==null?void 0:A.opener)&&k===((U=pe.safeWindow)==null?void 0:U.origin))&&ti(),!m)return Js({callbackUrl:h,onRedirect:o});try{if(I.dispatch(L()),u&&f)return setTimeout(()=>{d.logout({callbackUrl:h})},ol);r.hasConsentPopup&&_==="crossWindow"&&d.setShouldShowConsentPopup(!0),f&&(yield d.logout({callbackUrl:h}))}catch(D){console.error("Logging out error:",D)}finally{u||Js({callbackUrl:h,onRedirect:o})}})}var ep=()=>{let e=N(Wi),o=N($o),n=N(Bi),r=Z(),[a,p]=(0,Qs.useState)(),m=()=>x(void 0,null,function*(){let _=(n==null?void 0:n.index)!=null;try{if(r instanceof wr.HWProvider&&r.isInitialized())return _&&(yield r.setAddressIndex(n.index)),r;let u=new wr.HWProvider;return(yield u.init())?(_&&(yield u.setAddressIndex(n.index)),u):null}catch(u){return console.error("Failed to initialise Ledger Provider"),null}});function d(_){return x(this,null,function*(){let u,f=o&&!(_!=null&&_.isRelogin);try{if(u=yield m(),!u){console.warn("Could not initialise ledger app"),f&&Sr(e);return}let h=yield qc(u);return Xc(u),p(h),u}catch(h){console.error("Could not initialise ledger app",h),f&&Sr(e)}return null})}return{setLedgerProvider:d,ledgerData:a}};var op=()=>{let{provider:e}=ko(),{setLedgerProvider:o}=ep();return function(){return x(this,null,function*(){let r;try{let p=yield e.getAddress();r=Boolean(p)}catch(p){r=!1}if(r)return e;let a=yield o({isRelogin:!0});return a||e})}};t();function np({transactionDataLength:e,transactionGasLimit:o,ppu:n}){if(!n)return 1e9;let u=5e4+e*1500,f=o-u,h=o*n,b=Math.floor(h/(u+f*.01));return b<1e9?1e9:b>3e10?3e10:b}var ot={},tp=({isLedger:e=!1,transactionsToSign:o,egldLabel:n,address:r,activeGuardianAddress:a,hasGuardianScreen:p,onCancel:m,onSignTransaction:d,onTransactionsSignError:_,onTransactionsSignSuccess:u,onGetScamAddressData:f})=>{var Lr,Nr,Cr,Rr;let h=Boolean(a),[b,v]=(0,be.useState)(0),[k,F]=(0,be.useState)(),[A,U]=(0,be.useState)(null),[D,W]=(0,be.useState)({}),[C,Oe]=(0,be.useState)(!1),{getTxInfoByDataField:No,allTransactions:H}=ts({transactions:a?o==null?void 0:o.map(E=>{E.sender=Me.Address.newFromBech32(r),E.version=Me.TransactionVersion.withTxOptions().valueOf(),E.guardian=Me.Address.newFromBech32(a);let M=g({guarded:!0},e?{hashSign:!0}:{});return E.options=Me.TransactionOptions.withOptions(M).valueOf(),E}):o}),[ae,we]=(0,be.useState)(H),Fe=b===ae.length-1,ie=ae[b],ce=(Lr=A==null?void 0:A.transaction)==null?void 0:Lr.sender.toBech32(),{data:$}=Jn(!ce||ce===r?null:ce),Ve=()=>x(void 0,null,function*(){var Dr;if(ie==null)return;let{transaction:E,multiTxData:M,transactionIndex:J}=ie,_o=E.data.toString(),Ue=No(E.data.toString(),M),{tokenId:He}=Ue,le=E.receiver.toBech32();if(ce&&ce!==r&&!rs($,r))return console.error(ca),m==null?void 0:m();let ke=r!==le,se=le in ot;if(le&&ke&&!se&&f!=null){let Co=yield f(le);ot=g(g({},ot),Co!=null&&Co.scamInfo?{[le]:Co.scamInfo}:{})}let Q=Boolean(He&&wc({tokenId:He,erdLabel:n})),Tp=ie.needsSigning,yp=E.getNonce().valueOf();U({transaction:E,ppu:D[Number(yp)].ppu||bt,receiverScamInfo:((Dr=ot[le])==null?void 0:Dr.info)||null,transactionTokenInfo:Ue,isTokenTransaction:Q,dataField:_o,transactionIndex:J,needsSigning:Tp})}),ve=E=>{let M=ae[b].transaction,J=M.getNonce().valueOf(),_o=D[Number(J)].ppu,Ue=D[Number(J)].initialGasPrice;if(_o===E)return;W(se=>{let Q=g({},se);return Q[Number(J)]=S(g({},Q[Number(J)]),{ppu:E}),Q});let He=E?np({transactionDataLength:M.data.length,transactionGasLimit:Number(M.gasLimit),ppu:E}):Ue,le=Me.Transaction.fromPlainObject(S(g({},M.toPlainObject()),{gasPrice:He})),ke=S(g({},ae[b]),{transaction:le});we(se=>{let Q=[...se];return Q[b]=ke,Q}),U(se=>se&&S(g({},se),{transaction:le,ppu:E}))};(0,be.useEffect)(()=>{let E={};H.forEach(M=>{let J=M.transaction.nonce;E[Number(J)]={initialGasPrice:Number(M.transaction.gasPrice),ppu:bt}}),W(E),we(H)},[H]),(0,be.useEffect)(()=>{Ve()},[ie,$]);let Be=()=>{v(0),F(void 0),Oe(!1)},_e=()=>x(void 0,null,function*(){let E=Object.values(k!=null?k:{}),M=Zo({isGuarded:h,transactions:E});if(h&&M){u(E),Be();return}if(A==null)return;Oe(e);let J;try{J=yield d(A.transaction)}catch(ke){console.error(ke,"sign error");let{message:se}=ke,Q=e?si(ke).errorMessage:null;Be(),_(Q!=null?Q:se);return}if(!J)return;let _o={[b]:J},Ue=k?g(g({},k),_o):_o;if(F(Ue),!Fe){v(ke=>ke+1),Oe(!1);return}if(!Ue)return;let He=Object.values(Ue);!Zo({isGuarded:h,transactions:He})&&p||(u(He),Be())}),Ae=()=>x(void 0,null,function*(){try{if(A==null)return;if(A.transaction.getSignature().toString("hex")&&!Fe){v(M=>M+1);return}yield _e()}catch(E){console.error("Error during signing transaction"),yield _e()}}),nt=b===0,xp=()=>{nt?m==null||m():v(E=>E-1)},Pr=Boolean(((Nr=A==null?void 0:A.transactionTokenInfo)==null?void 0:Nr.type)&&((Cr=A==null?void 0:A.transactionTokenInfo)==null?void 0:Cr.multiTxData)&&!(A!=null&&A.dataField.endsWith((Rr=A==null?void 0:A.transactionTokenInfo)==null?void 0:Rr.multiTxData)));return{allTransactions:ae,onSignTransaction:()=>x(void 0,null,function*(){if(Pr){v(E=>E+1);return}yield Ae()}),onNext:()=>{v(E=>{let M=E+1;return M>(H==null?void 0:H.length)?E:M})},onPrev:()=>{v(E=>{let M=E-1;return M<0?E:M})},waitingForDevice:C,onAbort:xp,isLastTransaction:Fe,isFirstTransaction:b===0,hasMultipleTransactions:H.length>1,shouldContinueWithoutSigning:Pr,currentStep:b,signedTransactions:k,setSignedTransactions:F,updatePPU:ve,currentTransaction:A}};V();G();Sn();te();t();gt();V();function rp(){let e=$e();return o=>{e(yo()),e(Pn(o)),e(eo(cn))}}t();var Lo=require("react"),kr=require("@multiversx/sdk-extension-provider"),Ir=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Er=require("@multiversx/sdk-passkey-provider/out");t();t();var vr=()=>N(oo);t();t();t();t();t();t();t();var en=w(require("react"));var sL=(0,en.createContext)({}),pL=I.getState();t();var ip=w(require("react"));t();var Ar=w(require("react")),nl=w(require("axios"));t();t();t();var cp=require("react"),sp=require("@multiversx/sdk-web-wallet-provider"),pp=require("@multiversx/sdk-web-wallet-provider"),mp=w(require("qs"));P();V();oe();yt();te();var tl=R();function dp(e){let o=N(no),n=$e(),{shouldFetchWalletUrlOnLoad:r}=N(jt),{search:a}=r?tl:R();(0,cp.useEffect)(()=>{if(a!=null){let p=mp.default.parse(a.replace("?",""));if(p&&We in p){let m=String(p[We]),d=new sp.WalletProvider(`${o.walletAddress}${Na}`).getTransactionsFromWalletUrl(a);if(p.status==="cancelled"){n(bo({sessionId:m,status:"cancelled"})),e(),qn({removeParams:[...Object.keys(Tt),pp.WALLET_PROVIDER_CALLBACK_PARAM,We,"address"],search:a}),n(eo(cn));return}if(d.length>0){n(bo({sessionId:m,status:"signed",transactions:d.map(u=>qo(u))}));let[_]=d;Lc({transaction:_,search:a})}}}},[a])}Do();t();t();var up=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),gp=require("@multiversx/sdk-webview-provider/out/WebviewProvider");j();t();j();V();G();function _p(e,o=I.dispatch){let n=fr(e);if(n==null)return;let{signature:r,address:a}=n;r&&e&&a&&(o(qa({loginToken:e,signature:r,nativeAuthToken:e})),o(Ze({address:a,loginMethod:"extra"})))}t();t();t();P();V();function lp(){I.dispatch(yo())}t();Qe();t();V();G();t();t();var al=require("@multiversx/sdk-core"),il=w(require("bignumber.js"));P();t();var cl=w(require("bignumber.js"));P();var Se=class{constructor(){this.init=()=>x(this,null,function*(){return yield this._provider.init()});this.login=()=>x(this,null,function*(){var n;let o=yield this._provider.login();return{address:(n=o==null?void 0:o.address)!=null?n:""}});this.logout=()=>x(this,null,function*(){return I.dispatch(L()),yield this._provider.logout()});this.relogin=()=>x(this,null,function*(){let o=yield this._provider.relogin();return o?(_p(o),Yc(),o):(console.error("Unable to re-login. Missing accessToken."),null)});this.signTransactions=o=>x(this,null,function*(){let n=yield this._provider.signTransactions(o);return n||(lp(),this._provider.cancelAction(),null)});this.signTransaction=o=>x(this,null,function*(){return yield this._provider.signTransaction(o)});this.signMessage=o=>x(this,null,function*(){return yield this._provider.signMessage(o)});this.cancelAction=()=>x(this,null,function*(){return yield this._provider.cancelAction()});this.isInitialized=()=>this._provider.isInitialized();this.isConnected=()=>this._provider.isConnected();this.sendCustomRequest=o=>x(this,null,function*(){this._provider.sendPostMessage({type:o.request.method,payload:o.request.params})});this.getAddress=(0,up.providerNotInitializedError)("getAddress");this._provider=gp.WebviewProvider.getInstance({resetStateCallback:()=>I.dispatch(L())})}static getInstance(){return Se._instance||(Se._instance=new Se),Se._instance}getAccount(){var n;let o=this._provider.getAccount();return{address:(n=o==null?void 0:o.address)!=null?n:""}}setAccount(o){this._provider.setAccount(o)}};V();var fp=()=>{let e=$e(),{provider:o}=ko(),{nonce:n}=vr(),[r,a]=(0,Lo.useState)(null),[p,m]=(0,Lo.useState)(),[d,_]=(0,Lo.useState)([]),u=ss(),f=N(Yn),h=N(Dc),b=()=>x(void 0,null,function*(){var D,W;let A=Boolean(f==null?void 0:f.transactions),U=(D=f==null?void 0:f.transactions)!=null?D:[];if(A){let C=(W=f==null?void 0:f.customTransactionInformation)!=null&&W.skipUpdateNonces?U:yield u(U);_(C)}m(A)});(0,Lo.useEffect)(()=>{b()},[f,n]);let v=()=>{a(null),e(eo(null))},k=A=>{v(),F(A)};dp(k);function F(A){var H,ae,we,Fe,ie,ce,$,Ve,ve,Be,_e,Ae;let U=o instanceof kr.ExtensionProvider,D=o instanceof Er.PasskeyProvider,W=o instanceof Ne.CrossWindowProvider,C=o instanceof co.IframeProvider,Oe=o instanceof Ir.MetamaskProvider,No=o instanceof Se;e(yo()),e(Pn(A)),!(!U&&!W&&!C&&!D&&!Oe)&&(v(),U&&((ae=(H=kr.ExtensionProvider.getInstance())==null?void 0:H.cancelAction)==null||ae.call(H)),D&&((Fe=(we=Er.PasskeyProvider.getInstance())==null?void 0:we.cancelAction)==null||Fe.call(we)),Oe&&((ce=(ie=Ir.MetamaskProvider.getInstance())==null?void 0:ie.cancelAction)==null||ce.call(ie)),W&&((Ve=($=Ne.CrossWindowProvider.getInstance())==null?void 0:$.cancelAction)==null||Ve.call($)),C&&((Be=(ve=co.IframeProvider.getInstance())==null?void 0:ve.cancelAction)==null||Be.call(ve)),No&&((Ae=(_e=Se.getInstance())==null?void 0:_e.cancelAction)==null||Ae.call(_e)))}return{error:r,canceledTransactionsMessage:h,clearTransactionStatusMessage:v,onAbort:k,setError:a,hasTransactions:p,transactionsToSign:f&&S(g({},f),{transactions:d})}};function hp(e){let{onCancel:o,verifyReceiverScam:n=!0,hasGuardianScreen:r}=e,{transactionsToSign:a,hasTransactions:p}=fp(),m=N(ji),d=op(),_=N(zi),{account:u}=Bc(),{address:f,isGuarded:h,activeGuardianAddress:b}=u,{provider:v,providerType:k}=ko(),F=$e(),A=rp(),{transactions:U,sessionId:D,callbackRoute:W,customTransactionInformation:C}=a||{};function Oe($){D&&F(bo({sessionId:D,status:"cancelled"})),F(Ei($))}let{pathname:No}=R(),H=W!=null&&No.includes(W),ae=!(C!=null&&C.skipGuardian);function we($){return x(this,null,function*(){if(!as($))return;let{needs2FaSigning:ve,guardTransactions:Be}=Hs({transactions:$,sessionId:D,callbackRoute:W,isGuarded:h&&ae,walletAddress:m}),_e=$.map(Ae=>qo(Ae));if(ve){let Ae=yield Be();_e=Ae?Ae.map(nt=>qo(nt)):[]}!D||(F(bo({sessionId:D,status:"signed",transactions:_e})),W!=null&&(C==null?void 0:C.redirectAfterSign)&&!H&&Wo({url:W}))})}function Fe(){o(),A(D)}function ie($){return x(this,null,function*(){let Ve=k!=="ledger"?v:yield d();return $?yield Ve.signTransaction($):null})}let ce=tp({address:f,egldLabel:_,activeGuardianAddress:h&&ae?b:void 0,transactionsToSign:p?U:[],onGetScamAddressData:n?Oc:null,isLedger:mo("ledger"),hasGuardianScreen:r,onCancel:Fe,onSignTransaction:ie,onTransactionsSignError:Oe,onTransactionsSignSuccess:we});return S(g({},ce),{callbackRoute:W,waitingForDevice:ce.waitingForDevice})}function sl(e){return hp(e)}0&&(module.exports={useSignTransactionsWithLedger});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
