// @multiversx/sdk-dapp/__commonjs/hooks/toasts/index.js
"use strict";var sp=Object.create;var ze=Object.defineProperty,cp=Object.defineProperties,pp=Object.getOwnPropertyDescriptor,mp=Object.getOwnPropertyDescriptors,dp=Object.getOwnPropertyNames,Tn=Object.getOwnPropertySymbols,lp=Object.getPrototypeOf,uo=Object.prototype.hasOwnProperty,Ut=Object.prototype.propertyIsEnumerable;var Ft=(e,n,t)=>n in e?ze(e,n,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[n]=t,_=(e,n)=>{for(var t in n||(n={}))uo.call(n,t)&&Ft(e,t,n[t]);if(Tn)for(var t of Tn(n))Ut.call(n,t)&&Ft(e,t,n[t]);return e},w=(e,n)=>cp(e,mp(n));var ve=(e,n)=>{var t={};for(var a in e)uo.call(e,a)&&n.indexOf(a)<0&&(t[a]=e[a]);if(e!=null&&Tn)for(var a of Tn(e))n.indexOf(a)<0&&Ut.call(e,a)&&(t[a]=e[a]);return t};var g=(e,n)=>()=>(e&&(n=e(e=0)),n);var T=(e,n)=>()=>(n||e((n={exports:{}}).exports,n),n.exports),ee=(e,n)=>{for(var t in n)ze(e,t,{get:n[t],enumerable:!0})},Wt=(e,n,t,a)=>{if(n&&typeof n=="object"||typeof n=="function")for(let c of dp(n))!uo.call(e,c)&&c!==t&&ze(e,c,{get:()=>n[c],enumerable:!(a=pp(n,c))||a.enumerable});return e};var b=(e,n,t)=>(t=e!=null?sp(lp(e)):{},Wt(n||!e||!e.__esModule?ze(t,"default",{value:e,enumerable:!0}):t,e)),_e=e=>Wt(ze({},"__esModule",{value:!0}),e);var h=(e,n,t)=>new Promise((a,c)=>{var p=l=>{try{u(t.next(l))}catch(x){c(x)}},m=l=>{try{u(t.throw(l))}catch(x){c(x)}},u=l=>l.done?a(l.value):Promise.resolve(l.value).then(p,m);u((t=t.apply(e,n)).next())});var Ht=T(hn=>{"use strict";o();hn.byteLength=up;hn.toByteArray=fp;hn.fromByteArray=xp;var q=[],$=[],_p=typeof Uint8Array!="undefined"?Uint8Array:Array,go="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ue=0,Gt=go.length;ue<Gt;++ue)q[ue]=go[ue],$[go.charCodeAt(ue)]=ue;var ue,Gt;$["-".charCodeAt(0)]=62;$["_".charCodeAt(0)]=63;function $t(e){var n=e.length;if(n%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var t=e.indexOf("=");t===-1&&(t=n);var a=t===n?0:4-t%4;return[t,a]}function up(e){var n=$t(e),t=n[0],a=n[1];return(t+a)*3/4-a}function gp(e,n,t){return(n+t)*3/4-t}function fp(e){var n,t=$t(e),a=t[0],c=t[1],p=new _p(gp(e,a,c)),m=0,u=c>0?a-4:a,l;for(l=0;l<u;l+=4)n=$[e.charCodeAt(l)]<<18|$[e.charCodeAt(l+1)]<<12|$[e.charCodeAt(l+2)]<<6|$[e.charCodeAt(l+3)],p[m++]=n>>16&255,p[m++]=n>>8&255,p[m++]=n&255;return c===2&&(n=$[e.charCodeAt(l)]<<2|$[e.charCodeAt(l+1)]>>4,p[m++]=n&255),c===1&&(n=$[e.charCodeAt(l)]<<10|$[e.charCodeAt(l+1)]<<4|$[e.charCodeAt(l+2)]>>2,p[m++]=n>>8&255,p[m++]=n&255),p}function Tp(e){return q[e>>18&63]+q[e>>12&63]+q[e>>6&63]+q[e&63]}function hp(e,n,t){for(var a,c=[],p=n;p<t;p+=3)a=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),c.push(Tp(a));return c.join("")}function xp(e){for(var n,t=e.length,a=t%3,c=[],p=16383,m=0,u=t-a;m<u;m+=p)c.push(hp(e,m,m+p>u?u:m+p));return a===1?(n=e[t-1],c.push(q[n>>2]+q[n<<4&63]+"==")):a===2&&(n=(e[t-2]<<8)+e[t-1],c.push(q[n>>10]+q[n>>4&63]+q[n<<2&63]+"=")),c.join("")}});var Vt=T(fo=>{o();fo.read=function(e,n,t,a,c){var p,m,u=c*8-a-1,l=(1<<u)-1,x=l>>1,S=-7,y=t?c-1:0,L=t?-1:1,R=e[n+y];for(y+=L,p=R&(1<<-S)-1,R>>=-S,S+=u;S>0;p=p*256+e[n+y],y+=L,S-=8);for(m=p&(1<<-S)-1,p>>=-S,S+=a;S>0;m=m*256+e[n+y],y+=L,S-=8);if(p===0)p=1-x;else{if(p===l)return m?NaN:(R?-1:1)*(1/0);m=m+Math.pow(2,a),p=p-x}return(R?-1:1)*m*Math.pow(2,p-a)};fo.write=function(e,n,t,a,c,p){var m,u,l,x=p*8-c-1,S=(1<<x)-1,y=S>>1,L=c===23?Math.pow(2,-24)-Math.pow(2,-77):0,R=a?0:p-1,J=a?1:-1,fn=n<0||n===0&&1/n<0?1:0;for(n=Math.abs(n),isNaN(n)||n===1/0?(u=isNaN(n)?1:0,m=S):(m=Math.floor(Math.log(n)/Math.LN2),n*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+y>=1?n+=L/l:n+=L*Math.pow(2,1-y),n*l>=2&&(m++,l/=2),m+y>=S?(u=0,m=S):m+y>=1?(u=(n*l-1)*Math.pow(2,c),m=m+y):(u=n*Math.pow(2,y-1)*Math.pow(2,c),m=0));c>=8;e[t+R]=u&255,R+=J,u/=256,c-=8);for(m=m<<c|u,x+=c;x>0;e[t+R]=m&255,R+=J,m/=256,x-=8);e[t+R-J]|=fn*128}});var tr=T(Ie=>{"use strict";o();var To=Ht(),Ae=Vt(),qt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ie.Buffer=d;Ie.SlowBuffer=Ap;Ie.INSPECT_MAX_BYTES=50;var xn=2147483647;Ie.kMaxLength=xn;d.TYPED_ARRAY_SUPPORT=yp();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function yp(){try{var e=new Uint8Array(1),n={foo:function(){return 42}};return Object.setPrototypeOf(n,Uint8Array.prototype),Object.setPrototypeOf(e,n),e.foo()===42}catch(t){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function ne(e){if(e>xn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var n=new Uint8Array(e);return Object.setPrototypeOf(n,d.prototype),n}function d(e,n,t){if(typeof e=="number"){if(typeof n=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return bo(e)}return Kt(e,n,t)}d.poolSize=8192;function Kt(e,n,t){if(typeof e=="string")return Sp(e,n);if(ArrayBuffer.isView(e))return wp(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(j(e,ArrayBuffer)||e&&j(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(j(e,SharedArrayBuffer)||e&&j(e.buffer,SharedArrayBuffer)))return xo(e,n,t);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var a=e.valueOf&&e.valueOf();if(a!=null&&a!==e)return d.from(a,n,t);var c=vp(e);if(c)return c;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),n,t);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,n,t){return Kt(e,n,t)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function Xt(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function bp(e,n,t){return Xt(e),e<=0?ne(e):n!==void 0?typeof t=="string"?ne(e).fill(n,t):ne(e).fill(n):ne(e)}d.alloc=function(e,n,t){return bp(e,n,t)};function bo(e){return Xt(e),ne(e<0?0:So(e)|0)}d.allocUnsafe=function(e){return bo(e)};d.allocUnsafeSlow=function(e){return bo(e)};function Sp(e,n){if((typeof n!="string"||n==="")&&(n="utf8"),!d.isEncoding(n))throw new TypeError("Unknown encoding: "+n);var t=Yt(e,n)|0,a=ne(t),c=a.write(e,n);return c!==t&&(a=a.slice(0,c)),a}function ho(e){for(var n=e.length<0?0:So(e.length)|0,t=ne(n),a=0;a<n;a+=1)t[a]=e[a]&255;return t}function wp(e){if(j(e,Uint8Array)){var n=new Uint8Array(e);return xo(n.buffer,n.byteOffset,n.byteLength)}return ho(e)}function xo(e,n,t){if(n<0||e.byteLength<n)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<n+(t||0))throw new RangeError('"length" is outside of buffer bounds');var a;return n===void 0&&t===void 0?a=new Uint8Array(e):t===void 0?a=new Uint8Array(e,n):a=new Uint8Array(e,n,t),Object.setPrototypeOf(a,d.prototype),a}function vp(e){if(d.isBuffer(e)){var n=So(e.length)|0,t=ne(n);return t.length===0||e.copy(t,0,0,n),t}if(e.length!==void 0)return typeof e.length!="number"||wo(e.length)?ne(0):ho(e);if(e.type==="Buffer"&&Array.isArray(e.data))return ho(e.data)}function So(e){if(e>=xn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+xn.toString(16)+" bytes");return e|0}function Ap(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(n){return n!=null&&n._isBuffer===!0&&n!==d.prototype};d.compare=function(n,t){if(j(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),j(t,Uint8Array)&&(t=d.from(t,t.offset,t.byteLength)),!d.isBuffer(n)||!d.isBuffer(t))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(n===t)return 0;for(var a=n.length,c=t.length,p=0,m=Math.min(a,c);p<m;++p)if(n[p]!==t[p]){a=n[p],c=t[p];break}return a<c?-1:c<a?1:0};d.isEncoding=function(n){switch(String(n).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(n,t){if(!Array.isArray(n))throw new TypeError('"list" argument must be an Array of Buffers');if(n.length===0)return d.alloc(0);var a;if(t===void 0)for(t=0,a=0;a<n.length;++a)t+=n[a].length;var c=d.allocUnsafe(t),p=0;for(a=0;a<n.length;++a){var m=n[a];if(j(m,Uint8Array))p+m.length>c.length?d.from(m).copy(c,p):Uint8Array.prototype.set.call(c,m,p);else if(d.isBuffer(m))m.copy(c,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return c};function Yt(e,n){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||j(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var t=e.length,a=arguments.length>2&&arguments[2]===!0;if(!a&&t===0)return 0;for(var c=!1;;)switch(n){case"ascii":case"latin1":case"binary":return t;case"utf8":case"utf-8":return yo(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return t*2;case"hex":return t>>>1;case"base64":return or(e).length;default:if(c)return a?-1:yo(e).length;n=(""+n).toLowerCase(),c=!0}}d.byteLength=Yt;function Ip(e,n,t){var a=!1;if((n===void 0||n<0)&&(n=0),n>this.length||((t===void 0||t>this.length)&&(t=this.length),t<=0)||(t>>>=0,n>>>=0,t<=n))return"";for(e||(e="utf8");;)switch(e){case"hex":return Mp(this,n,t);case"utf8":case"utf-8":return Jt(this,n,t);case"ascii":return Dp(this,n,t);case"latin1":case"binary":return Op(this,n,t);case"base64":return Cp(this,n,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Bp(this,n,t);default:if(a)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),a=!0}}d.prototype._isBuffer=!0;function ge(e,n,t){var a=e[n];e[n]=e[t],e[t]=a}d.prototype.swap16=function(){var n=this.length;if(n%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var t=0;t<n;t+=2)ge(this,t,t+1);return this};d.prototype.swap32=function(){var n=this.length;if(n%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var t=0;t<n;t+=4)ge(this,t,t+3),ge(this,t+1,t+2);return this};d.prototype.swap64=function(){var n=this.length;if(n%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var t=0;t<n;t+=8)ge(this,t,t+7),ge(this,t+1,t+6),ge(this,t+2,t+5),ge(this,t+3,t+4);return this};d.prototype.toString=function(){var n=this.length;return n===0?"":arguments.length===0?Jt(this,0,n):Ip.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(n){if(!d.isBuffer(n))throw new TypeError("Argument must be a Buffer");return this===n?!0:d.compare(this,n)===0};d.prototype.inspect=function(){var n="",t=Ie.INSPECT_MAX_BYTES;return n=this.toString("hex",0,t).replace(/(.{2})/g,"$1 ").trim(),this.length>t&&(n+=" ... "),"<Buffer "+n+">"};qt&&(d.prototype[qt]=d.prototype.inspect);d.prototype.compare=function(n,t,a,c,p){if(j(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(n))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof n);if(t===void 0&&(t=0),a===void 0&&(a=n?n.length:0),c===void 0&&(c=0),p===void 0&&(p=this.length),t<0||a>n.length||c<0||p>this.length)throw new RangeError("out of range index");if(c>=p&&t>=a)return 0;if(c>=p)return-1;if(t>=a)return 1;if(t>>>=0,a>>>=0,c>>>=0,p>>>=0,this===n)return 0;for(var m=p-c,u=a-t,l=Math.min(m,u),x=this.slice(c,p),S=n.slice(t,a),y=0;y<l;++y)if(x[y]!==S[y]){m=x[y],u=S[y];break}return m<u?-1:u<m?1:0};function Zt(e,n,t,a,c){if(e.length===0)return-1;if(typeof t=="string"?(a=t,t=0):t>2147483647?t=2147483647:t<-2147483648&&(t=-2147483648),t=+t,wo(t)&&(t=c?0:e.length-1),t<0&&(t=e.length+t),t>=e.length){if(c)return-1;t=e.length-1}else if(t<0)if(c)t=0;else return-1;if(typeof n=="string"&&(n=d.from(n,a)),d.isBuffer(n))return n.length===0?-1:jt(e,n,t,a,c);if(typeof n=="number")return n=n&255,typeof Uint8Array.prototype.indexOf=="function"?c?Uint8Array.prototype.indexOf.call(e,n,t):Uint8Array.prototype.lastIndexOf.call(e,n,t):jt(e,[n],t,a,c);throw new TypeError("val must be string, number or Buffer")}function jt(e,n,t,a,c){var p=1,m=e.length,u=n.length;if(a!==void 0&&(a=String(a).toLowerCase(),a==="ucs2"||a==="ucs-2"||a==="utf16le"||a==="utf-16le")){if(e.length<2||n.length<2)return-1;p=2,m/=2,u/=2,t/=2}function l(R,J){return p===1?R[J]:R.readUInt16BE(J*p)}var x;if(c){var S=-1;for(x=t;x<m;x++)if(l(e,x)===l(n,S===-1?0:x-S)){if(S===-1&&(S=x),x-S+1===u)return S*p}else S!==-1&&(x-=x-S),S=-1}else for(t+u>m&&(t=m-u),x=t;x>=0;x--){for(var y=!0,L=0;L<u;L++)if(l(e,x+L)!==l(n,L)){y=!1;break}if(y)return x}return-1}d.prototype.includes=function(n,t,a){return this.indexOf(n,t,a)!==-1};d.prototype.indexOf=function(n,t,a){return Zt(this,n,t,a,!0)};d.prototype.lastIndexOf=function(n,t,a){return Zt(this,n,t,a,!1)};function kp(e,n,t,a){t=Number(t)||0;var c=e.length-t;a?(a=Number(a),a>c&&(a=c)):a=c;var p=n.length;a>p/2&&(a=p/2);for(var m=0;m<a;++m){var u=parseInt(n.substr(m*2,2),16);if(wo(u))return m;e[t+m]=u}return m}function Ep(e,n,t,a){return yn(yo(n,e.length-t),e,t,a)}function Pp(e,n,t,a){return yn(Wp(n),e,t,a)}function Np(e,n,t,a){return yn(or(n),e,t,a)}function Lp(e,n,t,a){return yn(Gp(n,e.length-t),e,t,a)}d.prototype.write=function(n,t,a,c){if(t===void 0)c="utf8",a=this.length,t=0;else if(a===void 0&&typeof t=="string")c=t,a=this.length,t=0;else if(isFinite(t))t=t>>>0,isFinite(a)?(a=a>>>0,c===void 0&&(c="utf8")):(c=a,a=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-t;if((a===void 0||a>p)&&(a=p),n.length>0&&(a<0||t<0)||t>this.length)throw new RangeError("Attempt to write outside buffer bounds");c||(c="utf8");for(var m=!1;;)switch(c){case"hex":return kp(this,n,t,a);case"utf8":case"utf-8":return Ep(this,n,t,a);case"ascii":case"latin1":case"binary":return Pp(this,n,t,a);case"base64":return Np(this,n,t,a);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Lp(this,n,t,a);default:if(m)throw new TypeError("Unknown encoding: "+c);c=(""+c).toLowerCase(),m=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Cp(e,n,t){return n===0&&t===e.length?To.fromByteArray(e):To.fromByteArray(e.slice(n,t))}function Jt(e,n,t){t=Math.min(e.length,t);for(var a=[],c=n;c<t;){var p=e[c],m=null,u=p>239?4:p>223?3:p>191?2:1;if(c+u<=t){var l,x,S,y;switch(u){case 1:p<128&&(m=p);break;case 2:l=e[c+1],(l&192)===128&&(y=(p&31)<<6|l&63,y>127&&(m=y));break;case 3:l=e[c+1],x=e[c+2],(l&192)===128&&(x&192)===128&&(y=(p&15)<<12|(l&63)<<6|x&63,y>2047&&(y<55296||y>57343)&&(m=y));break;case 4:l=e[c+1],x=e[c+2],S=e[c+3],(l&192)===128&&(x&192)===128&&(S&192)===128&&(y=(p&15)<<18|(l&63)<<12|(x&63)<<6|S&63,y>65535&&y<1114112&&(m=y))}}m===null?(m=65533,u=1):m>65535&&(m-=65536,a.push(m>>>10&1023|55296),m=56320|m&1023),a.push(m),c+=u}return Rp(a)}var zt=4096;function Rp(e){var n=e.length;if(n<=zt)return String.fromCharCode.apply(String,e);for(var t="",a=0;a<n;)t+=String.fromCharCode.apply(String,e.slice(a,a+=zt));return t}function Dp(e,n,t){var a="";t=Math.min(e.length,t);for(var c=n;c<t;++c)a+=String.fromCharCode(e[c]&127);return a}function Op(e,n,t){var a="";t=Math.min(e.length,t);for(var c=n;c<t;++c)a+=String.fromCharCode(e[c]);return a}function Mp(e,n,t){var a=e.length;(!n||n<0)&&(n=0),(!t||t<0||t>a)&&(t=a);for(var c="",p=n;p<t;++p)c+=$p[e[p]];return c}function Bp(e,n,t){for(var a=e.slice(n,t),c="",p=0;p<a.length-1;p+=2)c+=String.fromCharCode(a[p]+a[p+1]*256);return c}d.prototype.slice=function(n,t){var a=this.length;n=~~n,t=t===void 0?a:~~t,n<0?(n+=a,n<0&&(n=0)):n>a&&(n=a),t<0?(t+=a,t<0&&(t=0)):t>a&&(t=a),t<n&&(t=n);var c=this.subarray(n,t);return Object.setPrototypeOf(c,d.prototype),c};function C(e,n,t){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+n>t)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(n,t,a){n=n>>>0,t=t>>>0,a||C(n,t,this.length);for(var c=this[n],p=1,m=0;++m<t&&(p*=256);)c+=this[n+m]*p;return c};d.prototype.readUintBE=d.prototype.readUIntBE=function(n,t,a){n=n>>>0,t=t>>>0,a||C(n,t,this.length);for(var c=this[n+--t],p=1;t>0&&(p*=256);)c+=this[n+--t]*p;return c};d.prototype.readUint8=d.prototype.readUInt8=function(n,t){return n=n>>>0,t||C(n,1,this.length),this[n]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(n,t){return n=n>>>0,t||C(n,2,this.length),this[n]|this[n+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(n,t){return n=n>>>0,t||C(n,2,this.length),this[n]<<8|this[n+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(n,t){return n=n>>>0,t||C(n,4,this.length),(this[n]|this[n+1]<<8|this[n+2]<<16)+this[n+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(n,t){return n=n>>>0,t||C(n,4,this.length),this[n]*16777216+(this[n+1]<<16|this[n+2]<<8|this[n+3])};d.prototype.readIntLE=function(n,t,a){n=n>>>0,t=t>>>0,a||C(n,t,this.length);for(var c=this[n],p=1,m=0;++m<t&&(p*=256);)c+=this[n+m]*p;return p*=128,c>=p&&(c-=Math.pow(2,8*t)),c};d.prototype.readIntBE=function(n,t,a){n=n>>>0,t=t>>>0,a||C(n,t,this.length);for(var c=t,p=1,m=this[n+--c];c>0&&(p*=256);)m+=this[n+--c]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*t)),m};d.prototype.readInt8=function(n,t){return n=n>>>0,t||C(n,1,this.length),this[n]&128?(255-this[n]+1)*-1:this[n]};d.prototype.readInt16LE=function(n,t){n=n>>>0,t||C(n,2,this.length);var a=this[n]|this[n+1]<<8;return a&32768?a|4294901760:a};d.prototype.readInt16BE=function(n,t){n=n>>>0,t||C(n,2,this.length);var a=this[n+1]|this[n]<<8;return a&32768?a|4294901760:a};d.prototype.readInt32LE=function(n,t){return n=n>>>0,t||C(n,4,this.length),this[n]|this[n+1]<<8|this[n+2]<<16|this[n+3]<<24};d.prototype.readInt32BE=function(n,t){return n=n>>>0,t||C(n,4,this.length),this[n]<<24|this[n+1]<<16|this[n+2]<<8|this[n+3]};d.prototype.readFloatLE=function(n,t){return n=n>>>0,t||C(n,4,this.length),Ae.read(this,n,!0,23,4)};d.prototype.readFloatBE=function(n,t){return n=n>>>0,t||C(n,4,this.length),Ae.read(this,n,!1,23,4)};d.prototype.readDoubleLE=function(n,t){return n=n>>>0,t||C(n,8,this.length),Ae.read(this,n,!0,52,8)};d.prototype.readDoubleBE=function(n,t){return n=n>>>0,t||C(n,8,this.length),Ae.read(this,n,!1,52,8)};function B(e,n,t,a,c,p){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(n>c||n<p)throw new RangeError('"value" argument is out of bounds');if(t+a>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(n,t,a,c){if(n=+n,t=t>>>0,a=a>>>0,!c){var p=Math.pow(2,8*a)-1;B(this,n,t,a,p,0)}var m=1,u=0;for(this[t]=n&255;++u<a&&(m*=256);)this[t+u]=n/m&255;return t+a};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(n,t,a,c){if(n=+n,t=t>>>0,a=a>>>0,!c){var p=Math.pow(2,8*a)-1;B(this,n,t,a,p,0)}var m=a-1,u=1;for(this[t+m]=n&255;--m>=0&&(u*=256);)this[t+m]=n/u&255;return t+a};d.prototype.writeUint8=d.prototype.writeUInt8=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,1,255,0),this[t]=n&255,t+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,2,65535,0),this[t]=n&255,this[t+1]=n>>>8,t+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,2,65535,0),this[t]=n>>>8,this[t+1]=n&255,t+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,4,4294967295,0),this[t+3]=n>>>24,this[t+2]=n>>>16,this[t+1]=n>>>8,this[t]=n&255,t+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,4,4294967295,0),this[t]=n>>>24,this[t+1]=n>>>16,this[t+2]=n>>>8,this[t+3]=n&255,t+4};d.prototype.writeIntLE=function(n,t,a,c){if(n=+n,t=t>>>0,!c){var p=Math.pow(2,8*a-1);B(this,n,t,a,p-1,-p)}var m=0,u=1,l=0;for(this[t]=n&255;++m<a&&(u*=256);)n<0&&l===0&&this[t+m-1]!==0&&(l=1),this[t+m]=(n/u>>0)-l&255;return t+a};d.prototype.writeIntBE=function(n,t,a,c){if(n=+n,t=t>>>0,!c){var p=Math.pow(2,8*a-1);B(this,n,t,a,p-1,-p)}var m=a-1,u=1,l=0;for(this[t+m]=n&255;--m>=0&&(u*=256);)n<0&&l===0&&this[t+m+1]!==0&&(l=1),this[t+m]=(n/u>>0)-l&255;return t+a};d.prototype.writeInt8=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,1,127,-128),n<0&&(n=255+n+1),this[t]=n&255,t+1};d.prototype.writeInt16LE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,2,32767,-32768),this[t]=n&255,this[t+1]=n>>>8,t+2};d.prototype.writeInt16BE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,2,32767,-32768),this[t]=n>>>8,this[t+1]=n&255,t+2};d.prototype.writeInt32LE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,4,2147483647,-2147483648),this[t]=n&255,this[t+1]=n>>>8,this[t+2]=n>>>16,this[t+3]=n>>>24,t+4};d.prototype.writeInt32BE=function(n,t,a){return n=+n,t=t>>>0,a||B(this,n,t,4,2147483647,-2147483648),n<0&&(n=4294967295+n+1),this[t]=n>>>24,this[t+1]=n>>>16,this[t+2]=n>>>8,this[t+3]=n&255,t+4};function Qt(e,n,t,a,c,p){if(t+a>e.length)throw new RangeError("Index out of range");if(t<0)throw new RangeError("Index out of range")}function er(e,n,t,a,c){return n=+n,t=t>>>0,c||Qt(e,n,t,4,34028234663852886e22,-34028234663852886e22),Ae.write(e,n,t,a,23,4),t+4}d.prototype.writeFloatLE=function(n,t,a){return er(this,n,t,!0,a)};d.prototype.writeFloatBE=function(n,t,a){return er(this,n,t,!1,a)};function nr(e,n,t,a,c){return n=+n,t=t>>>0,c||Qt(e,n,t,8,17976931348623157e292,-17976931348623157e292),Ae.write(e,n,t,a,52,8),t+8}d.prototype.writeDoubleLE=function(n,t,a){return nr(this,n,t,!0,a)};d.prototype.writeDoubleBE=function(n,t,a){return nr(this,n,t,!1,a)};d.prototype.copy=function(n,t,a,c){if(!d.isBuffer(n))throw new TypeError("argument should be a Buffer");if(a||(a=0),!c&&c!==0&&(c=this.length),t>=n.length&&(t=n.length),t||(t=0),c>0&&c<a&&(c=a),c===a||n.length===0||this.length===0)return 0;if(t<0)throw new RangeError("targetStart out of bounds");if(a<0||a>=this.length)throw new RangeError("Index out of range");if(c<0)throw new RangeError("sourceEnd out of bounds");c>this.length&&(c=this.length),n.length-t<c-a&&(c=n.length-t+a);var p=c-a;return this===n&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(t,a,c):Uint8Array.prototype.set.call(n,this.subarray(a,c),t),p};d.prototype.fill=function(n,t,a,c){if(typeof n=="string"){if(typeof t=="string"?(c=t,t=0,a=this.length):typeof a=="string"&&(c=a,a=this.length),c!==void 0&&typeof c!="string")throw new TypeError("encoding must be a string");if(typeof c=="string"&&!d.isEncoding(c))throw new TypeError("Unknown encoding: "+c);if(n.length===1){var p=n.charCodeAt(0);(c==="utf8"&&p<128||c==="latin1")&&(n=p)}}else typeof n=="number"?n=n&255:typeof n=="boolean"&&(n=Number(n));if(t<0||this.length<t||this.length<a)throw new RangeError("Out of range index");if(a<=t)return this;t=t>>>0,a=a===void 0?this.length:a>>>0,n||(n=0);var m;if(typeof n=="number")for(m=t;m<a;++m)this[m]=n;else{var u=d.isBuffer(n)?n:d.from(n,c),l=u.length;if(l===0)throw new TypeError('The value "'+n+'" is invalid for argument "value"');for(m=0;m<a-t;++m)this[m+t]=u[m%l]}return this};var Fp=/[^+/0-9A-Za-z-_]/g;function Up(e){if(e=e.split("=")[0],e=e.trim().replace(Fp,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function yo(e,n){n=n||1/0;for(var t,a=e.length,c=null,p=[],m=0;m<a;++m){if(t=e.charCodeAt(m),t>55295&&t<57344){if(!c){if(t>56319){(n-=3)>-1&&p.push(239,191,189);continue}else if(m+1===a){(n-=3)>-1&&p.push(239,191,189);continue}c=t;continue}if(t<56320){(n-=3)>-1&&p.push(239,191,189),c=t;continue}t=(c-55296<<10|t-56320)+65536}else c&&(n-=3)>-1&&p.push(239,191,189);if(c=null,t<128){if((n-=1)<0)break;p.push(t)}else if(t<2048){if((n-=2)<0)break;p.push(t>>6|192,t&63|128)}else if(t<65536){if((n-=3)<0)break;p.push(t>>12|224,t>>6&63|128,t&63|128)}else if(t<1114112){if((n-=4)<0)break;p.push(t>>18|240,t>>12&63|128,t>>6&63|128,t&63|128)}else throw new Error("Invalid code point")}return p}function Wp(e){for(var n=[],t=0;t<e.length;++t)n.push(e.charCodeAt(t)&255);return n}function Gp(e,n){for(var t,a,c,p=[],m=0;m<e.length&&!((n-=2)<0);++m)t=e.charCodeAt(m),a=t>>8,c=t%256,p.push(c),p.push(a);return p}function or(e){return To.toByteArray(Up(e))}function yn(e,n,t,a){for(var c=0;c<a&&!(c+t>=n.length||c>=e.length);++c)n[c+t]=e[c];return c}function j(e,n){return e instanceof n||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===n.name}function wo(e){return e!==e}var $p=function(){for(var e="0123456789abcdef",n=new Array(256),t=0;t<16;++t)for(var a=t*16,c=0;c<16;++c)n[a+c]=e[t]+e[c];return n}()});var cr=T((qf,sr)=>{o();var E=sr.exports={},z,K;function vo(){throw new Error("setTimeout has not been defined")}function Ao(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?z=setTimeout:z=vo}catch(e){z=vo}try{typeof clearTimeout=="function"?K=clearTimeout:K=Ao}catch(e){K=Ao}})();function rr(e){if(z===setTimeout)return setTimeout(e,0);if((z===vo||!z)&&setTimeout)return z=setTimeout,setTimeout(e,0);try{return z(e,0)}catch(n){try{return z.call(null,e,0)}catch(t){return z.call(this,e,0)}}}function Hp(e){if(K===clearTimeout)return clearTimeout(e);if((K===Ao||!K)&&clearTimeout)return K=clearTimeout,clearTimeout(e);try{return K(e)}catch(n){try{return K.call(null,e)}catch(t){return K.call(this,e)}}}var oe=[],ke=!1,fe,bn=-1;function Vp(){!ke||!fe||(ke=!1,fe.length?oe=fe.concat(oe):bn=-1,oe.length&&ar())}function ar(){if(!ke){var e=rr(Vp);ke=!0;for(var n=oe.length;n;){for(fe=oe,oe=[];++bn<n;)fe&&fe[bn].run();bn=-1,n=oe.length}fe=null,ke=!1,Hp(e)}}E.nextTick=function(e){var n=new Array(arguments.length-1);if(arguments.length>1)for(var t=1;t<arguments.length;t++)n[t-1]=arguments[t];oe.push(new ir(e,n)),oe.length===1&&!ke&&rr(ar)};function ir(e,n){this.fun=e,this.array=n}ir.prototype.run=function(){this.fun.apply(null,this.array)};E.title="browser";E.browser=!0;E.env={};E.argv=[];E.version="";E.versions={};function te(){}E.on=te;E.addListener=te;E.once=te;E.off=te;E.removeListener=te;E.removeAllListeners=te;E.emit=te;E.prependListener=te;E.prependOnceListener=te;E.listeners=function(e){return[]};E.binding=function(e){throw new Error("process.binding is not supported")};E.cwd=function(){return"/"};E.chdir=function(e){throw new Error("process.chdir is not supported")};E.umask=function(){return 0}});var r,s,qp,i,o=g(()=>{r=b(tr()),s=b(cr()),qp=function(e){function n(){var a=this||self;return delete e.prototype.__magic__,a}if(typeof globalThis=="object")return globalThis;if(this)return n();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:n});var t=__magic__;return t}(Object),i=qp});var pr=g(()=>{"use strict";o()});var Io=g(()=>{"use strict";o()});var mr=g(()=>{"use strict";o()});var ko,dr=g(()=>{"use strict";o();ko=(a=>(a.SetGuardian="SetGuardian",a.GuardAccount="GuardAccount",a.UnGuardAccount="UnGuardAccount",a))(ko||{})});var lr=g(()=>{"use strict";o()});var Eo=g(()=>{"use strict";o()});var _r=g(()=>{"use strict";o()});var Po=g(()=>{"use strict";o()});var ur=g(()=>{"use strict";o()});var gr=g(()=>{"use strict";o()});var Ke,Ee,Pe=g(()=>{"use strict";o();Ke=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),Ee=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var fr,tT,Tr,rT,D=g(()=>{"use strict";o();Pe();fr=(t=>(t.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",t.reloginRequest="RELOGIN_REQUEST",t))(fr||{}),tT=_(_({},Ee.WindowProviderRequestEnums),fr),Tr=(n=>(n.reloginResponse="RELOGIN_RESPONSE",n))(Tr||{}),rT=_(_({},Ee.WindowProviderResponseEnums),Tr)});var hr=g(()=>{"use strict";o()});var xr=g(()=>{"use strict";o()});var F=g(()=>{"use strict";o()});var yr=g(()=>{"use strict";o()});var br=g(()=>{"use strict";o()});var Sr=g(()=>{"use strict";o()});var N=g(()=>{"use strict";o();Po();ur();gr();D();hr();xr();F();yr();br();Sr()});var Ne,wr,bT,vr,ST,Ar,wT,vT,jp,Le=g(()=>{"use strict";o();N();Ne={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:wr,egldLabel:bT}=Ne["devnet"],{chainId:vr,egldLabel:ST}=Ne["testnet"],{chainId:Ar,egldLabel:wT}=Ne["mainnet"],vT={["devnet"]:wr,["testnet"]:vr,["mainnet"]:Ar},jp={[wr]:"devnet",[vr]:"testnet",[Ar]:"mainnet"}});var Xe=g(()=>{"use strict";o()});var X,Ir=g(()=>{"use strict";o();X=require("@multiversx/sdk-web-wallet-provider")});var Ye,Sn=g(()=>{"use strict";o();Ye=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var kr,Er,No,NT,LT,Pr=g(()=>{"use strict";o();Sn();No=String((Er=(kr=Ye.safeWindow)==null?void 0:kr.navigator)==null?void 0:Er.userAgent),NT=/^((?!chrome|android).)*safari/i.test(No),LT=/firefox/i.test(No)&&/windows/i.test(No)});var wn,vn,Nr,An,Lr,Ce,A=g(()=>{"use strict";o();pr();Io();mr();dr();lr();Eo();_r();Le();Xe();Ir();Pr();wn=5e4,vn=1e9,Nr=1,An="logout",Lr="login",Ce="0"});var Ze,In=g(()=>{"use strict";o();Ze=()=>Date.now()/1e3});var Cr=g(()=>{"use strict";o()});var Rr=g(()=>{"use strict";o()});var Dr=g(()=>{"use strict";o();In();Cr();Rr()});var Lo={};ee(Lo,{clear:()=>Yp,getItem:()=>Kp,localStorageKeys:()=>re,removeItem:()=>Xp,setItem:()=>zp});var re,kn,zp,Kp,Xp,Yp,Te=g(()=>{"use strict";o();Dr();re={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},kn=typeof localStorage!="undefined",zp=({key:e,data:n,expires:t})=>{!kn||localStorage.setItem(String(e),JSON.stringify({expires:t,data:n}))},Kp=e=>{if(!kn)return;let n=localStorage.getItem(String(e));if(!n)return null;let t=JSON.parse(n);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Ze()>=t.expires?(localStorage.removeItem(String(e)),null):t.data},Xp=e=>{!kn||localStorage.removeItem(String(e))},Yp=()=>{!kn||localStorage.clear()}});var Co={};ee(Co,{clear:()=>Fr,getItem:()=>Mr,removeItem:()=>Br,setItem:()=>Or,storage:()=>Zp});var Or,Mr,Br,Fr,Zp,Ur=g(()=>{"use strict";o();Or=({key:e,data:n,expires:t})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:t,data:n}))},Mr=e=>{let n=sessionStorage.getItem(String(e));if(!n)return null;let t=JSON.parse(n);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Date.now()>=t.expires?(sessionStorage.removeItem(String(e)),null):t.data},Br=e=>sessionStorage.removeItem(String(e)),Fr=()=>sessionStorage.clear(),Zp={setItem:Or,getItem:Mr,removeItem:Br,clear:Fr}});var pe,Re=g(()=>{"use strict";o();Te();Ur();pe={session:Co,local:Lo}});var Ro,k,De,M=g(()=>{"use strict";o();Ro=require("@reduxjs/toolkit");A();k=(0,Ro.createAction)(An),De=(0,Ro.createAction)(Lr,e=>({payload:e}))});var Oo,Wr,Gr,En,Do,$r,Mo,ch,Bo,ph,mh,dh,lh,_h,uh,Jp,Qp,Pn,Nn=g(()=>{"use strict";o();Oo=require("@multiversx/sdk-core"),Wr=require("@reduxjs/toolkit"),Gr=require("redux-persist");A();Re();Te();M();En={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:Ce},Do={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":En},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},$r=(0,Wr.createSlice)({name:"accountInfoSlice",initialState:Do,reducers:{setAddress:(e,n)=>{let t=n.payload;e.address=t,e.publicKey=t?new Oo.Address(t).hex():""},setAccount:(e,n)=>{let t=e.address===n.payload.address;e.accounts={[e.address]:t?n.payload:En},e.shard=n.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,n)=>{let{address:t}=e;e.accounts[t].nonce=n.payload},setAccountShard:(e,n)=>{e.shard=n.payload},setLedgerAccount:(e,n)=>{e.ledgerAccount=n.payload},updateLedgerAccount:(e,n)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=n.payload.index,e.ledgerAccount.address=n.payload.address)},setWalletConnectAccount:(e,n)=>{e.walletConnectAccount=n.payload},setIsAccountLoading:(e,n)=>{e.isAccountLoading=n.payload,e.accountLoadingError=null},setAccountLoadingError:(e,n)=>{e.accountLoadingError=n.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,n)=>{e.websocketEvent={timestamp:Date.now(),message:n.payload}},setWebsocketBatchEvent:(e,n)=>{e.websocketBatchEvent={timestamp:Date.now(),data:n.payload}}},extraReducers:e=>{e.addCase(k,()=>(pe.local.removeItem(re.loginExpiresAt),Do)),e.addCase(De,(n,t)=>{let{address:a}=t.payload;n.address=a,n.publicKey=new Oo.Address(a).toHex()}),e.addCase(Gr.REHYDRATE,(n,t)=>{var x;if(!((x=t.payload)!=null&&x.account))return;let{account:a}=t.payload,{address:c,shard:p,accounts:m,publicKey:u}=a;n.address=c,n.shard=p;let l=m&&c in m;n.accounts=l?m:Do.accounts,n.publicKey=u})}}),{setAccount:Mo,setAddress:ch,setAccountNonce:Bo,setAccountShard:ph,setLedgerAccount:mh,updateLedgerAccount:dh,setWalletConnectAccount:lh,setIsAccountLoading:_h,setAccountLoadingError:uh,setWebsocketEvent:Jp,setWebsocketBatchEvent:Qp}=$r.actions,Pn=$r.reducer});function Je(){return new Date().setHours(new Date().getHours()+24)}function Qe(e){pe.local.setItem({key:re.loginExpiresAt,data:e,expires:e})}var Fo=g(()=>{"use strict";o();Re();Te()});var Vr,Hr,qr,vh,Ah,Ih,em,kh,Eh,jr,Ph,Nh,Lh,Ln,Cn=g(()=>{"use strict";o();Vr=require("@reduxjs/toolkit");Fo();D();M();Hr={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},qr=(0,Vr.createSlice)({name:"loginInfoSlice",initialState:Hr,reducers:{setLoginMethod:(e,n)=>{e.loginMethod=n.payload},setTokenLogin:(e,n)=>{e.tokenLogin=n.payload},setTokenLoginSignature:(e,n)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=n.payload)},setWalletLogin:(e,n)=>{e.walletLogin=n.payload},setWalletConnectLogin:(e,n)=>{e.walletConnectLogin=n.payload},setLedgerLogin:(e,n)=>{e.ledgerLogin=n.payload},setWebviewLogin:(e,n)=>{e.webviewLogin=n.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,n)=>{e.logoutRoute=n.payload},setIsWalletConnectV2Initialized:(e,n)=>{e.isWalletConnectV2Initialized=n.payload}},extraReducers:e=>{e.addCase(k,()=>Hr),e.addCase(De,(n,t)=>{n.isLoginSessionInvalid=!1,n.loginMethod=t.payload.loginMethod,n.iframeLoginType=t.payload.iframeLoginType,Qe(Je())})}}),{setLoginMethod:vh,setWalletConnectLogin:Ah,setLedgerLogin:Ih,setTokenLogin:em,setTokenLoginSignature:kh,setWalletLogin:Eh,invalidateLoginSession:jr,setLogoutRoute:Ph,setIsWalletConnectV2Initialized:Nh,setWebviewLogin:Lh}=qr.actions,Ln=qr.reducer});var Kr,zr,Xr,Oh,nm,Mh,Bh,Rn,Dn=g(()=>{"use strict";o();Kr=require("@reduxjs/toolkit");M();zr={},Xr=(0,Kr.createSlice)({name:"modalsSlice",initialState:zr,reducers:{setTxSubmittedModal:(e,n)=>{e.txSubmittedModal=n.payload},setNotificationModal:(e,n)=>{e.notificationModal=n.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(k,()=>zr)}}),{setTxSubmittedModal:Oh,setNotificationModal:nm,clearTxSubmittedModal:Mh,clearNotificationModal:Bh}=Xr.actions,Rn=Xr.reducer});var he,en=g(()=>{"use strict";o();he=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var O,W=g(()=>{"use strict";o();en();O=()=>{if(!he())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:n,hash:t,origin:a,href:c,search:p}}=window;return{pathname:n,hash:t,origin:a,href:c,search:p}}});var Yr=g(()=>{"use strict";o();W()});var Zr=g(()=>{"use strict";o();me()});var Jr=g(()=>{"use strict";o();Sn()});var me=g(()=>{"use strict";o();Yr();Zr();W();Jr()});var On=g(()=>{"use strict";o();me()});var ea=g(()=>{"use strict";o();On()});function na(e){return e[Math.floor(Math.random()*e.length)]}var oa=g(()=>{"use strict";o()});var Uo=g(()=>{"use strict";o();Eo()});var ta=g(()=>{"use strict";o();ea();oa();Uo()});var ra,aa,ia,Wo,tm,sa,fx,Tx,hx,Mn,Bn=g(()=>{"use strict";o();ra=require("@reduxjs/toolkit"),aa=b(require("lodash.omit")),ia=require("redux-persist");Xe();M();ta();Wo={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},tm={network:Wo},sa=(0,ra.createSlice)({name:"appConfig",initialState:tm,reducers:{initializeNetworkConfig:(e,n)=>{let t=na(n.payload.walletConnectV2RelayAddresses),a=(0,aa.default)(n.payload,"walletConnectV2RelayAddresses");e.network=w(_(_({},e.network),a),{walletConnectV2RelayAddress:t})},updateNetworkConfig:(e,n)=>{e.network=_(_({},e.network),n.payload)},setCustomWalletAddress:(e,n)=>{e.network.customWalletAddress=n.payload}},extraReducers:e=>{e.addCase(k,n=>{n.network.customWalletAddress=void 0}),e.addCase(ia.REHYDRATE,(n,t)=>{var c,p;if(!((p=(c=t.payload)==null?void 0:c.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:a}}=t.payload;n.network.customWalletAddress=a})}}),{initializeNetworkConfig:fx,updateNetworkConfig:Tx,setCustomWalletAddress:hx}=sa.actions,Mn=sa.reducer});var ca,Go,pa,vx,Ax,Ix,Fn,Un=g(()=>{"use strict";o();ca=require("@reduxjs/toolkit");N();M();Go={isSigning:!1,signedSessions:{}},pa=(0,ca.createSlice)({name:"signedMessageInfoSliceState",initialState:Go,reducers:{setSignSession:(e,n)=>{let{sessionId:t,signedSession:a,errorMessage:c}=n.payload;c&&(e.errorMessage=c),e.isSigning=a.status==="pending",e.signedSessions[t]=_(_({},e.signedSessions[t]),a)},setSignSessionState:(e,n)=>_(_({},e),n.payload),clearSignedMessageInfo:()=>Go},extraReducers:e=>{e.addCase(k,()=>Go)}}),{setSignSession:vx,clearSignedMessageInfo:Ax,setSignSessionState:Ix}=pa.actions,Fn=pa.reducer});var da,la,ma,_a,rm,am,Rx,im,Wn,Gn=g(()=>{"use strict";o();da=require("@reduxjs/toolkit"),la=require("redux-persist");N();In();M();ma={customToasts:[],transactionToasts:[]},_a=(0,da.createSlice)({name:"toastsSlice",initialState:ma,reducers:{addCustomToast:(e,n)=>{let t=n.payload.toastId||`custom-toast-${e.customToasts.length+1}`,a=e.customToasts.findIndex(c=>c.toastId===t);if(a!==-1){e.customToasts[a]=w(_(_({},e.customToasts[a]),n.payload),{type:"custom",toastId:t});return}e.customToasts.push(w(_({},n.payload),{type:"custom",toastId:t}))},removeCustomToast:(e,n)=>{e.customToasts=e.customToasts.filter(t=>t.toastId!==n.payload)},addTransactionToast:(e,n)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Ze(),toastId:n.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,n)=>{e.transactionToasts=e.transactionToasts.filter(t=>t.toastId!==n.payload)}},extraReducers:e=>{e.addCase(k,()=>ma),e.addCase(la.REHYDRATE,(n,t)=>{var c,p;let a=(p=(c=t.customToasts)==null?void 0:c.filter(m=>!("component"in m)))!=null?p:[];n.customToasts=a})}}),{addCustomToast:rm,removeCustomToast:am,addTransactionToast:Rx,removeTransactionToast:im}=_a.actions,Wn=_a.reducer});var ua,Ho,Vo,qo,sm,$o,ga,Bx,cm,jo,$n,Hn=g(()=>{"use strict";o();ua=require("@reduxjs/toolkit");M();Ho="Transaction failed",Vo="Transaction successful",qo="Processing transaction",sm="Transaction submitted",$o={},ga=(0,ua.createSlice)({name:"transactionsInfo",initialState:$o,reducers:{setTransactionsDisplayInfo(e,n){let{sessionId:t,transactionsDisplayInfo:a}=n.payload;t!=null&&(e[t]={errorMessage:(a==null?void 0:a.errorMessage)||Ho,successMessage:(a==null?void 0:a.successMessage)||Vo,processingMessage:(a==null?void 0:a.processingMessage)||qo,submittedMessage:(a==null?void 0:a.submittedMessage)||sm,transactionDuration:a==null?void 0:a.transactionDuration})},clearTransactionsInfoForSessionId(e,n){n.payload!=null&&delete e[n.payload]},clearTransactionsInfo:()=>$o},extraReducers:e=>{e.addCase(k,()=>$o)}}),{clearTransactionsInfo:Bx,setTransactionsDisplayInfo:cm,clearTransactionsInfoForSessionId:jo}=ga.actions,$n=ga.reducer});function Me(e){return e!=null&&(hm(e)||wm(e))}function Be(e){return e!=null&&(xm(e)||vm(e))}function Fe(e){return e!=null&&(ym(e)||Am(e))}function fa(e){return e!=null&&(bm(e)||Im(e))}function zo(e){return e!=null&&Sm(e)}function hm(e){return e!=null&&pm.includes(e)}function xm(e){return e!=null&&mm.includes(e)}function ym(e){return e!=null&&dm.includes(e)}function bm(e){return e!=null&&lm.includes(e)}function Sm(e){return e!=null&&_m.includes(e)}function wm(e){return e!=null&&um.includes(e)}function vm(e){return e!=null&&gm.includes(e)}function Am(e){return e!=null&&fm.includes(e)}function Im(e){return e!=null&&Tm.includes(e)}var pm,mm,dm,lm,_m,um,gm,fm,Tm,Ue=g(()=>{"use strict";o();D();pm=["sent"],mm=["success"],dm=["fail","cancelled","timedOut"],lm=["invalid"],_m=["timedOut"],um=["pending"],gm=["success"],fm=["fail","invalid"],Tm=["not executed"]});var Ta,ha,nn,km,xa,ya,ba,Em,Vn,Pm,Nm,jx,Lm,on,Ko,zx,qn,jn=g(()=>{"use strict";o();Ta=require("@reduxjs/toolkit"),ha=require("redux-persist");D();Ue();M();nn={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},km={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},xa=(0,Ta.createSlice)({name:"transactionsSlice",initialState:nn,reducers:{moveTransactionsToSignedState:(e,n)=>{var l,x;let{sessionId:t,transactions:a,errorMessage:c,status:p,redirectRoute:m,customTransactionInformation:u}=n.payload;e.customTransactionInformationForSessionId[t]=_(_(_({},km),((l=e.signedTransactions[t])==null?void 0:l.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[t]||{}),e.signedTransactions[t]={transactions:a,status:p,errorMessage:c,redirectRoute:m,customTransactionInformation:_(_({},e.customTransactionInformationForSessionId[t]),u!=null?u:{})},((x=e==null?void 0:e.transactionsToSign)==null?void 0:x.sessionId)===t&&(e.transactionsToSign=nn.transactionsToSign)},clearSignedTransaction:(e,n)=>{e.signedTransactions[n.payload]&&delete e.signedTransactions[n.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,n)=>{e.signedTransactions=_(_({},e.signedTransactions),n.payload)},updateSignedTransactions:(e,n)=>{let{sessionId:t,status:a,errorMessage:c,transactions:p}=n.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].status=a,c!=null&&(e.signedTransactions[t].errorMessage=c),p!=null&&(e.signedTransactions[t].transactions=p))},updateSignedTransactionStatus:(e,n)=>{var x,S,y,L,R,J,fn,Bt;let{sessionId:t,status:a,errorMessage:c,transactionHash:p,serverTransaction:m,inTransit:u}=n.payload,l=(S=(x=e.signedTransactions)==null?void 0:x[t])==null?void 0:S.transactions;if(l!=null){e.signedTransactions[t].transactions=l.map(Q=>Q.hash===p?w(_(_({},m!=null?m:{}),Q),{status:a,errorMessage:c,inTransit:u}):Q);let rp=(L=(y=e.signedTransactions[t])==null?void 0:y.transactions)==null?void 0:L.every(Q=>Be(Q.status)),ap=(J=(R=e.signedTransactions[t])==null?void 0:R.transactions)==null?void 0:J.some(Q=>Fe(Q.status)),ip=(Bt=(fn=e.signedTransactions[t])==null?void 0:fn.transactions)==null?void 0:Bt.every(Q=>fa(Q.status));rp&&(e.signedTransactions[t].status="success"),ap&&(e.signedTransactions[t].status="fail"),ip&&(e.signedTransactions[t].status="invalid")}},setTransactionsToSign:(e,n)=>{e.transactionsToSign=n.payload;let{sessionId:t,customTransactionInformation:a}=n.payload;e.customTransactionInformationForSessionId[t]=a,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=nn.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=nn.signedTransactions},setSignTransactionsError:(e,n)=>{e.signTransactionsError=n.payload},setSignTransactionsCancelMessage:(e,n)=>{e.signTransactionsCancelMessage=n.payload},updateSignedTransactionsCustomTransactionInformation:(e,n)=>{let{sessionId:t,customTransactionInformationOverrides:a}=n.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].customTransactionInformation=_(_({},e.signedTransactions[t].customTransactionInformation),a))}},extraReducers:e=>{e.addCase(k,()=>nn),e.addCase(ha.REHYDRATE,(n,t)=>{var m;if(n.signTransactionsCancelMessage=null,!((m=t.payload)!=null&&m.transactions))return;let{signedTransactions:a,customTransactionInformationForSessionId:c}=t.payload.transactions,p=Object.entries(a).reduce((u,[l,x])=>{let S=new Date(l),y=new Date;return y.setHours(y.getHours()+5),y-S>0||(u[l]=x),u},{});c!=null&&(n.customTransactionInformationForSessionId=c),a!=null&&(n.signedTransactions=p)})}}),{updateSignedTransactionStatus:ya,updateSignedTransactions:ba,setTransactionsToSign:Em,clearAllTransactionsToSign:Vn,clearAllSignedTransactions:Pm,clearSignedTransaction:Nm,clearTransactionToSign:jx,setSignTransactionsError:Lm,setSignTransactionsCancelMessage:on,moveTransactionsToSignedState:Ko,updateSignedTransactionsCustomTransactionInformation:zx}=xa.actions,qn=xa.reducer});var Sa,Xo,wa,Zx,Jx,Cm,Qx,zn,Kn=g(()=>{"use strict";o();Sa=require("@reduxjs/toolkit");M();Xo={},wa=(0,Sa.createSlice)({name:"batchTransactionsSlice",initialState:Xo,reducers:{setBatchTransactions:(e,n)=>{e[n.payload.id]=n.payload},updateBatchTransactions:(e,n)=>{e[n.payload.id]=n.payload},clearBatchTransactions:(e,n)=>{delete e[n.payload.batchId]},clearAllBatchTransactions:()=>Xo},extraReducers:e=>{e.addCase(k,()=>Xo)}}),{setBatchTransactions:Zx,updateBatchTransactions:Jx,clearBatchTransactions:Cm,clearAllBatchTransactions:Qx}=wa.actions,zn=wa.reducer});var Aa,va,Ia,ty,ka,Yo=g(()=>{"use strict";o();Aa=require("@reduxjs/toolkit");M();va={},Ia=(0,Aa.createSlice)({name:"dappConfigSlice",initialState:va,reducers:{setDappConfig:(e,n)=>n.payload},extraReducers:e=>{e.addCase(k,()=>va)}}),{setDappConfig:ty}=Ia.actions,ka=Ia.reducer});var P=g(()=>{"use strict";o();Nn();Cn();Dn();Bn();Un();Gn();Hn();jn();Kn();Yo()});var Zo=g(()=>{"use strict";o()});var Na,Mm,Bm,xe,Zn=g(()=>{"use strict";o();Na=require("@reduxjs/toolkit");Zo();Nn();Kn();Yo();Cn();Dn();Bn();Un();Gn();Hn();jn();Mm={["account"]:Pn,["dappConfig"]:ka,["loginInfo"]:Ln,["modals"]:Rn,["networkConfig"]:Mn,["signedMessageInfo"]:Fn,["toasts"]:Wn,["transactionsInfo"]:$n,["transactions"]:qn,["batchTransactions"]:zn},Bm=(e={})=>(0,Na.combineReducers)(_(_({},Mm),e)),xe=Bm});var Da={};ee(Da,{default:()=>Qm,sessionStorageReducers:()=>Jo});function ie(e,n=[]){return{key:e,version:1,storage:Ca.default,blacklist:n}}var U,La,Ca,Fm,tn,Um,Wm,Gm,$m,Hm,Vm,qm,jm,zm,Km,Ra,Xm,Jo,Ym,Zm,Jm,Qm,Oa=g(()=>{"use strict";o();U=require("redux-persist"),La=b(require("redux-persist/lib/storage")),Ca=b(require("redux-persist/lib/storage/session"));Zn();P();Nn();Kn();Cn();Dn();Bn();Un();Gn();Hn();jn();Zo();Fm={persistReducersStorageType:"localStorage"},tn={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Um=ie(tn["account"]),Wm=ie(tn["loginInfo"]),Gm=ie(tn["modals"]),$m=ie(tn["networkConfig"]),Hm={2:e=>w(_({},e),{networkConfig:Wo})};Vm=ie("sdk-dapp-transactionsInfo"),qm=ie("sdk-dapp-transactions",["transactionsToSign"]),jm=ie("sdk-dapp-batchTransactions",["batchTransactions"]),zm=ie("sdk-dapp-toasts"),Km=ie("sdk-dapp-signedMessageInfo"),Ra={key:"sdk-dapp-store",version:2,storage:La.default,whitelist:Object.keys(tn),migrate:(0,U.createMigrate)(Hm,{debug:!1})},Xm=w(_({},Ra),{whitelist:[]}),Jo={["toasts"]:(0,U.persistReducer)(zm,Wn),["transactions"]:(0,U.persistReducer)(qm,qn),["transactionsInfo"]:(0,U.persistReducer)(Vm,$n),["batchTransactions"]:(0,U.persistReducer)(jm,zn),["signedMessageInfo"]:(0,U.persistReducer)(Km,Fn)},Ym=w(_({},Jo),{["account"]:(0,U.persistReducer)(Um,Pn),["loginInfo"]:(0,U.persistReducer)(Wm,Ln),["modals"]:(0,U.persistReducer)(Gm,Rn),["networkConfig"]:(0,U.persistReducer)($m,Mn)}),Zm=Fm.persistReducersStorageType==="localStorage",Jm=Zm?(0,U.persistReducer)(Ra,xe(Jo)):(0,U.persistReducer)(Xm,xe(Ym)),Qm=Jm});var Ma={};ee(Ma,{default:()=>ed});var ed,Ba=g(()=>{"use strict";o();Zn();ed=xe()});var Fa={};ee(Fa,{default:()=>od});var H,nd,od,Ua=g(()=>{"use strict";o();H=require("redux-persist"),nd=[H.FLUSH,H.REHYDRATE,H.PAUSE,H.PERSIST,H.PURGE,H.REGISTER],od=nd});var $a={};ee($a,{default:()=>Ga});function Ga(e){return(0,Wa.persistStore)(e)}var Wa,Ha=g(()=>{"use strict";o();Wa=require("redux-persist")});var Ri=T((cR,Ci)=>{o();var Yl=typeof i=="object"&&i&&i.Object===Object&&i;Ci.exports=Yl});var pn=T((pR,Di)=>{o();var Zl=Ri(),Jl=typeof self=="object"&&self&&self.Object===Object&&self,Ql=Zl||Jl||Function("return this")();Di.exports=Ql});var xt=T((mR,Oi)=>{o();var e_=pn(),n_=e_.Symbol;Oi.exports=n_});var Ui=T((dR,Fi)=>{o();var Mi=xt(),Bi=Object.prototype,o_=Bi.hasOwnProperty,t_=Bi.toString,mn=Mi?Mi.toStringTag:void 0;function r_(e){var n=o_.call(e,mn),t=e[mn];try{e[mn]=void 0;var a=!0}catch(p){}var c=t_.call(e);return a&&(n?e[mn]=t:delete e[mn]),c}Fi.exports=r_});var Gi=T((lR,Wi)=>{o();var a_=Object.prototype,i_=a_.toString;function s_(e){return i_.call(e)}Wi.exports=s_});var yt=T((_R,Vi)=>{o();var $i=xt(),c_=Ui(),p_=Gi(),m_="[object Null]",d_="[object Undefined]",Hi=$i?$i.toStringTag:void 0;function l_(e){return e==null?e===void 0?d_:m_:Hi&&Hi in Object(e)?c_(e):p_(e)}Vi.exports=l_});var ji=T((uR,qi)=>{o();var __=Array.isArray;qi.exports=__});var Ki=T((gR,zi)=>{o();function u_(e){return e!=null&&typeof e=="object"}zi.exports=u_});var Yi=T((fR,Xi)=>{o();var g_=yt(),f_=ji(),T_=Ki(),h_="[object String]";function x_(e){return typeof e=="string"||!f_(e)&&T_(e)&&g_(e)==h_}Xi.exports=x_});var St={};ee(St,{css:()=>ns,default:()=>k_});var ns,k_,wt=g(()=>{"use strict";o();ns=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ns));k_={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var At={};ee(At,{css:()=>as,default:()=>N_});var as,N_,It=g(()=>{"use strict";o();as=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(as));N_={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var Et=T((YD,ps)=>{o();function G_(e){var n=typeof e;return e!=null&&(n=="object"||n=="function")}ps.exports=G_});var ds=T((ZD,ms)=>{o();var $_=yt(),H_=Et(),V_="[object AsyncFunction]",q_="[object Function]",j_="[object GeneratorFunction]",z_="[object Proxy]";function K_(e){if(!H_(e))return!1;var n=$_(e);return n==q_||n==j_||n==V_||n==z_}ms.exports=K_});var _s=T((JD,ls)=>{o();var X_=pn(),Y_=X_["__core-js_shared__"];ls.exports=Y_});var fs=T((QD,gs)=>{o();var Pt=_s(),us=function(){var e=/[^.]+$/.exec(Pt&&Pt.keys&&Pt.keys.IE_PROTO||"");return e?"Symbol(src)_1."+e:""}();function Z_(e){return!!us&&us in e}gs.exports=Z_});var hs=T((e3,Ts)=>{o();var J_=Function.prototype,Q_=J_.toString;function eu(e){if(e!=null){try{return Q_.call(e)}catch(n){}try{return e+""}catch(n){}}return""}Ts.exports=eu});var ys=T((n3,xs)=>{o();var nu=ds(),ou=fs(),tu=Et(),ru=hs(),au=/[\\^$.*+?()[\]{}|]/g,iu=/^\[object .+?Constructor\]$/,su=Function.prototype,cu=Object.prototype,pu=su.toString,mu=cu.hasOwnProperty,du=RegExp("^"+pu.call(mu).replace(au,"\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g,"$1.*?")+"$");function lu(e){if(!tu(e)||ou(e))return!1;var n=nu(e)?du:iu;return n.test(ru(e))}xs.exports=lu});var Ss=T((o3,bs)=>{o();function _u(e,n){return e==null?void 0:e[n]}bs.exports=_u});var mo=T((t3,ws)=>{o();var uu=ys(),gu=Ss();function fu(e,n){var t=gu(e,n);return uu(t)?t:void 0}ws.exports=fu});var ln=T((r3,vs)=>{o();var Tu=mo(),hu=Tu(Object,"create");vs.exports=hu});var ks=T((a3,Is)=>{o();var As=ln();function xu(){this.__data__=As?As(null):{},this.size=0}Is.exports=xu});var Ps=T((i3,Es)=>{o();function yu(e){var n=this.has(e)&&delete this.__data__[e];return this.size-=n?1:0,n}Es.exports=yu});var Ls=T((s3,Ns)=>{o();var bu=ln(),Su="__lodash_hash_undefined__",wu=Object.prototype,vu=wu.hasOwnProperty;function Au(e){var n=this.__data__;if(bu){var t=n[e];return t===Su?void 0:t}return vu.call(n,e)?n[e]:void 0}Ns.exports=Au});var Rs=T((c3,Cs)=>{o();var Iu=ln(),ku=Object.prototype,Eu=ku.hasOwnProperty;function Pu(e){var n=this.__data__;return Iu?n[e]!==void 0:Eu.call(n,e)}Cs.exports=Pu});var Os=T((p3,Ds)=>{o();var Nu=ln(),Lu="__lodash_hash_undefined__";function Cu(e,n){var t=this.__data__;return this.size+=this.has(e)?0:1,t[e]=Nu&&n===void 0?Lu:n,this}Ds.exports=Cu});var Bs=T((m3,Ms)=>{o();var Ru=ks(),Du=Ps(),Ou=Ls(),Mu=Rs(),Bu=Os();function Ve(e){var n=-1,t=e==null?0:e.length;for(this.clear();++n<t;){var a=e[n];this.set(a[0],a[1])}}Ve.prototype.clear=Ru;Ve.prototype.delete=Du;Ve.prototype.get=Ou;Ve.prototype.has=Mu;Ve.prototype.set=Bu;Ms.exports=Ve});var Us=T((d3,Fs)=>{o();function Fu(){this.__data__=[],this.size=0}Fs.exports=Fu});var Gs=T((l3,Ws)=>{o();function Uu(e,n){return e===n||e!==e&&n!==n}Ws.exports=Uu});var _n=T((_3,$s)=>{o();var Wu=Gs();function Gu(e,n){for(var t=e.length;t--;)if(Wu(e[t][0],n))return t;return-1}$s.exports=Gu});var Vs=T((u3,Hs)=>{o();var $u=_n(),Hu=Array.prototype,Vu=Hu.splice;function qu(e){var n=this.__data__,t=$u(n,e);if(t<0)return!1;var a=n.length-1;return t==a?n.pop():Vu.call(n,t,1),--this.size,!0}Hs.exports=qu});var js=T((g3,qs)=>{o();var ju=_n();function zu(e){var n=this.__data__,t=ju(n,e);return t<0?void 0:n[t][1]}qs.exports=zu});var Ks=T((f3,zs)=>{o();var Ku=_n();function Xu(e){return Ku(this.__data__,e)>-1}zs.exports=Xu});var Ys=T((T3,Xs)=>{o();var Yu=_n();function Zu(e,n){var t=this.__data__,a=Yu(t,e);return a<0?(++this.size,t.push([e,n])):t[a][1]=n,this}Xs.exports=Zu});var Js=T((h3,Zs)=>{o();var Ju=Us(),Qu=Vs(),eg=js(),ng=Ks(),og=Ys();function qe(e){var n=-1,t=e==null?0:e.length;for(this.clear();++n<t;){var a=e[n];this.set(a[0],a[1])}}qe.prototype.clear=Ju;qe.prototype.delete=Qu;qe.prototype.get=eg;qe.prototype.has=ng;qe.prototype.set=og;Zs.exports=qe});var ec=T((x3,Qs)=>{o();var tg=mo(),rg=pn(),ag=tg(rg,"Map");Qs.exports=ag});var tc=T((y3,oc)=>{o();var nc=Bs(),ig=Js(),sg=ec();function cg(){this.size=0,this.__data__={hash:new nc,map:new(sg||ig),string:new nc}}oc.exports=cg});var ac=T((b3,rc)=>{o();function pg(e){var n=typeof e;return n=="string"||n=="number"||n=="symbol"||n=="boolean"?e!=="__proto__":e===null}rc.exports=pg});var un=T((S3,ic)=>{o();var mg=ac();function dg(e,n){var t=e.__data__;return mg(n)?t[typeof n=="string"?"string":"hash"]:t.map}ic.exports=dg});var cc=T((w3,sc)=>{o();var lg=un();function _g(e){var n=lg(this,e).delete(e);return this.size-=n?1:0,n}sc.exports=_g});var mc=T((v3,pc)=>{o();var ug=un();function gg(e){return ug(this,e).get(e)}pc.exports=gg});var lc=T((A3,dc)=>{o();var fg=un();function Tg(e){return fg(this,e).has(e)}dc.exports=Tg});var uc=T((I3,_c)=>{o();var hg=un();function xg(e,n){var t=hg(this,e),a=t.size;return t.set(e,n),this.size+=t.size==a?0:1,this}_c.exports=xg});var fc=T((k3,gc)=>{o();var yg=tc(),bg=cc(),Sg=mc(),wg=lc(),vg=uc();function je(e){var n=-1,t=e==null?0:e.length;for(this.clear();++n<t;){var a=e[n];this.set(a[0],a[1])}}je.prototype.clear=yg;je.prototype.delete=bg;je.prototype.get=Sg;je.prototype.has=wg;je.prototype.set=vg;gc.exports=je});var hc=T((E3,Tc)=>{o();var Ag="__lodash_hash_undefined__";function Ig(e){return this.__data__.set(e,Ag),this}Tc.exports=Ig});var yc=T((P3,xc)=>{o();function kg(e){return this.__data__.has(e)}xc.exports=kg});var Sc=T((N3,bc)=>{o();var Eg=fc(),Pg=hc(),Ng=yc();function lo(e){var n=-1,t=e==null?0:e.length;for(this.__data__=new Eg;++n<t;)this.add(e[n])}lo.prototype.add=lo.prototype.push=Pg;lo.prototype.has=Ng;bc.exports=lo});var vc=T((L3,wc)=>{o();function Lg(e,n,t,a){for(var c=e.length,p=t+(a?1:-1);a?p--:++p<c;)if(n(e[p],p,e))return p;return-1}wc.exports=Lg});var Ic=T((C3,Ac)=>{o();function Cg(e){return e!==e}Ac.exports=Cg});var Ec=T((R3,kc)=>{o();function Rg(e,n,t){for(var a=t-1,c=e.length;++a<c;)if(e[a]===n)return a;return-1}kc.exports=Rg});var Nc=T((D3,Pc)=>{o();var Dg=vc(),Og=Ic(),Mg=Ec();function Bg(e,n,t){return n===n?Mg(e,n,t):Dg(e,Og,t)}Pc.exports=Bg});var Cc=T((O3,Lc)=>{o();var Fg=Nc();function Ug(e,n){var t=e==null?0:e.length;return!!t&&Fg(e,n,0)>-1}Lc.exports=Ug});var Dc=T((M3,Rc)=>{o();function Wg(e,n,t){for(var a=-1,c=e==null?0:e.length;++a<c;)if(t(n,e[a]))return!0;return!1}Rc.exports=Wg});var Mc=T((B3,Oc)=>{o();function Gg(e,n){return e.has(n)}Oc.exports=Gg});var Fc=T((F3,Bc)=>{o();var $g=mo(),Hg=pn(),Vg=$g(Hg,"Set");Bc.exports=Vg});var Wc=T((U3,Uc)=>{o();function qg(){}Uc.exports=qg});var Nt=T((W3,Gc)=>{o();function jg(e){var n=-1,t=Array(e.size);return e.forEach(function(a){t[++n]=a}),t}Gc.exports=jg});var Hc=T((G3,$c)=>{o();var Lt=Fc(),zg=Wc(),Kg=Nt(),Xg=1/0,Yg=Lt&&1/Kg(new Lt([,-0]))[1]==Xg?function(e){return new Lt(e)}:zg;$c.exports=Yg});var qc=T(($3,Vc)=>{o();var Zg=Sc(),Jg=Cc(),Qg=Dc(),ef=Mc(),nf=Hc(),of=Nt(),tf=200;function rf(e,n,t){var a=-1,c=Jg,p=e.length,m=!0,u=[],l=u;if(t)m=!1,c=Qg;else if(p>=tf){var x=n?null:nf(e);if(x)return of(x);m=!1,c=ef,l=new Zg}else l=n?[]:u;e:for(;++a<p;){var S=e[a],y=n?n(S):S;if(S=t||S!==0?S:0,m&&y===y){for(var L=l.length;L--;)if(l[L]===y)continue e;n&&l.push(y),u.push(S)}else c(l,y,t)||(l!==u&&l.push(y),u.push(S))}return u}Vc.exports=rf});var zc=T((H3,jc)=>{o();var af=qc();function sf(e){return e&&e.length?af(e):[]}jc.exports=sf});var Wf={};ee(Wf,{useGetToasts:()=>tp,useTransactionsToasts:()=>op});module.exports=_e(Wf);o();o();var Mt=require("react");o();var Xa=require("react"),We=require("react-redux");o();var Qo=require("@reduxjs/toolkit"),za=require("react-redux/lib/utils/Subscription");P();o();var Ea=b(require("lodash.throttle"));A();P();Fo();Re();Te();var Rm=[An],Xn=!1,Dm=(0,Ea.default)(()=>{Qe(Je())},5e3),Pa=e=>n=>t=>{if(Rm.includes(t.type))return n(t);let a=e.getState(),c=pe.local.getItem(re.loginExpiresAt);if(!Boolean(a==null?void 0:a.account.address))return n(t);if(c==null)return Qe(Je());let m=Date.now();return c-m<0&&!Xn?setTimeout(()=>{Xn=!0,e.dispatch(jr())},1e3):(Xn&&(Xn=!1),Dm()),n(t)};o();o();function Yn(){return typeof sessionStorage!="undefined"}var Va=Yn()?(Oa(),_e(Da)).default:(Ba(),_e(Ma)).default,qa=Yn()?(Ua(),_e(Fa)).default:[],ja=Yn()?(Ha(),_e($a)).default:e=>e;Zn();var v=(0,Qo.configureStore)({reducer:Va,middleware:e=>e({serializableCheck:{ignoredActions:[...qa,Bo.type,Mo.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Pa)}),Ka=(0,za.createSubscription)(v),nb=ja(v),ob=(0,Qo.configureStore)({reducer:xe});var td={store:v,subscription:Ka},et=(0,Xa.createContext)(td),ib=(0,We.createStoreHook)(et),Ge=(0,We.createDispatchHook)(et),I=(0,We.createSelectorHook)(et);o();o();P();o();var Ya=b(require("lodash.isequal")),Jn=require("reselect"),f=(0,Jn.createSelectorCreator)(Jn.defaultMemoize,Ya.default);var Y=e=>e.account,ye=f(Y,e=>e.address),le=f(Y,ye,(e,n)=>n in e.accounts?e.accounts[n]:En),rd=f(Y,le,(e,n)=>{let c=e,{accounts:t}=c,a=ve(c,["accounts"]);return w(_({},a),{address:n.address,account:n})}),db=f(le,e=>e.balance),ad=f(le,e=>{var n;return((n=e==null?void 0:e.nonce)==null?void 0:n.valueOf())||0}),lb=f(Y,e=>e.shard),_b=f(Y,e=>e.ledgerAccount),ub=f(Y,e=>e.walletConnectAccount),gb=f(Y,e=>e.isAccountLoading),fb=f(Y,e=>e.accountLoadingError),id=f(Y,e=>e.websocketEvent),sd=f(Y,e=>e.websocketBatchEvent);o();var Za=e=>e.dappConfig,xb=f(Za,e=>e.shouldUseWebViewProvider);o();var G=e=>e.loginInfo,wb=f(G,e=>e.loginMethod),nt=f(G,ye,(e,n)=>Boolean(n)),vb=f(G,e=>e.walletConnectLogin),cd=f(G,e=>e.ledgerLogin),pd=f(G,e=>e.walletLogin),Ab=f(G,e=>e.isLoginSessionInvalid),Ib=f(G,e=>e.tokenLogin),md=f(G,e=>e.logoutRoute),dd=f(G,e=>e.isWalletConnectV2Initialized);o();var Ja=e=>e.modals,Pb=f(Ja,e=>e.txSubmittedModal),Nb=f(Ja,e=>e.notificationModal);o();var Z=e=>e.networkConfig,Qa=f(Z,e=>e.network.chainId),ld=f(Z,e=>e.network.roundDuration),Rb=f(Z,e=>e.network.walletConnectBridgeAddress),Db=f(Z,e=>e.network.walletConnectV2RelayAddress),Ob=f(Z,e=>e.network.walletConnectV2ProjectId),Mb=f(Z,e=>e.network.walletConnectV2Options),Bb=f(Z,e=>e.network.walletConnectDeepLink),se=f(Z,e=>e.network),ei=f(se,e=>e.apiAddress),ni=f(se,e=>e.explorerAddress),oi=f(se,e=>{var n;return(n=e.customWalletAddress)!=null?n:e.walletAddress}),Fb=f(se,e=>e.xAliasAddress),_d=f(se,e=>e.egldLabel);o();var Qn=e=>e.signedMessageInfo,Gb=f(Qn,e=>e.isSigning),$b=f(Qn,e=>e.errorMessage),Hb=f(Qn,e=>{let n=Object.keys(e.signedSessions),t=n.length;return e.signedSessions[n[t-1]]}),Vb=f(Qn,e=>{let n=Object.keys(e.signedSessions),t=n.length;return n.length>0?n[t-1]:""});o();var ti=e=>e.toasts,ri=f(ti,e=>e.customToasts),eo=f(ti,e=>e.transactionToasts);o();P();var ai={errorMessage:Ho,successMessage:Vo,processingMessage:qo},ot=e=>e.transactionsInfo,ud=f(ot,(e,n)=>n,(e,n)=>n!=null&&(e==null?void 0:e[Number(n)])||ai);o();o();var be=require("@multiversx/sdk-core");A();o();o();o();function tt(e){try{let n=atob(e),t=btoa(n),a=r.Buffer.from(e,"base64").toString(),c=r.Buffer.from(a).toString("base64"),p=e===t||t.startsWith(e),m=e===c||c.startsWith(e);if(p&&m)return!0}catch(n){return!1}return!1}o();o();o();o();var ii=e=>{let n=e!=null?e:"";return tt(n)?r.Buffer.from(r.Buffer.from(n,"base64").toString().trim()):r.Buffer.from(n.trim())};o();A();var rn=({data:e,onlySetGuardian:n})=>e?n?e.startsWith("SetGuardian"):Object.values(ko).some(t=>e.startsWith(t)):!1;function no(e){var a,c,p;let n=_({},e);rn({data:n.data,onlySetGuardian:!0})&&(delete n.senderUsername,delete n.receiverUsername);let t=new be.Transaction(_(_(w(_(w(_(_({value:BigInt(n.value.valueOf()),data:ii(n.data),nonce:BigInt(n.nonce.valueOf()),receiver:new be.Address(n.receiver)},n.receiverUsername?{receiverUsername:n.receiverUsername}:{}),n.relayer?{relayer:new be.Address(n.relayer)}:{}),{sender:new be.Address(n.sender)}),n.senderUsername?{senderUsername:n.senderUsername}:{}),{gasLimit:BigInt((a=n.gasLimit.valueOf())!=null?a:wn),gasPrice:BigInt((c=n.gasPrice.valueOf())!=null?c:vn),chainID:n.chainID.valueOf(),version:(p=n.version)!=null?p:Nr}),n.options?{options:n.options}:{}),n.guardian?{guardian:new be.Address(n.guardian)}:{}));return n.relayerSignature&&(t.relayerSignature=r.Buffer.from(n.relayerSignature,"hex")),n.guardianSignature&&(t.guardianSignature=r.Buffer.from(n.guardianSignature,"hex")),n.signature&&(t.signature=r.Buffer.from(n.signature,"hex")),t}o();o();A();o();o();o();var hd=require("@multiversx/sdk-core");o();A();o();var yd=require("@multiversx/sdk-core");N();o();var xd=require("@multiversx/sdk-core");o();o();o();var bd=require("@multiversx/sdk-core"),Sd=b(require("bignumber.js"));D();o();o();o();var at="blocks";o();o();o();o();F();o();o();F();o();o();o();var Nd=b(require("bignumber.js"));A();o();var Pd=require("@multiversx/sdk-core"),st=b(require("bignumber.js"));A();o();var kd=b(require("bignumber.js"));o();st.default.config({ROUNDING_MODE:st.default.ROUND_FLOOR});o();o();o();A();o();o();o();o();A();o();A();o();var Ld=require("@multiversx/sdk-core"),Cd=b(require("bignumber.js"));A();o();var pt=require("@multiversx/sdk-core"),Od=b(require("bignumber.js"));A();o();o();var Rd=b(require("bignumber.js"));A();o();A();o();o();o();o();o();o();A();o();A();o();A();o();F();var Bd=["reDelegateRewards","claimRewards","unBond"],Fd=["wrapEgld","unwrapEgld"],Ud=["unStake"],Wd=["unDelegate"];o();o();o();A();o();var Vd=b(require("bignumber.js"));o();o();F();o();var jd=b(require("bignumber.js"));o();o();o();o();var Xd=b(require("bignumber.js"));N();o();o();var Yd=require("@multiversx/sdk-core");o();o();D();o();var el=require("@multiversx/sdk-web-wallet-provider");A();o();var Jd=b(require("qs"));o();me();en();o();en();var IA={search:he()?window.location.search:"",removeParams:[]};o();o();o();W();o();F();o();o();W();o();var nl=b(require("linkifyjs"));o();A();o();var ol=b(require("bignumber.js"));o();N();o();o();D();o();D();o();o();o();F();o();F();o();var tl=b(require("bignumber.js"));A();F();o();F();o();var _i=require("react");N();o();o();F();o();o();var rl=require("@multiversx/sdk-core"),al=b(require("bignumber.js"));F();o();N();o();o();N();var Tk=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];o();var ui=require("react");N();W();o();var sl=require("react");F();var Ak=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];Ue();o();var ro=e=>e.transactions,Se=f(ro,e=>e.signedTransactions),cl=f(ro,e=>e.signTransactionsError),gi=f(ro,e=>e.signTransactionsCancelMessage),ao=e=>n=>Object.entries(n).reduce((t,[a,c])=>(e(c.status)&&(t[a]=c),t),{}),fi=f(Se,ao(Me)),Ti=f(Se,ao(Be)),hi=f(Se,ao(Fe)),pl=f(Se,ao(zo)),xi=f(ro,e=>{var n;return(e==null?void 0:e.transactionsToSign)==null?null:w(_({},e.transactionsToSign),{transactions:((n=e==null?void 0:e.transactionsToSign)==null?void 0:n.transactions.map(t=>no(t)))||[]})}),ml=f(Se,(e,n)=>n,(e,n)=>n!=null?(e==null?void 0:e[n])||{}:{});N();o();o();o();o();o();o();o();o();var dl=require("@multiversx/sdk-extension-provider"),ll=require("@multiversx/sdk-hw-provider"),_l=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),ul=require("@multiversx/sdk-opera-provider"),gl=require("@multiversx/sdk-passkey-provider/out"),fl=require("@multiversx/sdk-web-wallet-provider");A();Pe();o();var cn=require("@multiversx/sdk-web-wallet-iframe-provider/out"),yi=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");D();o();var V=e=>`Unable to perform ${e}, Provider not initialized`,io=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(V("getAccount"))}setAccount(n){throw new Error(V(`setAccount: ${n}`))}login(n){throw new Error(V(`login with options: ${n}`))}logout(n){throw new Error(V(`logout with options: ${n}`))}getAddress(){throw new Error(V("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(n,t){throw new Error(V(`sendTransaction with transactions: ${n} options: ${t}`))}signTransaction(n,t){throw new Error(V(`signTransaction with transactions: ${n} options: ${t}`))}signTransactions(n,t){throw new Error(V(`signTransactions with transactions: ${n} options: ${t}`))}signMessage(n,t){throw new Error(V(`signTransactions with ${n} and options ${t}`))}sendCustomMessage({method:n,params:t}){throw new Error(V(`sendCustomMessage with method: ${n} params: ${t}`))}sendCustomRequest(n){throw new Error(V(`sendSessionEvent with options: ${n}`))}ping(){return Promise.resolve(!1)}},Tl=new io;P();o();o();o();o();o();var bi=require("@lifeomic/axios-fetch"),lt=b(require("axios")),_t=(0,bi.buildAxiosFetch)(lt.default),ut=(e,n)=>h(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let a=e.clone().json(),[c]=yield Promise.all([a]);return{data:c,status:e.status,statusText:e.statusText,headers:e.headers,config:n}});function hl(e,n,t){return h(this,null,function*(){try{let a=yield _t(e,_({method:"POST",body:n?JSON.stringify(n):void 0,headers:_({"Content-Type":"application/json"},(t==null?void 0:t.headers)||{})},t));return ut(a,t)}catch(a){throw console.error("Fetch Error:",a),a}})}function xl(e,n){return h(this,null,function*(){try{let t=yield _t(e,n);if(!t.ok)throw new Error(`HTTP error! Status: ${t.status}`);return ut(t,n)}catch(t){throw console.error("Fetch Error:",t),t}})}function yl(e,n,t){return h(this,null,function*(){try{let a=yield _t(e,_({method:"PATCH",body:n?JSON.stringify(n):void 0,headers:(t==null?void 0:t.headers)||{}},t));if(!a.ok)throw new Error(`HTTP error! Status: ${a.status}`);return ut(a,t)}catch(a){throw console.error("Fetch Error:",a),a}})}var $e=lt.default.create();$e.get=xl;$e.post=hl;$e.patch=yl;o();o();var bl=b(require("axios"));o();var Sl=b(require("swr"));o();D();o();o();o();W();o();Ue();o();o();o();var wl=b(require("axios"));o();var Al=b(require("axios"));Le();o();A();o();var Il=b(require("axios"));o();var El=b(require("axios"));o();o();var Pl=b(require("axios"));o();var Nl=b(require("axios"));o();o();P();D();o();o();o();o();W();o();M();N();o();P();o();var Ii=require("@multiversx/sdk-core");N();me();o();Ue();o();P();D();o();P();N();o();o();o();N();o();Xe();o();o();function ht(){let e=I(Se),n=Object.entries(e),t=(n==null?void 0:n.length)>0;return{signedTransactions:e,signedTransactionsArray:n,hasSignedTransactions:t}}o();o();var Ol=b(require("swr"));o();o();o();o();o();var Ei=require("react");o();o();var Ml=require("react"),Bl=require("@multiversx/sdk-web-wallet-provider"),Fl=require("@multiversx/sdk-web-wallet-provider"),Ul=b(require("qs"));A();P();D();Po();W();var rC=O();o();var cs=require("react"),po=require("@multiversx/sdk-core");o();A();Uo();o();o();o();o();o();o();o();o();o();o();Le();o();var Hl=b(require("axios"));o();var ql=b(require("axios"));o();var jl=b(require("axios"));o();Le();o();Le();o();o();o();o();var Kl=require("@multiversx/sdk-opera-provider");o();var Xl=require("@multiversx/sdk-extension-provider");o();Pe();o();o();o();o();var y_=b(Yi());o();var v_=require("@multiversx/sdk-native-auth-client");o();var Qi=b(require("axios"));o();o();o();function Zi(e){return h(this,null,function*(){return yield new Promise(n=>{setTimeout(()=>n(),e)})})}o();var Ji=(e,n,t,a=0)=>h(void 0,null,function*(){try{return yield e(...t)}catch(c){return a<n.retries?((n==null?void 0:n.delay)!=null&&(yield Zi(n.delay)),yield Ji(e,n,t,a+1)):null}}),bt=(e,n={retries:5,delay:500})=>(...t)=>h(void 0,null,function*(){return yield Ji(e,n,t)});var b_=4;var ER=bt((e,n,t)=>h(void 0,null,function*(){if(t){let p=Math.floor(Date.now()/1e3);return{hash:yield t(),timestamp:p}}let{data:a}=yield Qi.default.get(`${e}/${at}?from=${b_}&size=1&fields=hash,timestamp${n?"&shard="+n:""}`),[c]=a;return c}));o();o();In();o();o();W();var FR={origin:O().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};o();var A_=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");o();var I_=require("@multiversx/sdk-passkey-provider/out");o();var R_=b(require("react")),D_=require("@fortawesome/free-solid-svg-icons");o();o();var He=b(require("react")),is=require("@fortawesome/free-solid-svg-icons"),ss=require("@fortawesome/react-fontawesome");o();var ts=b(require("react"));o();var dn=b(require("react"));o();en();var es=()=>!he();var E_=()=>h(void 0,null,function*(){return yield Promise.resolve().then(()=>(wt(),St))}),P_=()=>(wt(),_e(St)).default,vt=es();function os({ssrGlobalImportCallback:e,ssrImportCallback:n,clientImportCallback:t}){let[a,c]=dn.default.useState(vt?void 0:P_()),[p,m]=dn.default.useState(vt||t==null?void 0:t()),u=()=>h(this,null,function*(){(e?e():E_()).then(l=>c(l.default)),n==null||n().then(l=>m(l.default))});return(0,dn.useEffect)(()=>{vt&&u()},[]),{globalStyles:a,styles:p}}function rs(e,n){return t=>{let{globalStyles:a,styles:c}=os({ssrGlobalImportCallback:n.ssrGlobalStyles,ssrImportCallback:n.ssrStyles,clientImportCallback:n.clientStyles});return ts.default.createElement(e,w(_({},t),{globalStyles:a!=null?a:{},styles:c!=null?c:{}}))}}var L_=({styles:e})=>He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},He.default.createElement(ss.FontAwesomeIcon,{icon:is.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),C_=rs(L_,{ssrStyles:()=>Promise.resolve().then(()=>(It(),At)),clientStyles:()=>(It(),_e(At)).default});Re();Te();o();o();P();var E2=24*60*60;A();W();o();o();o();var B_=require("react"),F_=require("@multiversx/sdk-hw-provider");o();Sn();M();N();o();On();Re();Te();me();o();o();o();var Zc=require("react"),Jc=require("@multiversx/sdk-core"),If=require("@multiversx/sdk-extension-provider"),kf=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Ef=require("@multiversx/sdk-passkey-provider/out"),Pf=b(zc());A();o();o();o();o();o();o();o();o();o();o();var gn=b(require("react"));var a5=(0,gn.createContext)({}),i5=v.getState();o();var Xc=b(require("react"));o();var Ct=b(require("react")),cf=b(require("axios"));o();o();Pe();o();o();var gf=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),ff=require("@multiversx/sdk-webview-provider/out/WebviewProvider");M();o();M();P();N();o();o();P();o();me();o();P();N();o();o();var mf=require("@multiversx/sdk-core"),df=b(require("bignumber.js"));A();o();var _f=b(require("bignumber.js"));A();P();D();me();W();o();var Yc=require("react"),yf=require("@multiversx/sdk-extension-provider"),bf=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Sf=require("@multiversx/sdk-passkey-provider/out");Pe();P();o();P();N();On();W();o();Io();P();o();o();var Nf=require("react");o();o();D();Ue();o();o();var Uf=require("react");o();o();var Qc=require("react");o();var Lf=b(require("axios"));o();var Of=require("react");o();var Dt=require("react"),Df=require("socket.io-client");P();o();var Rt=class{constructor(){this.current=null;this._status="NOT_INITIALIZED";this.listeners=[]}get status(){return this._status}set status(n){this._status!==n&&(this._status=n,this.notifyListeners())}subscribe(n){return this.listeners.push(n),()=>this.unsubscribe(n)}unsubscribe(n){this.listeners=this.listeners.filter(t=>t!==n)}notifyListeners(){this.listeners.forEach(n=>n(this._status))}},ep=new Rt;o();var Bf=require("react");Xe();o();o();var Ot=require("react");o();var Ff=require("react");var op=()=>{let e=I(eo),n=I(ot),{signedTransactions:t}=ht(),a=(0,Mt.useMemo)(()=>e.map(p=>{var u,l,x,S,y;let m=(u=t[p.toastId])==null?void 0:u.status;return{sessionId:p.toastId,baseTransactionsInfo:(l=n[p.toastId])!=null?l:{},customTransactionsInfo:(y=(S=(x=t[p.toastId])==null?void 0:x.customTransactionInformation)==null?void 0:S.sessionInformation)!=null?y:{},status:m,isPending:m!=="success"&&m!=="fail"&&m!=="cancelled"}}),[e,t,n]),c=(0,Mt.useMemo)(()=>a.filter(p=>p.isPending),[a]);return{transactionsToastsInfo:a,pendingTransactionsToastsInfo:c}};o();var tp=()=>{let e=I(ri),n=I(eo);return{customToasts:e,transactionToasts:n}};0&&(module.exports={useGetToasts,useTransactionsToasts});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
