// @multiversx/sdk-dapp/__commonjs/hooks/toasts/useTransactionsToasts.js
"use strict";var rp=Object.create;var ze=Object.defineProperty,ap=Object.defineProperties,ip=Object.getOwnPropertyDescriptor,sp=Object.getOwnPropertyDescriptors,cp=Object.getOwnPropertyNames,Tn=Object.getOwnPropertySymbols,pp=Object.getPrototypeOf,_o=Object.prototype.hasOwnProperty,Ft=Object.prototype.propertyIsEnumerable;var Bt=(e,n,t)=>n in e?ze(e,n,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[n]=t,_=(e,n)=>{for(var t in n||(n={}))_o.call(n,t)&&Bt(e,t,n[t]);if(Tn)for(var t of Tn(n))Ft.call(n,t)&&Bt(e,t,n[t]);return e},w=(e,n)=>ap(e,sp(n));var ve=(e,n)=>{var t={};for(var r in e)_o.call(e,r)&&n.indexOf(r)<0&&(t[r]=e[r]);if(e!=null&&Tn)for(var r of Tn(e))n.indexOf(r)<0&&Ft.call(e,r)&&(t[r]=e[r]);return t};var g=(e,n)=>()=>(e&&(n=e(e=0)),n);var T=(e,n)=>()=>(n||e((n={exports:{}}).exports,n),n.exports),ee=(e,n)=>{for(var t in n)ze(e,t,{get:n[t],enumerable:!0})},Ut=(e,n,t,r)=>{if(n&&typeof n=="object"||typeof n=="function")for(let c of cp(n))!_o.call(e,c)&&c!==t&&ze(e,c,{get:()=>n[c],enumerable:!(r=ip(n,c))||r.enumerable});return e};var b=(e,n,t)=>(t=e!=null?rp(pp(e)):{},Ut(n||!e||!e.__esModule?ze(t,"default",{value:e,enumerable:!0}):t,e)),_e=e=>Ut(ze({},"__esModule",{value:!0}),e);var h=(e,n,t)=>new Promise((r,c)=>{var p=l=>{try{u(t.next(l))}catch(x){c(x)}},m=l=>{try{u(t.throw(l))}catch(x){c(x)}},u=l=>l.done?r(l.value):Promise.resolve(l.value).then(p,m);u((t=t.apply(e,n)).next())});var $t=T(hn=>{"use strict";o();hn.byteLength=dp;hn.toByteArray=_p;hn.fromByteArray=fp;var q=[],$=[],mp=typeof Uint8Array!="undefined"?Uint8Array:Array,uo="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ue=0,Wt=uo.length;ue<Wt;++ue)q[ue]=uo[ue],$[uo.charCodeAt(ue)]=ue;var ue,Wt;$["-".charCodeAt(0)]=62;$["_".charCodeAt(0)]=63;function Gt(e){var n=e.length;if(n%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var t=e.indexOf("=");t===-1&&(t=n);var r=t===n?0:4-t%4;return[t,r]}function dp(e){var n=Gt(e),t=n[0],r=n[1];return(t+r)*3/4-r}function lp(e,n,t){return(n+t)*3/4-t}function _p(e){var n,t=Gt(e),r=t[0],c=t[1],p=new mp(lp(e,r,c)),m=0,u=c>0?r-4:r,l;for(l=0;l<u;l+=4)n=$[e.charCodeAt(l)]<<18|$[e.charCodeAt(l+1)]<<12|$[e.charCodeAt(l+2)]<<6|$[e.charCodeAt(l+3)],p[m++]=n>>16&255,p[m++]=n>>8&255,p[m++]=n&255;return c===2&&(n=$[e.charCodeAt(l)]<<2|$[e.charCodeAt(l+1)]>>4,p[m++]=n&255),c===1&&(n=$[e.charCodeAt(l)]<<10|$[e.charCodeAt(l+1)]<<4|$[e.charCodeAt(l+2)]>>2,p[m++]=n>>8&255,p[m++]=n&255),p}function up(e){return q[e>>18&63]+q[e>>12&63]+q[e>>6&63]+q[e&63]}function gp(e,n,t){for(var r,c=[],p=n;p<t;p+=3)r=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),c.push(up(r));return c.join("")}function fp(e){for(var n,t=e.length,r=t%3,c=[],p=16383,m=0,u=t-r;m<u;m+=p)c.push(gp(e,m,m+p>u?u:m+p));return r===1?(n=e[t-1],c.push(q[n>>2]+q[n<<4&63]+"==")):r===2&&(n=(e[t-2]<<8)+e[t-1],c.push(q[n>>10]+q[n>>4&63]+q[n<<2&63]+"=")),c.join("")}});var Ht=T(go=>{o();go.read=function(e,n,t,r,c){var p,m,u=c*8-r-1,l=(1<<u)-1,x=l>>1,S=-7,y=t?c-1:0,L=t?-1:1,R=e[n+y];for(y+=L,p=R&(1<<-S)-1,R>>=-S,S+=u;S>0;p=p*256+e[n+y],y+=L,S-=8);for(m=p&(1<<-S)-1,p>>=-S,S+=r;S>0;m=m*256+e[n+y],y+=L,S-=8);if(p===0)p=1-x;else{if(p===l)return m?NaN:(R?-1:1)*(1/0);m=m+Math.pow(2,r),p=p-x}return(R?-1:1)*m*Math.pow(2,p-r)};go.write=function(e,n,t,r,c,p){var m,u,l,x=p*8-c-1,S=(1<<x)-1,y=S>>1,L=c===23?Math.pow(2,-24)-Math.pow(2,-77):0,R=r?0:p-1,J=r?1:-1,fn=n<0||n===0&&1/n<0?1:0;for(n=Math.abs(n),isNaN(n)||n===1/0?(u=isNaN(n)?1:0,m=S):(m=Math.floor(Math.log(n)/Math.LN2),n*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+y>=1?n+=L/l:n+=L*Math.pow(2,1-y),n*l>=2&&(m++,l/=2),m+y>=S?(u=0,m=S):m+y>=1?(u=(n*l-1)*Math.pow(2,c),m=m+y):(u=n*Math.pow(2,y-1)*Math.pow(2,c),m=0));c>=8;e[t+R]=u&255,R+=J,u/=256,c-=8);for(m=m<<c|u,x+=c;x>0;e[t+R]=m&255,R+=J,m/=256,x-=8);e[t+R-J]|=fn*128}});var or=T(Ie=>{"use strict";o();var fo=$t(),Ae=Ht(),Vt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ie.Buffer=d;Ie.SlowBuffer=Sp;Ie.INSPECT_MAX_BYTES=50;var xn=2147483647;Ie.kMaxLength=xn;d.TYPED_ARRAY_SUPPORT=Tp();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Tp(){try{var e=new Uint8Array(1),n={foo:function(){return 42}};return Object.setPrototypeOf(n,Uint8Array.prototype),Object.setPrototypeOf(e,n),e.foo()===42}catch(t){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function ne(e){if(e>xn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var n=new Uint8Array(e);return Object.setPrototypeOf(n,d.prototype),n}function d(e,n,t){if(typeof e=="number"){if(typeof n=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return yo(e)}return zt(e,n,t)}d.poolSize=8192;function zt(e,n,t){if(typeof e=="string")return xp(e,n);if(ArrayBuffer.isView(e))return yp(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(j(e,ArrayBuffer)||e&&j(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(j(e,SharedArrayBuffer)||e&&j(e.buffer,SharedArrayBuffer)))return ho(e,n,t);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return d.from(r,n,t);var c=bp(e);if(c)return c;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),n,t);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,n,t){return zt(e,n,t)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function Kt(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function hp(e,n,t){return Kt(e),e<=0?ne(e):n!==void 0?typeof t=="string"?ne(e).fill(n,t):ne(e).fill(n):ne(e)}d.alloc=function(e,n,t){return hp(e,n,t)};function yo(e){return Kt(e),ne(e<0?0:bo(e)|0)}d.allocUnsafe=function(e){return yo(e)};d.allocUnsafeSlow=function(e){return yo(e)};function xp(e,n){if((typeof n!="string"||n==="")&&(n="utf8"),!d.isEncoding(n))throw new TypeError("Unknown encoding: "+n);var t=Xt(e,n)|0,r=ne(t),c=r.write(e,n);return c!==t&&(r=r.slice(0,c)),r}function To(e){for(var n=e.length<0?0:bo(e.length)|0,t=ne(n),r=0;r<n;r+=1)t[r]=e[r]&255;return t}function yp(e){if(j(e,Uint8Array)){var n=new Uint8Array(e);return ho(n.buffer,n.byteOffset,n.byteLength)}return To(e)}function ho(e,n,t){if(n<0||e.byteLength<n)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<n+(t||0))throw new RangeError('"length" is outside of buffer bounds');var r;return n===void 0&&t===void 0?r=new Uint8Array(e):t===void 0?r=new Uint8Array(e,n):r=new Uint8Array(e,n,t),Object.setPrototypeOf(r,d.prototype),r}function bp(e){if(d.isBuffer(e)){var n=bo(e.length)|0,t=ne(n);return t.length===0||e.copy(t,0,0,n),t}if(e.length!==void 0)return typeof e.length!="number"||So(e.length)?ne(0):To(e);if(e.type==="Buffer"&&Array.isArray(e.data))return To(e.data)}function bo(e){if(e>=xn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+xn.toString(16)+" bytes");return e|0}function Sp(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(n){return n!=null&&n._isBuffer===!0&&n!==d.prototype};d.compare=function(n,t){if(j(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),j(t,Uint8Array)&&(t=d.from(t,t.offset,t.byteLength)),!d.isBuffer(n)||!d.isBuffer(t))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(n===t)return 0;for(var r=n.length,c=t.length,p=0,m=Math.min(r,c);p<m;++p)if(n[p]!==t[p]){r=n[p],c=t[p];break}return r<c?-1:c<r?1:0};d.isEncoding=function(n){switch(String(n).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(n,t){if(!Array.isArray(n))throw new TypeError('"list" argument must be an Array of Buffers');if(n.length===0)return d.alloc(0);var r;if(t===void 0)for(t=0,r=0;r<n.length;++r)t+=n[r].length;var c=d.allocUnsafe(t),p=0;for(r=0;r<n.length;++r){var m=n[r];if(j(m,Uint8Array))p+m.length>c.length?d.from(m).copy(c,p):Uint8Array.prototype.set.call(c,m,p);else if(d.isBuffer(m))m.copy(c,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return c};function Xt(e,n){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||j(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var t=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&t===0)return 0;for(var c=!1;;)switch(n){case"ascii":case"latin1":case"binary":return t;case"utf8":case"utf-8":return xo(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return t*2;case"hex":return t>>>1;case"base64":return nr(e).length;default:if(c)return r?-1:xo(e).length;n=(""+n).toLowerCase(),c=!0}}d.byteLength=Xt;function wp(e,n,t){var r=!1;if((n===void 0||n<0)&&(n=0),n>this.length||((t===void 0||t>this.length)&&(t=this.length),t<=0)||(t>>>=0,n>>>=0,t<=n))return"";for(e||(e="utf8");;)switch(e){case"hex":return Rp(this,n,t);case"utf8":case"utf-8":return Zt(this,n,t);case"ascii":return Lp(this,n,t);case"latin1":case"binary":return Cp(this,n,t);case"base64":return Pp(this,n,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Dp(this,n,t);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}d.prototype._isBuffer=!0;function ge(e,n,t){var r=e[n];e[n]=e[t],e[t]=r}d.prototype.swap16=function(){var n=this.length;if(n%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var t=0;t<n;t+=2)ge(this,t,t+1);return this};d.prototype.swap32=function(){var n=this.length;if(n%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var t=0;t<n;t+=4)ge(this,t,t+3),ge(this,t+1,t+2);return this};d.prototype.swap64=function(){var n=this.length;if(n%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var t=0;t<n;t+=8)ge(this,t,t+7),ge(this,t+1,t+6),ge(this,t+2,t+5),ge(this,t+3,t+4);return this};d.prototype.toString=function(){var n=this.length;return n===0?"":arguments.length===0?Zt(this,0,n):wp.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(n){if(!d.isBuffer(n))throw new TypeError("Argument must be a Buffer");return this===n?!0:d.compare(this,n)===0};d.prototype.inspect=function(){var n="",t=Ie.INSPECT_MAX_BYTES;return n=this.toString("hex",0,t).replace(/(.{2})/g,"$1 ").trim(),this.length>t&&(n+=" ... "),"<Buffer "+n+">"};Vt&&(d.prototype[Vt]=d.prototype.inspect);d.prototype.compare=function(n,t,r,c,p){if(j(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(n))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof n);if(t===void 0&&(t=0),r===void 0&&(r=n?n.length:0),c===void 0&&(c=0),p===void 0&&(p=this.length),t<0||r>n.length||c<0||p>this.length)throw new RangeError("out of range index");if(c>=p&&t>=r)return 0;if(c>=p)return-1;if(t>=r)return 1;if(t>>>=0,r>>>=0,c>>>=0,p>>>=0,this===n)return 0;for(var m=p-c,u=r-t,l=Math.min(m,u),x=this.slice(c,p),S=n.slice(t,r),y=0;y<l;++y)if(x[y]!==S[y]){m=x[y],u=S[y];break}return m<u?-1:u<m?1:0};function Yt(e,n,t,r,c){if(e.length===0)return-1;if(typeof t=="string"?(r=t,t=0):t>2147483647?t=2147483647:t<-2147483648&&(t=-2147483648),t=+t,So(t)&&(t=c?0:e.length-1),t<0&&(t=e.length+t),t>=e.length){if(c)return-1;t=e.length-1}else if(t<0)if(c)t=0;else return-1;if(typeof n=="string"&&(n=d.from(n,r)),d.isBuffer(n))return n.length===0?-1:qt(e,n,t,r,c);if(typeof n=="number")return n=n&255,typeof Uint8Array.prototype.indexOf=="function"?c?Uint8Array.prototype.indexOf.call(e,n,t):Uint8Array.prototype.lastIndexOf.call(e,n,t):qt(e,[n],t,r,c);throw new TypeError("val must be string, number or Buffer")}function qt(e,n,t,r,c){var p=1,m=e.length,u=n.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||n.length<2)return-1;p=2,m/=2,u/=2,t/=2}function l(R,J){return p===1?R[J]:R.readUInt16BE(J*p)}var x;if(c){var S=-1;for(x=t;x<m;x++)if(l(e,x)===l(n,S===-1?0:x-S)){if(S===-1&&(S=x),x-S+1===u)return S*p}else S!==-1&&(x-=x-S),S=-1}else for(t+u>m&&(t=m-u),x=t;x>=0;x--){for(var y=!0,L=0;L<u;L++)if(l(e,x+L)!==l(n,L)){y=!1;break}if(y)return x}return-1}d.prototype.includes=function(n,t,r){return this.indexOf(n,t,r)!==-1};d.prototype.indexOf=function(n,t,r){return Yt(this,n,t,r,!0)};d.prototype.lastIndexOf=function(n,t,r){return Yt(this,n,t,r,!1)};function vp(e,n,t,r){t=Number(t)||0;var c=e.length-t;r?(r=Number(r),r>c&&(r=c)):r=c;var p=n.length;r>p/2&&(r=p/2);for(var m=0;m<r;++m){var u=parseInt(n.substr(m*2,2),16);if(So(u))return m;e[t+m]=u}return m}function Ap(e,n,t,r){return yn(xo(n,e.length-t),e,t,r)}function Ip(e,n,t,r){return yn(Bp(n),e,t,r)}function kp(e,n,t,r){return yn(nr(n),e,t,r)}function Ep(e,n,t,r){return yn(Fp(n,e.length-t),e,t,r)}d.prototype.write=function(n,t,r,c){if(t===void 0)c="utf8",r=this.length,t=0;else if(r===void 0&&typeof t=="string")c=t,r=this.length,t=0;else if(isFinite(t))t=t>>>0,isFinite(r)?(r=r>>>0,c===void 0&&(c="utf8")):(c=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-t;if((r===void 0||r>p)&&(r=p),n.length>0&&(r<0||t<0)||t>this.length)throw new RangeError("Attempt to write outside buffer bounds");c||(c="utf8");for(var m=!1;;)switch(c){case"hex":return vp(this,n,t,r);case"utf8":case"utf-8":return Ap(this,n,t,r);case"ascii":case"latin1":case"binary":return Ip(this,n,t,r);case"base64":return kp(this,n,t,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Ep(this,n,t,r);default:if(m)throw new TypeError("Unknown encoding: "+c);c=(""+c).toLowerCase(),m=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Pp(e,n,t){return n===0&&t===e.length?fo.fromByteArray(e):fo.fromByteArray(e.slice(n,t))}function Zt(e,n,t){t=Math.min(e.length,t);for(var r=[],c=n;c<t;){var p=e[c],m=null,u=p>239?4:p>223?3:p>191?2:1;if(c+u<=t){var l,x,S,y;switch(u){case 1:p<128&&(m=p);break;case 2:l=e[c+1],(l&192)===128&&(y=(p&31)<<6|l&63,y>127&&(m=y));break;case 3:l=e[c+1],x=e[c+2],(l&192)===128&&(x&192)===128&&(y=(p&15)<<12|(l&63)<<6|x&63,y>2047&&(y<55296||y>57343)&&(m=y));break;case 4:l=e[c+1],x=e[c+2],S=e[c+3],(l&192)===128&&(x&192)===128&&(S&192)===128&&(y=(p&15)<<18|(l&63)<<12|(x&63)<<6|S&63,y>65535&&y<1114112&&(m=y))}}m===null?(m=65533,u=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),c+=u}return Np(r)}var jt=4096;function Np(e){var n=e.length;if(n<=jt)return String.fromCharCode.apply(String,e);for(var t="",r=0;r<n;)t+=String.fromCharCode.apply(String,e.slice(r,r+=jt));return t}function Lp(e,n,t){var r="";t=Math.min(e.length,t);for(var c=n;c<t;++c)r+=String.fromCharCode(e[c]&127);return r}function Cp(e,n,t){var r="";t=Math.min(e.length,t);for(var c=n;c<t;++c)r+=String.fromCharCode(e[c]);return r}function Rp(e,n,t){var r=e.length;(!n||n<0)&&(n=0),(!t||t<0||t>r)&&(t=r);for(var c="",p=n;p<t;++p)c+=Up[e[p]];return c}function Dp(e,n,t){for(var r=e.slice(n,t),c="",p=0;p<r.length-1;p+=2)c+=String.fromCharCode(r[p]+r[p+1]*256);return c}d.prototype.slice=function(n,t){var r=this.length;n=~~n,t=t===void 0?r:~~t,n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),t<0?(t+=r,t<0&&(t=0)):t>r&&(t=r),t<n&&(t=n);var c=this.subarray(n,t);return Object.setPrototypeOf(c,d.prototype),c};function C(e,n,t){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+n>t)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(n,t,r){n=n>>>0,t=t>>>0,r||C(n,t,this.length);for(var c=this[n],p=1,m=0;++m<t&&(p*=256);)c+=this[n+m]*p;return c};d.prototype.readUintBE=d.prototype.readUIntBE=function(n,t,r){n=n>>>0,t=t>>>0,r||C(n,t,this.length);for(var c=this[n+--t],p=1;t>0&&(p*=256);)c+=this[n+--t]*p;return c};d.prototype.readUint8=d.prototype.readUInt8=function(n,t){return n=n>>>0,t||C(n,1,this.length),this[n]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(n,t){return n=n>>>0,t||C(n,2,this.length),this[n]|this[n+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(n,t){return n=n>>>0,t||C(n,2,this.length),this[n]<<8|this[n+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(n,t){return n=n>>>0,t||C(n,4,this.length),(this[n]|this[n+1]<<8|this[n+2]<<16)+this[n+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(n,t){return n=n>>>0,t||C(n,4,this.length),this[n]*16777216+(this[n+1]<<16|this[n+2]<<8|this[n+3])};d.prototype.readIntLE=function(n,t,r){n=n>>>0,t=t>>>0,r||C(n,t,this.length);for(var c=this[n],p=1,m=0;++m<t&&(p*=256);)c+=this[n+m]*p;return p*=128,c>=p&&(c-=Math.pow(2,8*t)),c};d.prototype.readIntBE=function(n,t,r){n=n>>>0,t=t>>>0,r||C(n,t,this.length);for(var c=t,p=1,m=this[n+--c];c>0&&(p*=256);)m+=this[n+--c]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*t)),m};d.prototype.readInt8=function(n,t){return n=n>>>0,t||C(n,1,this.length),this[n]&128?(255-this[n]+1)*-1:this[n]};d.prototype.readInt16LE=function(n,t){n=n>>>0,t||C(n,2,this.length);var r=this[n]|this[n+1]<<8;return r&32768?r|4294901760:r};d.prototype.readInt16BE=function(n,t){n=n>>>0,t||C(n,2,this.length);var r=this[n+1]|this[n]<<8;return r&32768?r|4294901760:r};d.prototype.readInt32LE=function(n,t){return n=n>>>0,t||C(n,4,this.length),this[n]|this[n+1]<<8|this[n+2]<<16|this[n+3]<<24};d.prototype.readInt32BE=function(n,t){return n=n>>>0,t||C(n,4,this.length),this[n]<<24|this[n+1]<<16|this[n+2]<<8|this[n+3]};d.prototype.readFloatLE=function(n,t){return n=n>>>0,t||C(n,4,this.length),Ae.read(this,n,!0,23,4)};d.prototype.readFloatBE=function(n,t){return n=n>>>0,t||C(n,4,this.length),Ae.read(this,n,!1,23,4)};d.prototype.readDoubleLE=function(n,t){return n=n>>>0,t||C(n,8,this.length),Ae.read(this,n,!0,52,8)};d.prototype.readDoubleBE=function(n,t){return n=n>>>0,t||C(n,8,this.length),Ae.read(this,n,!1,52,8)};function B(e,n,t,r,c,p){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(n>c||n<p)throw new RangeError('"value" argument is out of bounds');if(t+r>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(n,t,r,c){if(n=+n,t=t>>>0,r=r>>>0,!c){var p=Math.pow(2,8*r)-1;B(this,n,t,r,p,0)}var m=1,u=0;for(this[t]=n&255;++u<r&&(m*=256);)this[t+u]=n/m&255;return t+r};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(n,t,r,c){if(n=+n,t=t>>>0,r=r>>>0,!c){var p=Math.pow(2,8*r)-1;B(this,n,t,r,p,0)}var m=r-1,u=1;for(this[t+m]=n&255;--m>=0&&(u*=256);)this[t+m]=n/u&255;return t+r};d.prototype.writeUint8=d.prototype.writeUInt8=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,1,255,0),this[t]=n&255,t+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,2,65535,0),this[t]=n&255,this[t+1]=n>>>8,t+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,2,65535,0),this[t]=n>>>8,this[t+1]=n&255,t+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,4,4294967295,0),this[t+3]=n>>>24,this[t+2]=n>>>16,this[t+1]=n>>>8,this[t]=n&255,t+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,4,4294967295,0),this[t]=n>>>24,this[t+1]=n>>>16,this[t+2]=n>>>8,this[t+3]=n&255,t+4};d.prototype.writeIntLE=function(n,t,r,c){if(n=+n,t=t>>>0,!c){var p=Math.pow(2,8*r-1);B(this,n,t,r,p-1,-p)}var m=0,u=1,l=0;for(this[t]=n&255;++m<r&&(u*=256);)n<0&&l===0&&this[t+m-1]!==0&&(l=1),this[t+m]=(n/u>>0)-l&255;return t+r};d.prototype.writeIntBE=function(n,t,r,c){if(n=+n,t=t>>>0,!c){var p=Math.pow(2,8*r-1);B(this,n,t,r,p-1,-p)}var m=r-1,u=1,l=0;for(this[t+m]=n&255;--m>=0&&(u*=256);)n<0&&l===0&&this[t+m+1]!==0&&(l=1),this[t+m]=(n/u>>0)-l&255;return t+r};d.prototype.writeInt8=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,1,127,-128),n<0&&(n=255+n+1),this[t]=n&255,t+1};d.prototype.writeInt16LE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,2,32767,-32768),this[t]=n&255,this[t+1]=n>>>8,t+2};d.prototype.writeInt16BE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,2,32767,-32768),this[t]=n>>>8,this[t+1]=n&255,t+2};d.prototype.writeInt32LE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,4,2147483647,-2147483648),this[t]=n&255,this[t+1]=n>>>8,this[t+2]=n>>>16,this[t+3]=n>>>24,t+4};d.prototype.writeInt32BE=function(n,t,r){return n=+n,t=t>>>0,r||B(this,n,t,4,2147483647,-2147483648),n<0&&(n=4294967295+n+1),this[t]=n>>>24,this[t+1]=n>>>16,this[t+2]=n>>>8,this[t+3]=n&255,t+4};function Jt(e,n,t,r,c,p){if(t+r>e.length)throw new RangeError("Index out of range");if(t<0)throw new RangeError("Index out of range")}function Qt(e,n,t,r,c){return n=+n,t=t>>>0,c||Jt(e,n,t,4,34028234663852886e22,-34028234663852886e22),Ae.write(e,n,t,r,23,4),t+4}d.prototype.writeFloatLE=function(n,t,r){return Qt(this,n,t,!0,r)};d.prototype.writeFloatBE=function(n,t,r){return Qt(this,n,t,!1,r)};function er(e,n,t,r,c){return n=+n,t=t>>>0,c||Jt(e,n,t,8,17976931348623157e292,-17976931348623157e292),Ae.write(e,n,t,r,52,8),t+8}d.prototype.writeDoubleLE=function(n,t,r){return er(this,n,t,!0,r)};d.prototype.writeDoubleBE=function(n,t,r){return er(this,n,t,!1,r)};d.prototype.copy=function(n,t,r,c){if(!d.isBuffer(n))throw new TypeError("argument should be a Buffer");if(r||(r=0),!c&&c!==0&&(c=this.length),t>=n.length&&(t=n.length),t||(t=0),c>0&&c<r&&(c=r),c===r||n.length===0||this.length===0)return 0;if(t<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(c<0)throw new RangeError("sourceEnd out of bounds");c>this.length&&(c=this.length),n.length-t<c-r&&(c=n.length-t+r);var p=c-r;return this===n&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(t,r,c):Uint8Array.prototype.set.call(n,this.subarray(r,c),t),p};d.prototype.fill=function(n,t,r,c){if(typeof n=="string"){if(typeof t=="string"?(c=t,t=0,r=this.length):typeof r=="string"&&(c=r,r=this.length),c!==void 0&&typeof c!="string")throw new TypeError("encoding must be a string");if(typeof c=="string"&&!d.isEncoding(c))throw new TypeError("Unknown encoding: "+c);if(n.length===1){var p=n.charCodeAt(0);(c==="utf8"&&p<128||c==="latin1")&&(n=p)}}else typeof n=="number"?n=n&255:typeof n=="boolean"&&(n=Number(n));if(t<0||this.length<t||this.length<r)throw new RangeError("Out of range index");if(r<=t)return this;t=t>>>0,r=r===void 0?this.length:r>>>0,n||(n=0);var m;if(typeof n=="number")for(m=t;m<r;++m)this[m]=n;else{var u=d.isBuffer(n)?n:d.from(n,c),l=u.length;if(l===0)throw new TypeError('The value "'+n+'" is invalid for argument "value"');for(m=0;m<r-t;++m)this[m+t]=u[m%l]}return this};var Op=/[^+/0-9A-Za-z-_]/g;function Mp(e){if(e=e.split("=")[0],e=e.trim().replace(Op,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function xo(e,n){n=n||1/0;for(var t,r=e.length,c=null,p=[],m=0;m<r;++m){if(t=e.charCodeAt(m),t>55295&&t<57344){if(!c){if(t>56319){(n-=3)>-1&&p.push(239,191,189);continue}else if(m+1===r){(n-=3)>-1&&p.push(239,191,189);continue}c=t;continue}if(t<56320){(n-=3)>-1&&p.push(239,191,189),c=t;continue}t=(c-55296<<10|t-56320)+65536}else c&&(n-=3)>-1&&p.push(239,191,189);if(c=null,t<128){if((n-=1)<0)break;p.push(t)}else if(t<2048){if((n-=2)<0)break;p.push(t>>6|192,t&63|128)}else if(t<65536){if((n-=3)<0)break;p.push(t>>12|224,t>>6&63|128,t&63|128)}else if(t<1114112){if((n-=4)<0)break;p.push(t>>18|240,t>>12&63|128,t>>6&63|128,t&63|128)}else throw new Error("Invalid code point")}return p}function Bp(e){for(var n=[],t=0;t<e.length;++t)n.push(e.charCodeAt(t)&255);return n}function Fp(e,n){for(var t,r,c,p=[],m=0;m<e.length&&!((n-=2)<0);++m)t=e.charCodeAt(m),r=t>>8,c=t%256,p.push(c),p.push(r);return p}function nr(e){return fo.toByteArray(Mp(e))}function yn(e,n,t,r){for(var c=0;c<r&&!(c+t>=n.length||c>=e.length);++c)n[c+t]=e[c];return c}function j(e,n){return e instanceof n||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===n.name}function So(e){return e!==e}var Up=function(){for(var e="0123456789abcdef",n=new Array(256),t=0;t<16;++t)for(var r=t*16,c=0;c<16;++c)n[r+c]=e[t]+e[c];return n}()});var sr=T((Hf,ir)=>{o();var E=ir.exports={},z,K;function wo(){throw new Error("setTimeout has not been defined")}function vo(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?z=setTimeout:z=wo}catch(e){z=wo}try{typeof clearTimeout=="function"?K=clearTimeout:K=vo}catch(e){K=vo}})();function tr(e){if(z===setTimeout)return setTimeout(e,0);if((z===wo||!z)&&setTimeout)return z=setTimeout,setTimeout(e,0);try{return z(e,0)}catch(n){try{return z.call(null,e,0)}catch(t){return z.call(this,e,0)}}}function Wp(e){if(K===clearTimeout)return clearTimeout(e);if((K===vo||!K)&&clearTimeout)return K=clearTimeout,clearTimeout(e);try{return K(e)}catch(n){try{return K.call(null,e)}catch(t){return K.call(this,e)}}}var oe=[],ke=!1,fe,bn=-1;function Gp(){!ke||!fe||(ke=!1,fe.length?oe=fe.concat(oe):bn=-1,oe.length&&rr())}function rr(){if(!ke){var e=tr(Gp);ke=!0;for(var n=oe.length;n;){for(fe=oe,oe=[];++bn<n;)fe&&fe[bn].run();bn=-1,n=oe.length}fe=null,ke=!1,Wp(e)}}E.nextTick=function(e){var n=new Array(arguments.length-1);if(arguments.length>1)for(var t=1;t<arguments.length;t++)n[t-1]=arguments[t];oe.push(new ar(e,n)),oe.length===1&&!ke&&tr(rr)};function ar(e,n){this.fun=e,this.array=n}ar.prototype.run=function(){this.fun.apply(null,this.array)};E.title="browser";E.browser=!0;E.env={};E.argv=[];E.version="";E.versions={};function te(){}E.on=te;E.addListener=te;E.once=te;E.off=te;E.removeListener=te;E.removeAllListeners=te;E.emit=te;E.prependListener=te;E.prependOnceListener=te;E.listeners=function(e){return[]};E.binding=function(e){throw new Error("process.binding is not supported")};E.cwd=function(){return"/"};E.chdir=function(e){throw new Error("process.chdir is not supported")};E.umask=function(){return 0}});var a,s,$p,i,o=g(()=>{a=b(or()),s=b(sr()),$p=function(e){function n(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return n();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:n});var t=__magic__;return t}(Object),i=$p});var cr=g(()=>{"use strict";o()});var Ao=g(()=>{"use strict";o()});var pr=g(()=>{"use strict";o()});var Io,mr=g(()=>{"use strict";o();Io=(r=>(r.SetGuardian="SetGuardian",r.GuardAccount="GuardAccount",r.UnGuardAccount="UnGuardAccount",r))(Io||{})});var dr=g(()=>{"use strict";o()});var ko=g(()=>{"use strict";o()});var lr=g(()=>{"use strict";o()});var Eo=g(()=>{"use strict";o()});var _r=g(()=>{"use strict";o()});var ur=g(()=>{"use strict";o()});var Ke,Ee,Pe=g(()=>{"use strict";o();Ke=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),Ee=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var gr,nT,fr,oT,D=g(()=>{"use strict";o();Pe();gr=(t=>(t.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",t.reloginRequest="RELOGIN_REQUEST",t))(gr||{}),nT=_(_({},Ee.WindowProviderRequestEnums),gr),fr=(n=>(n.reloginResponse="RELOGIN_RESPONSE",n))(fr||{}),oT=_(_({},Ee.WindowProviderResponseEnums),fr)});var Tr=g(()=>{"use strict";o()});var hr=g(()=>{"use strict";o()});var F=g(()=>{"use strict";o()});var xr=g(()=>{"use strict";o()});var yr=g(()=>{"use strict";o()});var br=g(()=>{"use strict";o()});var N=g(()=>{"use strict";o();Eo();_r();ur();D();Tr();hr();F();xr();yr();br()});var Ne,Sr,xT,wr,yT,vr,bT,ST,Hp,Le=g(()=>{"use strict";o();N();Ne={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:Sr,egldLabel:xT}=Ne["devnet"],{chainId:wr,egldLabel:yT}=Ne["testnet"],{chainId:vr,egldLabel:bT}=Ne["mainnet"],ST={["devnet"]:Sr,["testnet"]:wr,["mainnet"]:vr},Hp={[Sr]:"devnet",[wr]:"testnet",[vr]:"mainnet"}});var Xe=g(()=>{"use strict";o()});var X,Ar=g(()=>{"use strict";o();X=require("@multiversx/sdk-web-wallet-provider")});var Ye,Sn=g(()=>{"use strict";o();Ye=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var Ir,kr,Po,ET,PT,Er=g(()=>{"use strict";o();Sn();Po=String((kr=(Ir=Ye.safeWindow)==null?void 0:Ir.navigator)==null?void 0:kr.userAgent),ET=/^((?!chrome|android).)*safari/i.test(Po),PT=/firefox/i.test(Po)&&/windows/i.test(Po)});var wn,vn,Pr,An,Nr,Ce,A=g(()=>{"use strict";o();cr();Ao();pr();mr();dr();ko();lr();Le();Xe();Ar();Er();wn=5e4,vn=1e9,Pr=1,An="logout",Nr="login",Ce="0"});var Ze,In=g(()=>{"use strict";o();Ze=()=>Date.now()/1e3});var Lr=g(()=>{"use strict";o()});var Cr=g(()=>{"use strict";o()});var Rr=g(()=>{"use strict";o();In();Lr();Cr()});var No={};ee(No,{clear:()=>zp,getItem:()=>qp,localStorageKeys:()=>re,removeItem:()=>jp,setItem:()=>Vp});var re,kn,Vp,qp,jp,zp,Te=g(()=>{"use strict";o();Rr();re={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},kn=typeof localStorage!="undefined",Vp=({key:e,data:n,expires:t})=>{!kn||localStorage.setItem(String(e),JSON.stringify({expires:t,data:n}))},qp=e=>{if(!kn)return;let n=localStorage.getItem(String(e));if(!n)return null;let t=JSON.parse(n);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Ze()>=t.expires?(localStorage.removeItem(String(e)),null):t.data},jp=e=>{!kn||localStorage.removeItem(String(e))},zp=()=>{!kn||localStorage.clear()}});var Lo={};ee(Lo,{clear:()=>Br,getItem:()=>Or,removeItem:()=>Mr,setItem:()=>Dr,storage:()=>Kp});var Dr,Or,Mr,Br,Kp,Fr=g(()=>{"use strict";o();Dr=({key:e,data:n,expires:t})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:t,data:n}))},Or=e=>{let n=sessionStorage.getItem(String(e));if(!n)return null;let t=JSON.parse(n);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Date.now()>=t.expires?(sessionStorage.removeItem(String(e)),null):t.data},Mr=e=>sessionStorage.removeItem(String(e)),Br=()=>sessionStorage.clear(),Kp={setItem:Dr,getItem:Or,removeItem:Mr,clear:Br}});var pe,Re=g(()=>{"use strict";o();Te();Fr();pe={session:Lo,local:No}});var Co,k,De,M=g(()=>{"use strict";o();Co=require("@reduxjs/toolkit");A();k=(0,Co.createAction)(An),De=(0,Co.createAction)(Nr,e=>({payload:e}))});var Do,Ur,Wr,En,Ro,Gr,Oo,ih,Mo,sh,ch,ph,mh,dh,lh,Xp,Yp,Pn,Nn=g(()=>{"use strict";o();Do=require("@multiversx/sdk-core"),Ur=require("@reduxjs/toolkit"),Wr=require("redux-persist");A();Re();Te();M();En={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:Ce},Ro={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":En},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Gr=(0,Ur.createSlice)({name:"accountInfoSlice",initialState:Ro,reducers:{setAddress:(e,n)=>{let t=n.payload;e.address=t,e.publicKey=t?new Do.Address(t).hex():""},setAccount:(e,n)=>{let t=e.address===n.payload.address;e.accounts={[e.address]:t?n.payload:En},e.shard=n.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,n)=>{let{address:t}=e;e.accounts[t].nonce=n.payload},setAccountShard:(e,n)=>{e.shard=n.payload},setLedgerAccount:(e,n)=>{e.ledgerAccount=n.payload},updateLedgerAccount:(e,n)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=n.payload.index,e.ledgerAccount.address=n.payload.address)},setWalletConnectAccount:(e,n)=>{e.walletConnectAccount=n.payload},setIsAccountLoading:(e,n)=>{e.isAccountLoading=n.payload,e.accountLoadingError=null},setAccountLoadingError:(e,n)=>{e.accountLoadingError=n.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,n)=>{e.websocketEvent={timestamp:Date.now(),message:n.payload}},setWebsocketBatchEvent:(e,n)=>{e.websocketBatchEvent={timestamp:Date.now(),data:n.payload}}},extraReducers:e=>{e.addCase(k,()=>(pe.local.removeItem(re.loginExpiresAt),Ro)),e.addCase(De,(n,t)=>{let{address:r}=t.payload;n.address=r,n.publicKey=new Do.Address(r).toHex()}),e.addCase(Wr.REHYDRATE,(n,t)=>{var x;if(!((x=t.payload)!=null&&x.account))return;let{account:r}=t.payload,{address:c,shard:p,accounts:m,publicKey:u}=r;n.address=c,n.shard=p;let l=m&&c in m;n.accounts=l?m:Ro.accounts,n.publicKey=u})}}),{setAccount:Oo,setAddress:ih,setAccountNonce:Mo,setAccountShard:sh,setLedgerAccount:ch,updateLedgerAccount:ph,setWalletConnectAccount:mh,setIsAccountLoading:dh,setAccountLoadingError:lh,setWebsocketEvent:Xp,setWebsocketBatchEvent:Yp}=Gr.actions,Pn=Gr.reducer});function Je(){return new Date().setHours(new Date().getHours()+24)}function Qe(e){pe.local.setItem({key:re.loginExpiresAt,data:e,expires:e})}var Bo=g(()=>{"use strict";o();Re();Te()});var Hr,$r,Vr,Sh,wh,vh,Zp,Ah,Ih,qr,kh,Eh,Ph,Ln,Cn=g(()=>{"use strict";o();Hr=require("@reduxjs/toolkit");Bo();D();M();$r={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},Vr=(0,Hr.createSlice)({name:"loginInfoSlice",initialState:$r,reducers:{setLoginMethod:(e,n)=>{e.loginMethod=n.payload},setTokenLogin:(e,n)=>{e.tokenLogin=n.payload},setTokenLoginSignature:(e,n)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=n.payload)},setWalletLogin:(e,n)=>{e.walletLogin=n.payload},setWalletConnectLogin:(e,n)=>{e.walletConnectLogin=n.payload},setLedgerLogin:(e,n)=>{e.ledgerLogin=n.payload},setWebviewLogin:(e,n)=>{e.webviewLogin=n.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,n)=>{e.logoutRoute=n.payload},setIsWalletConnectV2Initialized:(e,n)=>{e.isWalletConnectV2Initialized=n.payload}},extraReducers:e=>{e.addCase(k,()=>$r),e.addCase(De,(n,t)=>{n.isLoginSessionInvalid=!1,n.loginMethod=t.payload.loginMethod,n.iframeLoginType=t.payload.iframeLoginType,Qe(Je())})}}),{setLoginMethod:Sh,setWalletConnectLogin:wh,setLedgerLogin:vh,setTokenLogin:Zp,setTokenLoginSignature:Ah,setWalletLogin:Ih,invalidateLoginSession:qr,setLogoutRoute:kh,setIsWalletConnectV2Initialized:Eh,setWebviewLogin:Ph}=Vr.actions,Ln=Vr.reducer});var zr,jr,Kr,Rh,Jp,Dh,Oh,Rn,Dn=g(()=>{"use strict";o();zr=require("@reduxjs/toolkit");M();jr={},Kr=(0,zr.createSlice)({name:"modalsSlice",initialState:jr,reducers:{setTxSubmittedModal:(e,n)=>{e.txSubmittedModal=n.payload},setNotificationModal:(e,n)=>{e.notificationModal=n.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(k,()=>jr)}}),{setTxSubmittedModal:Rh,setNotificationModal:Jp,clearTxSubmittedModal:Dh,clearNotificationModal:Oh}=Kr.actions,Rn=Kr.reducer});var he,en=g(()=>{"use strict";o();he=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var O,W=g(()=>{"use strict";o();en();O=()=>{if(!he())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:n,hash:t,origin:r,href:c,search:p}}=window;return{pathname:n,hash:t,origin:r,href:c,search:p}}});var Xr=g(()=>{"use strict";o();W()});var Yr=g(()=>{"use strict";o();me()});var Zr=g(()=>{"use strict";o();Sn()});var me=g(()=>{"use strict";o();Xr();Yr();W();Zr()});var On=g(()=>{"use strict";o();me()});var Qr=g(()=>{"use strict";o();On()});function ea(e){return e[Math.floor(Math.random()*e.length)]}var na=g(()=>{"use strict";o()});var Fo=g(()=>{"use strict";o();ko()});var oa=g(()=>{"use strict";o();Qr();na();Fo()});var ta,ra,aa,Uo,em,ia,ux,gx,fx,Mn,Bn=g(()=>{"use strict";o();ta=require("@reduxjs/toolkit"),ra=b(require("lodash.omit")),aa=require("redux-persist");Xe();M();oa();Uo={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},em={network:Uo},ia=(0,ta.createSlice)({name:"appConfig",initialState:em,reducers:{initializeNetworkConfig:(e,n)=>{let t=ea(n.payload.walletConnectV2RelayAddresses),r=(0,ra.default)(n.payload,"walletConnectV2RelayAddresses");e.network=w(_(_({},e.network),r),{walletConnectV2RelayAddress:t})},updateNetworkConfig:(e,n)=>{e.network=_(_({},e.network),n.payload)},setCustomWalletAddress:(e,n)=>{e.network.customWalletAddress=n.payload}},extraReducers:e=>{e.addCase(k,n=>{n.network.customWalletAddress=void 0}),e.addCase(aa.REHYDRATE,(n,t)=>{var c,p;if(!((p=(c=t.payload)==null?void 0:c.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:r}}=t.payload;n.network.customWalletAddress=r})}}),{initializeNetworkConfig:ux,updateNetworkConfig:gx,setCustomWalletAddress:fx}=ia.actions,Mn=ia.reducer});var sa,Wo,ca,Sx,wx,vx,Fn,Un=g(()=>{"use strict";o();sa=require("@reduxjs/toolkit");N();M();Wo={isSigning:!1,signedSessions:{}},ca=(0,sa.createSlice)({name:"signedMessageInfoSliceState",initialState:Wo,reducers:{setSignSession:(e,n)=>{let{sessionId:t,signedSession:r,errorMessage:c}=n.payload;c&&(e.errorMessage=c),e.isSigning=r.status==="pending",e.signedSessions[t]=_(_({},e.signedSessions[t]),r)},setSignSessionState:(e,n)=>_(_({},e),n.payload),clearSignedMessageInfo:()=>Wo},extraReducers:e=>{e.addCase(k,()=>Wo)}}),{setSignSession:Sx,clearSignedMessageInfo:wx,setSignSessionState:vx}=ca.actions,Fn=ca.reducer});var ma,da,pa,la,nm,om,Lx,tm,Wn,Gn=g(()=>{"use strict";o();ma=require("@reduxjs/toolkit"),da=require("redux-persist");N();In();M();pa={customToasts:[],transactionToasts:[]},la=(0,ma.createSlice)({name:"toastsSlice",initialState:pa,reducers:{addCustomToast:(e,n)=>{let t=n.payload.toastId||`custom-toast-${e.customToasts.length+1}`,r=e.customToasts.findIndex(c=>c.toastId===t);if(r!==-1){e.customToasts[r]=w(_(_({},e.customToasts[r]),n.payload),{type:"custom",toastId:t});return}e.customToasts.push(w(_({},n.payload),{type:"custom",toastId:t}))},removeCustomToast:(e,n)=>{e.customToasts=e.customToasts.filter(t=>t.toastId!==n.payload)},addTransactionToast:(e,n)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Ze(),toastId:n.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,n)=>{e.transactionToasts=e.transactionToasts.filter(t=>t.toastId!==n.payload)}},extraReducers:e=>{e.addCase(k,()=>pa),e.addCase(da.REHYDRATE,(n,t)=>{var c,p;let r=(p=(c=t.customToasts)==null?void 0:c.filter(m=>!("component"in m)))!=null?p:[];n.customToasts=r})}}),{addCustomToast:nm,removeCustomToast:om,addTransactionToast:Lx,removeTransactionToast:tm}=la.actions,Wn=la.reducer});var _a,$o,Ho,Vo,rm,Go,ua,Ox,am,qo,$n,Hn=g(()=>{"use strict";o();_a=require("@reduxjs/toolkit");M();$o="Transaction failed",Ho="Transaction successful",Vo="Processing transaction",rm="Transaction submitted",Go={},ua=(0,_a.createSlice)({name:"transactionsInfo",initialState:Go,reducers:{setTransactionsDisplayInfo(e,n){let{sessionId:t,transactionsDisplayInfo:r}=n.payload;t!=null&&(e[t]={errorMessage:(r==null?void 0:r.errorMessage)||$o,successMessage:(r==null?void 0:r.successMessage)||Ho,processingMessage:(r==null?void 0:r.processingMessage)||Vo,submittedMessage:(r==null?void 0:r.submittedMessage)||rm,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,n){n.payload!=null&&delete e[n.payload]},clearTransactionsInfo:()=>Go},extraReducers:e=>{e.addCase(k,()=>Go)}}),{clearTransactionsInfo:Ox,setTransactionsDisplayInfo:am,clearTransactionsInfoForSessionId:qo}=ua.actions,$n=ua.reducer});function Me(e){return e!=null&&(gm(e)||ym(e))}function Be(e){return e!=null&&(fm(e)||bm(e))}function Fe(e){return e!=null&&(Tm(e)||Sm(e))}function ga(e){return e!=null&&(hm(e)||wm(e))}function jo(e){return e!=null&&xm(e)}function gm(e){return e!=null&&im.includes(e)}function fm(e){return e!=null&&sm.includes(e)}function Tm(e){return e!=null&&cm.includes(e)}function hm(e){return e!=null&&pm.includes(e)}function xm(e){return e!=null&&mm.includes(e)}function ym(e){return e!=null&&dm.includes(e)}function bm(e){return e!=null&&lm.includes(e)}function Sm(e){return e!=null&&_m.includes(e)}function wm(e){return e!=null&&um.includes(e)}var im,sm,cm,pm,mm,dm,lm,_m,um,Ue=g(()=>{"use strict";o();D();im=["sent"],sm=["success"],cm=["fail","cancelled","timedOut"],pm=["invalid"],mm=["timedOut"],dm=["pending"],lm=["success"],_m=["fail","invalid"],um=["not executed"]});var fa,Ta,nn,vm,ha,xa,ya,Am,Vn,Im,km,Vx,Em,on,zo,qx,qn,jn=g(()=>{"use strict";o();fa=require("@reduxjs/toolkit"),Ta=require("redux-persist");D();Ue();M();nn={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},vm={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},ha=(0,fa.createSlice)({name:"transactionsSlice",initialState:nn,reducers:{moveTransactionsToSignedState:(e,n)=>{var l,x;let{sessionId:t,transactions:r,errorMessage:c,status:p,redirectRoute:m,customTransactionInformation:u}=n.payload;e.customTransactionInformationForSessionId[t]=_(_(_({},vm),((l=e.signedTransactions[t])==null?void 0:l.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[t]||{}),e.signedTransactions[t]={transactions:r,status:p,errorMessage:c,redirectRoute:m,customTransactionInformation:_(_({},e.customTransactionInformationForSessionId[t]),u!=null?u:{})},((x=e==null?void 0:e.transactionsToSign)==null?void 0:x.sessionId)===t&&(e.transactionsToSign=nn.transactionsToSign)},clearSignedTransaction:(e,n)=>{e.signedTransactions[n.payload]&&delete e.signedTransactions[n.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,n)=>{e.signedTransactions=_(_({},e.signedTransactions),n.payload)},updateSignedTransactions:(e,n)=>{let{sessionId:t,status:r,errorMessage:c,transactions:p}=n.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].status=r,c!=null&&(e.signedTransactions[t].errorMessage=c),p!=null&&(e.signedTransactions[t].transactions=p))},updateSignedTransactionStatus:(e,n)=>{var x,S,y,L,R,J,fn,Mt;let{sessionId:t,status:r,errorMessage:c,transactionHash:p,serverTransaction:m,inTransit:u}=n.payload,l=(S=(x=e.signedTransactions)==null?void 0:x[t])==null?void 0:S.transactions;if(l!=null){e.signedTransactions[t].transactions=l.map(Q=>Q.hash===p?w(_(_({},m!=null?m:{}),Q),{status:r,errorMessage:c,inTransit:u}):Q);let np=(L=(y=e.signedTransactions[t])==null?void 0:y.transactions)==null?void 0:L.every(Q=>Be(Q.status)),op=(J=(R=e.signedTransactions[t])==null?void 0:R.transactions)==null?void 0:J.some(Q=>Fe(Q.status)),tp=(Mt=(fn=e.signedTransactions[t])==null?void 0:fn.transactions)==null?void 0:Mt.every(Q=>ga(Q.status));np&&(e.signedTransactions[t].status="success"),op&&(e.signedTransactions[t].status="fail"),tp&&(e.signedTransactions[t].status="invalid")}},setTransactionsToSign:(e,n)=>{e.transactionsToSign=n.payload;let{sessionId:t,customTransactionInformation:r}=n.payload;e.customTransactionInformationForSessionId[t]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=nn.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=nn.signedTransactions},setSignTransactionsError:(e,n)=>{e.signTransactionsError=n.payload},setSignTransactionsCancelMessage:(e,n)=>{e.signTransactionsCancelMessage=n.payload},updateSignedTransactionsCustomTransactionInformation:(e,n)=>{let{sessionId:t,customTransactionInformationOverrides:r}=n.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].customTransactionInformation=_(_({},e.signedTransactions[t].customTransactionInformation),r))}},extraReducers:e=>{e.addCase(k,()=>nn),e.addCase(Ta.REHYDRATE,(n,t)=>{var m;if(n.signTransactionsCancelMessage=null,!((m=t.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:c}=t.payload.transactions,p=Object.entries(r).reduce((u,[l,x])=>{let S=new Date(l),y=new Date;return y.setHours(y.getHours()+5),y-S>0||(u[l]=x),u},{});c!=null&&(n.customTransactionInformationForSessionId=c),r!=null&&(n.signedTransactions=p)})}}),{updateSignedTransactionStatus:xa,updateSignedTransactions:ya,setTransactionsToSign:Am,clearAllTransactionsToSign:Vn,clearAllSignedTransactions:Im,clearSignedTransaction:km,clearTransactionToSign:Vx,setSignTransactionsError:Em,setSignTransactionsCancelMessage:on,moveTransactionsToSignedState:zo,updateSignedTransactionsCustomTransactionInformation:qx}=ha.actions,qn=ha.reducer});var ba,Ko,Sa,Xx,Yx,Pm,Zx,zn,Kn=g(()=>{"use strict";o();ba=require("@reduxjs/toolkit");M();Ko={},Sa=(0,ba.createSlice)({name:"batchTransactionsSlice",initialState:Ko,reducers:{setBatchTransactions:(e,n)=>{e[n.payload.id]=n.payload},updateBatchTransactions:(e,n)=>{e[n.payload.id]=n.payload},clearBatchTransactions:(e,n)=>{delete e[n.payload.batchId]},clearAllBatchTransactions:()=>Ko},extraReducers:e=>{e.addCase(k,()=>Ko)}}),{setBatchTransactions:Xx,updateBatchTransactions:Yx,clearBatchTransactions:Pm,clearAllBatchTransactions:Zx}=Sa.actions,zn=Sa.reducer});var va,wa,Aa,ny,Ia,Xo=g(()=>{"use strict";o();va=require("@reduxjs/toolkit");M();wa={},Aa=(0,va.createSlice)({name:"dappConfigSlice",initialState:wa,reducers:{setDappConfig:(e,n)=>n.payload},extraReducers:e=>{e.addCase(k,()=>wa)}}),{setDappConfig:ny}=Aa.actions,Ia=Aa.reducer});var P=g(()=>{"use strict";o();Nn();Cn();Dn();Bn();Un();Gn();Hn();jn();Kn();Xo()});var Yo=g(()=>{"use strict";o()});var Pa,Rm,Dm,xe,Zn=g(()=>{"use strict";o();Pa=require("@reduxjs/toolkit");Yo();Nn();Kn();Xo();Cn();Dn();Bn();Un();Gn();Hn();jn();Rm={["account"]:Pn,["dappConfig"]:Ia,["loginInfo"]:Ln,["modals"]:Rn,["networkConfig"]:Mn,["signedMessageInfo"]:Fn,["toasts"]:Wn,["transactionsInfo"]:$n,["transactions"]:qn,["batchTransactions"]:zn},Dm=(e={})=>(0,Pa.combineReducers)(_(_({},Rm),e)),xe=Dm});var Ra={};ee(Ra,{default:()=>Ym,sessionStorageReducers:()=>Zo});function ie(e,n=[]){return{key:e,version:1,storage:La.default,blacklist:n}}var U,Na,La,Om,tn,Mm,Bm,Fm,Um,Wm,Gm,$m,Hm,Vm,qm,Ca,jm,Zo,zm,Km,Xm,Ym,Da=g(()=>{"use strict";o();U=require("redux-persist"),Na=b(require("redux-persist/lib/storage")),La=b(require("redux-persist/lib/storage/session"));Zn();P();Nn();Kn();Cn();Dn();Bn();Un();Gn();Hn();jn();Yo();Om={persistReducersStorageType:"localStorage"},tn={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Mm=ie(tn["account"]),Bm=ie(tn["loginInfo"]),Fm=ie(tn["modals"]),Um=ie(tn["networkConfig"]),Wm={2:e=>w(_({},e),{networkConfig:Uo})};Gm=ie("sdk-dapp-transactionsInfo"),$m=ie("sdk-dapp-transactions",["transactionsToSign"]),Hm=ie("sdk-dapp-batchTransactions",["batchTransactions"]),Vm=ie("sdk-dapp-toasts"),qm=ie("sdk-dapp-signedMessageInfo"),Ca={key:"sdk-dapp-store",version:2,storage:Na.default,whitelist:Object.keys(tn),migrate:(0,U.createMigrate)(Wm,{debug:!1})},jm=w(_({},Ca),{whitelist:[]}),Zo={["toasts"]:(0,U.persistReducer)(Vm,Wn),["transactions"]:(0,U.persistReducer)($m,qn),["transactionsInfo"]:(0,U.persistReducer)(Gm,$n),["batchTransactions"]:(0,U.persistReducer)(Hm,zn),["signedMessageInfo"]:(0,U.persistReducer)(qm,Fn)},zm=w(_({},Zo),{["account"]:(0,U.persistReducer)(Mm,Pn),["loginInfo"]:(0,U.persistReducer)(Bm,Ln),["modals"]:(0,U.persistReducer)(Fm,Rn),["networkConfig"]:(0,U.persistReducer)(Um,Mn)}),Km=Om.persistReducersStorageType==="localStorage",Xm=Km?(0,U.persistReducer)(Ca,xe(Zo)):(0,U.persistReducer)(jm,xe(zm)),Ym=Xm});var Oa={};ee(Oa,{default:()=>Zm});var Zm,Ma=g(()=>{"use strict";o();Zn();Zm=xe()});var Ba={};ee(Ba,{default:()=>Qm});var H,Jm,Qm,Fa=g(()=>{"use strict";o();H=require("redux-persist"),Jm=[H.FLUSH,H.REHYDRATE,H.PAUSE,H.PERSIST,H.PURGE,H.REGISTER],Qm=Jm});var Ga={};ee(Ga,{default:()=>Wa});function Wa(e){return(0,Ua.persistStore)(e)}var Ua,$a=g(()=>{"use strict";o();Ua=require("redux-persist")});var Ci=T((sR,Li)=>{o();var zl=typeof i=="object"&&i&&i.Object===Object&&i;Li.exports=zl});var pn=T((cR,Ri)=>{o();var Kl=Ci(),Xl=typeof self=="object"&&self&&self.Object===Object&&self,Yl=Kl||Xl||Function("return this")();Ri.exports=Yl});var ht=T((pR,Di)=>{o();var Zl=pn(),Jl=Zl.Symbol;Di.exports=Jl});var Fi=T((mR,Bi)=>{o();var Oi=ht(),Mi=Object.prototype,Ql=Mi.hasOwnProperty,e_=Mi.toString,mn=Oi?Oi.toStringTag:void 0;function n_(e){var n=Ql.call(e,mn),t=e[mn];try{e[mn]=void 0;var r=!0}catch(p){}var c=e_.call(e);return r&&(n?e[mn]=t:delete e[mn]),c}Bi.exports=n_});var Wi=T((dR,Ui)=>{o();var o_=Object.prototype,t_=o_.toString;function r_(e){return t_.call(e)}Ui.exports=r_});var xt=T((lR,Hi)=>{o();var Gi=ht(),a_=Fi(),i_=Wi(),s_="[object Null]",c_="[object Undefined]",$i=Gi?Gi.toStringTag:void 0;function p_(e){return e==null?e===void 0?c_:s_:$i&&$i in Object(e)?a_(e):i_(e)}Hi.exports=p_});var qi=T((_R,Vi)=>{o();var m_=Array.isArray;Vi.exports=m_});var zi=T((uR,ji)=>{o();function d_(e){return e!=null&&typeof e=="object"}ji.exports=d_});var Xi=T((gR,Ki)=>{o();var l_=xt(),__=qi(),u_=zi(),g_="[object String]";function f_(e){return typeof e=="string"||!__(e)&&u_(e)&&l_(e)==g_}Ki.exports=f_});var bt={};ee(bt,{css:()=>es,default:()=>v_});var es,v_,St=g(()=>{"use strict";o();es=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(es));v_={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var vt={};ee(vt,{css:()=>rs,default:()=>k_});var rs,k_,At=g(()=>{"use strict";o();rs=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(rs));k_={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var kt=T((XD,cs)=>{o();function F_(e){var n=typeof e;return e!=null&&(n=="object"||n=="function")}cs.exports=F_});var ms=T((YD,ps)=>{o();var U_=xt(),W_=kt(),G_="[object AsyncFunction]",$_="[object Function]",H_="[object GeneratorFunction]",V_="[object Proxy]";function q_(e){if(!W_(e))return!1;var n=U_(e);return n==$_||n==H_||n==G_||n==V_}ps.exports=q_});var ls=T((ZD,ds)=>{o();var j_=pn(),z_=j_["__core-js_shared__"];ds.exports=z_});var gs=T((JD,us)=>{o();var Et=ls(),_s=function(){var e=/[^.]+$/.exec(Et&&Et.keys&&Et.keys.IE_PROTO||"");return e?"Symbol(src)_1."+e:""}();function K_(e){return!!_s&&_s in e}us.exports=K_});var Ts=T((QD,fs)=>{o();var X_=Function.prototype,Y_=X_.toString;function Z_(e){if(e!=null){try{return Y_.call(e)}catch(n){}try{return e+""}catch(n){}}return""}fs.exports=Z_});var xs=T((e3,hs)=>{o();var J_=ms(),Q_=gs(),eu=kt(),nu=Ts(),ou=/[\\^$.*+?()[\]{}|]/g,tu=/^\[object .+?Constructor\]$/,ru=Function.prototype,au=Object.prototype,iu=ru.toString,su=au.hasOwnProperty,cu=RegExp("^"+iu.call(su).replace(ou,"\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g,"$1.*?")+"$");function pu(e){if(!eu(e)||Q_(e))return!1;var n=J_(e)?cu:tu;return n.test(nu(e))}hs.exports=pu});var bs=T((n3,ys)=>{o();function mu(e,n){return e==null?void 0:e[n]}ys.exports=mu});var po=T((o3,Ss)=>{o();var du=xs(),lu=bs();function _u(e,n){var t=lu(e,n);return du(t)?t:void 0}Ss.exports=_u});var ln=T((t3,ws)=>{o();var uu=po(),gu=uu(Object,"create");ws.exports=gu});var Is=T((r3,As)=>{o();var vs=ln();function fu(){this.__data__=vs?vs(null):{},this.size=0}As.exports=fu});var Es=T((a3,ks)=>{o();function Tu(e){var n=this.has(e)&&delete this.__data__[e];return this.size-=n?1:0,n}ks.exports=Tu});var Ns=T((i3,Ps)=>{o();var hu=ln(),xu="__lodash_hash_undefined__",yu=Object.prototype,bu=yu.hasOwnProperty;function Su(e){var n=this.__data__;if(hu){var t=n[e];return t===xu?void 0:t}return bu.call(n,e)?n[e]:void 0}Ps.exports=Su});var Cs=T((s3,Ls)=>{o();var wu=ln(),vu=Object.prototype,Au=vu.hasOwnProperty;function Iu(e){var n=this.__data__;return wu?n[e]!==void 0:Au.call(n,e)}Ls.exports=Iu});var Ds=T((c3,Rs)=>{o();var ku=ln(),Eu="__lodash_hash_undefined__";function Pu(e,n){var t=this.__data__;return this.size+=this.has(e)?0:1,t[e]=ku&&n===void 0?Eu:n,this}Rs.exports=Pu});var Ms=T((p3,Os)=>{o();var Nu=Is(),Lu=Es(),Cu=Ns(),Ru=Cs(),Du=Ds();function Ve(e){var n=-1,t=e==null?0:e.length;for(this.clear();++n<t;){var r=e[n];this.set(r[0],r[1])}}Ve.prototype.clear=Nu;Ve.prototype.delete=Lu;Ve.prototype.get=Cu;Ve.prototype.has=Ru;Ve.prototype.set=Du;Os.exports=Ve});var Fs=T((m3,Bs)=>{o();function Ou(){this.__data__=[],this.size=0}Bs.exports=Ou});var Ws=T((d3,Us)=>{o();function Mu(e,n){return e===n||e!==e&&n!==n}Us.exports=Mu});var _n=T((l3,Gs)=>{o();var Bu=Ws();function Fu(e,n){for(var t=e.length;t--;)if(Bu(e[t][0],n))return t;return-1}Gs.exports=Fu});var Hs=T((_3,$s)=>{o();var Uu=_n(),Wu=Array.prototype,Gu=Wu.splice;function $u(e){var n=this.__data__,t=Uu(n,e);if(t<0)return!1;var r=n.length-1;return t==r?n.pop():Gu.call(n,t,1),--this.size,!0}$s.exports=$u});var qs=T((u3,Vs)=>{o();var Hu=_n();function Vu(e){var n=this.__data__,t=Hu(n,e);return t<0?void 0:n[t][1]}Vs.exports=Vu});var zs=T((g3,js)=>{o();var qu=_n();function ju(e){return qu(this.__data__,e)>-1}js.exports=ju});var Xs=T((f3,Ks)=>{o();var zu=_n();function Ku(e,n){var t=this.__data__,r=zu(t,e);return r<0?(++this.size,t.push([e,n])):t[r][1]=n,this}Ks.exports=Ku});var Zs=T((T3,Ys)=>{o();var Xu=Fs(),Yu=Hs(),Zu=qs(),Ju=zs(),Qu=Xs();function qe(e){var n=-1,t=e==null?0:e.length;for(this.clear();++n<t;){var r=e[n];this.set(r[0],r[1])}}qe.prototype.clear=Xu;qe.prototype.delete=Yu;qe.prototype.get=Zu;qe.prototype.has=Ju;qe.prototype.set=Qu;Ys.exports=qe});var Qs=T((h3,Js)=>{o();var eg=po(),ng=pn(),og=eg(ng,"Map");Js.exports=og});var oc=T((x3,nc)=>{o();var ec=Ms(),tg=Zs(),rg=Qs();function ag(){this.size=0,this.__data__={hash:new ec,map:new(rg||tg),string:new ec}}nc.exports=ag});var rc=T((y3,tc)=>{o();function ig(e){var n=typeof e;return n=="string"||n=="number"||n=="symbol"||n=="boolean"?e!=="__proto__":e===null}tc.exports=ig});var un=T((b3,ac)=>{o();var sg=rc();function cg(e,n){var t=e.__data__;return sg(n)?t[typeof n=="string"?"string":"hash"]:t.map}ac.exports=cg});var sc=T((S3,ic)=>{o();var pg=un();function mg(e){var n=pg(this,e).delete(e);return this.size-=n?1:0,n}ic.exports=mg});var pc=T((w3,cc)=>{o();var dg=un();function lg(e){return dg(this,e).get(e)}cc.exports=lg});var dc=T((v3,mc)=>{o();var _g=un();function ug(e){return _g(this,e).has(e)}mc.exports=ug});var _c=T((A3,lc)=>{o();var gg=un();function fg(e,n){var t=gg(this,e),r=t.size;return t.set(e,n),this.size+=t.size==r?0:1,this}lc.exports=fg});var gc=T((I3,uc)=>{o();var Tg=oc(),hg=sc(),xg=pc(),yg=dc(),bg=_c();function je(e){var n=-1,t=e==null?0:e.length;for(this.clear();++n<t;){var r=e[n];this.set(r[0],r[1])}}je.prototype.clear=Tg;je.prototype.delete=hg;je.prototype.get=xg;je.prototype.has=yg;je.prototype.set=bg;uc.exports=je});var Tc=T((k3,fc)=>{o();var Sg="__lodash_hash_undefined__";function wg(e){return this.__data__.set(e,Sg),this}fc.exports=wg});var xc=T((E3,hc)=>{o();function vg(e){return this.__data__.has(e)}hc.exports=vg});var bc=T((P3,yc)=>{o();var Ag=gc(),Ig=Tc(),kg=xc();function mo(e){var n=-1,t=e==null?0:e.length;for(this.__data__=new Ag;++n<t;)this.add(e[n])}mo.prototype.add=mo.prototype.push=Ig;mo.prototype.has=kg;yc.exports=mo});var wc=T((N3,Sc)=>{o();function Eg(e,n,t,r){for(var c=e.length,p=t+(r?1:-1);r?p--:++p<c;)if(n(e[p],p,e))return p;return-1}Sc.exports=Eg});var Ac=T((L3,vc)=>{o();function Pg(e){return e!==e}vc.exports=Pg});var kc=T((C3,Ic)=>{o();function Ng(e,n,t){for(var r=t-1,c=e.length;++r<c;)if(e[r]===n)return r;return-1}Ic.exports=Ng});var Pc=T((R3,Ec)=>{o();var Lg=wc(),Cg=Ac(),Rg=kc();function Dg(e,n,t){return n===n?Rg(e,n,t):Lg(e,Cg,t)}Ec.exports=Dg});var Lc=T((D3,Nc)=>{o();var Og=Pc();function Mg(e,n){var t=e==null?0:e.length;return!!t&&Og(e,n,0)>-1}Nc.exports=Mg});var Rc=T((O3,Cc)=>{o();function Bg(e,n,t){for(var r=-1,c=e==null?0:e.length;++r<c;)if(t(n,e[r]))return!0;return!1}Cc.exports=Bg});var Oc=T((M3,Dc)=>{o();function Fg(e,n){return e.has(n)}Dc.exports=Fg});var Bc=T((B3,Mc)=>{o();var Ug=po(),Wg=pn(),Gg=Ug(Wg,"Set");Mc.exports=Gg});var Uc=T((F3,Fc)=>{o();function $g(){}Fc.exports=$g});var Pt=T((U3,Wc)=>{o();function Hg(e){var n=-1,t=Array(e.size);return e.forEach(function(r){t[++n]=r}),t}Wc.exports=Hg});var $c=T((W3,Gc)=>{o();var Nt=Bc(),Vg=Uc(),qg=Pt(),jg=1/0,zg=Nt&&1/qg(new Nt([,-0]))[1]==jg?function(e){return new Nt(e)}:Vg;Gc.exports=zg});var Vc=T((G3,Hc)=>{o();var Kg=bc(),Xg=Lc(),Yg=Rc(),Zg=Oc(),Jg=$c(),Qg=Pt(),ef=200;function nf(e,n,t){var r=-1,c=Xg,p=e.length,m=!0,u=[],l=u;if(t)m=!1,c=Yg;else if(p>=ef){var x=n?null:Jg(e);if(x)return Qg(x);m=!1,c=Zg,l=new Kg}else l=n?[]:u;e:for(;++r<p;){var S=e[r],y=n?n(S):S;if(S=t||S!==0?S:0,m&&y===y){for(var L=l.length;L--;)if(l[L]===y)continue e;n&&l.push(y),u.push(S)}else c(l,y,t)||(l!==u&&l.push(y),u.push(S))}return u}Hc.exports=nf});var jc=T(($3,qc)=>{o();var of=Vc();function tf(e){return e&&e.length?of(e):[]}qc.exports=tf});var Ff={};ee(Ff,{useTransactionsToasts:()=>Bf});module.exports=_e(Ff);o();var Ot=require("react");o();var Ka=require("react"),We=require("react-redux");o();var Jo=require("@reduxjs/toolkit"),ja=require("react-redux/lib/utils/Subscription");P();o();var ka=b(require("lodash.throttle"));A();P();Bo();Re();Te();var Nm=[An],Xn=!1,Lm=(0,ka.default)(()=>{Qe(Je())},5e3),Ea=e=>n=>t=>{if(Nm.includes(t.type))return n(t);let r=e.getState(),c=pe.local.getItem(re.loginExpiresAt);if(!Boolean(r==null?void 0:r.account.address))return n(t);if(c==null)return Qe(Je());let m=Date.now();return c-m<0&&!Xn?setTimeout(()=>{Xn=!0,e.dispatch(qr())},1e3):(Xn&&(Xn=!1),Lm()),n(t)};o();o();function Yn(){return typeof sessionStorage!="undefined"}var Ha=Yn()?(Da(),_e(Ra)).default:(Ma(),_e(Oa)).default,Va=Yn()?(Fa(),_e(Ba)).default:[],qa=Yn()?($a(),_e(Ga)).default:e=>e;Zn();var v=(0,Jo.configureStore)({reducer:Ha,middleware:e=>e({serializableCheck:{ignoredActions:[...Va,Mo.type,Oo.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Ea)}),za=(0,ja.createSubscription)(v),Qy=qa(v),eb=(0,Jo.configureStore)({reducer:xe});var ed={store:v,subscription:za},Qo=(0,Ka.createContext)(ed),rb=(0,We.createStoreHook)(Qo),Ge=(0,We.createDispatchHook)(Qo),I=(0,We.createSelectorHook)(Qo);o();o();P();o();var Xa=b(require("lodash.isequal")),Jn=require("reselect"),f=(0,Jn.createSelectorCreator)(Jn.defaultMemoize,Xa.default);var Y=e=>e.account,ye=f(Y,e=>e.address),le=f(Y,ye,(e,n)=>n in e.accounts?e.accounts[n]:En),nd=f(Y,le,(e,n)=>{let c=e,{accounts:t}=c,r=ve(c,["accounts"]);return w(_({},r),{address:n.address,account:n})}),pb=f(le,e=>e.balance),od=f(le,e=>{var n;return((n=e==null?void 0:e.nonce)==null?void 0:n.valueOf())||0}),mb=f(Y,e=>e.shard),db=f(Y,e=>e.ledgerAccount),lb=f(Y,e=>e.walletConnectAccount),_b=f(Y,e=>e.isAccountLoading),ub=f(Y,e=>e.accountLoadingError),td=f(Y,e=>e.websocketEvent),rd=f(Y,e=>e.websocketBatchEvent);o();var Ya=e=>e.dappConfig,Tb=f(Ya,e=>e.shouldUseWebViewProvider);o();var G=e=>e.loginInfo,bb=f(G,e=>e.loginMethod),et=f(G,ye,(e,n)=>Boolean(n)),Sb=f(G,e=>e.walletConnectLogin),ad=f(G,e=>e.ledgerLogin),id=f(G,e=>e.walletLogin),wb=f(G,e=>e.isLoginSessionInvalid),vb=f(G,e=>e.tokenLogin),sd=f(G,e=>e.logoutRoute),cd=f(G,e=>e.isWalletConnectV2Initialized);o();var Za=e=>e.modals,kb=f(Za,e=>e.txSubmittedModal),Eb=f(Za,e=>e.notificationModal);o();var Z=e=>e.networkConfig,Ja=f(Z,e=>e.network.chainId),pd=f(Z,e=>e.network.roundDuration),Lb=f(Z,e=>e.network.walletConnectBridgeAddress),Cb=f(Z,e=>e.network.walletConnectV2RelayAddress),Rb=f(Z,e=>e.network.walletConnectV2ProjectId),Db=f(Z,e=>e.network.walletConnectV2Options),Ob=f(Z,e=>e.network.walletConnectDeepLink),se=f(Z,e=>e.network),Qa=f(se,e=>e.apiAddress),ei=f(se,e=>e.explorerAddress),ni=f(se,e=>{var n;return(n=e.customWalletAddress)!=null?n:e.walletAddress}),Mb=f(se,e=>e.xAliasAddress),md=f(se,e=>e.egldLabel);o();var Qn=e=>e.signedMessageInfo,Ub=f(Qn,e=>e.isSigning),Wb=f(Qn,e=>e.errorMessage),Gb=f(Qn,e=>{let n=Object.keys(e.signedSessions),t=n.length;return e.signedSessions[n[t-1]]}),$b=f(Qn,e=>{let n=Object.keys(e.signedSessions),t=n.length;return n.length>0?n[t-1]:""});o();var oi=e=>e.toasts,qb=f(oi,e=>e.customToasts),ti=f(oi,e=>e.transactionToasts);o();P();var ri={errorMessage:$o,successMessage:Ho,processingMessage:Vo},nt=e=>e.transactionsInfo,dd=f(nt,(e,n)=>n,(e,n)=>n!=null&&(e==null?void 0:e[Number(n)])||ri);o();o();var be=require("@multiversx/sdk-core");A();o();o();o();function ot(e){try{let n=atob(e),t=btoa(n),r=a.Buffer.from(e,"base64").toString(),c=a.Buffer.from(r).toString("base64"),p=e===t||t.startsWith(e),m=e===c||c.startsWith(e);if(p&&m)return!0}catch(n){return!1}return!1}o();o();o();o();var ai=e=>{let n=e!=null?e:"";return ot(n)?a.Buffer.from(a.Buffer.from(n,"base64").toString().trim()):a.Buffer.from(n.trim())};o();A();var rn=({data:e,onlySetGuardian:n})=>e?n?e.startsWith("SetGuardian"):Object.values(Io).some(t=>e.startsWith(t)):!1;function eo(e){var r,c,p;let n=_({},e);rn({data:n.data,onlySetGuardian:!0})&&(delete n.senderUsername,delete n.receiverUsername);let t=new be.Transaction(_(_(w(_(w(_(_({value:BigInt(n.value.valueOf()),data:ai(n.data),nonce:BigInt(n.nonce.valueOf()),receiver:new be.Address(n.receiver)},n.receiverUsername?{receiverUsername:n.receiverUsername}:{}),n.relayer?{relayer:new be.Address(n.relayer)}:{}),{sender:new be.Address(n.sender)}),n.senderUsername?{senderUsername:n.senderUsername}:{}),{gasLimit:BigInt((r=n.gasLimit.valueOf())!=null?r:wn),gasPrice:BigInt((c=n.gasPrice.valueOf())!=null?c:vn),chainID:n.chainID.valueOf(),version:(p=n.version)!=null?p:Pr}),n.options?{options:n.options}:{}),n.guardian?{guardian:new be.Address(n.guardian)}:{}));return n.relayerSignature&&(t.relayerSignature=a.Buffer.from(n.relayerSignature,"hex")),n.guardianSignature&&(t.guardianSignature=a.Buffer.from(n.guardianSignature,"hex")),n.signature&&(t.signature=a.Buffer.from(n.signature,"hex")),t}o();o();A();o();o();o();var gd=require("@multiversx/sdk-core");o();A();o();var Td=require("@multiversx/sdk-core");N();o();var fd=require("@multiversx/sdk-core");o();o();o();var hd=require("@multiversx/sdk-core"),xd=b(require("bignumber.js"));D();o();o();o();var rt="blocks";o();o();o();o();F();o();o();F();o();o();o();var kd=b(require("bignumber.js"));A();o();var Id=require("@multiversx/sdk-core"),it=b(require("bignumber.js"));A();o();var vd=b(require("bignumber.js"));o();it.default.config({ROUNDING_MODE:it.default.ROUND_FLOOR});o();o();o();A();o();o();o();o();A();o();A();o();var Ed=require("@multiversx/sdk-core"),Pd=b(require("bignumber.js"));A();o();var ct=require("@multiversx/sdk-core"),Cd=b(require("bignumber.js"));A();o();o();var Nd=b(require("bignumber.js"));A();o();A();o();o();o();o();o();o();A();o();A();o();A();o();F();var Dd=["reDelegateRewards","claimRewards","unBond"],Od=["wrapEgld","unwrapEgld"],Md=["unStake"],Bd=["unDelegate"];o();o();o();A();o();var Gd=b(require("bignumber.js"));o();o();F();o();var Hd=b(require("bignumber.js"));o();o();o();o();var jd=b(require("bignumber.js"));N();o();o();var zd=require("@multiversx/sdk-core");o();o();D();o();var Zd=require("@multiversx/sdk-web-wallet-provider");A();o();var Xd=b(require("qs"));o();me();en();o();en();var AA={search:he()?window.location.search:"",removeParams:[]};o();o();o();W();o();F();o();o();W();o();var Jd=b(require("linkifyjs"));o();A();o();var Qd=b(require("bignumber.js"));o();N();o();o();D();o();D();o();o();o();F();o();F();o();var el=b(require("bignumber.js"));A();F();o();F();o();var li=require("react");N();o();o();F();o();o();var nl=require("@multiversx/sdk-core"),ol=b(require("bignumber.js"));F();o();N();o();o();N();var fk=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];o();var _i=require("react");N();W();o();var rl=require("react");F();var vk=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];Ue();o();var to=e=>e.transactions,Se=f(to,e=>e.signedTransactions),al=f(to,e=>e.signTransactionsError),ui=f(to,e=>e.signTransactionsCancelMessage),ro=e=>n=>Object.entries(n).reduce((t,[r,c])=>(e(c.status)&&(t[r]=c),t),{}),gi=f(Se,ro(Me)),fi=f(Se,ro(Be)),Ti=f(Se,ro(Fe)),il=f(Se,ro(jo)),hi=f(to,e=>{var n;return(e==null?void 0:e.transactionsToSign)==null?null:w(_({},e.transactionsToSign),{transactions:((n=e==null?void 0:e.transactionsToSign)==null?void 0:n.transactions.map(t=>eo(t)))||[]})}),sl=f(Se,(e,n)=>n,(e,n)=>n!=null?(e==null?void 0:e[n])||{}:{});N();o();o();o();o();o();o();o();o();var cl=require("@multiversx/sdk-extension-provider"),pl=require("@multiversx/sdk-hw-provider"),ml=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),dl=require("@multiversx/sdk-opera-provider"),ll=require("@multiversx/sdk-passkey-provider/out"),_l=require("@multiversx/sdk-web-wallet-provider");A();Pe();o();var cn=require("@multiversx/sdk-web-wallet-iframe-provider/out"),xi=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");D();o();var V=e=>`Unable to perform ${e}, Provider not initialized`,ao=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(V("getAccount"))}setAccount(n){throw new Error(V(`setAccount: ${n}`))}login(n){throw new Error(V(`login with options: ${n}`))}logout(n){throw new Error(V(`logout with options: ${n}`))}getAddress(){throw new Error(V("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(n,t){throw new Error(V(`sendTransaction with transactions: ${n} options: ${t}`))}signTransaction(n,t){throw new Error(V(`signTransaction with transactions: ${n} options: ${t}`))}signTransactions(n,t){throw new Error(V(`signTransactions with transactions: ${n} options: ${t}`))}signMessage(n,t){throw new Error(V(`signTransactions with ${n} and options ${t}`))}sendCustomMessage({method:n,params:t}){throw new Error(V(`sendCustomMessage with method: ${n} params: ${t}`))}sendCustomRequest(n){throw new Error(V(`sendSessionEvent with options: ${n}`))}ping(){return Promise.resolve(!1)}},ul=new ao;P();o();o();o();o();o();var yi=require("@lifeomic/axios-fetch"),dt=b(require("axios")),lt=(0,yi.buildAxiosFetch)(dt.default),_t=(e,n)=>h(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let r=e.clone().json(),[c]=yield Promise.all([r]);return{data:c,status:e.status,statusText:e.statusText,headers:e.headers,config:n}});function gl(e,n,t){return h(this,null,function*(){try{let r=yield lt(e,_({method:"POST",body:n?JSON.stringify(n):void 0,headers:_({"Content-Type":"application/json"},(t==null?void 0:t.headers)||{})},t));return _t(r,t)}catch(r){throw console.error("Fetch Error:",r),r}})}function fl(e,n){return h(this,null,function*(){try{let t=yield lt(e,n);if(!t.ok)throw new Error(`HTTP error! Status: ${t.status}`);return _t(t,n)}catch(t){throw console.error("Fetch Error:",t),t}})}function Tl(e,n,t){return h(this,null,function*(){try{let r=yield lt(e,_({method:"PATCH",body:n?JSON.stringify(n):void 0,headers:(t==null?void 0:t.headers)||{}},t));if(!r.ok)throw new Error(`HTTP error! Status: ${r.status}`);return _t(r,t)}catch(r){throw console.error("Fetch Error:",r),r}})}var $e=dt.default.create();$e.get=fl;$e.post=gl;$e.patch=Tl;o();o();var hl=b(require("axios"));o();var xl=b(require("swr"));o();D();o();o();o();W();o();Ue();o();o();o();var yl=b(require("axios"));o();var Sl=b(require("axios"));Le();o();A();o();var wl=b(require("axios"));o();var Al=b(require("axios"));o();o();var Il=b(require("axios"));o();var kl=b(require("axios"));o();o();P();D();o();o();o();o();W();o();M();N();o();P();o();var Ai=require("@multiversx/sdk-core");N();me();o();Ue();o();P();D();o();P();N();o();o();o();N();o();Xe();o();o();function Tt(){let e=I(Se),n=Object.entries(e),t=(n==null?void 0:n.length)>0;return{signedTransactions:e,signedTransactionsArray:n,hasSignedTransactions:t}}o();o();var Cl=b(require("swr"));o();o();o();o();o();var ki=require("react");o();o();var Rl=require("react"),Dl=require("@multiversx/sdk-web-wallet-provider"),Ol=require("@multiversx/sdk-web-wallet-provider"),Ml=b(require("qs"));A();P();D();Eo();W();var tC=O();o();var ss=require("react"),co=require("@multiversx/sdk-core");o();A();Fo();o();o();o();o();o();o();o();o();o();o();Le();o();var Wl=b(require("axios"));o();var $l=b(require("axios"));o();var Hl=b(require("axios"));o();Le();o();Le();o();o();o();o();var ql=require("@multiversx/sdk-opera-provider");o();var jl=require("@multiversx/sdk-extension-provider");o();Pe();o();o();o();o();var T_=b(Xi());o();var b_=require("@multiversx/sdk-native-auth-client");o();var Ji=b(require("axios"));o();o();o();function Yi(e){return h(this,null,function*(){return yield new Promise(n=>{setTimeout(()=>n(),e)})})}o();var Zi=(e,n,t,r=0)=>h(void 0,null,function*(){try{return yield e(...t)}catch(c){return r<n.retries?((n==null?void 0:n.delay)!=null&&(yield Yi(n.delay)),yield Zi(e,n,t,r+1)):null}}),yt=(e,n={retries:5,delay:500})=>(...t)=>h(void 0,null,function*(){return yield Zi(e,n,t)});var h_=4;var kR=yt((e,n,t)=>h(void 0,null,function*(){if(t){let p=Math.floor(Date.now()/1e3);return{hash:yield t(),timestamp:p}}let{data:r}=yield Ji.default.get(`${e}/${rt}?from=${h_}&size=1&fields=hash,timestamp${n?"&shard="+n:""}`),[c]=r;return c}));o();o();In();o();o();W();var BR={origin:O().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};o();var S_=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");o();var w_=require("@multiversx/sdk-passkey-provider/out");o();var N_=b(require("react")),L_=require("@fortawesome/free-solid-svg-icons");o();o();var He=b(require("react")),as=require("@fortawesome/free-solid-svg-icons"),is=require("@fortawesome/react-fontawesome");o();var os=b(require("react"));o();var dn=b(require("react"));o();en();var Qi=()=>!he();var A_=()=>h(void 0,null,function*(){return yield Promise.resolve().then(()=>(St(),bt))}),I_=()=>(St(),_e(bt)).default,wt=Qi();function ns({ssrGlobalImportCallback:e,ssrImportCallback:n,clientImportCallback:t}){let[r,c]=dn.default.useState(wt?void 0:I_()),[p,m]=dn.default.useState(wt||t==null?void 0:t()),u=()=>h(this,null,function*(){(e?e():A_()).then(l=>c(l.default)),n==null||n().then(l=>m(l.default))});return(0,dn.useEffect)(()=>{wt&&u()},[]),{globalStyles:r,styles:p}}function ts(e,n){return t=>{let{globalStyles:r,styles:c}=ns({ssrGlobalImportCallback:n.ssrGlobalStyles,ssrImportCallback:n.ssrStyles,clientImportCallback:n.clientStyles});return os.default.createElement(e,w(_({},t),{globalStyles:r!=null?r:{},styles:c!=null?c:{}}))}}var E_=({styles:e})=>He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},He.default.createElement(is.FontAwesomeIcon,{icon:as.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),He.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),P_=ts(E_,{ssrStyles:()=>Promise.resolve().then(()=>(At(),vt)),clientStyles:()=>(At(),_e(vt)).default});Re();Te();o();o();P();var k2=24*60*60;A();W();o();o();o();var D_=require("react"),O_=require("@multiversx/sdk-hw-provider");o();Sn();M();N();o();On();Re();Te();me();o();o();o();var Yc=require("react"),Zc=require("@multiversx/sdk-core"),wf=require("@multiversx/sdk-extension-provider"),vf=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),Af=require("@multiversx/sdk-passkey-provider/out"),If=b(jc());A();o();o();o();o();o();o();o();o();o();o();var gn=b(require("react"));var r5=(0,gn.createContext)({}),a5=v.getState();o();var Kc=b(require("react"));o();var Lt=b(require("react")),rf=b(require("axios"));o();o();Pe();o();o();var lf=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),_f=require("@multiversx/sdk-webview-provider/out/WebviewProvider");M();o();M();P();N();o();o();P();o();me();o();P();N();o();o();var sf=require("@multiversx/sdk-core"),cf=b(require("bignumber.js"));A();o();var mf=b(require("bignumber.js"));A();P();D();me();W();o();var Xc=require("react"),Tf=require("@multiversx/sdk-extension-provider"),hf=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),xf=require("@multiversx/sdk-passkey-provider/out");Pe();P();o();P();N();On();W();o();Ao();P();o();o();var kf=require("react");o();o();D();Ue();o();o();var Mf=require("react");o();o();var Jc=require("react");o();var Ef=b(require("axios"));o();var Cf=require("react");o();var Rt=require("react"),Lf=require("socket.io-client");P();o();var Ct=class{constructor(){this.current=null;this._status="NOT_INITIALIZED";this.listeners=[]}get status(){return this._status}set status(n){this._status!==n&&(this._status=n,this.notifyListeners())}subscribe(n){return this.listeners.push(n),()=>this.unsubscribe(n)}unsubscribe(n){this.listeners=this.listeners.filter(t=>t!==n)}notifyListeners(){this.listeners.forEach(n=>n(this._status))}},Qc=new Ct;o();var Df=require("react");Xe();o();o();var Dt=require("react");o();var Of=require("react");var Bf=()=>{let e=I(ti),n=I(nt),{signedTransactions:t}=Tt(),r=(0,Ot.useMemo)(()=>e.map(p=>{var u,l,x,S,y;let m=(u=t[p.toastId])==null?void 0:u.status;return{sessionId:p.toastId,baseTransactionsInfo:(l=n[p.toastId])!=null?l:{},customTransactionsInfo:(y=(S=(x=t[p.toastId])==null?void 0:x.customTransactionInformation)==null?void 0:S.sessionInformation)!=null?y:{},status:m,isPending:m!=="success"&&m!=="fail"&&m!=="cancelled"}}),[e,t,n]),c=(0,Ot.useMemo)(()=>r.filter(p=>p.isPending),[r]);return{transactionsToastsInfo:r,pendingTransactionsToastsInfo:c}};0&&(module.exports={useTransactionsToasts});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
