// @multiversx/sdk-core/out/tokenManagement/tokenManagementTransactionsFactory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenManagementTransactionsFactory = void 0;
const abi_1 = require("../abi");
const address_1 = require("../core/address");
const constants_1 = require("../core/constants");
const errors_1 = require("../core/errors");
const logger_1 = require("../core/logger");
const transactionBuilder_1 = require("../core/transactionBuilder");
/**
 * Use this class to create token management transactions like issuing ESDTs, creating NFTs, setting roles, etc.
 */
class TokenManagementTransactionsFactory {
    constructor(options) {
        this.config = options.config;
        this.argSerializer = new abi_1.ArgSerializer();
        this.trueAsString = "true";
        this.falseAsString = "false";
        this.esdtContractAddress = address_1.Address.newFromHex(constants_1.ESDT_CONTRACT_ADDRESS_HEX, this.config.addressHrp);
    }
    createTransactionForIssuingFungible(sender, options) {
        this.notifyAboutUnsettingBurnRoleGlobally();
        const args = [
            new abi_1.StringValue(options.tokenName),
            new abi_1.StringValue(options.tokenTicker),
            new abi_1.BigUIntValue(options.initialSupply),
            new abi_1.BigUIntValue(options.numDecimals),
            new abi_1.StringValue("canFreeze"),
            new abi_1.StringValue(this.boolToString(options.canFreeze)),
            new abi_1.StringValue("canWipe"),
            new abi_1.StringValue(this.boolToString(options.canWipe)),
            new abi_1.StringValue("canPause"),
            new abi_1.StringValue(this.boolToString(options.canPause)),
            new abi_1.StringValue("canChangeOwner"),
            new abi_1.StringValue(this.boolToString(options.canChangeOwner)),
            new abi_1.StringValue("canUpgrade"),
            new abi_1.StringValue(this.boolToString(options.canUpgrade)),
            new abi_1.StringValue("canAddSpecialRoles"),
            new abi_1.StringValue(this.boolToString(options.canAddSpecialRoles)),
        ];
        const dataParts = ["issue", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitIssue,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    createTransactionForIssuingSemiFungible(sender, options) {
        this.notifyAboutUnsettingBurnRoleGlobally();
        const args = [
            new abi_1.StringValue(options.tokenName),
            new abi_1.StringValue(options.tokenTicker),
            new abi_1.StringValue("canFreeze"),
            new abi_1.StringValue(this.boolToString(options.canFreeze)),
            new abi_1.StringValue("canWipe"),
            new abi_1.StringValue(this.boolToString(options.canWipe)),
            new abi_1.StringValue("canPause"),
            new abi_1.StringValue(this.boolToString(options.canPause)),
            new abi_1.StringValue("canTransferNFTCreateRole"),
            new abi_1.StringValue(this.boolToString(options.canTransferNFTCreateRole)),
            new abi_1.StringValue("canChangeOwner"),
            new abi_1.StringValue(this.boolToString(options.canChangeOwner)),
            new abi_1.StringValue("canUpgrade"),
            new abi_1.StringValue(this.boolToString(options.canUpgrade)),
            new abi_1.StringValue("canAddSpecialRoles"),
            new abi_1.StringValue(this.boolToString(options.canAddSpecialRoles)),
        ];
        const dataParts = ["issueSemiFungible", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitIssue,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    createTransactionForIssuingNonFungible(sender, options) {
        this.notifyAboutUnsettingBurnRoleGlobally();
        const args = [
            new abi_1.StringValue(options.tokenName),
            new abi_1.StringValue(options.tokenTicker),
            new abi_1.StringValue("canFreeze"),
            new abi_1.StringValue(this.boolToString(options.canFreeze)),
            new abi_1.StringValue("canWipe"),
            new abi_1.StringValue(this.boolToString(options.canWipe)),
            new abi_1.StringValue("canPause"),
            new abi_1.StringValue(this.boolToString(options.canPause)),
            new abi_1.StringValue("canTransferNFTCreateRole"),
            new abi_1.StringValue(this.boolToString(options.canTransferNFTCreateRole)),
            new abi_1.StringValue("canChangeOwner"),
            new abi_1.StringValue(this.boolToString(options.canChangeOwner)),
            new abi_1.StringValue("canUpgrade"),
            new abi_1.StringValue(this.boolToString(options.canUpgrade)),
            new abi_1.StringValue("canAddSpecialRoles"),
            new abi_1.StringValue(this.boolToString(options.canAddSpecialRoles)),
        ];
        const dataParts = ["issueNonFungible", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitIssue,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    createTransactionForRegisteringMetaESDT(sender, options) {
        this.notifyAboutUnsettingBurnRoleGlobally();
        const args = [
            new abi_1.StringValue(options.tokenName),
            new abi_1.StringValue(options.tokenTicker),
            new abi_1.BigUIntValue(options.numDecimals),
            new abi_1.StringValue("canFreeze"),
            new abi_1.StringValue(this.boolToString(options.canFreeze)),
            new abi_1.StringValue("canWipe"),
            new abi_1.StringValue(this.boolToString(options.canWipe)),
            new abi_1.StringValue("canPause"),
            new abi_1.StringValue(this.boolToString(options.canPause)),
            new abi_1.StringValue("canTransferNFTCreateRole"),
            new abi_1.StringValue(this.boolToString(options.canTransferNFTCreateRole)),
            new abi_1.StringValue("canChangeOwner"),
            new abi_1.StringValue(this.boolToString(options.canChangeOwner)),
            new abi_1.StringValue("canUpgrade"),
            new abi_1.StringValue(this.boolToString(options.canUpgrade)),
            new abi_1.StringValue("canAddSpecialRoles"),
            new abi_1.StringValue(this.boolToString(options.canAddSpecialRoles)),
        ];
        const dataParts = ["registerMetaESDT", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitIssue,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    createTransactionForRegisteringAndSettingRoles(sender, options) {
        this.notifyAboutUnsettingBurnRoleGlobally();
        const dataParts = [
            "registerAndSetAllRoles",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenName),
                new abi_1.StringValue(options.tokenTicker),
                new abi_1.StringValue(options.tokenType),
                new abi_1.BigUIntValue(options.numDecimals),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitIssue,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    createTransactionForSettingBurnRoleGlobally(sender, options) {
        const dataParts = [
            "setBurnRoleGlobally",
            ...this.argSerializer.valuesToStrings([new abi_1.StringValue(options.tokenIdentifier)]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitToggleBurnRoleGlobally,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnsettingBurnRoleGlobally(sender, options) {
        const dataParts = [
            "unsetBurnRoleGlobally",
            ...this.argSerializer.valuesToStrings([new abi_1.StringValue(options.tokenIdentifier)]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitToggleBurnRoleGlobally,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingSpecialRoleOnFungibleToken(sender, options) {
        const args = [new abi_1.StringValue(options.tokenIdentifier), new abi_1.AddressValue(options.user)];
        options.addRoleLocalMint ? args.push(new abi_1.StringValue("ESDTRoleLocalMint")) : 0;
        options.addRoleLocalBurn ? args.push(new abi_1.StringValue("ESDTRoleLocalBurn")) : 0;
        options.addRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTTransferRole")) : 0;
        const dataParts = ["setSpecialRole", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetSpecialRole,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnsettingSpecialRoleOnFungibleToken(sender, options) {
        const args = [new abi_1.StringValue(options.tokenIdentifier), new abi_1.AddressValue(options.user)];
        options.removeRoleLocalMint ? args.push(new abi_1.StringValue("ESDTRoleLocalMint")) : 0;
        options.removeRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTRoleLocalBurn")) : 0;
        options.removeRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTTransferRole")) : 0;
        const dataParts = ["unSetSpecialRole", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetSpecialRole,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingSpecialRoleOnSemiFungibleToken(sender, options) {
        const args = [new abi_1.StringValue(options.tokenIdentifier), new abi_1.AddressValue(options.user)];
        options.addRoleNFTCreate ? args.push(new abi_1.StringValue("ESDTRoleNFTCreate")) : 0;
        options.addRoleNFTBurn ? args.push(new abi_1.StringValue("ESDTRoleNFTBurn")) : 0;
        options.addRoleNFTAddQuantity ? args.push(new abi_1.StringValue("ESDTRoleNFTAddQuantity")) : 0;
        options.addRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTTransferRole")) : 0;
        options.addRoleNFTUpdate ? args.push(new abi_1.StringValue("ESDTRoleNFTUpdate")) : 0;
        options.addRoleESDTModifyRoyalties ? args.push(new abi_1.StringValue("ESDTRoleModifyRoyalties")) : 0;
        options.addRoleESDTSetNewUri ? args.push(new abi_1.StringValue("ESDTRoleSetNewURI")) : 0;
        options.addRoleESDTModifyCreator ? args.push(new abi_1.StringValue("ESDTRoleModifyCreator")) : 0;
        options.addRoleNFTRecreate ? args.push(new abi_1.StringValue("ESDTRoleNFTRecreate")) : 0;
        const dataParts = ["setSpecialRole", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetSpecialRole,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnsettingSpecialRoleOnSemiFungibleToken(sender, options) {
        const args = [new abi_1.StringValue(options.tokenIdentifier), new abi_1.AddressValue(options.user)];
        options.removeRoleNFTBurn ? args.push(new abi_1.StringValue("ESDTRoleNFTBurn")) : 0;
        options.removeRoleNFTAddQuantity ? args.push(new abi_1.StringValue("ESDTRoleNFTAddQuantity")) : 0;
        options.removeRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTTransferRole")) : 0;
        options.removeRoleNFTUpdate ? args.push(new abi_1.StringValue("ESDTRoleNFTUpdate")) : 0;
        options.removeRoleESDTModifyRoyalties ? args.push(new abi_1.StringValue("ESDTRoleModifyRoyalties")) : 0;
        options.removeRoleESDTSetNewUri ? args.push(new abi_1.StringValue("ESDTRoleSetNewURI")) : 0;
        options.removeRoleESDTModifyCreator ? args.push(new abi_1.StringValue("ESDTRoleModifyCreator")) : 0;
        options.removeRoleNFTRecreate ? args.push(new abi_1.StringValue("ESDTRoleNFTRecreate")) : 0;
        const dataParts = ["unSetSpecialRole", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetSpecialRole,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingSpecialRoleOnMetaESDT(sender, options) {
        return this.createTransactionForSettingSpecialRoleOnSemiFungibleToken(sender, options);
    }
    createTransactionForUnsettingSpecialRoleOnMetaESDT(sender, options) {
        return this.createTransactionForUnsettingSpecialRoleOnSemiFungibleToken(sender, options);
    }
    createTransactionForSettingSpecialRoleOnNonFungibleToken(sender, options) {
        const args = [new abi_1.StringValue(options.tokenIdentifier), new abi_1.AddressValue(options.user)];
        options.addRoleNFTCreate ? args.push(new abi_1.StringValue("ESDTRoleNFTCreate")) : 0;
        options.addRoleNFTBurn ? args.push(new abi_1.StringValue("ESDTRoleNFTBurn")) : 0;
        options.addRoleNFTUpdateAttributes ? args.push(new abi_1.StringValue("ESDTRoleNFTUpdateAttributes")) : 0;
        options.addRoleNFTAddURI ? args.push(new abi_1.StringValue("ESDTRoleNFTAddURI")) : 0;
        options.addRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTTransferRole")) : 0;
        options.addRoleESDTModifyCreator ? args.push(new abi_1.StringValue("ESDTRoleModifyCreator")) : 0;
        options.addRoleNFTRecreate ? args.push(new abi_1.StringValue("ESDTRoleNFTRecreate")) : 0;
        options.addRoleESDTSetNewURI ? args.push(new abi_1.StringValue("ESDTRoleSetNewURI")) : 0;
        options.addRoleESDTModifyRoyalties ? args.push(new abi_1.StringValue("ESDTRoleModifyRoyalties")) : 0;
        const dataParts = ["setSpecialRole", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetSpecialRole,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnsettingSpecialRoleOnNonFungibleToken(sender, options) {
        const args = [new abi_1.StringValue(options.tokenIdentifier), new abi_1.AddressValue(options.user)];
        options.removeRoleNFTBurn ? args.push(new abi_1.StringValue("ESDTRoleNFTBurn")) : 0;
        options.removeRoleNFTUpdateAttributes ? args.push(new abi_1.StringValue("ESDTRoleNFTUpdateAttributes")) : 0;
        options.removeRoleNFTAddURI ? args.push(new abi_1.StringValue("ESDTRoleNFTAddURI")) : 0;
        options.removeRoleESDTTransferRole ? args.push(new abi_1.StringValue("ESDTTransferRole")) : 0;
        options.removeRoleESDTModifyCreator ? args.push(new abi_1.StringValue("ESDTRoleModifyCreator")) : 0;
        options.removeRoleNFTRecreate ? args.push(new abi_1.StringValue("ESDTRoleNFTRecreate")) : 0;
        options.removeRoleESDTSetNewURI ? args.push(new abi_1.StringValue("ESDTRoleSetNewURI")) : 0;
        options.removeRoleESDTModifyRoyalties ? args.push(new abi_1.StringValue("ESDTRoleModifyRoyalties")) : 0;
        const dataParts = ["unSetSpecialRole", ...this.argSerializer.valuesToStrings(args)];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetSpecialRole,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForCreatingNFT(sender, options) {
        const dataParts = [
            "ESDTNFTCreate",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.initialQuantity ?? 1n),
                new abi_1.StringValue(options.name),
                new abi_1.BigUIntValue(options.royalties),
                new abi_1.StringValue(options.hash),
                new abi_1.BytesValue(Buffer.from(options.attributes)),
                ...options.uris.map((uri) => new abi_1.StringValue(uri)),
            ]),
        ];
        // Note that the following is an approximation (a reasonable one):
        const nftData = options.name + options.hash + options.attributes + options.uris.join("");
        const storageGasLimit = this.config.gasLimitStorePerByte + BigInt(nftData.length);
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtNftCreate + storageGasLimit,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForPausing(sender, options) {
        const dataParts = ["pause", ...this.argSerializer.valuesToStrings([new abi_1.StringValue(options.tokenIdentifier)])];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitPausing,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnpausing(sender, options) {
        const dataParts = [
            "unPause",
            ...this.argSerializer.valuesToStrings([new abi_1.StringValue(options.tokenIdentifier)]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitPausing,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForFreezing(sender, options) {
        const dataParts = [
            "freeze",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.AddressValue(options.user),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitFreezing,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUnfreezing(sender, options) {
        const dataParts = [
            "UnFreeze",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.AddressValue(options.user),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitFreezing,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForWiping(sender, options) {
        const dataParts = [
            "wipe",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.AddressValue(options.user),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitWiping,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForLocalMint(sender, options) {
        const dataParts = [
            "ESDTLocalMint",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.supplyToMint),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtLocalMint,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForLocalBurning(sender, options) {
        const dataParts = [
            "ESDTLocalBurn",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.supplyToBurn),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtLocalBurn,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUpdatingAttributes(sender, options) {
        const dataParts = [
            "ESDTNFTUpdateAttributes",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                new abi_1.BytesValue(Buffer.from(options.attributes)),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtNftUpdateAttributes,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForAddingQuantity(sender, options) {
        const dataParts = [
            "ESDTNFTAddQuantity",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                new abi_1.BigUIntValue(options.quantity),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtNftAddQuantity,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForBurningQuantity(sender, options) {
        const dataParts = [
            "ESDTNFTBurn",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                new abi_1.BigUIntValue(options.quantity),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtNftBurn,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForModifyingRoyalties(sender, options) {
        const dataParts = [
            "ESDTModifyRoyalties",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                new abi_1.BigUIntValue(options.newRoyalties),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtModifyRoyalties,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForSettingNewUris(sender, options) {
        if (!options.newUris.length) {
            throw new errors_1.ErrBadUsage("No URIs provided");
        }
        const dataParts = [
            "ESDTSetNewURIs",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                ...options.newUris.map((uri) => new abi_1.StringValue(uri)),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitSetNewUris,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForModifyingCreator(sender, options) {
        const dataParts = [
            "ESDTModifyCreator",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtModifyCreator,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUpdatingMetadata(sender, options) {
        const dataParts = [
            "ESDTMetaDataUpdate",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                ...(options.newTokenName ? [new abi_1.StringValue(options.newTokenName)] : []),
                ...(options.newRoyalties ? [new abi_1.BigUIntValue(options.newRoyalties)] : []),
                ...(options.newHash ? [new abi_1.StringValue(options.newHash)] : []),
                ...(options.newAttributes ? [new abi_1.BytesValue(Buffer.from(options.newAttributes))] : []),
                ...(options.newUris ? options.newUris.map((uri) => new abi_1.StringValue(uri)) : []),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitEsdtMetadataUpdate,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForMetadataRecreate(sender, options) {
        const dataParts = [
            "ESDTMetaDataRecreate",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenIdentifier),
                new abi_1.BigUIntValue(options.tokenNonce),
                ...(options.newTokenName ? [new abi_1.StringValue(options.newTokenName)] : []),
                ...(options.newRoyalties ? [new abi_1.BigUIntValue(options.newRoyalties)] : []),
                ...(options.newHash ? [new abi_1.StringValue(options.newHash)] : []),
                ...(options.newAttributes ? [new abi_1.BytesValue(Buffer.from(options.newAttributes))] : []),
                ...(options.newUris ? options.newUris.map((uri) => new abi_1.StringValue(uri)) : []),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: sender,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitNftMetadataRecreate,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForChangingTokenToDynamic(sender, options) {
        const dataParts = [
            "changeToDynamic",
            ...this.argSerializer.valuesToStrings([new abi_1.StringValue(options.tokenIdentifier)]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitNftChangeToDynamic,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForUpdatingTokenId(sender, options) {
        const dataParts = [
            "updateTokenID",
            ...this.argSerializer.valuesToStrings([new abi_1.StringValue(options.tokenIdentifier)]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitUpdateTokenId,
            addDataMovementGas: true,
        }).build();
    }
    createTransactionForRegisteringDynamicToken(sender, options) {
        const dataParts = [
            "registerDynamic",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenName),
                new abi_1.StringValue(options.tokenTicker),
                new abi_1.StringValue(options.tokenType),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitRegisterDynamic,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    createTransactionForRegisteringDynamicAndSettingRoles(sender, options) {
        const dataParts = [
            "registerAndSetAllRolesDynamic",
            ...this.argSerializer.valuesToStrings([
                new abi_1.StringValue(options.tokenName),
                new abi_1.StringValue(options.tokenTicker),
                new abi_1.StringValue(options.tokenType),
            ]),
        ];
        return new transactionBuilder_1.TransactionBuilder({
            config: this.config,
            sender: sender,
            receiver: this.esdtContractAddress,
            dataParts: dataParts,
            gasLimit: this.config.gasLimitRegisterDynamic,
            addDataMovementGas: true,
            amount: this.config.issueCost,
        }).build();
    }
    notifyAboutUnsettingBurnRoleGlobally() {
        logger_1.Logger.info(`
==========
IMPORTANT!
==========
You are about to issue (register) a new token. This will set the role "ESDTRoleBurnForAll" (globally).
Once the token is registered, you can unset this role by calling "unsetBurnRoleGlobally" (in a separate transaction).`);
    }
    boolToString(value) {
        if (value) {
            return this.trueAsString;
        }
        return this.falseAsString;
    }
}
exports.TokenManagementTransactionsFactory = TokenManagementTransactionsFactory;
