// @multiversx/sdk-core/out/core/transaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction = void 0;
const bignumber_js_1 = require("bignumber.js");
const address_1 = require("./address");
const constants_1 = require("./constants");
const signature_1 = require("./signature");
const transactionComputer_1 = require("./transactionComputer");
/**
 * An abstraction for creating and signing transactions.
 */
class Transaction {
    /**
     * Creates a new Transaction object.
     */
    constructor(options) {
        this.nonce = options.nonce ?? 0n;
        this.value = options.value ?? 0n;
        this.sender = options.sender;
        this.receiver = options.receiver;
        this.senderUsername = options.senderUsername || "";
        this.receiverUsername = options.receiverUsername || "";
        this.gasPrice = options.gasPrice ?? BigInt(constants_1.TRANSACTION_MIN_GAS_PRICE);
        this.gasLimit = options.gasLimit;
        this.data = options.data ?? new Uint8Array();
        this.chainID = options.chainID.valueOf();
        this.version = options.version ?? constants_1.TRANSACTION_VERSION_DEFAULT;
        this.options = options.options ?? constants_1.TRANSACTION_OPTIONS_DEFAULT;
        this.guardian = options.guardian ?? address_1.Address.empty();
        this.relayer = options.relayer ? options.relayer : address_1.Address.empty();
        this.signature = options.signature || Buffer.from([]);
        this.guardianSignature = options.guardianSignature || Buffer.from([]);
        this.relayerSignature = options.relayerSignature || Buffer.from([]);
    }
    /**
     * @deprecated method, use {@link nonce} property instead.
     */
    getNonce() {
        return this.nonce;
    }
    /**
     * @deprecated method, use {@link nonce} property instead.
     * Sets the account sequence number of the sender. Must be done prior signing.
     */
    setNonce(nonce) {
        this.nonce = nonce;
    }
    /**
     * @deprecated method, use {@link value} property instead.
     */
    getValue() {
        return this.value;
    }
    /**
     * @deprecated method, use {@link value} property instead.
     */
    setValue(value) {
        this.value = value;
    }
    /**
     * @deprecated method, use {@link sender} property instead.
     */
    getSender() {
        return this.sender;
    }
    /**
     * @deprecated method, use {@link sender} property instead.
     */
    setSender(sender) {
        this.sender = sender;
    }
    /**
     * @deprecated method, use {@link receiver} property instead.
     */
    getReceiver() {
        return this.receiver;
    }
    /**
     * @deprecated method, use {@link senderUsername} property instead.
     */
    getSenderUsername() {
        return this.senderUsername;
    }
    /**
     * @deprecated method, use {@link senderUsername} property instead.
     */
    setSenderUsername(senderUsername) {
        this.senderUsername = senderUsername;
    }
    /**
     * @deprecated method, use {@link receiverUsername} property instead.
     */
    getReceiverUsername() {
        return this.receiverUsername;
    }
    /**
     * @deprecated method, use {@link receiverUsername} property instead.
     */
    setReceiverUsername(receiverUsername) {
        this.receiverUsername = receiverUsername;
    }
    /**
     * @deprecated method, use {@link guardian} property instead.
     */
    getGuardian() {
        return this.guardian;
    }
    /**
     * @deprecated method, use {@link gasPrice} property instead.
     */
    getGasPrice() {
        return this.gasPrice;
    }
    /**
     * @deprecated method, use {@link gasPrice} property instead.
     */
    setGasPrice(gasPrice) {
        this.gasPrice = gasPrice;
    }
    /**
     * @deprecated method, use {@link gasLimit} property instead.
     */
    getGasLimit() {
        return this.gasLimit;
    }
    /**
     * @deprecated method, use {@link gasLimit} property instead.
     */
    setGasLimit(gasLimit) {
        this.gasLimit = gasLimit;
    }
    /**
     * @deprecated method, use {@link data} property instead.
     */
    getData() {
        return this.data;
    }
    /**
     * @deprecated method, use {@link chainID} property instead.
     */
    getChainID() {
        return this.chainID;
    }
    /**
     * @deprecated method, use {@link chainID} property instead.
     */
    setChainID(chainID) {
        this.chainID = chainID;
    }
    /**
     * @deprecated method, use {@link version} property instead.
     */
    getVersion() {
        return this.version;
    }
    /**
     * @deprecated method, use {@link version} property instead.
     */
    setVersion(version) {
        this.version = version;
    }
    /**
     * @deprecated method, use {@link options} property instead.
     */
    getOptions() {
        return this.options;
    }
    /**
     * @deprecated method, use {@link options} property instead.
     *
     * Question for review: check how the options are set by sdk-dapp, wallet, ledger, extension.
     */
    setOptions(options) {
        this.options = options;
    }
    /**
     * @deprecated method, use{@link signature} property instead.
     */
    getSignature() {
        return Buffer.from(this.signature);
    }
    /**
     * @deprecated method, use {@link guardianSignature} property instead.
     */
    getGuardianSignature() {
        return Buffer.from(this.guardianSignature);
    }
    /**
     * @deprecated method, use {@link guardian} property instead.
     */
    setGuardian(guardian) {
        this.guardian = guardian;
    }
    /**
     * @deprecated method, use "TransactionComputer.computeBytesForSigning()" instead.
     * Serializes a transaction to a sequence of bytes, ready to be signed.
     * This function is called internally by signers.
     */
    serializeForSigning() {
        const computer = new transactionComputer_1.TransactionComputer();
        const bytes = computer.computeBytesForSigning(this);
        return Buffer.from(bytes);
    }
    /**
     * Checks the integrity of the guarded transaction
     */
    isGuardedTransaction() {
        const computer = new transactionComputer_1.TransactionComputer();
        const hasGuardian = !this.guardian.isEmpty();
        const hasGuardianSignature = this.guardianSignature.length > 0;
        return computer.hasOptionsSetForGuardedTransaction(this) && hasGuardian && hasGuardianSignature;
    }
    /**
     * Converts the transaction object into a ready-to-serialize, plain JavaScript object.
     * This function is called internally within the signing procedure.
     */
    toPlainObject() {
        const plainObject = {
            nonce: Number(this.nonce),
            value: this.value.toString(),
            receiver: this.receiver.toBech32(),
            sender: this.sender.toBech32(),
            senderUsername: this.toBase64OrUndefined(this.senderUsername),
            receiverUsername: this.toBase64OrUndefined(this.receiverUsername),
            gasPrice: Number(this.gasPrice),
            gasLimit: Number(this.gasLimit),
            data: this.toBase64OrUndefined(this.data),
            chainID: this.chainID.valueOf(),
            version: this.version,
            options: this.options == 0 ? undefined : this.options,
            guardian: this.guardian.isEmpty() ? undefined : this.guardian.toBech32(),
            relayer: this.relayer.isEmpty() ? undefined : this.relayer.toBech32(),
            signature: this.toHexOrUndefined(this.signature),
            guardianSignature: this.toHexOrUndefined(this.guardianSignature),
            relayerSignature: this.toHexOrUndefined(this.relayerSignature),
        };
        return plainObject;
    }
    /**
     * @deprecated method, use {@link toPlainObject} instead.
     * Converts a plain object transaction into a Transaction Object.
     *
     * @param plainObjectTransaction Raw data of a transaction, usually obtained by calling toPlainObject()
     */
    static fromPlainObject(plainObjectTransaction) {
        return Transaction.newFromPlainObject(plainObjectTransaction);
    }
    /**
     * Converts a plain object transaction into a Transaction Object.
     *
     * @param plainObjectTransaction Raw data of a transaction, usually obtained by calling toPlainObject()
     */
    static newFromPlainObject(plainObjectTransaction) {
        const transaction = new Transaction({
            nonce: BigInt(plainObjectTransaction.nonce),
            value: BigInt(plainObjectTransaction.value || ""),
            receiver: address_1.Address.newFromBech32(plainObjectTransaction.receiver),
            receiverUsername: Buffer.from(plainObjectTransaction.receiverUsername || "", "base64").toString(),
            sender: address_1.Address.newFromBech32(plainObjectTransaction.sender),
            senderUsername: Buffer.from(plainObjectTransaction.senderUsername || "", "base64").toString(),
            guardian: plainObjectTransaction.guardian
                ? address_1.Address.newFromBech32(plainObjectTransaction.guardian)
                : address_1.Address.empty(),
            relayer: plainObjectTransaction.relayer
                ? address_1.Address.newFromBech32(plainObjectTransaction.relayer)
                : address_1.Address.empty(),
            gasPrice: BigInt(plainObjectTransaction.gasPrice),
            gasLimit: BigInt(plainObjectTransaction.gasLimit),
            data: Buffer.from(plainObjectTransaction.data || "", "base64"),
            chainID: String(plainObjectTransaction.chainID),
            version: Number(plainObjectTransaction.version),
            options: plainObjectTransaction.options ? Number(plainObjectTransaction.options) : undefined,
            signature: Buffer.from(plainObjectTransaction.signature || "", "hex"),
            guardianSignature: Buffer.from(plainObjectTransaction.guardianSignature || "", "hex"),
            relayerSignature: Buffer.from(plainObjectTransaction.relayerSignature || "", "hex"),
        });
        return transaction;
    }
    /**
     * @deprecated method, use {@link signature} property instead.
     * Applies the signature on the transaction.
     *
     * @param signature The signature, as computed by a signer.
     */
    applySignature(signature) {
        this.signature = signature_1.interpretSignatureAsBuffer(signature);
    }
    /**
     * @deprecated method, use {@link guardianSignature} property instead.
     * Applies the guardian signature on the transaction.
     *
     * @param guardianSignature The signature, as computed by a signer.
     */
    applyGuardianSignature(guardianSignature) {
        this.guardianSignature = signature_1.interpretSignatureAsBuffer(guardianSignature);
    }
    /**
     * Converts a transaction to a ready-to-broadcast object.
     * Called internally by the network provider.
     */
    toSendable() {
        return this.toPlainObject();
    }
    /**
     * @deprecated method, use "TransactionComputer.computeTransactionFee()" instead.
     *
     * Computes the current transaction fee based on the {@link NetworkConfig} and transaction properties
     * @param networkConfig {@link NetworkConfig}
     */
    computeFee(networkConfig) {
        const computer = new transactionComputer_1.TransactionComputer();
        const fee = computer.computeTransactionFee(this, networkConfig);
        return new bignumber_js_1.BigNumber(fee.toString());
    }
    toBase64OrUndefined(value) {
        return value && value.length ? Buffer.from(value).toString("base64") : undefined;
    }
    toHexOrUndefined(value) {
        return value && value.length ? Buffer.from(value).toString("hex") : undefined;
    }
}
exports.Transaction = Transaction;
