// @multiversx/sdk-dapp/__commonjs/UI/webWallet/XaliasCrossWindowLoginButton/XaliasCrossWindowLoginButton.js
"use strict";var Rm=Object.create;var Lo=Object.defineProperty,Dm=Object.defineProperties,Mm=Object.getOwnPropertyDescriptor,Om=Object.getOwnPropertyDescriptors,Bm=Object.getOwnPropertyNames,cn=Object.getOwnPropertySymbols,Fm=Object.getPrototypeOf,ct=Object.prototype.hasOwnProperty,Gr=Object.prototype.propertyIsEnumerable;var Ur=(e,o,t)=>o in e?Lo(e,o,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[o]=t,u=(e,o)=>{for(var t in o||(o={}))ct.call(o,t)&&Ur(e,t,o[t]);if(cn)for(var t of cn(o))Gr.call(o,t)&&Ur(e,t,o[t]);return e},v=(e,o)=>Dm(e,Om(o));var Re=(e,o)=>{var t={};for(var s in e)ct.call(e,s)&&o.indexOf(s)<0&&(t[s]=e[s]);if(e!=null&&cn)for(var s of cn(e))o.indexOf(s)<0&&Gr.call(e,s)&&(t[s]=e[s]);return t};var h=(e,o)=>()=>(e&&(o=e(e=0)),o);var S=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),ie=(e,o)=>{for(var t in o)Lo(e,t,{get:o[t],enumerable:!0})},Hr=(e,o,t,s)=>{if(o&&typeof o=="object"||typeof o=="function")for(let c of Bm(o))!ct.call(e,c)&&c!==t&&Lo(e,c,{get:()=>o[c],enumerable:!(s=Mm(o,c))||s.enumerable});return e};var x=(e,o,t)=>(t=e!=null?Rm(Fm(e)):{},Hr(o||!e||!e.__esModule?Lo(t,"default",{value:e,enumerable:!0}):t,e)),ee=e=>Hr(Lo({},"__esModule",{value:!0}),e);var f=(e,o,t)=>new Promise((s,c)=>{var p=l=>{try{d(t.next(l))}catch(g){c(g)}},m=l=>{try{d(t.throw(l))}catch(g){c(g)}},d=l=>l.done?s(l.value):Promise.resolve(l.value).then(p,m);d((t=t.apply(e,o)).next())});var qr=S(pn=>{"use strict";n();pn.byteLength=Um;pn.toByteArray=Hm;pn.fromByteArray=qm;var me=[],oe=[],Wm=typeof Uint8Array!="undefined"?Uint8Array:Array,pt="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Ge=0,$r=pt.length;Ge<$r;++Ge)me[Ge]=pt[Ge],oe[pt.charCodeAt(Ge)]=Ge;var Ge,$r;oe["-".charCodeAt(0)]=62;oe["_".charCodeAt(0)]=63;function Vr(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var t=e.indexOf("=");t===-1&&(t=o);var s=t===o?0:4-t%4;return[t,s]}function Um(e){var o=Vr(e),t=o[0],s=o[1];return(t+s)*3/4-s}function Gm(e,o,t){return(o+t)*3/4-t}function Hm(e){var o,t=Vr(e),s=t[0],c=t[1],p=new Wm(Gm(e,s,c)),m=0,d=c>0?s-4:s,l;for(l=0;l<d;l+=4)o=oe[e.charCodeAt(l)]<<18|oe[e.charCodeAt(l+1)]<<12|oe[e.charCodeAt(l+2)]<<6|oe[e.charCodeAt(l+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return c===2&&(o=oe[e.charCodeAt(l)]<<2|oe[e.charCodeAt(l+1)]>>4,p[m++]=o&255),c===1&&(o=oe[e.charCodeAt(l)]<<10|oe[e.charCodeAt(l+1)]<<4|oe[e.charCodeAt(l+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function $m(e){return me[e>>18&63]+me[e>>12&63]+me[e>>6&63]+me[e&63]}function Vm(e,o,t){for(var s,c=[],p=o;p<t;p+=3)s=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),c.push($m(s));return c.join("")}function qm(e){for(var o,t=e.length,s=t%3,c=[],p=16383,m=0,d=t-s;m<d;m+=p)c.push(Vm(e,m,m+p>d?d:m+p));return s===1?(o=e[t-1],c.push(me[o>>2]+me[o<<4&63]+"==")):s===2&&(o=(e[t-2]<<8)+e[t-1],c.push(me[o>>10]+me[o>>4&63]+me[o<<2&63]+"=")),c.join("")}});var jr=S(mt=>{n();mt.read=function(e,o,t,s,c){var p,m,d=c*8-s-1,l=(1<<d)-1,g=l>>1,b=-7,T=t?c-1:0,w=t?-1:1,E=e[o+T];for(T+=w,p=E&(1<<-b)-1,E>>=-b,b+=d;b>0;p=p*256+e[o+T],T+=w,b-=8);for(m=p&(1<<-b)-1,p>>=-b,b+=s;b>0;m=m*256+e[o+T],T+=w,b-=8);if(p===0)p=1-g;else{if(p===l)return m?NaN:(E?-1:1)*(1/0);m=m+Math.pow(2,s),p=p-g}return(E?-1:1)*m*Math.pow(2,p-s)};mt.write=function(e,o,t,s,c,p){var m,d,l,g=p*8-c-1,b=(1<<g)-1,T=b>>1,w=c===23?Math.pow(2,-24)-Math.pow(2,-77):0,E=s?0:p-1,L=s?1:-1,N=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(d=isNaN(o)?1:0,m=b):(m=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+T>=1?o+=w/l:o+=w*Math.pow(2,1-T),o*l>=2&&(m++,l/=2),m+T>=b?(d=0,m=b):m+T>=1?(d=(o*l-1)*Math.pow(2,c),m=m+T):(d=o*Math.pow(2,T-1)*Math.pow(2,c),m=0));c>=8;e[t+E]=d&255,E+=L,d/=256,c-=8);for(m=m<<c|d,g+=c;g>0;e[t+E]=m&255,E+=L,m/=256,g-=8);e[t+E-L]|=N*128}});var aa=S(ro=>{"use strict";n();var dt=qr(),to=jr(),zr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;ro.Buffer=_;ro.SlowBuffer=Zm;ro.INSPECT_MAX_BYTES=50;var mn=2147483647;ro.kMaxLength=mn;_.TYPED_ARRAY_SUPPORT=jm();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function jm(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(t){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function xe(e){if(e>mn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,_.prototype),o}function _(e,o,t){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return gt(e)}return Yr(e,o,t)}_.poolSize=8192;function Yr(e,o,t){if(typeof e=="string")return Km(e,o);if(ArrayBuffer.isView(e))return Xm(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(de(e,ArrayBuffer)||e&&de(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(de(e,SharedArrayBuffer)||e&&de(e.buffer,SharedArrayBuffer)))return _t(e,o,t);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var s=e.valueOf&&e.valueOf();if(s!=null&&s!==e)return _.from(s,o,t);var c=Ym(e);if(c)return c;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return _.from(e[Symbol.toPrimitive]("string"),o,t);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}_.from=function(e,o,t){return Yr(e,o,t)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function Zr(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function zm(e,o,t){return Zr(e),e<=0?xe(e):o!==void 0?typeof t=="string"?xe(e).fill(o,t):xe(e).fill(o):xe(e)}_.alloc=function(e,o,t){return zm(e,o,t)};function gt(e){return Zr(e),xe(e<0?0:ft(e)|0)}_.allocUnsafe=function(e){return gt(e)};_.allocUnsafeSlow=function(e){return gt(e)};function Km(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!_.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var t=Jr(e,o)|0,s=xe(t),c=s.write(e,o);return c!==t&&(s=s.slice(0,c)),s}function lt(e){for(var o=e.length<0?0:ft(e.length)|0,t=xe(o),s=0;s<o;s+=1)t[s]=e[s]&255;return t}function Xm(e){if(de(e,Uint8Array)){var o=new Uint8Array(e);return _t(o.buffer,o.byteOffset,o.byteLength)}return lt(e)}function _t(e,o,t){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(t||0))throw new RangeError('"length" is outside of buffer bounds');var s;return o===void 0&&t===void 0?s=new Uint8Array(e):t===void 0?s=new Uint8Array(e,o):s=new Uint8Array(e,o,t),Object.setPrototypeOf(s,_.prototype),s}function Ym(e){if(_.isBuffer(e)){var o=ft(e.length)|0,t=xe(o);return t.length===0||e.copy(t,0,0,o),t}if(e.length!==void 0)return typeof e.length!="number"||Tt(e.length)?xe(0):lt(e);if(e.type==="Buffer"&&Array.isArray(e.data))return lt(e.data)}function ft(e){if(e>=mn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+mn.toString(16)+" bytes");return e|0}function Zm(e){return+e!=e&&(e=0),_.alloc(+e)}_.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==_.prototype};_.compare=function(o,t){if(de(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),de(t,Uint8Array)&&(t=_.from(t,t.offset,t.byteLength)),!_.isBuffer(o)||!_.isBuffer(t))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===t)return 0;for(var s=o.length,c=t.length,p=0,m=Math.min(s,c);p<m;++p)if(o[p]!==t[p]){s=o[p],c=t[p];break}return s<c?-1:c<s?1:0};_.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(o,t){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return _.alloc(0);var s;if(t===void 0)for(t=0,s=0;s<o.length;++s)t+=o[s].length;var c=_.allocUnsafe(t),p=0;for(s=0;s<o.length;++s){var m=o[s];if(de(m,Uint8Array))p+m.length>c.length?_.from(m).copy(c,p):Uint8Array.prototype.set.call(c,m,p);else if(_.isBuffer(m))m.copy(c,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return c};function Jr(e,o){if(_.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||de(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var t=e.length,s=arguments.length>2&&arguments[2]===!0;if(!s&&t===0)return 0;for(var c=!1;;)switch(o){case"ascii":case"latin1":case"binary":return t;case"utf8":case"utf-8":return ut(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return t*2;case"hex":return t>>>1;case"base64":return ra(e).length;default:if(c)return s?-1:ut(e).length;o=(""+o).toLowerCase(),c=!0}}_.byteLength=Jr;function Jm(e,o,t){var s=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((t===void 0||t>this.length)&&(t=this.length),t<=0)||(t>>>=0,o>>>=0,t<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return cd(this,o,t);case"utf8":case"utf-8":return ea(this,o,t);case"ascii":return id(this,o,t);case"latin1":case"binary":return sd(this,o,t);case"base64":return rd(this,o,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return pd(this,o,t);default:if(s)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),s=!0}}_.prototype._isBuffer=!0;function He(e,o,t){var s=e[o];e[o]=e[t],e[t]=s}_.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var t=0;t<o;t+=2)He(this,t,t+1);return this};_.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var t=0;t<o;t+=4)He(this,t,t+3),He(this,t+1,t+2);return this};_.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var t=0;t<o;t+=8)He(this,t,t+7),He(this,t+1,t+6),He(this,t+2,t+5),He(this,t+3,t+4);return this};_.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?ea(this,0,o):Jm.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(o){if(!_.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:_.compare(this,o)===0};_.prototype.inspect=function(){var o="",t=ro.INSPECT_MAX_BYTES;return o=this.toString("hex",0,t).replace(/(.{2})/g,"$1 ").trim(),this.length>t&&(o+=" ... "),"<Buffer "+o+">"};zr&&(_.prototype[zr]=_.prototype.inspect);_.prototype.compare=function(o,t,s,c,p){if(de(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),!_.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(t===void 0&&(t=0),s===void 0&&(s=o?o.length:0),c===void 0&&(c=0),p===void 0&&(p=this.length),t<0||s>o.length||c<0||p>this.length)throw new RangeError("out of range index");if(c>=p&&t>=s)return 0;if(c>=p)return-1;if(t>=s)return 1;if(t>>>=0,s>>>=0,c>>>=0,p>>>=0,this===o)return 0;for(var m=p-c,d=s-t,l=Math.min(m,d),g=this.slice(c,p),b=o.slice(t,s),T=0;T<l;++T)if(g[T]!==b[T]){m=g[T],d=b[T];break}return m<d?-1:d<m?1:0};function Qr(e,o,t,s,c){if(e.length===0)return-1;if(typeof t=="string"?(s=t,t=0):t>2147483647?t=2147483647:t<-2147483648&&(t=-2147483648),t=+t,Tt(t)&&(t=c?0:e.length-1),t<0&&(t=e.length+t),t>=e.length){if(c)return-1;t=e.length-1}else if(t<0)if(c)t=0;else return-1;if(typeof o=="string"&&(o=_.from(o,s)),_.isBuffer(o))return o.length===0?-1:Kr(e,o,t,s,c);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?c?Uint8Array.prototype.indexOf.call(e,o,t):Uint8Array.prototype.lastIndexOf.call(e,o,t):Kr(e,[o],t,s,c);throw new TypeError("val must be string, number or Buffer")}function Kr(e,o,t,s,c){var p=1,m=e.length,d=o.length;if(s!==void 0&&(s=String(s).toLowerCase(),s==="ucs2"||s==="ucs-2"||s==="utf16le"||s==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,d/=2,t/=2}function l(E,L){return p===1?E[L]:E.readUInt16BE(L*p)}var g;if(c){var b=-1;for(g=t;g<m;g++)if(l(e,g)===l(o,b===-1?0:g-b)){if(b===-1&&(b=g),g-b+1===d)return b*p}else b!==-1&&(g-=g-b),b=-1}else for(t+d>m&&(t=m-d),g=t;g>=0;g--){for(var T=!0,w=0;w<d;w++)if(l(e,g+w)!==l(o,w)){T=!1;break}if(T)return g}return-1}_.prototype.includes=function(o,t,s){return this.indexOf(o,t,s)!==-1};_.prototype.indexOf=function(o,t,s){return Qr(this,o,t,s,!0)};_.prototype.lastIndexOf=function(o,t,s){return Qr(this,o,t,s,!1)};function Qm(e,o,t,s){t=Number(t)||0;var c=e.length-t;s?(s=Number(s),s>c&&(s=c)):s=c;var p=o.length;s>p/2&&(s=p/2);for(var m=0;m<s;++m){var d=parseInt(o.substr(m*2,2),16);if(Tt(d))return m;e[t+m]=d}return m}function ed(e,o,t,s){return dn(ut(o,e.length-t),e,t,s)}function od(e,o,t,s){return dn(ld(o),e,t,s)}function nd(e,o,t,s){return dn(ra(o),e,t,s)}function td(e,o,t,s){return dn(_d(o,e.length-t),e,t,s)}_.prototype.write=function(o,t,s,c){if(t===void 0)c="utf8",s=this.length,t=0;else if(s===void 0&&typeof t=="string")c=t,s=this.length,t=0;else if(isFinite(t))t=t>>>0,isFinite(s)?(s=s>>>0,c===void 0&&(c="utf8")):(c=s,s=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-t;if((s===void 0||s>p)&&(s=p),o.length>0&&(s<0||t<0)||t>this.length)throw new RangeError("Attempt to write outside buffer bounds");c||(c="utf8");for(var m=!1;;)switch(c){case"hex":return Qm(this,o,t,s);case"utf8":case"utf-8":return ed(this,o,t,s);case"ascii":case"latin1":case"binary":return od(this,o,t,s);case"base64":return nd(this,o,t,s);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return td(this,o,t,s);default:if(m)throw new TypeError("Unknown encoding: "+c);c=(""+c).toLowerCase(),m=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function rd(e,o,t){return o===0&&t===e.length?dt.fromByteArray(e):dt.fromByteArray(e.slice(o,t))}function ea(e,o,t){t=Math.min(e.length,t);for(var s=[],c=o;c<t;){var p=e[c],m=null,d=p>239?4:p>223?3:p>191?2:1;if(c+d<=t){var l,g,b,T;switch(d){case 1:p<128&&(m=p);break;case 2:l=e[c+1],(l&192)===128&&(T=(p&31)<<6|l&63,T>127&&(m=T));break;case 3:l=e[c+1],g=e[c+2],(l&192)===128&&(g&192)===128&&(T=(p&15)<<12|(l&63)<<6|g&63,T>2047&&(T<55296||T>57343)&&(m=T));break;case 4:l=e[c+1],g=e[c+2],b=e[c+3],(l&192)===128&&(g&192)===128&&(b&192)===128&&(T=(p&15)<<18|(l&63)<<12|(g&63)<<6|b&63,T>65535&&T<1114112&&(m=T))}}m===null?(m=65533,d=1):m>65535&&(m-=65536,s.push(m>>>10&1023|55296),m=56320|m&1023),s.push(m),c+=d}return ad(s)}var Xr=4096;function ad(e){var o=e.length;if(o<=Xr)return String.fromCharCode.apply(String,e);for(var t="",s=0;s<o;)t+=String.fromCharCode.apply(String,e.slice(s,s+=Xr));return t}function id(e,o,t){var s="";t=Math.min(e.length,t);for(var c=o;c<t;++c)s+=String.fromCharCode(e[c]&127);return s}function sd(e,o,t){var s="";t=Math.min(e.length,t);for(var c=o;c<t;++c)s+=String.fromCharCode(e[c]);return s}function cd(e,o,t){var s=e.length;(!o||o<0)&&(o=0),(!t||t<0||t>s)&&(t=s);for(var c="",p=o;p<t;++p)c+=ud[e[p]];return c}function pd(e,o,t){for(var s=e.slice(o,t),c="",p=0;p<s.length-1;p+=2)c+=String.fromCharCode(s[p]+s[p+1]*256);return c}_.prototype.slice=function(o,t){var s=this.length;o=~~o,t=t===void 0?s:~~t,o<0?(o+=s,o<0&&(o=0)):o>s&&(o=s),t<0?(t+=s,t<0&&(t=0)):t>s&&(t=s),t<o&&(t=o);var c=this.subarray(o,t);return Object.setPrototypeOf(c,_.prototype),c};function U(e,o,t){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>t)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=this[o],p=1,m=0;++m<t&&(p*=256);)c+=this[o+m]*p;return c};_.prototype.readUintBE=_.prototype.readUIntBE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=this[o+--t],p=1;t>0&&(p*=256);)c+=this[o+--t]*p;return c};_.prototype.readUint8=_.prototype.readUInt8=function(o,t){return o=o>>>0,t||U(o,1,this.length),this[o]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(o,t){return o=o>>>0,t||U(o,2,this.length),this[o]|this[o+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(o,t){return o=o>>>0,t||U(o,2,this.length),this[o]<<8|this[o+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(o,t){return o=o>>>0,t||U(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(o,t){return o=o>>>0,t||U(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};_.prototype.readIntLE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=this[o],p=1,m=0;++m<t&&(p*=256);)c+=this[o+m]*p;return p*=128,c>=p&&(c-=Math.pow(2,8*t)),c};_.prototype.readIntBE=function(o,t,s){o=o>>>0,t=t>>>0,s||U(o,t,this.length);for(var c=t,p=1,m=this[o+--c];c>0&&(p*=256);)m+=this[o+--c]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*t)),m};_.prototype.readInt8=function(o,t){return o=o>>>0,t||U(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};_.prototype.readInt16LE=function(o,t){o=o>>>0,t||U(o,2,this.length);var s=this[o]|this[o+1]<<8;return s&32768?s|4294901760:s};_.prototype.readInt16BE=function(o,t){o=o>>>0,t||U(o,2,this.length);var s=this[o+1]|this[o]<<8;return s&32768?s|4294901760:s};_.prototype.readInt32LE=function(o,t){return o=o>>>0,t||U(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};_.prototype.readInt32BE=function(o,t){return o=o>>>0,t||U(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};_.prototype.readFloatLE=function(o,t){return o=o>>>0,t||U(o,4,this.length),to.read(this,o,!0,23,4)};_.prototype.readFloatBE=function(o,t){return o=o>>>0,t||U(o,4,this.length),to.read(this,o,!1,23,4)};_.prototype.readDoubleLE=function(o,t){return o=o>>>0,t||U(o,8,this.length),to.read(this,o,!0,52,8)};_.prototype.readDoubleBE=function(o,t){return o=o>>>0,t||U(o,8,this.length),to.read(this,o,!1,52,8)};function j(e,o,t,s,c,p){if(!_.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>c||o<p)throw new RangeError('"value" argument is out of bounds');if(t+s>e.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(o,t,s,c){if(o=+o,t=t>>>0,s=s>>>0,!c){var p=Math.pow(2,8*s)-1;j(this,o,t,s,p,0)}var m=1,d=0;for(this[t]=o&255;++d<s&&(m*=256);)this[t+d]=o/m&255;return t+s};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(o,t,s,c){if(o=+o,t=t>>>0,s=s>>>0,!c){var p=Math.pow(2,8*s)-1;j(this,o,t,s,p,0)}var m=s-1,d=1;for(this[t+m]=o&255;--m>=0&&(d*=256);)this[t+m]=o/d&255;return t+s};_.prototype.writeUint8=_.prototype.writeUInt8=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,1,255,0),this[t]=o&255,t+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,65535,0),this[t]=o&255,this[t+1]=o>>>8,t+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,65535,0),this[t]=o>>>8,this[t+1]=o&255,t+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,4294967295,0),this[t+3]=o>>>24,this[t+2]=o>>>16,this[t+1]=o>>>8,this[t]=o&255,t+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,4294967295,0),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};_.prototype.writeIntLE=function(o,t,s,c){if(o=+o,t=t>>>0,!c){var p=Math.pow(2,8*s-1);j(this,o,t,s,p-1,-p)}var m=0,d=1,l=0;for(this[t]=o&255;++m<s&&(d*=256);)o<0&&l===0&&this[t+m-1]!==0&&(l=1),this[t+m]=(o/d>>0)-l&255;return t+s};_.prototype.writeIntBE=function(o,t,s,c){if(o=+o,t=t>>>0,!c){var p=Math.pow(2,8*s-1);j(this,o,t,s,p-1,-p)}var m=s-1,d=1,l=0;for(this[t+m]=o&255;--m>=0&&(d*=256);)o<0&&l===0&&this[t+m+1]!==0&&(l=1),this[t+m]=(o/d>>0)-l&255;return t+s};_.prototype.writeInt8=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,1,127,-128),o<0&&(o=255+o+1),this[t]=o&255,t+1};_.prototype.writeInt16LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,32767,-32768),this[t]=o&255,this[t+1]=o>>>8,t+2};_.prototype.writeInt16BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,2,32767,-32768),this[t]=o>>>8,this[t+1]=o&255,t+2};_.prototype.writeInt32LE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,2147483647,-2147483648),this[t]=o&255,this[t+1]=o>>>8,this[t+2]=o>>>16,this[t+3]=o>>>24,t+4};_.prototype.writeInt32BE=function(o,t,s){return o=+o,t=t>>>0,s||j(this,o,t,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};function oa(e,o,t,s,c,p){if(t+s>e.length)throw new RangeError("Index out of range");if(t<0)throw new RangeError("Index out of range")}function na(e,o,t,s,c){return o=+o,t=t>>>0,c||oa(e,o,t,4,34028234663852886e22,-34028234663852886e22),to.write(e,o,t,s,23,4),t+4}_.prototype.writeFloatLE=function(o,t,s){return na(this,o,t,!0,s)};_.prototype.writeFloatBE=function(o,t,s){return na(this,o,t,!1,s)};function ta(e,o,t,s,c){return o=+o,t=t>>>0,c||oa(e,o,t,8,17976931348623157e292,-17976931348623157e292),to.write(e,o,t,s,52,8),t+8}_.prototype.writeDoubleLE=function(o,t,s){return ta(this,o,t,!0,s)};_.prototype.writeDoubleBE=function(o,t,s){return ta(this,o,t,!1,s)};_.prototype.copy=function(o,t,s,c){if(!_.isBuffer(o))throw new TypeError("argument should be a Buffer");if(s||(s=0),!c&&c!==0&&(c=this.length),t>=o.length&&(t=o.length),t||(t=0),c>0&&c<s&&(c=s),c===s||o.length===0||this.length===0)return 0;if(t<0)throw new RangeError("targetStart out of bounds");if(s<0||s>=this.length)throw new RangeError("Index out of range");if(c<0)throw new RangeError("sourceEnd out of bounds");c>this.length&&(c=this.length),o.length-t<c-s&&(c=o.length-t+s);var p=c-s;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(t,s,c):Uint8Array.prototype.set.call(o,this.subarray(s,c),t),p};_.prototype.fill=function(o,t,s,c){if(typeof o=="string"){if(typeof t=="string"?(c=t,t=0,s=this.length):typeof s=="string"&&(c=s,s=this.length),c!==void 0&&typeof c!="string")throw new TypeError("encoding must be a string");if(typeof c=="string"&&!_.isEncoding(c))throw new TypeError("Unknown encoding: "+c);if(o.length===1){var p=o.charCodeAt(0);(c==="utf8"&&p<128||c==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(t<0||this.length<t||this.length<s)throw new RangeError("Out of range index");if(s<=t)return this;t=t>>>0,s=s===void 0?this.length:s>>>0,o||(o=0);var m;if(typeof o=="number")for(m=t;m<s;++m)this[m]=o;else{var d=_.isBuffer(o)?o:_.from(o,c),l=d.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<s-t;++m)this[m+t]=d[m%l]}return this};var md=/[^+/0-9A-Za-z-_]/g;function dd(e){if(e=e.split("=")[0],e=e.trim().replace(md,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function ut(e,o){o=o||1/0;for(var t,s=e.length,c=null,p=[],m=0;m<s;++m){if(t=e.charCodeAt(m),t>55295&&t<57344){if(!c){if(t>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===s){(o-=3)>-1&&p.push(239,191,189);continue}c=t;continue}if(t<56320){(o-=3)>-1&&p.push(239,191,189),c=t;continue}t=(c-55296<<10|t-56320)+65536}else c&&(o-=3)>-1&&p.push(239,191,189);if(c=null,t<128){if((o-=1)<0)break;p.push(t)}else if(t<2048){if((o-=2)<0)break;p.push(t>>6|192,t&63|128)}else if(t<65536){if((o-=3)<0)break;p.push(t>>12|224,t>>6&63|128,t&63|128)}else if(t<1114112){if((o-=4)<0)break;p.push(t>>18|240,t>>12&63|128,t>>6&63|128,t&63|128)}else throw new Error("Invalid code point")}return p}function ld(e){for(var o=[],t=0;t<e.length;++t)o.push(e.charCodeAt(t)&255);return o}function _d(e,o){for(var t,s,c,p=[],m=0;m<e.length&&!((o-=2)<0);++m)t=e.charCodeAt(m),s=t>>8,c=t%256,p.push(c),p.push(s);return p}function ra(e){return dt.toByteArray(dd(e))}function dn(e,o,t,s){for(var c=0;c<s&&!(c+t>=o.length||c>=e.length);++c)o[c+t]=e[c];return c}function de(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Tt(e){return e!==e}var ud=function(){for(var e="0123456789abcdef",o=new Array(256),t=0;t<16;++t)for(var s=t*16,c=0;c<16;++c)o[s+c]=e[t]+e[c];return o}()});var ma=S((lx,pa)=>{n();var O=pa.exports={},le,_e;function ht(){throw new Error("setTimeout has not been defined")}function xt(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?le=setTimeout:le=ht}catch(e){le=ht}try{typeof clearTimeout=="function"?_e=clearTimeout:_e=xt}catch(e){_e=xt}})();function ia(e){if(le===setTimeout)return setTimeout(e,0);if((le===ht||!le)&&setTimeout)return le=setTimeout,setTimeout(e,0);try{return le(e,0)}catch(o){try{return le.call(null,e,0)}catch(t){return le.call(this,e,0)}}}function gd(e){if(_e===clearTimeout)return clearTimeout(e);if((_e===xt||!_e)&&clearTimeout)return _e=clearTimeout,clearTimeout(e);try{return _e(e)}catch(o){try{return _e.call(null,e)}catch(t){return _e.call(this,e)}}}var ye=[],ao=!1,$e,ln=-1;function fd(){!ao||!$e||(ao=!1,$e.length?ye=$e.concat(ye):ln=-1,ye.length&&sa())}function sa(){if(!ao){var e=ia(fd);ao=!0;for(var o=ye.length;o;){for($e=ye,ye=[];++ln<o;)$e&&$e[ln].run();ln=-1,o=ye.length}$e=null,ao=!1,gd(e)}}O.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var t=1;t<arguments.length;t++)o[t-1]=arguments[t];ye.push(new ca(e,o)),ye.length===1&&!ao&&ia(sa)};function ca(e,o){this.fun=e,this.array=o}ca.prototype.run=function(){this.fun.apply(null,this.array)};O.title="browser";O.browser=!0;O.env={};O.argv=[];O.version="";O.versions={};function Se(){}O.on=Se;O.addListener=Se;O.once=Se;O.off=Se;O.removeListener=Se;O.removeAllListeners=Se;O.emit=Se;O.prependListener=Se;O.prependOnceListener=Se;O.listeners=function(e){return[]};O.binding=function(e){throw new Error("process.binding is not supported")};O.cwd=function(){return"/"};O.chdir=function(e){throw new Error("process.chdir is not supported")};O.umask=function(){return 0}});var r,i,Td,a,n=h(()=>{r=x(aa()),i=x(ma()),Td=function(e){function o(){var s=this||self;return delete e.prototype.__magic__,s}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var t=__magic__;return t}(Object),a=Td});var yt=h(()=>{"use strict";n()});var be,ue=h(()=>{"use strict";n();be="Action not allowed. User is logged in. Call logout() first"});var da=h(()=>{"use strict";n()});var St,la=h(()=>{"use strict";n();St=(s=>(s.SetGuardian="SetGuardian",s.GuardAccount="GuardAccount",s.UnGuardAccount="UnGuardAccount",s))(St||{})});var _a=h(()=>{"use strict";n()});var bt=h(()=>{"use strict";n()});var ua=h(()=>{"use strict";n()});var wt=h(()=>{"use strict";n()});var ga=h(()=>{"use strict";n()});var fa=h(()=>{"use strict";n()});var se,io,De=h(()=>{"use strict";n();se=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),io=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var W,Ta,vx,ha,Ax,C=h(()=>{"use strict";n();De();W=(w=>(w.ledger="ledger",w.walletconnect="walletconnect",w.walletconnectv2="walletconnectv2",w.wallet="wallet",w.crossWindow="crossWindow",w.iframe="iframe",w.extension="extension",w.passkey="passkey",w.metamask="metamask",w.opera="opera",w.extra="extra",w.none="",w))(W||{}),Ta=(t=>(t.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",t.reloginRequest="RELOGIN_REQUEST",t))(Ta||{}),vx=u(u({},io.WindowProviderRequestEnums),Ta),ha=(o=>(o.reloginResponse="RELOGIN_RESPONSE",o))(ha||{}),Ax=u(u({},io.WindowProviderResponseEnums),ha)});var xa=h(()=>{"use strict";n()});var ya=h(()=>{"use strict";n()});var z=h(()=>{"use strict";n()});var Sa=h(()=>{"use strict";n()});var ba=h(()=>{"use strict";n()});var wa=h(()=>{"use strict";n()});var B=h(()=>{"use strict";n();wt();ga();fa();C();xa();ya();z();Sa();ba();wa()});var so,va,Hx,Aa,$x,Ia,Vx,qx,hd,ka,co=h(()=>{"use strict";n();B();so={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:va,egldLabel:Hx}=so["devnet"],{chainId:Aa,egldLabel:$x}=so["testnet"],{chainId:Ia,egldLabel:Vx}=so["mainnet"],qx={["devnet"]:va,["testnet"]:Aa,["mainnet"]:Ia},hd={[va]:"devnet",[Aa]:"testnet",[Ia]:"mainnet"},ka="multiversx://"});var Po=h(()=>{"use strict";n()});var ge,La=h(()=>{"use strict";n();ge=require("@multiversx/sdk-web-wallet-provider")});var ne,po=h(()=>{"use strict";n();ne=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var Pa,Ea,vt,xd,yd,Ca,At=h(()=>{"use strict";n();po();vt=String((Ea=(Pa=ne.safeWindow)==null?void 0:Pa.navigator)==null?void 0:Ea.userAgent),xd=/^((?!chrome|android).)*safari/i.test(vt),yd=/firefox/i.test(vt)&&/windows/i.test(vt),Ca=xd||yd});var _n,un,Na,gn,Ra,mo,k=h(()=>{"use strict";n();yt();ue();da();la();_a();bt();ua();co();Po();La();At();_n=5e4,un=1e9,Na=1,gn="logout",Ra="login",mo="0"});var lo,fn=h(()=>{"use strict";n();lo=()=>Date.now()/1e3});var Da=h(()=>{"use strict";n()});var Ma=h(()=>{"use strict";n()});var It=h(()=>{"use strict";n();fn();Da();Ma()});var kt={};ie(kt,{clear:()=>vd,getItem:()=>bd,localStorageKeys:()=>we,removeItem:()=>wd,setItem:()=>Sd});var we,Tn,Sd,bd,wd,vd,Ve=h(()=>{"use strict";n();It();we={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},Tn=typeof localStorage!="undefined",Sd=({key:e,data:o,expires:t})=>{!Tn||localStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},bd=e=>{if(!Tn)return;let o=localStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:lo()>=t.expires?(localStorage.removeItem(String(e)),null):t.data},wd=e=>{!Tn||localStorage.removeItem(String(e))},vd=()=>{!Tn||localStorage.clear()}});var Lt={};ie(Lt,{clear:()=>Wa,getItem:()=>Ba,removeItem:()=>Fa,setItem:()=>Oa,storage:()=>Ad});var Oa,Ba,Fa,Wa,Ad,Ua=h(()=>{"use strict";n();Oa=({key:e,data:o,expires:t})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},Ba=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Date.now()>=t.expires?(sessionStorage.removeItem(String(e)),null):t.data},Fa=e=>sessionStorage.removeItem(String(e)),Wa=()=>sessionStorage.clear(),Ad={setItem:Oa,getItem:Ba,removeItem:Fa,clear:Wa}});var Me,qe=h(()=>{"use strict";n();Ve();Ua();Me={session:Lt,local:kt}});var Pt,R,K,D=h(()=>{"use strict";n();Pt=require("@reduxjs/toolkit");k();R=(0,Pt.createAction)(gn),K=(0,Pt.createAction)(Ra,e=>({payload:e}))});var Ct,Ga,Ha,X,Et,$a,J,fe,Nt,ky,Id,kd,Ly,Py,Ey,Ld,Pd,hn,xn=h(()=>{"use strict";n();Ct=require("@multiversx/sdk-core"),Ga=require("@reduxjs/toolkit"),Ha=require("redux-persist");k();qe();Ve();D();X={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:mo},Et={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":X},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},$a=(0,Ga.createSlice)({name:"accountInfoSlice",initialState:Et,reducers:{setAddress:(e,o)=>{let t=o.payload;e.address=t,e.publicKey=t?new Ct.Address(t).hex():""},setAccount:(e,o)=>{let t=e.address===o.payload.address;e.accounts={[e.address]:t?o.payload:X},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:t}=e;e.accounts[t].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}},setWebsocketBatchEvent:(e,o)=>{e.websocketBatchEvent={timestamp:Date.now(),data:o.payload}}},extraReducers:e=>{e.addCase(R,()=>(Me.local.removeItem(we.loginExpiresAt),Et)),e.addCase(K,(o,t)=>{let{address:s}=t.payload;o.address=s,o.publicKey=new Ct.Address(s).toHex()}),e.addCase(Ha.REHYDRATE,(o,t)=>{var g;if(!((g=t.payload)!=null&&g.account))return;let{account:s}=t.payload,{address:c,shard:p,accounts:m,publicKey:d}=s;o.address=c,o.shard=p;let l=m&&c in m;o.accounts=l?m:Et.accounts,o.publicKey=d})}}),{setAccount:J,setAddress:fe,setAccountNonce:Nt,setAccountShard:ky,setLedgerAccount:Id,updateLedgerAccount:kd,setWalletConnectAccount:Ly,setIsAccountLoading:Py,setAccountLoadingError:Ey,setWebsocketEvent:Ld,setWebsocketBatchEvent:Pd}=$a.actions,hn=$a.reducer});function Eo(){return new Date().setHours(new Date().getHours()+24)}function Co(e){Me.local.setItem({key:we.loginExpiresAt,data:e,expires:e})}var Rt=h(()=>{"use strict";n();qe();Ve()});var qa,Va,ja,Uy,Ed,Cd,No,Gy,Nd,za,Hy,$y,Vy,yn,Sn=h(()=>{"use strict";n();qa=require("@reduxjs/toolkit");Rt();C();D();Va={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},ja=(0,qa.createSlice)({name:"loginInfoSlice",initialState:Va,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},setWebviewLogin:(e,o)=>{e.webviewLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload},setIsWalletConnectV2Initialized:(e,o)=>{e.isWalletConnectV2Initialized=o.payload}},extraReducers:e=>{e.addCase(R,()=>Va),e.addCase(K,(o,t)=>{o.isLoginSessionInvalid=!1,o.loginMethod=t.payload.loginMethod,o.iframeLoginType=t.payload.iframeLoginType,Co(Eo())})}}),{setLoginMethod:Uy,setWalletConnectLogin:Ed,setLedgerLogin:Cd,setTokenLogin:No,setTokenLoginSignature:Gy,setWalletLogin:Nd,invalidateLoginSession:za,setLogoutRoute:Hy,setIsWalletConnectV2Initialized:$y,setWebviewLogin:Vy}=ja.actions,yn=ja.reducer});var Xa,Ka,Ya,Ky,Rd,Xy,Dd,bn,wn=h(()=>{"use strict";n();Xa=require("@reduxjs/toolkit");D();Ka={},Ya=(0,Xa.createSlice)({name:"modalsSlice",initialState:Ka,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(R,()=>Ka)}}),{setTxSubmittedModal:Ky,setNotificationModal:Rd,clearTxSubmittedModal:Xy,clearNotificationModal:Dd}=Ya.actions,bn=Ya.reducer});var je,_o=h(()=>{"use strict";n();je=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var F,H=h(()=>{"use strict";n();_o();F=()=>{if(!je())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:o,hash:t,origin:s,href:c,search:p}}=window;return{pathname:o,hash:t,origin:s,href:c,search:p}}});var Za=h(()=>{"use strict";n();H()});var Ja=h(()=>{"use strict";n();$()});var Qa,ei=h(()=>{"use strict";n();po();Qa=e=>{if(!ne.safeWindow.location)return;if(!new URL(ne.safeWindow.location.href).protocol.startsWith("http"))return ne.safeWindow.location.reload();ne.safeWindow.location.assign(e)}});var $=h(()=>{"use strict";n();Za();Ja();H();ei()});var Md,vn,Ro=h(()=>{"use strict";n();$();Md=!1,vn=({timeout:e=0,url:o})=>{Md||setTimeout(()=>{if(!!window)return Qa(o)},e)}});function Oe({callbackRoute:e,onLoginRedirect:o,options:t}){let s=Boolean(e),c=typeof o=="function";if(s&&e!=null){if(c)return o(e,t);vn({url:e,timeout:Od})}}var Od,oi=h(()=>{"use strict";n();Ro();Od=200});function ni(e){return e[Math.floor(Math.random()*e.length)]}var ti=h(()=>{"use strict";n()});var Dt=h(()=>{"use strict";n();bt()});var ve=h(()=>{"use strict";n();oi();ti();Dt()});var ri,ai,ii,Mt,Fd,si,kS,LS,Ot,An,In=h(()=>{"use strict";n();ri=require("@reduxjs/toolkit"),ai=x(require("lodash.omit")),ii=require("redux-persist");Po();D();ve();Mt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},Fd={network:Mt},si=(0,ri.createSlice)({name:"appConfig",initialState:Fd,reducers:{initializeNetworkConfig:(e,o)=>{let t=ni(o.payload.walletConnectV2RelayAddresses),s=(0,ai.default)(o.payload,"walletConnectV2RelayAddresses");e.network=v(u(u({},e.network),s),{walletConnectV2RelayAddress:t})},updateNetworkConfig:(e,o)=>{e.network=u(u({},e.network),o.payload)},setCustomWalletAddress:(e,o)=>{e.network.customWalletAddress=o.payload}},extraReducers:e=>{e.addCase(R,o=>{o.network.customWalletAddress=void 0}),e.addCase(ii.REHYDRATE,(o,t)=>{var c,p;if(!((p=(c=t.payload)==null?void 0:c.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:s}}=t.payload;o.network.customWalletAddress=s})}}),{initializeNetworkConfig:kS,updateNetworkConfig:LS,setCustomWalletAddress:Ot}=si.actions,An=si.reducer});var ci,Bt,pi,DS,MS,OS,kn,Ln=h(()=>{"use strict";n();ci=require("@reduxjs/toolkit");B();D();Bt={isSigning:!1,signedSessions:{}},pi=(0,ci.createSlice)({name:"signedMessageInfoSliceState",initialState:Bt,reducers:{setSignSession:(e,o)=>{let{sessionId:t,signedSession:s,errorMessage:c}=o.payload;c&&(e.errorMessage=c),e.isSigning=s.status==="pending",e.signedSessions[t]=u(u({},e.signedSessions[t]),s)},setSignSessionState:(e,o)=>u(u({},e),o.payload),clearSignedMessageInfo:()=>Bt},extraReducers:e=>{e.addCase(R,()=>Bt)}}),{setSignSession:DS,clearSignedMessageInfo:MS,setSignSessionState:OS}=pi.actions,kn=pi.reducer});var di,li,mi,_i,Wd,Ud,$S,Gd,Pn,En=h(()=>{"use strict";n();di=require("@reduxjs/toolkit"),li=require("redux-persist");B();fn();D();mi={customToasts:[],transactionToasts:[]},_i=(0,di.createSlice)({name:"toastsSlice",initialState:mi,reducers:{addCustomToast:(e,o)=>{let t=o.payload.toastId||`custom-toast-${e.customToasts.length+1}`,s=e.customToasts.findIndex(c=>c.toastId===t);if(s!==-1){e.customToasts[s]=v(u(u({},e.customToasts[s]),o.payload),{type:"custom",toastId:t});return}e.customToasts.push(v(u({},o.payload),{type:"custom",toastId:t}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(t=>t.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:lo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(t=>t.toastId!==o.payload)}},extraReducers:e=>{e.addCase(R,()=>mi),e.addCase(li.REHYDRATE,(o,t)=>{var c,p;let s=(p=(c=t.customToasts)==null?void 0:c.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=s})}}),{addCustomToast:Wd,removeCustomToast:Ud,addTransactionToast:$S,removeTransactionToast:Gd}=_i.actions,Pn=_i.reducer});var ui,Wt,Ut,Gt,Hd,Ft,gi,zS,$d,Ht,Cn,Nn=h(()=>{"use strict";n();ui=require("@reduxjs/toolkit");D();Wt="Transaction failed",Ut="Transaction successful",Gt="Processing transaction",Hd="Transaction submitted",Ft={},gi=(0,ui.createSlice)({name:"transactionsInfo",initialState:Ft,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:t,transactionsDisplayInfo:s}=o.payload;t!=null&&(e[t]={errorMessage:(s==null?void 0:s.errorMessage)||Wt,successMessage:(s==null?void 0:s.successMessage)||Ut,processingMessage:(s==null?void 0:s.processingMessage)||Gt,submittedMessage:(s==null?void 0:s.submittedMessage)||Hd,transactionDuration:s==null?void 0:s.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>Ft},extraReducers:e=>{e.addCase(R,()=>Ft)}}),{clearTransactionsInfo:zS,setTransactionsDisplayInfo:$d,clearTransactionsInfoForSessionId:Ht}=gi.actions,Cn=gi.reducer});function go(e){return e!=null&&(Qd(e)||rl(e))}function fo(e){return e!=null&&(el(e)||al(e))}function To(e){return e!=null&&(ol(e)||il(e))}function fi(e){return e!=null&&(nl(e)||sl(e))}function $t(e){return e!=null&&tl(e)}function Qd(e){return e!=null&&Vd.includes(e)}function el(e){return e!=null&&qd.includes(e)}function ol(e){return e!=null&&jd.includes(e)}function nl(e){return e!=null&&zd.includes(e)}function tl(e){return e!=null&&Kd.includes(e)}function rl(e){return e!=null&&Xd.includes(e)}function al(e){return e!=null&&Yd.includes(e)}function il(e){return e!=null&&Zd.includes(e)}function sl(e){return e!=null&&Jd.includes(e)}var Vd,qd,jd,zd,Kd,Xd,Yd,Zd,Jd,ho=h(()=>{"use strict";n();C();Vd=["sent"],qd=["success"],jd=["fail","cancelled","timedOut"],zd=["invalid"],Kd=["timedOut"],Xd=["pending"],Yd=["success"],Zd=["fail","invalid"],Jd=["not executed"]});var Ti,hi,Do,cl,xi,yi,Si,pl,Rn,ml,dl,nb,ll,Mo,Vt,tb,Dn,Mn=h(()=>{"use strict";n();Ti=require("@reduxjs/toolkit"),hi=require("redux-persist");C();ho();D();Do={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},cl={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},xi=(0,Ti.createSlice)({name:"transactionsSlice",initialState:Do,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,g;let{sessionId:t,transactions:s,errorMessage:c,status:p,redirectRoute:m,customTransactionInformation:d}=o.payload;e.customTransactionInformationForSessionId[t]=u(u(u({},cl),((l=e.signedTransactions[t])==null?void 0:l.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[t]||{}),e.signedTransactions[t]={transactions:s,status:p,errorMessage:c,redirectRoute:m,customTransactionInformation:u(u({},e.customTransactionInformationForSessionId[t]),d!=null?d:{})},((g=e==null?void 0:e.transactionsToSign)==null?void 0:g.sessionId)===t&&(e.transactionsToSign=Do.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=u(u({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:t,status:s,errorMessage:c,transactions:p}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].status=s,c!=null&&(e.signedTransactions[t].errorMessage=c),p!=null&&(e.signedTransactions[t].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var g,b,T,w,E,L,N,Z;let{sessionId:t,status:s,errorMessage:c,transactionHash:p,serverTransaction:m,inTransit:d}=o.payload,l=(b=(g=e.signedTransactions)==null?void 0:g[t])==null?void 0:b.transactions;if(l!=null){e.signedTransactions[t].transactions=l.map(M=>M.hash===p?v(u(u({},m!=null?m:{}),M),{status:s,errorMessage:c,inTransit:d}):M);let q=(w=(T=e.signedTransactions[t])==null?void 0:T.transactions)==null?void 0:w.every(M=>fo(M.status)),ae=(L=(E=e.signedTransactions[t])==null?void 0:E.transactions)==null?void 0:L.some(M=>To(M.status)),pe=(Z=(N=e.signedTransactions[t])==null?void 0:N.transactions)==null?void 0:Z.every(M=>fi(M.status));q&&(e.signedTransactions[t].status="success"),ae&&(e.signedTransactions[t].status="fail"),pe&&(e.signedTransactions[t].status="invalid")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:t,customTransactionInformation:s}=o.payload;e.customTransactionInformationForSessionId[t]=s,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Do.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Do.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload},updateSignedTransactionsCustomTransactionInformation:(e,o)=>{let{sessionId:t,customTransactionInformationOverrides:s}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].customTransactionInformation=u(u({},e.signedTransactions[t].customTransactionInformation),s))}},extraReducers:e=>{e.addCase(R,()=>Do),e.addCase(hi.REHYDRATE,(o,t)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=t.payload)!=null&&m.transactions))return;let{signedTransactions:s,customTransactionInformationForSessionId:c}=t.payload.transactions,p=Object.entries(s).reduce((d,[l,g])=>{let b=new Date(l),T=new Date;return T.setHours(T.getHours()+5),T-b>0||(d[l]=g),d},{});c!=null&&(o.customTransactionInformationForSessionId=c),s!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:yi,updateSignedTransactions:Si,setTransactionsToSign:pl,clearAllTransactionsToSign:Rn,clearAllSignedTransactions:ml,clearSignedTransaction:dl,clearTransactionToSign:nb,setSignTransactionsError:ll,setSignTransactionsCancelMessage:Mo,moveTransactionsToSignedState:Vt,updateSignedTransactionsCustomTransactionInformation:tb}=xi.actions,Dn=xi.reducer});var bi,qt,wi,sb,cb,_l,pb,On,Bn=h(()=>{"use strict";n();bi=require("@reduxjs/toolkit");D();qt={},wi=(0,bi.createSlice)({name:"batchTransactionsSlice",initialState:qt,reducers:{setBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},updateBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},clearBatchTransactions:(e,o)=>{delete e[o.payload.batchId]},clearAllBatchTransactions:()=>qt},extraReducers:e=>{e.addCase(R,()=>qt)}}),{setBatchTransactions:sb,updateBatchTransactions:cb,clearBatchTransactions:_l,clearAllBatchTransactions:pb}=wi.actions,On=wi.reducer});var Ai,vi,Ii,_b,ki,jt=h(()=>{"use strict";n();Ai=require("@reduxjs/toolkit");D();vi={},Ii=(0,Ai.createSlice)({name:"dappConfigSlice",initialState:vi,reducers:{setDappConfig:(e,o)=>o.payload},extraReducers:e=>{e.addCase(R,()=>vi)}}),{setDappConfig:_b}=Ii.actions,ki=Ii.reducer});var P=h(()=>{"use strict";n();xn();Sn();wn();In();Ln();En();Nn();Mn();Bn();jt()});var zt=h(()=>{"use strict";n()});var Ei,Tl,hl,ze,Un=h(()=>{"use strict";n();Ei=require("@reduxjs/toolkit");zt();xn();Bn();jt();Sn();wn();In();Ln();En();Nn();Mn();Tl={["account"]:hn,["dappConfig"]:ki,["loginInfo"]:yn,["modals"]:bn,["networkConfig"]:An,["signedMessageInfo"]:kn,["toasts"]:Pn,["transactionsInfo"]:Cn,["transactions"]:Dn,["batchTransactions"]:On},hl=(e={})=>(0,Ei.combineReducers)(u(u({},Tl),e)),ze=hl});var Di={};ie(Di,{default:()=>Dl,sessionStorageReducers:()=>Kt});function Ae(e,o=[]){return{key:e,version:1,storage:Ni.default,blacklist:o}}var Y,Ci,Ni,xl,Oo,yl,Sl,bl,wl,vl,Al,Il,kl,Ll,Pl,Ri,El,Kt,Cl,Nl,Rl,Dl,Mi=h(()=>{"use strict";n();Y=require("redux-persist"),Ci=x(require("redux-persist/lib/storage")),Ni=x(require("redux-persist/lib/storage/session"));Un();P();xn();Bn();Sn();wn();In();Ln();En();Nn();Mn();zt();xl={persistReducersStorageType:"localStorage"},Oo={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},yl=Ae(Oo["account"]),Sl=Ae(Oo["loginInfo"]),bl=Ae(Oo["modals"]),wl=Ae(Oo["networkConfig"]),vl={2:e=>v(u({},e),{networkConfig:Mt})};Al=Ae("sdk-dapp-transactionsInfo"),Il=Ae("sdk-dapp-transactions",["transactionsToSign"]),kl=Ae("sdk-dapp-batchTransactions",["batchTransactions"]),Ll=Ae("sdk-dapp-toasts"),Pl=Ae("sdk-dapp-signedMessageInfo"),Ri={key:"sdk-dapp-store",version:2,storage:Ci.default,whitelist:Object.keys(Oo),migrate:(0,Y.createMigrate)(vl,{debug:!1})},El=v(u({},Ri),{whitelist:[]}),Kt={["toasts"]:(0,Y.persistReducer)(Ll,Pn),["transactions"]:(0,Y.persistReducer)(Il,Dn),["transactionsInfo"]:(0,Y.persistReducer)(Al,Cn),["batchTransactions"]:(0,Y.persistReducer)(kl,On),["signedMessageInfo"]:(0,Y.persistReducer)(Pl,kn)},Cl=v(u({},Kt),{["account"]:(0,Y.persistReducer)(yl,hn),["loginInfo"]:(0,Y.persistReducer)(Sl,yn),["modals"]:(0,Y.persistReducer)(bl,bn),["networkConfig"]:(0,Y.persistReducer)(wl,An)}),Nl=xl.persistReducersStorageType==="localStorage",Rl=Nl?(0,Y.persistReducer)(Ri,ze(Kt)):(0,Y.persistReducer)(El,ze(Cl)),Dl=Rl});var Oi={};ie(Oi,{default:()=>Ml});var Ml,Bi=h(()=>{"use strict";n();Un();Ml=ze()});var Fi={};ie(Fi,{default:()=>Bl});var te,Ol,Bl,Wi=h(()=>{"use strict";n();te=require("redux-persist"),Ol=[te.FLUSH,te.REHYDRATE,te.PAUSE,te.PERSIST,te.PURGE,te.REGISTER],Bl=Ol});var Hi={};ie(Hi,{default:()=>Gi});function Gi(e){return(0,Ui.persistStore)(e)}var Ui,$i=h(()=>{"use strict";n();Ui=require("redux-persist")});var Os=S((l2,Ms)=>{n();var Ou=typeof a=="object"&&a&&a.Object===Object&&a;Ms.exports=Ou});var qo=S((_2,Bs)=>{n();var Bu=Os(),Fu=typeof self=="object"&&self&&self.Object===Object&&self,Wu=Bu||Fu||Function("return this")();Bs.exports=Wu});var dr=S((u2,Fs)=>{n();var Uu=qo(),Gu=Uu.Symbol;Fs.exports=Gu});var Hs=S((g2,Gs)=>{n();var Ws=dr(),Us=Object.prototype,Hu=Us.hasOwnProperty,$u=Us.toString,jo=Ws?Ws.toStringTag:void 0;function Vu(e){var o=Hu.call(e,jo),t=e[jo];try{e[jo]=void 0;var s=!0}catch(p){}var c=$u.call(e);return s&&(o?e[jo]=t:delete e[jo]),c}Gs.exports=Vu});var Vs=S((f2,$s)=>{n();var qu=Object.prototype,ju=qu.toString;function zu(e){return ju.call(e)}$s.exports=zu});var lr=S((T2,zs)=>{n();var qs=dr(),Ku=Hs(),Xu=Vs(),Yu="[object Null]",Zu="[object Undefined]",js=qs?qs.toStringTag:void 0;function Ju(e){return e==null?e===void 0?Zu:Yu:js&&js in Object(e)?Ku(e):Xu(e)}zs.exports=Ju});var Xs=S((h2,Ks)=>{n();var Qu=Array.isArray;Ks.exports=Qu});var Zs=S((x2,Ys)=>{n();function eg(e){return e!=null&&typeof e=="object"}Ys.exports=eg});var Qs=S((y2,Js)=>{n();var og=lr(),ng=Xs(),tg=Zs(),rg="[object String]";function ag(e){return typeof e=="string"||!ng(e)&&tg(e)&&og(e)==rg}Js.exports=ag});var fr={};ie(fr,{css:()=>mc,default:()=>lg});var mc,lg,Tr=h(()=>{"use strict";n();mc=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(mc));lg={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var xr={};ie(xr,{css:()=>_c,default:()=>gg});var _c,gg,yr=h(()=>{"use strict";n();_c=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(_c));gg={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var nt=S((H3,Tc)=>{n();function Eg(e){var o=typeof e;return e!=null&&(o=="object"||o=="function")}Tc.exports=Eg});var xc=S(($3,hc)=>{n();var Cg=lr(),Ng=nt(),Rg="[object AsyncFunction]",Dg="[object Function]",Mg="[object GeneratorFunction]",Og="[object Proxy]";function Bg(e){if(!Ng(e))return!1;var o=Cg(e);return o==Dg||o==Mg||o==Rg||o==Og}hc.exports=Bg});var Sc=S((V3,yc)=>{n();var Fg=qo(),Wg=Fg["__core-js_shared__"];yc.exports=Wg});var vc=S((q3,wc)=>{n();var Sr=Sc(),bc=function(){var e=/[^.]+$/.exec(Sr&&Sr.keys&&Sr.keys.IE_PROTO||"");return e?"Symbol(src)_1."+e:""}();function Ug(e){return!!bc&&bc in e}wc.exports=Ug});var Ic=S((j3,Ac)=>{n();var Gg=Function.prototype,Hg=Gg.toString;function $g(e){if(e!=null){try{return Hg.call(e)}catch(o){}try{return e+""}catch(o){}}return""}Ac.exports=$g});var Lc=S((z3,kc)=>{n();var Vg=xc(),qg=vc(),jg=nt(),zg=Ic(),Kg=/[\\^$.*+?()[\]{}|]/g,Xg=/^\[object .+?Constructor\]$/,Yg=Function.prototype,Zg=Object.prototype,Jg=Yg.toString,Qg=Zg.hasOwnProperty,ef=RegExp("^"+Jg.call(Qg).replace(Kg,"\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g,"$1.*?")+"$");function of(e){if(!jg(e)||qg(e))return!1;var o=Vg(e)?ef:Xg;return o.test(zg(e))}kc.exports=of});var Ec=S((K3,Pc)=>{n();function nf(e,o){return e==null?void 0:e[o]}Pc.exports=nf});var tt=S((X3,Cc)=>{n();var tf=Lc(),rf=Ec();function af(e,o){var t=rf(e,o);return tf(t)?t:void 0}Cc.exports=af});var Zo=S((Y3,Nc)=>{n();var sf=tt(),cf=sf(Object,"create");Nc.exports=cf});var Mc=S((Z3,Dc)=>{n();var Rc=Zo();function pf(){this.__data__=Rc?Rc(null):{},this.size=0}Dc.exports=pf});var Bc=S((J3,Oc)=>{n();function mf(e){var o=this.has(e)&&delete this.__data__[e];return this.size-=o?1:0,o}Oc.exports=mf});var Wc=S((Q3,Fc)=>{n();var df=Zo(),lf="__lodash_hash_undefined__",_f=Object.prototype,uf=_f.hasOwnProperty;function gf(e){var o=this.__data__;if(df){var t=o[e];return t===lf?void 0:t}return uf.call(o,e)?o[e]:void 0}Fc.exports=gf});var Gc=S((eO,Uc)=>{n();var ff=Zo(),Tf=Object.prototype,hf=Tf.hasOwnProperty;function xf(e){var o=this.__data__;return ff?o[e]!==void 0:hf.call(o,e)}Uc.exports=xf});var $c=S((oO,Hc)=>{n();var yf=Zo(),Sf="__lodash_hash_undefined__";function bf(e,o){var t=this.__data__;return this.size+=this.has(e)?0:1,t[e]=yf&&o===void 0?Sf:o,this}Hc.exports=bf});var qc=S((nO,Vc)=>{n();var wf=Mc(),vf=Bc(),Af=Wc(),If=Gc(),kf=$c();function vo(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}vo.prototype.clear=wf;vo.prototype.delete=vf;vo.prototype.get=Af;vo.prototype.has=If;vo.prototype.set=kf;Vc.exports=vo});var zc=S((tO,jc)=>{n();function Lf(){this.__data__=[],this.size=0}jc.exports=Lf});var Xc=S((rO,Kc)=>{n();function Pf(e,o){return e===o||e!==e&&o!==o}Kc.exports=Pf});var Jo=S((aO,Yc)=>{n();var Ef=Xc();function Cf(e,o){for(var t=e.length;t--;)if(Ef(e[t][0],o))return t;return-1}Yc.exports=Cf});var Jc=S((iO,Zc)=>{n();var Nf=Jo(),Rf=Array.prototype,Df=Rf.splice;function Mf(e){var o=this.__data__,t=Nf(o,e);if(t<0)return!1;var s=o.length-1;return t==s?o.pop():Df.call(o,t,1),--this.size,!0}Zc.exports=Mf});var ep=S((sO,Qc)=>{n();var Of=Jo();function Bf(e){var o=this.__data__,t=Of(o,e);return t<0?void 0:o[t][1]}Qc.exports=Bf});var np=S((cO,op)=>{n();var Ff=Jo();function Wf(e){return Ff(this.__data__,e)>-1}op.exports=Wf});var rp=S((pO,tp)=>{n();var Uf=Jo();function Gf(e,o){var t=this.__data__,s=Uf(t,e);return s<0?(++this.size,t.push([e,o])):t[s][1]=o,this}tp.exports=Gf});var ip=S((mO,ap)=>{n();var Hf=zc(),$f=Jc(),Vf=ep(),qf=np(),jf=rp();function Ao(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Ao.prototype.clear=Hf;Ao.prototype.delete=$f;Ao.prototype.get=Vf;Ao.prototype.has=qf;Ao.prototype.set=jf;ap.exports=Ao});var cp=S((dO,sp)=>{n();var zf=tt(),Kf=qo(),Xf=zf(Kf,"Map");sp.exports=Xf});var dp=S((lO,mp)=>{n();var pp=qc(),Yf=ip(),Zf=cp();function Jf(){this.size=0,this.__data__={hash:new pp,map:new(Zf||Yf),string:new pp}}mp.exports=Jf});var _p=S((_O,lp)=>{n();function Qf(e){var o=typeof e;return o=="string"||o=="number"||o=="symbol"||o=="boolean"?e!=="__proto__":e===null}lp.exports=Qf});var Qo=S((uO,up)=>{n();var eT=_p();function oT(e,o){var t=e.__data__;return eT(o)?t[typeof o=="string"?"string":"hash"]:t.map}up.exports=oT});var fp=S((gO,gp)=>{n();var nT=Qo();function tT(e){var o=nT(this,e).delete(e);return this.size-=o?1:0,o}gp.exports=tT});var hp=S((fO,Tp)=>{n();var rT=Qo();function aT(e){return rT(this,e).get(e)}Tp.exports=aT});var yp=S((TO,xp)=>{n();var iT=Qo();function sT(e){return iT(this,e).has(e)}xp.exports=sT});var bp=S((hO,Sp)=>{n();var cT=Qo();function pT(e,o){var t=cT(this,e),s=t.size;return t.set(e,o),this.size+=t.size==s?0:1,this}Sp.exports=pT});var vp=S((xO,wp)=>{n();var mT=dp(),dT=fp(),lT=hp(),_T=yp(),uT=bp();function Io(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Io.prototype.clear=mT;Io.prototype.delete=dT;Io.prototype.get=lT;Io.prototype.has=_T;Io.prototype.set=uT;wp.exports=Io});var Ip=S((yO,Ap)=>{n();var gT="__lodash_hash_undefined__";function fT(e){return this.__data__.set(e,gT),this}Ap.exports=fT});var Lp=S((SO,kp)=>{n();function TT(e){return this.__data__.has(e)}kp.exports=TT});var Ep=S((bO,Pp)=>{n();var hT=vp(),xT=Ip(),yT=Lp();function rt(e){var o=-1,t=e==null?0:e.length;for(this.__data__=new hT;++o<t;)this.add(e[o])}rt.prototype.add=rt.prototype.push=xT;rt.prototype.has=yT;Pp.exports=rt});var Np=S((wO,Cp)=>{n();function ST(e,o,t,s){for(var c=e.length,p=t+(s?1:-1);s?p--:++p<c;)if(o(e[p],p,e))return p;return-1}Cp.exports=ST});var Dp=S((vO,Rp)=>{n();function bT(e){return e!==e}Rp.exports=bT});var Op=S((AO,Mp)=>{n();function wT(e,o,t){for(var s=t-1,c=e.length;++s<c;)if(e[s]===o)return s;return-1}Mp.exports=wT});var Fp=S((IO,Bp)=>{n();var vT=Np(),AT=Dp(),IT=Op();function kT(e,o,t){return o===o?IT(e,o,t):vT(e,AT,t)}Bp.exports=kT});var Up=S((kO,Wp)=>{n();var LT=Fp();function PT(e,o){var t=e==null?0:e.length;return!!t&&LT(e,o,0)>-1}Wp.exports=PT});var Hp=S((LO,Gp)=>{n();function ET(e,o,t){for(var s=-1,c=e==null?0:e.length;++s<c;)if(t(o,e[s]))return!0;return!1}Gp.exports=ET});var Vp=S((PO,$p)=>{n();function CT(e,o){return e.has(o)}$p.exports=CT});var jp=S((EO,qp)=>{n();var NT=tt(),RT=qo(),DT=NT(RT,"Set");qp.exports=DT});var Kp=S((CO,zp)=>{n();function MT(){}zp.exports=MT});var br=S((NO,Xp)=>{n();function OT(e){var o=-1,t=Array(e.size);return e.forEach(function(s){t[++o]=s}),t}Xp.exports=OT});var Zp=S((RO,Yp)=>{n();var wr=jp(),BT=Kp(),FT=br(),WT=1/0,UT=wr&&1/FT(new wr([,-0]))[1]==WT?function(e){return new wr(e)}:BT;Yp.exports=UT});var Qp=S((DO,Jp)=>{n();var GT=Ep(),HT=Up(),$T=Hp(),VT=Vp(),qT=Zp(),jT=br(),zT=200;function KT(e,o,t){var s=-1,c=HT,p=e.length,m=!0,d=[],l=d;if(t)m=!1,c=$T;else if(p>=zT){var g=o?null:qT(e);if(g)return jT(g);m=!1,c=VT,l=new GT}else l=o?[]:d;e:for(;++s<p;){var b=e[s],T=o?o(b):b;if(b=t||b!==0?b:0,m&&T===T){for(var w=l.length;w--;)if(l[w]===T)continue e;o&&l.push(T),d.push(b)}else c(l,T,t)||(l!==d&&l.push(T),d.push(b))}return d}Jp.exports=KT});var om=S((MO,em)=>{n();var XT=Qp();function YT(e){return e&&e.length?XT(e):[]}em.exports=YT});var Ce={};ie(Ce,{css:()=>gm,default:()=>Wh});var gm,Wh,Ne=h(()=>{"use strict";n();gm=`.dapp-core-component__dappModalStyles__dappModal {
  position: fixed;
  z-index: 999;
  padding-top: 100px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
}

.dapp-core-component__dappModalStyles__dappModalContent {
  background: #fff;
  border: 2px solid #aaa;
  border-radius: 5px;
  z-index: 999;
  max-width: 500px;
  margin: auto;
  padding: 0;
  position: relative;
  width: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.dapp-core-component__dappModalStyles__dappModalHeader {
  padding: 2px 16px;
  background-color: #fff;
  font-size: 1.25rem;
}

.dapp-core-component__dappModalStyles__dappModalHeaderText {
  padding: 0 1rem;
}

.dapp-core-component__dappModalStyles__dappModalBody {
  padding: 16px 0;
}

.dapp-core-component__dappModalStyles__dappModalFooter {
  background-color: #fff;
  padding: 1rem 0.5rem;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton {
  position: absolute;
  top: 0;
  right: 0;
  float: right;
  font-weight: bold;
  border-radius: 5px;
  font-size: 1rem;
  padding: 0.5rem 1rem;
  border: 0;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton:hover,
.dapp-core-component__dappModalStyles__dappModalCloseButton:focus {
  text-decoration: none;
  cursor: pointer;
  color: #212529;
  background-color: #dee2e6;
  border-color: #e9ecef;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(gm));Wh={dappModal:"dapp-core-component__dappModalStyles__dappModal",dappModalContent:"dapp-core-component__dappModalStyles__dappModalContent",dappModalHeader:"dapp-core-component__dappModalStyles__dappModalHeader",dappModalHeaderText:"dapp-core-component__dappModalStyles__dappModalHeaderText",dappModalBody:"dapp-core-component__dappModalStyles__dappModalBody",dappModalFooter:"dapp-core-component__dappModalStyles__dappModalFooter",dappModalCloseButton:"dapp-core-component__dappModalStyles__dappModalCloseButton"}});var Br={};ie(Br,{css:()=>km,default:()=>rx});var km,rx,Fr=h(()=>{"use strict";n();km=`.dapp-core-component__loginButtonStyles__default-login-button {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__wrapper {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__login-text {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-wrapper {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-content {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-title {
  color: inherit;
}

.dapp-core-component__loginButtonStyles__button-icon {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(km));rx={"default-login-button":"dapp-core-component__loginButtonStyles__default-login-button",defaultLoginButton:"dapp-core-component__loginButtonStyles__default-login-button",wrapper:"dapp-core-component__loginButtonStyles__wrapper","login-text":"dapp-core-component__loginButtonStyles__login-text",loginText:"dapp-core-component__loginButtonStyles__login-text","button-wrapper":"dapp-core-component__loginButtonStyles__button-wrapper",buttonWrapper:"dapp-core-component__loginButtonStyles__button-wrapper","button-content":"dapp-core-component__loginButtonStyles__button-content",buttonContent:"dapp-core-component__loginButtonStyles__button-content","button-title":"dapp-core-component__loginButtonStyles__button-title",buttonTitle:"dapp-core-component__loginButtonStyles__button-title","button-icon":"dapp-core-component__loginButtonStyles__button-icon",buttonIcon:"dapp-core-component__loginButtonStyles__button-icon"}});var sx={};ie(sx,{XaliasCrossWindowLoginButton:()=>ix});module.exports=ee(sx);n();var Em=x(require("react"));yt();n();n();n();var Xi=require("react"),xo=require("react-redux");n();var Xt=require("@reduxjs/toolkit"),zi=require("react-redux/lib/utils/Subscription");P();n();var Li=x(require("lodash.throttle"));k();P();Rt();qe();Ve();var ul=[gn],Fn=!1,gl=(0,Li.default)(()=>{Co(Eo())},5e3),Pi=e=>o=>t=>{if(ul.includes(t.type))return o(t);let s=e.getState(),c=Me.local.getItem(we.loginExpiresAt);if(!Boolean(s==null?void 0:s.account.address))return o(t);if(c==null)return Co(Eo());let m=Date.now();return c-m<0&&!Fn?setTimeout(()=>{Fn=!0,e.dispatch(za())},1e3):(Fn&&(Fn=!1),gl()),o(t)};n();n();function Wn(){return typeof sessionStorage!="undefined"}var Vi=Wn()?(Mi(),ee(Di)).default:(Bi(),ee(Oi)).default,qi=Wn()?(Wi(),ee(Fi)).default:[],ji=Wn()?($i(),ee(Hi)).default:e=>e;Un();var A=(0,Xt.configureStore)({reducer:Vi,middleware:e=>e({serializableCheck:{ignoredActions:[...qi,Nt.type,J.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Pi)}),Ki=(0,zi.createSubscription)(A),d0=ji(A),l0=(0,Xt.configureStore)({reducer:ze});var Fl={store:A,subscription:Ki},Yt=(0,Xi.createContext)(Fl),f0=(0,xo.createStoreHook)(Yt),G=(0,xo.createDispatchHook)(Yt),I=(0,xo.createSelectorHook)(Yt);n();n();P();n();var Yi=x(require("lodash.isequal")),Gn=require("reselect"),y=(0,Gn.createSelectorCreator)(Gn.defaultMemoize,Yi.default);var Te=e=>e.account,Ke=y(Te,e=>e.address),Ie=y(Te,Ke,(e,o)=>o in e.accounts?e.accounts[o]:X),Wl=y(Te,Ie,(e,o)=>{let c=e,{accounts:t}=c,s=Re(c,["accounts"]);return v(u({},s),{address:o.address,account:o})}),S0=y(Ie,e=>e.balance),Zi=y(Ie,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),b0=y(Te,e=>e.shard),Ul=y(Te,e=>e.ledgerAccount),w0=y(Te,e=>e.walletConnectAccount),v0=y(Te,e=>e.isAccountLoading),A0=y(Te,e=>e.accountLoadingError),Gl=y(Te,e=>e.websocketEvent),Hl=y(Te,e=>e.websocketBatchEvent);n();var Ji=e=>e.dappConfig,L0=y(Ji,e=>e.shouldUseWebViewProvider);n();var Q=e=>e.loginInfo,$l=y(Q,e=>e.loginMethod),Bo=y(Q,Ke,(e,o)=>Boolean(o)),N0=y(Q,e=>e.walletConnectLogin),Vl=y(Q,e=>e.ledgerLogin),ql=y(Q,e=>e.walletLogin),R0=y(Q,e=>e.isLoginSessionInvalid),Fo=y(Q,e=>e.tokenLogin),Qi=y(Q,e=>e.logoutRoute),jl=y(Q,e=>e.isWalletConnectV2Initialized);n();var es=e=>e.modals,O0=y(es,e=>e.txSubmittedModal),zl=y(es,e=>e.notificationModal);n();var he=e=>e.networkConfig,Hn=y(he,e=>e.network.chainId),Kl=y(he,e=>e.network.roundDuration),W0=y(he,e=>e.network.walletConnectBridgeAddress),Xl=y(he,e=>e.network.walletConnectV2RelayAddress),Yl=y(he,e=>e.network.walletConnectV2ProjectId),Zl=y(he,e=>e.network.walletConnectV2Options),Jl=y(he,e=>e.network.walletConnectDeepLink),V=y(he,e=>e.network),Zt=y(V,e=>e.apiAddress),os=y(V,e=>e.explorerAddress),ns=y(V,e=>{var o;return(o=e.customWalletAddress)!=null?o:e.walletAddress}),Jt=y(V,e=>e.xAliasAddress),ts=y(V,e=>e.egldLabel);n();var $n=e=>e.signedMessageInfo,H0=y($n,e=>e.isSigning),$0=y($n,e=>e.errorMessage),V0=y($n,e=>{let o=Object.keys(e.signedSessions),t=o.length;return e.signedSessions[o[t-1]]}),q0=y($n,e=>{let o=Object.keys(e.signedSessions),t=o.length;return o.length>0?o[t-1]:""});n();var rs=e=>e.toasts,Ql=y(rs,e=>e.customToasts),as=y(rs,e=>e.transactionToasts);n();P();var is={errorMessage:Wt,successMessage:Ut,processingMessage:Gt},ss=e=>e.transactionsInfo,e_=y(ss,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||is);n();n();var Ye=require("@multiversx/sdk-core");k();n();n();n();function Vn(e){try{let o=atob(e),t=btoa(o),s=r.Buffer.from(e,"base64").toString(),c=r.Buffer.from(s).toString("base64"),p=e===t||t.startsWith(e),m=e===c||c.startsWith(e);if(p&&m)return!0}catch(o){return!1}return!1}function Xe(e){return Vn(e)?atob(e):e}n();n();n();n();var cs=e=>{let o=e!=null?e:"";return Vn(o)?r.Buffer.from(r.Buffer.from(o,"base64").toString().trim()):r.Buffer.from(o.trim())};n();k();var Wo=({data:e,onlySetGuardian:o})=>e?o?e.startsWith("SetGuardian"):Object.values(St).some(t=>e.startsWith(t)):!1;function qn(e){var s,c,p;let o=u({},e);Wo({data:o.data,onlySetGuardian:!0})&&(delete o.senderUsername,delete o.receiverUsername);let t=new Ye.Transaction(u(u(v(u(v(u(u({value:BigInt(o.value.valueOf()),data:cs(o.data),nonce:BigInt(o.nonce.valueOf()),receiver:new Ye.Address(o.receiver)},o.receiverUsername?{receiverUsername:o.receiverUsername}:{}),o.relayer?{relayer:new Ye.Address(o.relayer)}:{}),{sender:new Ye.Address(o.sender)}),o.senderUsername?{senderUsername:o.senderUsername}:{}),{gasLimit:BigInt((s=o.gasLimit.valueOf())!=null?s:_n),gasPrice:BigInt((c=o.gasPrice.valueOf())!=null?c:un),chainID:o.chainID.valueOf(),version:(p=o.version)!=null?p:Na}),o.options?{options:o.options}:{}),o.guardian?{guardian:new Ye.Address(o.guardian)}:{}));return o.relayerSignature&&(t.relayerSignature=r.Buffer.from(o.relayerSignature,"hex")),o.guardianSignature&&(t.guardianSignature=r.Buffer.from(o.guardianSignature,"hex")),o.signature&&(t.signature=r.Buffer.from(o.signature,"hex")),t}n();n();k();n();n();n();var r_=require("@multiversx/sdk-core");n();k();n();var i_=require("@multiversx/sdk-core");B();n();var a_=require("@multiversx/sdk-core");n();n();n();var s_=require("@multiversx/sdk-core"),c_=x(require("bignumber.js"));C();n();n();n();var Uo="accounts";var er="blocks";n();n();n();n();z();n();n();z();n();n();n();var f_=x(require("bignumber.js"));k();n();var g_=require("@multiversx/sdk-core"),or=x(require("bignumber.js"));k();n();var __=x(require("bignumber.js"));n();or.default.config({ROUNDING_MODE:or.default.ROUND_FLOOR});n();n();n();k();n();n();n();n();k();n();k();n();var T_=require("@multiversx/sdk-core"),h_=x(require("bignumber.js"));k();n();var tr=require("@multiversx/sdk-core"),S_=x(require("bignumber.js"));k();n();n();var x_=x(require("bignumber.js"));k();n();k();n();n();n();n();n();n();k();n();k();n();k();n();z();var w_=["reDelegateRewards","claimRewards","unBond"],v_=["wrapEgld","unwrapEgld"],A_=["unStake"],I_=["unDelegate"];n();n();n();k();n();var E_=x(require("bignumber.js"));n();n();z();n();var N_=x(require("bignumber.js"));n();n();n();n();var M_=x(require("bignumber.js"));B();n();n();var O_=require("@multiversx/sdk-core");n();n();C();n();var U_=require("@multiversx/sdk-web-wallet-provider");k();n();var F_=x(require("qs"));n();$();_o();n();_o();var kk={search:je()?window.location.search:"",removeParams:[]};n();n();n();H();n();z();n();n();H();n();var G_=x(require("linkifyjs"));n();k();n();var H_=x(require("bignumber.js"));n();B();n();n();C();n();C();n();n();n();z();n();z();n();var $_=x(require("bignumber.js"));k();z();n();z();n();var us=require("react");B();n();n();z();n();n();var V_=require("@multiversx/sdk-core"),q_=x(require("bignumber.js"));z();n();B();n();n();B();var hP=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];n();var gs=require("react");B();H();n();var z_=require("react");z();var IP=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];ho();n();var Kn=e=>e.transactions,yo=y(Kn,e=>e.signedTransactions),K_=y(Kn,e=>e.signTransactionsError),fs=y(Kn,e=>e.signTransactionsCancelMessage),Xn=e=>o=>Object.entries(o).reduce((t,[s,c])=>(e(c.status)&&(t[s]=c),t),{}),Ts=y(yo,Xn(go)),hs=y(yo,Xn(fo)),xs=y(yo,Xn(To)),X_=y(yo,Xn($t)),ys=y(Kn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:v(u({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(t=>qn(t)))||[]})}),Y_=y(yo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});n();var Ss=require("react");n();P();n();n();n();n();n();n();n();n();var J_=require("@multiversx/sdk-extension-provider"),Q_=require("@multiversx/sdk-hw-provider"),eu=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),ou=require("@multiversx/sdk-opera-provider"),nu=require("@multiversx/sdk-passkey-provider/out"),tu=require("@multiversx/sdk-web-wallet-provider");k();De();n();var Ze=require("@multiversx/sdk-web-wallet-iframe-provider/out"),bs=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");C();n();var ce=e=>`Unable to perform ${e}, Provider not initialized`,Yn=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(ce("getAccount"))}setAccount(o){throw new Error(ce(`setAccount: ${o}`))}login(o){throw new Error(ce(`login with options: ${o}`))}logout(o){throw new Error(ce(`logout with options: ${o}`))}getAddress(){throw new Error(ce("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(o,t){throw new Error(ce(`sendTransaction with transactions: ${o} options: ${t}`))}signTransaction(o,t){throw new Error(ce(`signTransaction with transactions: ${o} options: ${t}`))}signTransactions(o,t){throw new Error(ce(`signTransactions with transactions: ${o} options: ${t}`))}signMessage(o,t){throw new Error(ce(`signTransactions with ${o} and options ${t}`))}sendCustomMessage({method:o,params:t}){throw new Error(ce(`sendCustomMessage with method: ${o} params: ${t}`))}sendCustomRequest(o){throw new Error(ce(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},ir=new Yn;var ru=ir;function Le(e){ru=e}P();n();n();n();n();n();var ws=require("@lifeomic/axios-fetch"),sr=x(require("axios")),cr=(0,ws.buildAxiosFetch)(sr.default),pr=(e,o)=>f(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let s=e.clone().json(),[c]=yield Promise.all([s]);return{data:c,status:e.status,statusText:e.statusText,headers:e.headers,config:o}});function au(e,o,t){return f(this,null,function*(){try{let s=yield cr(e,u({method:"POST",body:o?JSON.stringify(o):void 0,headers:u({"Content-Type":"application/json"},(t==null?void 0:t.headers)||{})},t));return pr(s,t)}catch(s){throw console.error("Fetch Error:",s),s}})}function iu(e,o){return f(this,null,function*(){try{let t=yield cr(e,o);if(!t.ok)throw new Error(`HTTP error! Status: ${t.status}`);return pr(t,o)}catch(t){throw console.error("Fetch Error:",t),t}})}function su(e,o,t){return f(this,null,function*(){try{let s=yield cr(e,u({method:"PATCH",body:o?JSON.stringify(o):void 0,headers:(t==null?void 0:t.headers)||{}},t));if(!s.ok)throw new Error(`HTTP error! Status: ${s.status}`);return pr(s,t)}catch(s){throw console.error("Fetch Error:",s),s}})}var Je=sr.default.create();Je.get=iu;Je.post=au;Je.patch=su;n();var So=e=>{let o=e!=null?e:Zt(A.getState());return o.endsWith("/")?o.slice(0,-1):o};n();var cu=x(require("axios"));n();var pu=x(require("swr"));var mu=(e,o)=>{var c;let s=`${(c=o==null?void 0:o.apiAddress)!=null?c:So()}/${Uo}/${e}?withGuardianInfo=true`;return Je.get(s)},vs=(e,o)=>f(void 0,null,function*(){if(!e)return null;try{let{data:t}=yield mu(e,o);return t}catch(t){console.error("error fetching configuration for ",e,t)}return null});var bo=(e,o)=>vs(e,o);n();C();n();n();n();H();n();function Vo(e){let o=A.getState(),t=Zi(o);return e?t&&!isNaN(t)?Math.max(t,e.nonce):e.nonce:t}ho();n();n();n();var du=x(require("axios"));n();var _u=x(require("axios"));co();n();k();n();var uu=x(require("axios"));n();var fu=x(require("axios"));n();n();var Tu=x(require("axios"));n();var hu=x(require("axios"));n();n();P();C();n();n();n();n();H();n();D();B();n();P();n();var Ls=require("@multiversx/sdk-core");B();$();n();ho();n();P();C();n();P();B();n();n();n();B();n();Po();n();n();n();n();var Su=x(require("swr"));n();n();n();n();var Cs=require("react");n();n();var bu=require("react"),wu=require("@multiversx/sdk-web-wallet-provider"),vu=require("@multiversx/sdk-web-wallet-provider"),Au=x(require("qs"));k();P();C();wt();H();var cD=F();n();var fc=require("react"),ot=require("@multiversx/sdk-core");n();k();Dt();n();n();n();n();n();n();n();n();n();n();co();n();var Pu=x(require("axios"));n();var Cu=x(require("axios"));n();var Nu=x(require("axios"));n();co();n();co();n();n();n();n();var Du=require("@multiversx/sdk-opera-provider");n();var Mu=require("@multiversx/sdk-extension-provider");n();De();n();n();n();n();var ec=x(Qs());var Qe=e=>{if(!e||!(0,ec.default)(e))return null;let o=e.split(".");if(o.length!==4)return null;try{let[t,s,c,p]=o,m=JSON.parse(Xe(p)),d=Xe(t);return{ttl:Number(c),extraInfo:m,origin:d,blockHash:s}}catch(t){return console.error(`Error trying to decode ${e}:`,t),null}};n();var ic=require("@multiversx/sdk-native-auth-client");n();var tc=x(require("axios"));n();n();n();function oc(e){return f(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}n();var nc=(e,o,t,s=0)=>f(void 0,null,function*(){try{return yield e(...t)}catch(c){return s<o.retries?((o==null?void 0:o.delay)!=null&&(yield oc(o.delay)),yield nc(e,o,t,s+1)):null}}),_r=(e,o={retries:5,delay:500})=>(...t)=>f(void 0,null,function*(){return yield nc(e,o,t)});var ig=4,sg=3e4,Qn={current:null},eo={current:null},cg=_r((e,o,t)=>f(void 0,null,function*(){if(t){let p=Math.floor(Date.now()/1e3);return{hash:yield t(),timestamp:p}}let{data:s}=yield tc.default.get(`${e}/${er}?from=${ig}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[c]=s;return c}));function rc(e,o,t,s){return f(this,null,function*(){if(e==null)throw new Error("missing api url");let c=Date.now();if(Qn.current!=null&&c<Qn.current.timestamp*1e3+sg&&!s)return Qn.current;if(eo.current!=null)return yield eo.current;eo.current=cg(e,o,t);try{let p=yield eo.current;if(p==null)throw eo.current=null,new Error("could not get block hash");return Qn.current={hash:p.hash,timestamp:p.timestamp},eo.current=null,p}catch(p){return eo.current=null,null}})}n();n();fn();n();function pg(e){return Object.prototype.toString.call(e)==="[object String]"}var ur=e=>{var t;if(!e||!pg(e))return null;let o=e.split(".");if(o.length!==3)return console.error("Invalid nativeAuthToken. You may be trying to decode a loginToken. Try using decodeLoginToken method instead"),null;try{let[s,c,p]=o,m=Xe(s),d=Xe(c),l=Qe(d);if(!l)return{address:m,body:d,signature:p,blockHash:"",origin:"",ttl:0};let g=v(u({},l),{address:m,body:d,signature:p});return(t=l.extraInfo)!=null&&t.timestamp||delete g.extraInfo,g}catch(s){return null}};var gr={isExpired:!1},ac=e=>{if(!e)return gr;let o=ur(e);if(!o)return gr;let t=lo(),{ttl:s,extraInfo:c}=o,p=c==null?void 0:c.timestamp;if(!p)return gr;let m=p+s,d=t>m,l=m-t;return{isExpired:d,expiresAt:m,secondsUntilExpires:l}};n();H();var zo={origin:F().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60},Ko=e=>{var t,s,c,p,m,d;return e===!0?zo:{origin:(t=e==null?void 0:e.origin)!=null?t:zo.origin,blockHashShard:e==null?void 0:e.blockHashShard,expirySeconds:(s=e==null?void 0:e.expirySeconds)!=null?s:zo.expirySeconds,apiAddress:(c=e==null?void 0:e.apiAddress)!=null?c:zo.apiAddress,tokenExpirationToastWarningSeconds:(p=e==null?void 0:e.tokenExpirationToastWarningSeconds)!=null?p:zo.tokenExpirationToastWarningSeconds,extraInfo:(m=e==null?void 0:e.extraInfo)!=null?m:{},gatewayUrl:e==null?void 0:e.gatewayUrl,extraRequestHeaders:(d=e==null?void 0:e.extraRequestHeaders)!=null?d:{}}};var Xo=e=>{let{origin:o,apiAddress:t,expirySeconds:s,blockHashShard:c,extraInfo:p,gatewayUrl:m,extraRequestHeaders:d}=Ko(e),l=new ic.NativeAuthClient({origin:o,apiUrl:t,expirySeconds:s,blockHashShard:c,gatewayUrl:m,extraRequestHeaders:d});return{getNativeAuthConfig:Ko,initialize:T=>f(void 0,null,function*(){var E,L;if(!t||!o)return"";let w=()=>l.getCurrentBlockHash();try{let N=(E=T==null?void 0:T.latestBlockInfo)!=null?E:yield rc(t,c,w,T==null?void 0:T.noCache);if(!N)return"";let{hash:Z,timestamp:q}=N,ae=l.encodeValue(JSON.stringify(u(u({},(L=T==null?void 0:T.extraInfo)!=null?L:p),q?{timestamp:q}:{})));return`${l.encodeValue(o)}.${Z}.${s}.${ae}`}catch(N){return console.error("Error getting native auth token: ",N.toString()),""}}),getToken:({address:T,token:w,signature:E})=>l.getToken(T,w,E),getTokenExpiration:ac}};var sc=t=>f(void 0,[t],function*({loginToken:e,extraInfoData:o}){if(e==null||Object.keys(o).length===0)return null;let s=Qe(String(e)),g=(s==null?void 0:s.extraInfo)||{},{timestamp:c}=g,p=Re(g,["timestamp"]);if(!(s&&c!=null))return null;let d={hash:String(s==null?void 0:s.blockHash),timestamp:Number(c)};return yield Xo({extraInfo:u(u({},p),o),expirySeconds:s==null?void 0:s.ttl,origin:s==null?void 0:s.origin}).initialize({latestBlockInfo:d})});var cc=p=>f(void 0,[p],function*({loginToken:e,extraInfoData:o,address:t,signature:s,loginService:c}){let m=yield sc({loginToken:e,extraInfoData:o}),d=o.multisig||o.impersonate||t,l=m!=null?d:t;return m!=null&&c.setLoginToken(m),s&&c.setTokenLoginInfo({signature:s,address:t}),yield bo(l)});n();var mg=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");n();var dg=require("@multiversx/sdk-passkey-provider/out");n();var hg=x(require("react")),xg=require("@fortawesome/free-solid-svg-icons");n();n();var wo=x(require("react")),uc=require("@fortawesome/free-solid-svg-icons"),gc=require("@fortawesome/react-fontawesome");n();var lc=x(require("react"));n();var Yo=x(require("react"));n();_o();var pc=()=>!je();var _g=()=>f(void 0,null,function*(){return yield Promise.resolve().then(()=>(Tr(),fr))}),ug=()=>(Tr(),ee(fr)).default,hr=pc();function dc({ssrGlobalImportCallback:e,ssrImportCallback:o,clientImportCallback:t}){let[s,c]=Yo.default.useState(hr?void 0:ug()),[p,m]=Yo.default.useState(hr||t==null?void 0:t()),d=()=>f(this,null,function*(){(e?e():_g()).then(l=>c(l.default)),o==null||o().then(l=>m(l.default))});return(0,Yo.useEffect)(()=>{hr&&d()},[]),{globalStyles:s,styles:p}}function re(e,o){return t=>{let{globalStyles:s,styles:c}=dc({ssrGlobalImportCallback:o.ssrGlobalStyles,ssrImportCallback:o.ssrStyles,clientImportCallback:o.clientStyles});return lc.default.createElement(e,v(u({},t),{globalStyles:s!=null?s:{},styles:c!=null?c:{}}))}}var fg=({styles:e})=>wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},wo.default.createElement(gc.FontAwesomeIcon,{icon:uc.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),wo.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),Tg=re(fg,{ssrStyles:()=>Promise.resolve().then(()=>(yr(),xr)),clientStyles:()=>(yr(),ee(xr)).default});qe();Ve();n();n();P();var AM=24*60*60;k();H();n();n();n();var Ag=require("react"),Ig=require("@multiversx/sdk-hw-provider");n();po();D();B();n();Ro();qe();Ve();$();n();n();n();var am=require("react"),im=require("@multiversx/sdk-core"),uh=require("@multiversx/sdk-extension-provider"),gh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),fh=require("@multiversx/sdk-passkey-provider/out"),Th=x(om());k();n();n();var oo=()=>I(Ie);n();n();n();n();n();n();n();n();var en=x(require("react"));var ZO=(0,en.createContext)({}),JO=A.getState();n();var tm=x(require("react"));n();var vr=x(require("react")),ZT=x(require("axios"));n();n();De();n();n();var th=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),rh=require("@multiversx/sdk-webview-provider/out/WebviewProvider");D();n();D();P();B();n();n();P();n();$();n();P();B();n();n();var QT=require("@multiversx/sdk-core"),eh=x(require("bignumber.js"));k();n();var oh=x(require("bignumber.js"));k();P();C();$();H();n();var rm=require("react"),ch=require("@multiversx/sdk-extension-provider"),ph=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),mh=require("@multiversx/sdk-passkey-provider/out");De();P();n();P();B();Ro();H();n();ue();P();n();n();var xh=require("react");n();n();C();ho();n();n();var Lh=require("react");n();n();var sm=require("react");n();var yh=x(require("axios"));n();var vh=require("react");n();var Ir=require("react"),wh=require("socket.io-client");P();n();var Ar=class{constructor(){this.current=null;this._status="NOT_INITIALIZED";this.listeners=[]}get status(){return this._status}set status(o){this._status!==o&&(this._status=o,this.notifyListeners())}subscribe(o){return this.listeners.push(o),()=>this.unsubscribe(o)}unsubscribe(o){this.listeners=this.listeners.filter(t=>t!==o)}notifyListeners(){this.listeners.forEach(o=>o(this._status))}},cm=new Ar;n();var Ih=require("react");Po();n();n();var kr=require("react");n();var kh=require("react");n();var mm=require("react");n();n();var Eh=require("react"),Ch=require("@multiversx/sdk-extension-provider");ue();D();P();C();n();function Pe(){return Bo(A.getState())}ve();$();$();n();n();De();B();var We=e=>(Object.values(W).forEach(o=>{if(o!==(e==null?void 0:e.skipLoginMethod))switch(o){case"crossWindow":{let t=se.CrossWindowProvider.getInstance();t.isInitialized()&&t.dispose();break}case"iframe":{let t=Ze.IframeProvider.getInstance();t.isInitialized()&&t.dispose();break}default:break}}),null);n();var dm=require("react"),at=require("@multiversx/sdk-core");P();var Ph=(e,o)=>{var t;return o?o===!0?e:(t=o.apiAddress)!=null?t:e:null},Ee=e=>{let o=I(V),t=I(Fo),s=(0,dm.useRef)(t==null?void 0:t.loginToken),c=Ph(o.apiAddress,e),p=Ko(u(u({},e===!0?{}:e),c?{apiAddress:c}:{})),m=Boolean(e),d=Xo(p),{address:l}=oo(),g=G(),b=L=>{s.current=L,g(No(u(v(u({},t),{loginToken:L}),c?{nativeAuthConfig:p}:{})))},T=()=>{try{return d.initialize()}catch(L){console.error("Unable to get block. Login failed.",L);return}},w=({address:L,signature:N})=>{let Z=s.current;if(!Z)throw"Token not found. Call `setLoginToken` first.";if(!m){g(No({loginToken:Z,signature:N}));return}let q=d.getToken({address:L,token:Z,signature:N});return g(No(u({loginToken:Z,signature:N,nativeAuthToken:q},c?{nativeAuthConfig:p}:{}))),q};return{configuration:p,setLoginToken:b,getNativeAuthLoginToken:T,setTokenLoginInfo:w,refreshNativeAuthTokenLogin:Z=>f(void 0,[Z],function*({signMessageCallback:L,nativeAuthClientConfig:N}){let ae=yield Xo(N||p).initialize({noCache:Boolean(N)});if(s.current=ae,!ae)return;let pe=new at.Message({address:new at.Address(l),data:r.Buffer.from(`${l}${ae}`)}),M=yield L(pe,{});if(!(M!=null&&M.signature))throw"Message not signed";return w({address:l,signature:r.Buffer.from(M.signature).toString("hex")})})}};n();var Nh=require("react"),Rh=require("@multiversx/sdk-opera-provider");ue();D();P();C();ve();$();H();n();var Lr=require("react");At();ue();De();D();P();P();C();ve();H();var Pr=({callbackRoute:e,token:o,nativeAuth:t,onLoginRedirect:s,hasConsentPopup:c,walletAddress:p})=>{let[m,d]=(0,Lr.useState)(""),[l,g]=(0,Lr.useState)(!1),b=t!=null,T=Ee(t),w=o,E=I(V),L=G(),N=Pe();function Z(){return f(this,null,function*(){if(N)throw new Error(be);We({skipLoginMethod:"crossWindow"}),L(fe(X.address)),L(J(X)),g(!0);let ae=yield se.CrossWindowProvider.getInstance().init();p&&L(Ot(p));let pe=se.CrossWindowProvider.getInstance().setWalletUrl(p!=null?p:E.walletAddress);try{if(!ae){console.warn("Something went wrong trying to redirect to wallet login..");return}let{origin:M,pathname:rn}=F(),it=encodeURIComponent(`${M}${e!=null?e:rn}`);if(b&&!w&&(w=yield T.getNativeAuthLoginToken(),!w)){console.warn("Fetching block failed. Login cancelled.");return}w&&T.setLoginToken(w);let st=u({callbackUrl:it},w&&{token:w});(Ca&&b||c)&&pe.setShouldShowConsentPopup(!0);let{signature:an,address:sn,multisig:Cm,impersonate:Nm}=yield pe.login(st);if(!sn){g(!1),se.CrossWindowProvider.getInstance().onDestroy(),console.warn("Login cancelled.");return}Le(pe);let no=yield cc({loginToken:w,extraInfoData:{multisig:Cm,impersonate:Nm},address:sn,signature:an,loginService:T});if(!no)return;L(K({address:no.address,loginMethod:"crossWindow"})),L(fe(no.address)),L(J(v(u({},no),{nonce:Vo(no)}))),Oe({callbackRoute:e,onLoginRedirect:s,options:{signature:an,address:no.address}})}catch(M){console.error("error loging in",M),d("error logging in"+M.message)}finally{g(!1)}})}let q=Boolean(m);return[Z,{loginFailed:q,error:m,isLoading:l&&!q,isLoggedIn:N&&!q}]};n();var Bh=require("react");ue();P();n();n();It();n();n();var Dh=x(require("platform"));_o();n();n();n();n();n();n();n();po();C();n();n();var Mh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");n();n();Ro();qe();n();n();$();n();n();var Oh=x(nt());po();var lm,_m,um,FG=(um=(_m=(lm=ne.safeWindow)==null?void 0:lm.location)==null?void 0:_m.origin)==null?void 0:um.includes("localhost");n();H();n();n();var zh=require("react");k();n();n();k();var Er=e=>{var c;let o=I(Fo),t=(c=o==null?void 0:o.loginToken)!=null?c:e;if(!t)return;let s=Qe(t);if(!!s)return s.origin.toLowerCase().startsWith(ka)};n();n();n();var Fh=x(require("bignumber.js"));n();n();n();var Ue=x(require("react")),Sm=require("react"),bm=require("react"),Mr=x(require("classnames")),wm=require("react-dom");k();n();var fm=x(require("react")),Tm=x(require("classnames"));var Uh=({className:e,children:o,styles:t})=>fm.default.createElement("div",{className:(0,Tm.default)(t==null?void 0:t.dappModalBody,e)},o),Cr=re(Uh,{ssrStyles:()=>Promise.resolve().then(()=>(Ne(),Ce)),clientStyles:()=>(Ne(),ee(Ce)).default});n();var Nr=x(require("react")),hm=x(require("classnames"));var Gh=({visible:e,customFooter:o,className:t,footerText:s,styles:c})=>e?Nr.default.createElement("div",{className:(0,hm.default)(c==null?void 0:c.dappModalFooter,t)},o!=null?o:Nr.default.createElement("div",null,s)):null,Rr=re(Gh,{ssrStyles:()=>Promise.resolve().then(()=>(Ne(),Ce)),clientStyles:()=>(Ne(),ee(Ce)).default});n();var ko=x(require("react")),xm=require("@fortawesome/free-solid-svg-icons"),ym=require("@fortawesome/react-fontawesome"),nn=x(require("classnames"));var Hh=({visible:e,headerText:o,customHeader:t,className:s,closeButtonClassName:c,headerTextClassName:p,onHide:m,globalStyles:d,styles:l})=>e?t?ko.default.createElement("div",{className:(0,nn.default)(l==null?void 0:l.dappModalHeader,s)},t):ko.default.createElement("div",{className:(0,nn.default)(l==null?void 0:l.dappModalHeader,s)},ko.default.createElement("div",{className:(0,nn.default)(l==null?void 0:l.dappModalHeaderText,p)},o),ko.default.createElement("button",{onClick:m,className:(0,nn.default)(l==null?void 0:l.dappModalCloseButton,d==null?void 0:d.btn,d==null?void 0:d.btnLight,c)},ko.default.createElement(ym.FontAwesomeIcon,{size:"lg",icon:xm.faTimes}))):null,Dr=re(Hh,{ssrStyles:()=>Promise.resolve().then(()=>(Ne(),Ce)),clientStyles:()=>(Ne(),ee(Ce)).default});var Vh={showHeader:!0,showFooter:!1,headerText:"",footerText:""},qh=({"data-testid":e="dappModal",children:o,className:t="dapp-modal-dialog-wrapper",closeOnEscape:s,config:c=Vh,id:p="dapp-modal",onHide:m,parentElement:d,visible:l,styles:g})=>{let[b,T]=(0,Sm.useState)(!1);if((0,bm.useEffect)(()=>{T(!0)},[]),!l)return null;let{showHeader:w,showFooter:E,headerText:L,footerText:N,modalDialogClassName:Z="dapp-modal-dialog",modalContentClassName:q="dapp-modal-dialog-content",modalHeaderClassName:ae="dapp-modal-dialog-header",modalHeaderTextClassName:pe="dapp-modal-dialog-header-text",modalCloseButtonClassName:M="dapp-modal-dialog-close-button",modalBodyClassName:rn="dapp-modal-dialog-content-body",modalFooterClassName:it="dapp-modal-dialog-footer",customModalHeader:st,customModalFooter:Wr}=c,an=sn=>{sn.key==="Escape"&&s&&(m==null||m())};return Ue.default.createElement(Ue.default.Fragment,null,b&&(0,wm.createPortal)(Ue.default.createElement("div",{id:p,role:"dialog","aria-modal":"true",className:(0,Mr.default)(Z,g==null?void 0:g.dappModal,t),"data-testid":e,onKeyDown:an},Ue.default.createElement("div",{className:(0,Mr.default)(g==null?void 0:g.dappModalContent,q)},Ue.default.createElement(Dr,{visible:w,headerText:L,customHeader:st,className:ae,headerTextClassName:pe,closeButtonClassName:M,onHide:m}),Ue.default.createElement(Cr,{className:rn},o),Ue.default.createElement(Rr,{visible:E,customFooter:Wr,footerText:N,className:it}))),d!=null?d:document==null?void 0:document.body))},jh=re(qh,{ssrStyles:()=>Promise.resolve().then(()=>(Ne(),Ce)),clientStyles:()=>(Ne(),ee(Ce)).default});n();n();var vm=require("react");n();n();var Am=require("react"),Xh=require("@multiversx/sdk-hw-provider");ue();D();P();C();ve();n();var Kh=require("react");n();n();var Or=require("react");D();P();C();ve();H();n();var Zh=require("react"),Jh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");ue();D();P();C();ve();$();$();n();var Qh=require("react"),ex=require("@multiversx/sdk-passkey-provider/out");ue();D();P();C();ve();$();$();n();n();var ox=require("react");B();n();n();n();var tx=x(require("swr"));n();n();n();n();var nx=x(require("axios"));n();n();var tn=x(require("react")),Lm=x(require("classnames"));var ax=({onLogin:e,text:o="Default Login Button",className:t="dapp-login-button",btnClassName:s="dapp-default-login-button",disabled:c,"data-testid":p,children:m,globalStyles:d,styles:l})=>{let g=(0,tn.useMemo)(()=>({wrapper:(0,Lm.default)(d==null?void 0:d.btn,d==null?void 0:d.btnPrimary,d==null?void 0:d.px4,d==null?void 0:d.m1,d==null?void 0:d.mx3,{[s]:s!=null},t),loginText:l==null?void 0:l.loginText,wrapperClassName:t}),[d,l]);return tn.default.createElement("button",{"data-testid":p,disabled:c,className:g.wrapper,onClick:e},m||tn.default.createElement("span",{className:g.loginText},o))},Pm=re(ax,{ssrStyles:()=>Promise.resolve().then(()=>(Fr(),Br)),clientStyles:()=>(Fr(),ee(Br)).default});var ix=({token:e,className:o="dapp-cross-window-xalias-login",children:t,callbackRoute:s,buttonClassName:c,nativeAuth:p,loginButtonText:m="xAlias",onLoginRedirect:d,disabled:l,hasConsentPopup:g,customWalletAddress:b,"data-testid":T="accessCrossWindowWalletBtn"})=>{let w=I(Jt),E=Er(e),[L]=Pr({callbackRoute:s,token:e,onLoginRedirect:d,nativeAuth:p,hasConsentPopup:g,walletAddress:b!=null?b:w});return Em.default.createElement(Pm,{onLogin:()=>{L()},className:o,btnClassName:c,text:m,disabled:l||E,"data-testid":T},t)};0&&(module.exports={XaliasCrossWindowLoginButton});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
