// @multiversx/sdk-dapp/__commonjs/UI/SignTransactionsModals/components/DeviceConfirmationScreen.js
"use strict";var Td=Object.create;var Go=Object.defineProperty,hd=Object.defineProperties,xd=Object.getOwnPropertyDescriptor,yd=Object.getOwnPropertyDescriptors,Sd=Object.getOwnPropertyNames,yn=Object.getOwnPropertySymbols,bd=Object.getPrototypeOf,bt=Object.prototype.hasOwnProperty,pa=Object.prototype.propertyIsEnumerable;var ca=(e,o,t)=>o in e?Go(e,o,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[o]=t,u=(e,o)=>{for(var t in o||(o={}))bt.call(o,t)&&ca(e,t,o[t]);if(yn)for(var t of yn(o))pa.call(o,t)&&ca(e,t,o[t]);return e},v=(e,o)=>hd(e,yd(o));var Ne=(e,o)=>{var t={};for(var s in e)bt.call(e,s)&&o.indexOf(s)<0&&(t[s]=e[s]);if(e!=null&&yn)for(var s of yn(e))o.indexOf(s)<0&&pa.call(e,s)&&(t[s]=e[s]);return t};var T=(e,o)=>()=>(e&&(o=e(e=0)),o);var S=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),ne=(e,o)=>{for(var t in o)Go(e,t,{get:o[t],enumerable:!0})},ma=(e,o,t,s)=>{if(o&&typeof o=="object"||typeof o=="function")for(let c of Sd(o))!bt.call(e,c)&&c!==t&&Go(e,c,{get:()=>o[c],enumerable:!(s=xd(o,c))||s.enumerable});return e};var h=(e,o,t)=>(t=e!=null?Td(bd(e)):{},ma(o||!e||!e.__esModule?Go(t,"default",{value:e,enumerable:!0}):t,e)),Y=e=>ma(Go({},"__esModule",{value:!0}),e);var g=(e,o,t)=>new Promise((s,c)=>{var p=_=>{try{l(t.next(_))}catch(f){c(f)}},m=_=>{try{l(t.throw(_))}catch(f){c(f)}},l=_=>_.done?s(_.value):Promise.resolve(_.value).then(p,m);l((t=t.apply(e,o)).next())});var _a=S(Sn=>{"use strict";n();Sn.byteLength=vd;Sn.toByteArray=Ad;Sn.fromByteArray=Pd;var se=[],Q=[],wd=typeof Uint8Array!="undefined"?Uint8Array:Array,wt="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Ve=0,da=wt.length;Ve<da;++Ve)se[Ve]=wt[Ve],Q[wt.charCodeAt(Ve)]=Ve;var Ve,da;Q["-".charCodeAt(0)]=62;Q["_".charCodeAt(0)]=63;function la(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var t=e.indexOf("=");t===-1&&(t=o);var s=t===o?0:4-t%4;return[t,s]}function vd(e){var o=la(e),t=o[0],s=o[1];return(t+s)*3/4-s}function Id(e,o,t){return(o+t)*3/4-t}function Ad(e){var o,t=la(e),s=t[0],c=t[1],p=new wd(Id(e,s,c)),m=0,l=c>0?s-4:s,_;for(_=0;_<l;_+=4)o=Q[e.charCodeAt(_)]<<18|Q[e.charCodeAt(_+1)]<<12|Q[e.charCodeAt(_+2)]<<6|Q[e.charCodeAt(_+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return c===2&&(o=Q[e.charCodeAt(_)]<<2|Q[e.charCodeAt(_+1)]>>4,p[m++]=o&255),c===1&&(o=Q[e.charCodeAt(_)]<<10|Q[e.charCodeAt(_+1)]<<4|Q[e.charCodeAt(_+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function kd(e){return se[e>>18&63]+se[e>>12&63]+se[e>>6&63]+se[e&63]}function Ld(e,o,t){for(var s,c=[],p=o;p<t;p+=3)s=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),c.push(kd(s));return c.join("")}function Pd(e){for(var o,t=e.length,s=t%3,c=[],p=16383,m=0,l=t-s;m<l;m+=p)c.push(Ld(e,m,m+p>l?l:m+p));return s===1?(o=e[t-1],c.push(se[o>>2]+se[o<<4&63]+"==")):s===2&&(o=(e[t-2]<<8)+e[t-1],c.push(se[o>>10]+se[o>>4&63]+se[o<<2&63]+"=")),c.join("")}});var ua=S(vt=>{n();vt.read=function(e,o,t,s,c){var p,m,l=c*8-s-1,_=(1<<l)-1,f=_>>1,x=-7,b=t?c-1:0,P=t?-1:1,L=e[o+b];for(b+=P,p=L&(1<<-x)-1,L>>=-x,x+=l;x>0;p=p*256+e[o+b],b+=P,x-=8);for(m=p&(1<<-x)-1,p>>=-x,x+=s;x>0;m=m*256+e[o+b],b+=P,x-=8);if(p===0)p=1-f;else{if(p===_)return m?NaN:(L?-1:1)*(1/0);m=m+Math.pow(2,s),p=p-f}return(L?-1:1)*m*Math.pow(2,p-s)};vt.write=function(e,o,t,s,c,p){var m,l,_,f=p*8-c-1,x=(1<<f)-1,b=x>>1,P=c===23?Math.pow(2,-24)-Math.pow(2,-77):0,L=s?0:p-1,G=s?1:-1,ie=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(l=isNaN(o)?1:0,m=x):(m=Math.floor(Math.log(o)/Math.LN2),o*(_=Math.pow(2,-m))<1&&(m--,_*=2),m+b>=1?o+=P/_:o+=P*Math.pow(2,1-b),o*_>=2&&(m++,_/=2),m+b>=x?(l=0,m=x):m+b>=1?(l=(o*_-1)*Math.pow(2,c),m=m+b):(l=o*Math.pow(2,b-1)*Math.pow(2,c),m=0));c>=8;e[t+L]=l&255,L+=G,l/=256,c-=8);for(m=m<<c|l,f+=c;f>0;e[t+L]=m&255,L+=G,m/=256,f-=8);e[t+L-G]|=ie*128}});var ka=S(mo=>{"use strict";n();var It=_a(),po=ua(),ga=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;mo.Buffer=d;mo.SlowBuffer=Md;mo.INSPECT_MAX_BYTES=50;var bn=2147483647;mo.kMaxLength=bn;d.TYPED_ARRAY_SUPPORT=Ed();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Ed(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(t){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function xe(e){if(e>bn)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,d.prototype),o}function d(e,o,t){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Pt(e)}return ha(e,o,t)}d.poolSize=8192;function ha(e,o,t){if(typeof e=="string")return Nd(e,o);if(ArrayBuffer.isView(e))return Rd(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(ce(e,ArrayBuffer)||e&&ce(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(ce(e,SharedArrayBuffer)||e&&ce(e.buffer,SharedArrayBuffer)))return kt(e,o,t);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var s=e.valueOf&&e.valueOf();if(s!=null&&s!==e)return d.from(s,o,t);var c=Dd(e);if(c)return c;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),o,t);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,o,t){return ha(e,o,t)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function xa(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Cd(e,o,t){return xa(e),e<=0?xe(e):o!==void 0?typeof t=="string"?xe(e).fill(o,t):xe(e).fill(o):xe(e)}d.alloc=function(e,o,t){return Cd(e,o,t)};function Pt(e){return xa(e),xe(e<0?0:Et(e)|0)}d.allocUnsafe=function(e){return Pt(e)};d.allocUnsafeSlow=function(e){return Pt(e)};function Nd(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!d.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var t=ya(e,o)|0,s=xe(t),c=s.write(e,o);return c!==t&&(s=s.slice(0,c)),s}function At(e){for(var o=e.length<0?0:Et(e.length)|0,t=xe(o),s=0;s<o;s+=1)t[s]=e[s]&255;return t}function Rd(e){if(ce(e,Uint8Array)){var o=new Uint8Array(e);return kt(o.buffer,o.byteOffset,o.byteLength)}return At(e)}function kt(e,o,t){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(t||0))throw new RangeError('"length" is outside of buffer bounds');var s;return o===void 0&&t===void 0?s=new Uint8Array(e):t===void 0?s=new Uint8Array(e,o):s=new Uint8Array(e,o,t),Object.setPrototypeOf(s,d.prototype),s}function Dd(e){if(d.isBuffer(e)){var o=Et(e.length)|0,t=xe(o);return t.length===0||e.copy(t,0,0,o),t}if(e.length!==void 0)return typeof e.length!="number"||Ct(e.length)?xe(0):At(e);if(e.type==="Buffer"&&Array.isArray(e.data))return At(e.data)}function Et(e){if(e>=bn)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+bn.toString(16)+" bytes");return e|0}function Md(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==d.prototype};d.compare=function(o,t){if(ce(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),ce(t,Uint8Array)&&(t=d.from(t,t.offset,t.byteLength)),!d.isBuffer(o)||!d.isBuffer(t))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===t)return 0;for(var s=o.length,c=t.length,p=0,m=Math.min(s,c);p<m;++p)if(o[p]!==t[p]){s=o[p],c=t[p];break}return s<c?-1:c<s?1:0};d.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(o,t){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return d.alloc(0);var s;if(t===void 0)for(t=0,s=0;s<o.length;++s)t+=o[s].length;var c=d.allocUnsafe(t),p=0;for(s=0;s<o.length;++s){var m=o[s];if(ce(m,Uint8Array))p+m.length>c.length?d.from(m).copy(c,p):Uint8Array.prototype.set.call(c,m,p);else if(d.isBuffer(m))m.copy(c,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return c};function ya(e,o){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||ce(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var t=e.length,s=arguments.length>2&&arguments[2]===!0;if(!s&&t===0)return 0;for(var c=!1;;)switch(o){case"ascii":case"latin1":case"binary":return t;case"utf8":case"utf-8":return Lt(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return t*2;case"hex":return t>>>1;case"base64":return Aa(e).length;default:if(c)return s?-1:Lt(e).length;o=(""+o).toLowerCase(),c=!0}}d.byteLength=ya;function Od(e,o,t){var s=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((t===void 0||t>this.length)&&(t=this.length),t<=0)||(t>>>=0,o>>>=0,t<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return jd(this,o,t);case"utf8":case"utf-8":return ba(this,o,t);case"ascii":return Vd(this,o,t);case"latin1":case"binary":return qd(this,o,t);case"base64":return Hd(this,o,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return zd(this,o,t);default:if(s)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),s=!0}}d.prototype._isBuffer=!0;function qe(e,o,t){var s=e[o];e[o]=e[t],e[t]=s}d.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var t=0;t<o;t+=2)qe(this,t,t+1);return this};d.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var t=0;t<o;t+=4)qe(this,t,t+3),qe(this,t+1,t+2);return this};d.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var t=0;t<o;t+=8)qe(this,t,t+7),qe(this,t+1,t+6),qe(this,t+2,t+5),qe(this,t+3,t+4);return this};d.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?ba(this,0,o):Od.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(o){if(!d.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:d.compare(this,o)===0};d.prototype.inspect=function(){var o="",t=mo.INSPECT_MAX_BYTES;return o=this.toString("hex",0,t).replace(/(.{2})/g,"$1 ").trim(),this.length>t&&(o+=" ... "),"<Buffer "+o+">"};ga&&(d.prototype[ga]=d.prototype.inspect);d.prototype.compare=function(o,t,s,c,p){if(ce(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(t===void 0&&(t=0),s===void 0&&(s=o?o.length:0),c===void 0&&(c=0),p===void 0&&(p=this.length),t<0||s>o.length||c<0||p>this.length)throw new RangeError("out of range index");if(c>=p&&t>=s)return 0;if(c>=p)return-1;if(t>=s)return 1;if(t>>>=0,s>>>=0,c>>>=0,p>>>=0,this===o)return 0;for(var m=p-c,l=s-t,_=Math.min(m,l),f=this.slice(c,p),x=o.slice(t,s),b=0;b<_;++b)if(f[b]!==x[b]){m=f[b],l=x[b];break}return m<l?-1:l<m?1:0};function Sa(e,o,t,s,c){if(e.length===0)return-1;if(typeof t=="string"?(s=t,t=0):t>2147483647?t=2147483647:t<-2147483648&&(t=-2147483648),t=+t,Ct(t)&&(t=c?0:e.length-1),t<0&&(t=e.length+t),t>=e.length){if(c)return-1;t=e.length-1}else if(t<0)if(c)t=0;else return-1;if(typeof o=="string"&&(o=d.from(o,s)),d.isBuffer(o))return o.length===0?-1:fa(e,o,t,s,c);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?c?Uint8Array.prototype.indexOf.call(e,o,t):Uint8Array.prototype.lastIndexOf.call(e,o,t):fa(e,[o],t,s,c);throw new TypeError("val must be string, number or Buffer")}function fa(e,o,t,s,c){var p=1,m=e.length,l=o.length;if(s!==void 0&&(s=String(s).toLowerCase(),s==="ucs2"||s==="ucs-2"||s==="utf16le"||s==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,l/=2,t/=2}function _(L,G){return p===1?L[G]:L.readUInt16BE(G*p)}var f;if(c){var x=-1;for(f=t;f<m;f++)if(_(e,f)===_(o,x===-1?0:f-x)){if(x===-1&&(x=f),f-x+1===l)return x*p}else x!==-1&&(f-=f-x),x=-1}else for(t+l>m&&(t=m-l),f=t;f>=0;f--){for(var b=!0,P=0;P<l;P++)if(_(e,f+P)!==_(o,P)){b=!1;break}if(b)return f}return-1}d.prototype.includes=function(o,t,s){return this.indexOf(o,t,s)!==-1};d.prototype.indexOf=function(o,t,s){return Sa(this,o,t,s,!0)};d.prototype.lastIndexOf=function(o,t,s){return Sa(this,o,t,s,!1)};function Fd(e,o,t,s){t=Number(t)||0;var c=e.length-t;s?(s=Number(s),s>c&&(s=c)):s=c;var p=o.length;s>p/2&&(s=p/2);for(var m=0;m<s;++m){var l=parseInt(o.substr(m*2,2),16);if(Ct(l))return m;e[t+m]=l}return m}function Bd(e,o,t,s){return wn(Lt(o,e.length-t),e,t,s)}function Wd(e,o,t,s){return wn(Yd(o),e,t,s)}function Ud(e,o,t,s){return wn(Aa(o),e,t,s)}function Gd(e,o,t,s){return wn(Zd(o,e.length-t),e,t,s)}d.prototype.write=function(o,t,s,c){if(t===void 0)c="utf8",s=this.length,t=0;else if(s===void 0&&typeof t=="string")c=t,s=this.length,t=0;else if(isFinite(t))t=t>>>0,isFinite(s)?(s=s>>>0,c===void 0&&(c="utf8")):(c=s,s=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-t;if((s===void 0||s>p)&&(s=p),o.length>0&&(s<0||t<0)||t>this.length)throw new RangeError("Attempt to write outside buffer bounds");c||(c="utf8");for(var m=!1;;)switch(c){case"hex":return Fd(this,o,t,s);case"utf8":case"utf-8":return Bd(this,o,t,s);case"ascii":case"latin1":case"binary":return Wd(this,o,t,s);case"base64":return Ud(this,o,t,s);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Gd(this,o,t,s);default:if(m)throw new TypeError("Unknown encoding: "+c);c=(""+c).toLowerCase(),m=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Hd(e,o,t){return o===0&&t===e.length?It.fromByteArray(e):It.fromByteArray(e.slice(o,t))}function ba(e,o,t){t=Math.min(e.length,t);for(var s=[],c=o;c<t;){var p=e[c],m=null,l=p>239?4:p>223?3:p>191?2:1;if(c+l<=t){var _,f,x,b;switch(l){case 1:p<128&&(m=p);break;case 2:_=e[c+1],(_&192)===128&&(b=(p&31)<<6|_&63,b>127&&(m=b));break;case 3:_=e[c+1],f=e[c+2],(_&192)===128&&(f&192)===128&&(b=(p&15)<<12|(_&63)<<6|f&63,b>2047&&(b<55296||b>57343)&&(m=b));break;case 4:_=e[c+1],f=e[c+2],x=e[c+3],(_&192)===128&&(f&192)===128&&(x&192)===128&&(b=(p&15)<<18|(_&63)<<12|(f&63)<<6|x&63,b>65535&&b<1114112&&(m=b))}}m===null?(m=65533,l=1):m>65535&&(m-=65536,s.push(m>>>10&1023|55296),m=56320|m&1023),s.push(m),c+=l}return $d(s)}var Ta=4096;function $d(e){var o=e.length;if(o<=Ta)return String.fromCharCode.apply(String,e);for(var t="",s=0;s<o;)t+=String.fromCharCode.apply(String,e.slice(s,s+=Ta));return t}function Vd(e,o,t){var s="";t=Math.min(e.length,t);for(var c=o;c<t;++c)s+=String.fromCharCode(e[c]&127);return s}function qd(e,o,t){var s="";t=Math.min(e.length,t);for(var c=o;c<t;++c)s+=String.fromCharCode(e[c]);return s}function jd(e,o,t){var s=e.length;(!o||o<0)&&(o=0),(!t||t<0||t>s)&&(t=s);for(var c="",p=o;p<t;++p)c+=Jd[e[p]];return c}function zd(e,o,t){for(var s=e.slice(o,t),c="",p=0;p<s.length-1;p+=2)c+=String.fromCharCode(s[p]+s[p+1]*256);return c}d.prototype.slice=function(o,t){var s=this.length;o=~~o,t=t===void 0?s:~~t,o<0?(o+=s,o<0&&(o=0)):o>s&&(o=s),t<0?(t+=s,t<0&&(t=0)):t>s&&(t=s),t<o&&(t=o);var c=this.subarray(o,t);return Object.setPrototypeOf(c,d.prototype),c};function O(e,o,t){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>t)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(o,t,s){o=o>>>0,t=t>>>0,s||O(o,t,this.length);for(var c=this[o],p=1,m=0;++m<t&&(p*=256);)c+=this[o+m]*p;return c};d.prototype.readUintBE=d.prototype.readUIntBE=function(o,t,s){o=o>>>0,t=t>>>0,s||O(o,t,this.length);for(var c=this[o+--t],p=1;t>0&&(p*=256);)c+=this[o+--t]*p;return c};d.prototype.readUint8=d.prototype.readUInt8=function(o,t){return o=o>>>0,t||O(o,1,this.length),this[o]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(o,t){return o=o>>>0,t||O(o,2,this.length),this[o]|this[o+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(o,t){return o=o>>>0,t||O(o,2,this.length),this[o]<<8|this[o+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(o,t){return o=o>>>0,t||O(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(o,t){return o=o>>>0,t||O(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};d.prototype.readIntLE=function(o,t,s){o=o>>>0,t=t>>>0,s||O(o,t,this.length);for(var c=this[o],p=1,m=0;++m<t&&(p*=256);)c+=this[o+m]*p;return p*=128,c>=p&&(c-=Math.pow(2,8*t)),c};d.prototype.readIntBE=function(o,t,s){o=o>>>0,t=t>>>0,s||O(o,t,this.length);for(var c=t,p=1,m=this[o+--c];c>0&&(p*=256);)m+=this[o+--c]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*t)),m};d.prototype.readInt8=function(o,t){return o=o>>>0,t||O(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};d.prototype.readInt16LE=function(o,t){o=o>>>0,t||O(o,2,this.length);var s=this[o]|this[o+1]<<8;return s&32768?s|4294901760:s};d.prototype.readInt16BE=function(o,t){o=o>>>0,t||O(o,2,this.length);var s=this[o+1]|this[o]<<8;return s&32768?s|4294901760:s};d.prototype.readInt32LE=function(o,t){return o=o>>>0,t||O(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};d.prototype.readInt32BE=function(o,t){return o=o>>>0,t||O(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};d.prototype.readFloatLE=function(o,t){return o=o>>>0,t||O(o,4,this.length),po.read(this,o,!0,23,4)};d.prototype.readFloatBE=function(o,t){return o=o>>>0,t||O(o,4,this.length),po.read(this,o,!1,23,4)};d.prototype.readDoubleLE=function(o,t){return o=o>>>0,t||O(o,8,this.length),po.read(this,o,!0,52,8)};d.prototype.readDoubleBE=function(o,t){return o=o>>>0,t||O(o,8,this.length),po.read(this,o,!1,52,8)};function V(e,o,t,s,c,p){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>c||o<p)throw new RangeError('"value" argument is out of bounds');if(t+s>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(o,t,s,c){if(o=+o,t=t>>>0,s=s>>>0,!c){var p=Math.pow(2,8*s)-1;V(this,o,t,s,p,0)}var m=1,l=0;for(this[t]=o&255;++l<s&&(m*=256);)this[t+l]=o/m&255;return t+s};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(o,t,s,c){if(o=+o,t=t>>>0,s=s>>>0,!c){var p=Math.pow(2,8*s)-1;V(this,o,t,s,p,0)}var m=s-1,l=1;for(this[t+m]=o&255;--m>=0&&(l*=256);)this[t+m]=o/l&255;return t+s};d.prototype.writeUint8=d.prototype.writeUInt8=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,1,255,0),this[t]=o&255,t+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,2,65535,0),this[t]=o&255,this[t+1]=o>>>8,t+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,2,65535,0),this[t]=o>>>8,this[t+1]=o&255,t+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,4,4294967295,0),this[t+3]=o>>>24,this[t+2]=o>>>16,this[t+1]=o>>>8,this[t]=o&255,t+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,4,4294967295,0),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};d.prototype.writeIntLE=function(o,t,s,c){if(o=+o,t=t>>>0,!c){var p=Math.pow(2,8*s-1);V(this,o,t,s,p-1,-p)}var m=0,l=1,_=0;for(this[t]=o&255;++m<s&&(l*=256);)o<0&&_===0&&this[t+m-1]!==0&&(_=1),this[t+m]=(o/l>>0)-_&255;return t+s};d.prototype.writeIntBE=function(o,t,s,c){if(o=+o,t=t>>>0,!c){var p=Math.pow(2,8*s-1);V(this,o,t,s,p-1,-p)}var m=s-1,l=1,_=0;for(this[t+m]=o&255;--m>=0&&(l*=256);)o<0&&_===0&&this[t+m+1]!==0&&(_=1),this[t+m]=(o/l>>0)-_&255;return t+s};d.prototype.writeInt8=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,1,127,-128),o<0&&(o=255+o+1),this[t]=o&255,t+1};d.prototype.writeInt16LE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,2,32767,-32768),this[t]=o&255,this[t+1]=o>>>8,t+2};d.prototype.writeInt16BE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,2,32767,-32768),this[t]=o>>>8,this[t+1]=o&255,t+2};d.prototype.writeInt32LE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,4,2147483647,-2147483648),this[t]=o&255,this[t+1]=o>>>8,this[t+2]=o>>>16,this[t+3]=o>>>24,t+4};d.prototype.writeInt32BE=function(o,t,s){return o=+o,t=t>>>0,s||V(this,o,t,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};function wa(e,o,t,s,c,p){if(t+s>e.length)throw new RangeError("Index out of range");if(t<0)throw new RangeError("Index out of range")}function va(e,o,t,s,c){return o=+o,t=t>>>0,c||wa(e,o,t,4,34028234663852886e22,-34028234663852886e22),po.write(e,o,t,s,23,4),t+4}d.prototype.writeFloatLE=function(o,t,s){return va(this,o,t,!0,s)};d.prototype.writeFloatBE=function(o,t,s){return va(this,o,t,!1,s)};function Ia(e,o,t,s,c){return o=+o,t=t>>>0,c||wa(e,o,t,8,17976931348623157e292,-17976931348623157e292),po.write(e,o,t,s,52,8),t+8}d.prototype.writeDoubleLE=function(o,t,s){return Ia(this,o,t,!0,s)};d.prototype.writeDoubleBE=function(o,t,s){return Ia(this,o,t,!1,s)};d.prototype.copy=function(o,t,s,c){if(!d.isBuffer(o))throw new TypeError("argument should be a Buffer");if(s||(s=0),!c&&c!==0&&(c=this.length),t>=o.length&&(t=o.length),t||(t=0),c>0&&c<s&&(c=s),c===s||o.length===0||this.length===0)return 0;if(t<0)throw new RangeError("targetStart out of bounds");if(s<0||s>=this.length)throw new RangeError("Index out of range");if(c<0)throw new RangeError("sourceEnd out of bounds");c>this.length&&(c=this.length),o.length-t<c-s&&(c=o.length-t+s);var p=c-s;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(t,s,c):Uint8Array.prototype.set.call(o,this.subarray(s,c),t),p};d.prototype.fill=function(o,t,s,c){if(typeof o=="string"){if(typeof t=="string"?(c=t,t=0,s=this.length):typeof s=="string"&&(c=s,s=this.length),c!==void 0&&typeof c!="string")throw new TypeError("encoding must be a string");if(typeof c=="string"&&!d.isEncoding(c))throw new TypeError("Unknown encoding: "+c);if(o.length===1){var p=o.charCodeAt(0);(c==="utf8"&&p<128||c==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(t<0||this.length<t||this.length<s)throw new RangeError("Out of range index");if(s<=t)return this;t=t>>>0,s=s===void 0?this.length:s>>>0,o||(o=0);var m;if(typeof o=="number")for(m=t;m<s;++m)this[m]=o;else{var l=d.isBuffer(o)?o:d.from(o,c),_=l.length;if(_===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<s-t;++m)this[m+t]=l[m%_]}return this};var Kd=/[^+/0-9A-Za-z-_]/g;function Xd(e){if(e=e.split("=")[0],e=e.trim().replace(Kd,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Lt(e,o){o=o||1/0;for(var t,s=e.length,c=null,p=[],m=0;m<s;++m){if(t=e.charCodeAt(m),t>55295&&t<57344){if(!c){if(t>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===s){(o-=3)>-1&&p.push(239,191,189);continue}c=t;continue}if(t<56320){(o-=3)>-1&&p.push(239,191,189),c=t;continue}t=(c-55296<<10|t-56320)+65536}else c&&(o-=3)>-1&&p.push(239,191,189);if(c=null,t<128){if((o-=1)<0)break;p.push(t)}else if(t<2048){if((o-=2)<0)break;p.push(t>>6|192,t&63|128)}else if(t<65536){if((o-=3)<0)break;p.push(t>>12|224,t>>6&63|128,t&63|128)}else if(t<1114112){if((o-=4)<0)break;p.push(t>>18|240,t>>12&63|128,t>>6&63|128,t&63|128)}else throw new Error("Invalid code point")}return p}function Yd(e){for(var o=[],t=0;t<e.length;++t)o.push(e.charCodeAt(t)&255);return o}function Zd(e,o){for(var t,s,c,p=[],m=0;m<e.length&&!((o-=2)<0);++m)t=e.charCodeAt(m),s=t>>8,c=t%256,p.push(c),p.push(s);return p}function Aa(e){return It.toByteArray(Xd(e))}function wn(e,o,t,s){for(var c=0;c<s&&!(c+t>=o.length||c>=e.length);++c)o[c+t]=e[c];return c}function ce(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Ct(e){return e!==e}var Jd=function(){for(var e="0123456789abcdef",o=new Array(256),t=0;t<16;++t)for(var s=t*16,c=0;c<16;++c)o[s+c]=e[t]+e[c];return o}()});var Na=S((wx,Ca)=>{n();var D=Ca.exports={},pe,me;function Nt(){throw new Error("setTimeout has not been defined")}function Rt(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?pe=setTimeout:pe=Nt}catch(e){pe=Nt}try{typeof clearTimeout=="function"?me=clearTimeout:me=Rt}catch(e){me=Rt}})();function La(e){if(pe===setTimeout)return setTimeout(e,0);if((pe===Nt||!pe)&&setTimeout)return pe=setTimeout,setTimeout(e,0);try{return pe(e,0)}catch(o){try{return pe.call(null,e,0)}catch(t){return pe.call(this,e,0)}}}function Qd(e){if(me===clearTimeout)return clearTimeout(e);if((me===Rt||!me)&&clearTimeout)return me=clearTimeout,clearTimeout(e);try{return me(e)}catch(o){try{return me.call(null,e)}catch(t){return me.call(this,e)}}}var ye=[],lo=!1,je,vn=-1;function el(){!lo||!je||(lo=!1,je.length?ye=je.concat(ye):vn=-1,ye.length&&Pa())}function Pa(){if(!lo){var e=La(el);lo=!0;for(var o=ye.length;o;){for(je=ye,ye=[];++vn<o;)je&&je[vn].run();vn=-1,o=ye.length}je=null,lo=!1,Qd(e)}}D.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var t=1;t<arguments.length;t++)o[t-1]=arguments[t];ye.push(new Ea(e,o)),ye.length===1&&!lo&&La(Pa)};function Ea(e,o){this.fun=e,this.array=o}Ea.prototype.run=function(){this.fun.apply(null,this.array)};D.title="browser";D.browser=!0;D.env={};D.argv=[];D.version="";D.versions={};function Se(){}D.on=Se;D.addListener=Se;D.once=Se;D.off=Se;D.removeListener=Se;D.removeAllListeners=Se;D.emit=Se;D.prependListener=Se;D.prependOnceListener=Se;D.listeners=function(e){return[]};D.binding=function(e){throw new Error("process.binding is not supported")};D.cwd=function(){return"/"};D.chdir=function(e){throw new Error("process.chdir is not supported")};D.umask=function(){return 0}});var r,i,ol,a,n=T(()=>{r=h(ka()),i=h(Na()),ol=function(e){function o(){var s=this||self;return delete e.prototype.__magic__,s}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var t=__magic__;return t}(Object),a=ol});var Ra=T(()=>{"use strict";n()});var In,Da,de=T(()=>{"use strict";n();In="Transaction canceled",Da="Undefined transaction status"});var Ma=T(()=>{"use strict";n()});var Dt,Oa=T(()=>{"use strict";n();Dt=(s=>(s.SetGuardian="SetGuardian",s.GuardAccount="GuardAccount",s.UnGuardAccount="UnGuardAccount",s))(Dt||{})});var Fa=T(()=>{"use strict";n()});var Mt=T(()=>{"use strict";n()});var Ba=T(()=>{"use strict";n()});var Ot,Ft=T(()=>{"use strict";n();Ot=(c=>(c.signature="signature",c.sessionId="sessionId",c.status="status",c.address="address",c))(Ot||{})});var Wa=T(()=>{"use strict";n()});var Ua=T(()=>{"use strict";n()});var le,_o,Re=T(()=>{"use strict";n();le=require("@multiversx/sdk-web-wallet-cross-window-provider/out/CrossWindowProvider/CrossWindowProvider"),_o=require("@multiversx/sdk-web-wallet-cross-window-provider/out/enums")});var Ga,Mx,Ha,Ox,E=T(()=>{"use strict";n();Re();Ga=(t=>(t.signTransactionsWithGuardianResponse="SIGN_TRANSACTIONS_WITH_GUARDIAN_RESPONSE",t.reloginRequest="RELOGIN_REQUEST",t))(Ga||{}),Mx=u(u({},_o.WindowProviderRequestEnums),Ga),Ha=(o=>(o.reloginResponse="RELOGIN_RESPONSE",o))(Ha||{}),Ox=u(u({},_o.WindowProviderResponseEnums),Ha)});var $a=T(()=>{"use strict";n()});var Va=T(()=>{"use strict";n()});var q=T(()=>{"use strict";n()});var qa=T(()=>{"use strict";n()});var ja=T(()=>{"use strict";n()});var za=T(()=>{"use strict";n()});var M=T(()=>{"use strict";n();Ft();Wa();Ua();E();$a();Va();q();qa();ja();za()});var uo,Ka,Qx,Xa,ey,Ya,oy,ny,nl,go=T(()=>{"use strict";n();M();uo={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",xAliasAddress:"https://devnet.xalias.com",apiAddress:"https://devnet-api.multiversx.com",extrasApiAddress:"https://devnet-extras-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://devnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",xAliasAddress:"https://testnet.xalias.com",apiAddress:"https://testnet-api.multiversx.com",extrasApiAddress:"https://testnet-extras-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",metamaskSnapWalletAddress:"https://testnet-iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",xAliasAddress:"https://xalias.com",apiAddress:"https://api.multiversx.com",extrasApiAddress:"https://extras-api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",metamaskSnapWalletAddress:"https://iframe-wallet.multiversx.com",apiTimeout:"4000",roundDuration:6e3}},{chainId:Ka,egldLabel:Qx}=uo["devnet"],{chainId:Xa,egldLabel:ey}=uo["testnet"],{chainId:Ya,egldLabel:oy}=uo["mainnet"],ny={["devnet"]:Ka,["testnet"]:Xa,["mainnet"]:Ya},nl={[Ka]:"devnet",[Xa]:"testnet",[Ya]:"mainnet"}});var Za,Ho=T(()=>{"use strict";n();Za="cancel-transaction-toast"});var _e,Ja=T(()=>{"use strict";n();_e=require("@multiversx/sdk-web-wallet-provider")});var De,fo=T(()=>{"use strict";n();De=require("@multiversx/sdk-dapp-utils/out/constants/browserConstants")});var Qa,ei,Bt,cy,py,Wt=T(()=>{"use strict";n();fo();Bt=String((ei=(Qa=De.safeWindow)==null?void 0:Qa.navigator)==null?void 0:ei.userAgent),cy=/^((?!chrome|android).)*safari/i.test(Bt),py=/firefox/i.test(Bt)&&/windows/i.test(Bt)});var An,kn,oi,ni,Me,Ut,Ln,ti,To,A=T(()=>{"use strict";n();Ra();de();Ma();Oa();Fa();Mt();Ba();go();Ho();Ja();Wt();An=5e4,kn=1e9,oi=1,ni="/dapp/init",Me="signSession",Ut="sdk-dapp-version",Ln="logout",ti="login",To="0"});var $o,Pn=T(()=>{"use strict";n();$o=()=>Date.now()/1e3});var ri=T(()=>{"use strict";n()});var ai=T(()=>{"use strict";n()});var Gt=T(()=>{"use strict";n();Pn();ri();ai()});var Ht={};ne(Ht,{clear:()=>il,getItem:()=>rl,localStorageKeys:()=>be,removeItem:()=>al,setItem:()=>tl});var be,En,tl,rl,al,il,ze=T(()=>{"use strict";n();Gt();be={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event",guardianBreachToastDismissTimestamp:"sdk-dapp-guardian-breach-toast-dismiss-timestamp"},En=typeof localStorage!="undefined",tl=({key:e,data:o,expires:t})=>{!En||localStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},rl=e=>{if(!En)return;let o=localStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:$o()>=t.expires?(localStorage.removeItem(String(e)),null):t.data},al=e=>{!En||localStorage.removeItem(String(e))},il=()=>{!En||localStorage.clear()}});var $t={};ne($t,{clear:()=>pi,getItem:()=>si,removeItem:()=>ci,setItem:()=>ii,storage:()=>sl});var ii,si,ci,pi,sl,mi=T(()=>{"use strict";n();ii=({key:e,data:o,expires:t})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},si=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Date.now()>=t.expires?(sessionStorage.removeItem(String(e)),null):t.data},ci=e=>sessionStorage.removeItem(String(e)),pi=()=>sessionStorage.clear(),sl={setItem:ii,getItem:si,removeItem:ci,clear:pi}});var Oe,Ke=T(()=>{"use strict";n();ze();mi();Oe={session:$t,local:Ht}});var Vt,C,j,R=T(()=>{"use strict";n();Vt=require("@reduxjs/toolkit");A();C=(0,Vt.createAction)(Ln),j=(0,Vt.createAction)(ti,e=>({payload:e}))});var jt,di,li,ee,qt,_i,te,Fe,zt,Uy,cl,pl,Gy,Hy,$y,ml,dl,Cn,Nn=T(()=>{"use strict";n();jt=require("@multiversx/sdk-core"),di=require("@reduxjs/toolkit"),li=require("redux-persist");A();Ke();ze();R();ee={balance:"...",address:"",isGuarded:!1,nonce:0,txCount:0,scrCount:0,claimableRewards:To},qt={address:"",websocketEvent:null,websocketBatchEvent:null,accounts:{"":ee},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},_i=(0,di.createSlice)({name:"accountInfoSlice",initialState:qt,reducers:{setAddress:(e,o)=>{let t=o.payload;e.address=t,e.publicKey=t?new jt.Address(t).hex():""},setAccount:(e,o)=>{let t=e.address===o.payload.address;e.accounts={[e.address]:t?o.payload:ee},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:t}=e;e.accounts[t].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}},setWebsocketBatchEvent:(e,o)=>{e.websocketBatchEvent={timestamp:Date.now(),data:o.payload}}},extraReducers:e=>{e.addCase(C,()=>(Oe.local.removeItem(be.loginExpiresAt),qt)),e.addCase(j,(o,t)=>{let{address:s}=t.payload;o.address=s,o.publicKey=new jt.Address(s).toHex()}),e.addCase(li.REHYDRATE,(o,t)=>{var f;if(!((f=t.payload)!=null&&f.account))return;let{account:s}=t.payload,{address:c,shard:p,accounts:m,publicKey:l}=s;o.address=c,o.shard=p;let _=m&&c in m;o.accounts=_?m:qt.accounts,o.publicKey=l})}}),{setAccount:te,setAddress:Fe,setAccountNonce:zt,setAccountShard:Uy,setLedgerAccount:cl,updateLedgerAccount:pl,setWalletConnectAccount:Gy,setIsAccountLoading:Hy,setAccountLoadingError:$y,setWebsocketEvent:ml,setWebsocketBatchEvent:dl}=_i.actions,Cn=_i.reducer});function Vo(){return new Date().setHours(new Date().getHours()+24)}function qo(e){Oe.local.setItem({key:be.loginExpiresAt,data:e,expires:e})}var Kt=T(()=>{"use strict";n();Ke();ze()});var gi,ui,fi,Qy,ll,_l,Xt,eS,ul,Ti,oS,nS,tS,Rn,Dn=T(()=>{"use strict";n();gi=require("@reduxjs/toolkit");Kt();E();R();ui={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,passkeyLogin:null,operaLogin:null,crossWindowLogin:null,iframeWindowLogin:null,isLoginSessionInvalid:!1,webviewLogin:null},fi=(0,gi.createSlice)({name:"loginInfoSlice",initialState:ui,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},setWebviewLogin:(e,o)=>{e.webviewLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload},setIsWalletConnectV2Initialized:(e,o)=>{e.isWalletConnectV2Initialized=o.payload}},extraReducers:e=>{e.addCase(C,()=>ui),e.addCase(j,(o,t)=>{o.isLoginSessionInvalid=!1,o.loginMethod=t.payload.loginMethod,o.iframeLoginType=t.payload.iframeLoginType,qo(Vo())})}}),{setLoginMethod:Qy,setWalletConnectLogin:ll,setLedgerLogin:_l,setTokenLogin:Xt,setTokenLoginSignature:eS,setWalletLogin:ul,invalidateLoginSession:Ti,setLogoutRoute:oS,setIsWalletConnectV2Initialized:nS,setWebviewLogin:tS}=fi.actions,Rn=fi.reducer});var xi,hi,yi,sS,gl,cS,fl,Mn,On=T(()=>{"use strict";n();xi=require("@reduxjs/toolkit");R();hi={},yi=(0,xi.createSlice)({name:"modalsSlice",initialState:hi,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(C,()=>hi)}}),{setTxSubmittedModal:sS,setNotificationModal:gl,clearTxSubmittedModal:cS,clearNotificationModal:fl}=yi.actions,Mn=yi.reducer});var we,ho=T(()=>{"use strict";n();we=()=>typeof window!="undefined"&&typeof(window==null?void 0:window.location)!="undefined"});var N,W=T(()=>{"use strict";n();ho();N=()=>{if(!we())return{pathname:"",hash:"",origin:"",href:"",search:""};let{location:{pathname:o,hash:t,origin:s,href:c,search:p}}=window;return{pathname:o,hash:t,origin:s,href:c,search:p}}});var Si=T(()=>{"use strict";n();W()});var Be,bi=T(()=>{"use strict";n();U();Be=()=>{let{pathname:e,search:o,hash:t}=N();return`${e!=null?e:""}${o!=null?o:""}${t!=null?t:""}`}});var wi=T(()=>{"use strict";n();fo()});var U=T(()=>{"use strict";n();Si();bi();W();wi()});var jo=T(()=>{"use strict";n();U()});var Ii=T(()=>{"use strict";n();jo()});function Ai(e){return e[Math.floor(Math.random()*e.length)]}var ki=T(()=>{"use strict";n()});var Yt=T(()=>{"use strict";n();Mt()});var ve=T(()=>{"use strict";n();Ii();ki();Yt()});var Li,Pi,Ei,Zt,hl,Ci,GS,HS,Ni,Fn,Bn=T(()=>{"use strict";n();Li=require("@reduxjs/toolkit"),Pi=h(require("lodash.omit")),Ei=require("redux-persist");Ho();R();ve();Zt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000",roundDuration:6e3},hl={network:Zt},Ci=(0,Li.createSlice)({name:"appConfig",initialState:hl,reducers:{initializeNetworkConfig:(e,o)=>{let t=Ai(o.payload.walletConnectV2RelayAddresses),s=(0,Pi.default)(o.payload,"walletConnectV2RelayAddresses");e.network=v(u(u({},e.network),s),{walletConnectV2RelayAddress:t})},updateNetworkConfig:(e,o)=>{e.network=u(u({},e.network),o.payload)},setCustomWalletAddress:(e,o)=>{e.network.customWalletAddress=o.payload}},extraReducers:e=>{e.addCase(C,o=>{o.network.customWalletAddress=void 0}),e.addCase(Ei.REHYDRATE,(o,t)=>{var c,p;if(!((p=(c=t.payload)==null?void 0:c.network)!=null&&p.customWalletAddress))return;let{network:{customWalletAddress:s}}=t.payload;o.network.customWalletAddress=s})}}),{initializeNetworkConfig:GS,updateNetworkConfig:HS,setCustomWalletAddress:Ni}=Ci.actions,Fn=Ci.reducer});var Ri,Jt,Di,KS,XS,YS,Wn,Un=T(()=>{"use strict";n();Ri=require("@reduxjs/toolkit");M();R();Jt={isSigning:!1,signedSessions:{}},Di=(0,Ri.createSlice)({name:"signedMessageInfoSliceState",initialState:Jt,reducers:{setSignSession:(e,o)=>{let{sessionId:t,signedSession:s,errorMessage:c}=o.payload;c&&(e.errorMessage=c),e.isSigning=s.status==="pending",e.signedSessions[t]=u(u({},e.signedSessions[t]),s)},setSignSessionState:(e,o)=>u(u({},e),o.payload),clearSignedMessageInfo:()=>Jt},extraReducers:e=>{e.addCase(C,()=>Jt)}}),{setSignSession:KS,clearSignedMessageInfo:XS,setSignSessionState:YS}=Di.actions,Wn=Di.reducer});var Oi,Fi,Mi,Bi,Qt,xl,tb,yl,Gn,Hn=T(()=>{"use strict";n();Oi=require("@reduxjs/toolkit"),Fi=require("redux-persist");M();Pn();R();Mi={customToasts:[],transactionToasts:[]},Bi=(0,Oi.createSlice)({name:"toastsSlice",initialState:Mi,reducers:{addCustomToast:(e,o)=>{let t=o.payload.toastId||`custom-toast-${e.customToasts.length+1}`,s=e.customToasts.findIndex(c=>c.toastId===t);if(s!==-1){e.customToasts[s]=v(u(u({},e.customToasts[s]),o.payload),{type:"custom",toastId:t});return}e.customToasts.push(v(u({},o.payload),{type:"custom",toastId:t}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(t=>t.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:$o(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(t=>t.toastId!==o.payload)}},extraReducers:e=>{e.addCase(C,()=>Mi),e.addCase(Fi.REHYDRATE,(o,t)=>{var c,p;let s=(p=(c=t.customToasts)==null?void 0:c.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=s})}}),{addCustomToast:Qt,removeCustomToast:xl,addTransactionToast:tb,removeTransactionToast:yl}=Bi.actions,Gn=Bi.reducer});var Wi,or,nr,tr,Sl,er,Ui,sb,bl,$n,Vn,qn=T(()=>{"use strict";n();Wi=require("@reduxjs/toolkit");R();or="Transaction failed",nr="Transaction successful",tr="Processing transaction",Sl="Transaction submitted",er={},Ui=(0,Wi.createSlice)({name:"transactionsInfo",initialState:er,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:t,transactionsDisplayInfo:s}=o.payload;t!=null&&(e[t]={errorMessage:(s==null?void 0:s.errorMessage)||or,successMessage:(s==null?void 0:s.successMessage)||nr,processingMessage:(s==null?void 0:s.processingMessage)||tr,submittedMessage:(s==null?void 0:s.submittedMessage)||Sl,transactionDuration:s==null?void 0:s.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>er},extraReducers:e=>{e.addCase(C,()=>er)}}),{clearTransactionsInfo:sb,setTransactionsDisplayInfo:bl,clearTransactionsInfoForSessionId:$n}=Ui.actions,Vn=Ui.reducer});function yo(e){return e!=null&&(Nl(e)||Fl(e))}function So(e){return e!=null&&(Rl(e)||Bl(e))}function bo(e){return e!=null&&(Dl(e)||Wl(e))}function Gi(e){return e!=null&&(Ml(e)||Ul(e))}function rr(e){return e!=null&&Ol(e)}function Nl(e){return e!=null&&wl.includes(e)}function Rl(e){return e!=null&&vl.includes(e)}function Dl(e){return e!=null&&Il.includes(e)}function Ml(e){return e!=null&&Al.includes(e)}function Ol(e){return e!=null&&kl.includes(e)}function Fl(e){return e!=null&&Ll.includes(e)}function Bl(e){return e!=null&&Pl.includes(e)}function Wl(e){return e!=null&&El.includes(e)}function Ul(e){return e!=null&&Cl.includes(e)}var wl,vl,Il,Al,kl,Ll,Pl,El,Cl,wo=T(()=>{"use strict";n();E();wl=["sent"],vl=["success"],Il=["fail","cancelled","timedOut"],Al=["invalid"],kl=["timedOut"],Ll=["pending"],Pl=["success"],El=["fail","invalid"],Cl=["not executed"]});var Hi,$i,zo,Gl,Vi,qi,ji,Hl,vo,$l,Vl,fb,ql,Xe,Ko,Tb,jn,zn=T(()=>{"use strict";n();Hi=require("@reduxjs/toolkit"),$i=require("redux-persist");E();wo();R();zo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Gl={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Vi=(0,Hi.createSlice)({name:"transactionsSlice",initialState:zo,reducers:{moveTransactionsToSignedState:(e,o)=>{var _,f;let{sessionId:t,transactions:s,errorMessage:c,status:p,redirectRoute:m,customTransactionInformation:l}=o.payload;e.customTransactionInformationForSessionId[t]=u(u(u({},Gl),((_=e.signedTransactions[t])==null?void 0:_.customTransactionInformation)||{}),e.customTransactionInformationForSessionId[t]||{}),e.signedTransactions[t]={transactions:s,status:p,errorMessage:c,redirectRoute:m,customTransactionInformation:u(u({},e.customTransactionInformationForSessionId[t]),l!=null?l:{})},((f=e==null?void 0:e.transactionsToSign)==null?void 0:f.sessionId)===t&&(e.transactionsToSign=zo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=u(u({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:t,status:s,errorMessage:c,transactions:p}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].status=s,c!=null&&(e.signedTransactions[t].errorMessage=c),p!=null&&(e.signedTransactions[t].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var f,x,b,P,L,G,ie,$;let{sessionId:t,status:s,errorMessage:c,transactionHash:p,serverTransaction:m,inTransit:l}=o.payload,_=(x=(f=e.signedTransactions)==null?void 0:f[t])==null?void 0:x.transactions;if(_!=null){e.signedTransactions[t].transactions=_.map(H=>H.hash===p?v(u(u({},m!=null?m:{}),H),{status:s,errorMessage:c,inTransit:l}):H);let J=(P=(b=e.signedTransactions[t])==null?void 0:b.transactions)==null?void 0:P.every(H=>So(H.status)),Te=(G=(L=e.signedTransactions[t])==null?void 0:L.transactions)==null?void 0:G.some(H=>bo(H.status)),he=($=(ie=e.signedTransactions[t])==null?void 0:ie.transactions)==null?void 0:$.every(H=>Gi(H.status));J&&(e.signedTransactions[t].status="success"),Te&&(e.signedTransactions[t].status="fail"),he&&(e.signedTransactions[t].status="invalid")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:t,customTransactionInformation:s}=o.payload;e.customTransactionInformationForSessionId[t]=s,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=zo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=zo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload},updateSignedTransactionsCustomTransactionInformation:(e,o)=>{let{sessionId:t,customTransactionInformationOverrides:s}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].customTransactionInformation=u(u({},e.signedTransactions[t].customTransactionInformation),s))}},extraReducers:e=>{e.addCase(C,()=>zo),e.addCase($i.REHYDRATE,(o,t)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=t.payload)!=null&&m.transactions))return;let{signedTransactions:s,customTransactionInformationForSessionId:c}=t.payload.transactions,p=Object.entries(s).reduce((l,[_,f])=>{let x=new Date(_),b=new Date;return b.setHours(b.getHours()+5),b-x>0||(l[_]=f),l},{});c!=null&&(o.customTransactionInformationForSessionId=c),s!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:qi,updateSignedTransactions:ji,setTransactionsToSign:Hl,clearAllTransactionsToSign:vo,clearAllSignedTransactions:$l,clearSignedTransaction:Vl,clearTransactionToSign:fb,setSignTransactionsError:ql,setSignTransactionsCancelMessage:Xe,moveTransactionsToSignedState:Ko,updateSignedTransactionsCustomTransactionInformation:Tb}=Vi.actions,jn=Vi.reducer});var zi,ar,Ki,Sb,bb,jl,wb,Kn,Xn=T(()=>{"use strict";n();zi=require("@reduxjs/toolkit");R();ar={},Ki=(0,zi.createSlice)({name:"batchTransactionsSlice",initialState:ar,reducers:{setBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},updateBatchTransactions:(e,o)=>{e[o.payload.id]=o.payload},clearBatchTransactions:(e,o)=>{delete e[o.payload.batchId]},clearAllBatchTransactions:()=>ar},extraReducers:e=>{e.addCase(C,()=>ar)}}),{setBatchTransactions:Sb,updateBatchTransactions:bb,clearBatchTransactions:jl,clearAllBatchTransactions:wb}=Ki.actions,Kn=Ki.reducer});var Yi,Xi,Zi,kb,Ji,ir=T(()=>{"use strict";n();Yi=require("@reduxjs/toolkit");R();Xi={},Zi=(0,Yi.createSlice)({name:"dappConfigSlice",initialState:Xi,reducers:{setDappConfig:(e,o)=>o.payload},extraReducers:e=>{e.addCase(C,()=>Xi)}}),{setDappConfig:kb}=Zi.actions,Ji=Zi.reducer});var k=T(()=>{"use strict";n();Nn();Dn();On();Bn();Un();Hn();qn();zn();Xn();ir()});var sr=T(()=>{"use strict";n()});var os,Yl,Zl,Ye,Jn=T(()=>{"use strict";n();os=require("@reduxjs/toolkit");sr();Nn();Xn();ir();Dn();On();Bn();Un();Hn();qn();zn();Yl={["account"]:Cn,["dappConfig"]:Ji,["loginInfo"]:Rn,["modals"]:Mn,["networkConfig"]:Fn,["signedMessageInfo"]:Wn,["toasts"]:Gn,["transactionsInfo"]:Vn,["transactions"]:jn,["batchTransactions"]:Kn},Zl=(e={})=>(0,os.combineReducers)(u(u({},Yl),e)),Ye=Zl});var as={};ne(as,{default:()=>__,sessionStorageReducers:()=>cr});function Ie(e,o=[]){return{key:e,version:1,storage:ts.default,blacklist:o}}var z,ns,ts,Jl,Xo,Ql,e_,o_,n_,t_,r_,a_,i_,s_,c_,rs,p_,cr,m_,d_,l_,__,is=T(()=>{"use strict";n();z=require("redux-persist"),ns=h(require("redux-persist/lib/storage")),ts=h(require("redux-persist/lib/storage/session"));Jn();k();Nn();Xn();Dn();On();Bn();Un();Hn();qn();zn();sr();Jl={persistReducersStorageType:"localStorage"},Xo={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Ql=Ie(Xo["account"]),e_=Ie(Xo["loginInfo"]),o_=Ie(Xo["modals"]),n_=Ie(Xo["networkConfig"]),t_={2:e=>v(u({},e),{networkConfig:Zt})};r_=Ie("sdk-dapp-transactionsInfo"),a_=Ie("sdk-dapp-transactions",["transactionsToSign"]),i_=Ie("sdk-dapp-batchTransactions",["batchTransactions"]),s_=Ie("sdk-dapp-toasts"),c_=Ie("sdk-dapp-signedMessageInfo"),rs={key:"sdk-dapp-store",version:2,storage:ns.default,whitelist:Object.keys(Xo),migrate:(0,z.createMigrate)(t_,{debug:!1})},p_=v(u({},rs),{whitelist:[]}),cr={["toasts"]:(0,z.persistReducer)(s_,Gn),["transactions"]:(0,z.persistReducer)(a_,jn),["transactionsInfo"]:(0,z.persistReducer)(r_,Vn),["batchTransactions"]:(0,z.persistReducer)(i_,Kn),["signedMessageInfo"]:(0,z.persistReducer)(c_,Wn)},m_=v(u({},cr),{["account"]:(0,z.persistReducer)(Ql,Cn),["loginInfo"]:(0,z.persistReducer)(e_,Rn),["modals"]:(0,z.persistReducer)(o_,Mn),["networkConfig"]:(0,z.persistReducer)(n_,Fn)}),d_=Jl.persistReducersStorageType==="localStorage",l_=d_?(0,z.persistReducer)(rs,Ye(cr)):(0,z.persistReducer)(p_,Ye(m_)),__=l_});var ss={};ne(ss,{default:()=>u_});var u_,cs=T(()=>{"use strict";n();Jn();u_=Ye()});var ps={};ne(ps,{default:()=>f_});var oe,g_,f_,ms=T(()=>{"use strict";n();oe=require("redux-persist"),g_=[oe.FLUSH,oe.REHYDRATE,oe.PAUSE,oe.PERSIST,oe.PURGE,oe.REGISTER],f_=g_});var _s={};ne(_s,{default:()=>ls});function ls(e){return(0,ds.persistStore)(e)}var ds,us=T(()=>{"use strict";n();ds=require("redux-persist")});var vc=S((h2,wc)=>{n();var Zu=typeof a=="object"&&a&&a.Object===Object&&a;wc.exports=Zu});var on=S((x2,Ic)=>{n();var Ju=vc(),Qu=typeof self=="object"&&self&&self.Object===Object&&self,eg=Ju||Qu||Function("return this")();Ic.exports=eg});var Pr=S((y2,Ac)=>{n();var og=on(),ng=og.Symbol;Ac.exports=ng});var Ec=S((S2,Pc)=>{n();var kc=Pr(),Lc=Object.prototype,tg=Lc.hasOwnProperty,rg=Lc.toString,nn=kc?kc.toStringTag:void 0;function ag(e){var o=tg.call(e,nn),t=e[nn];try{e[nn]=void 0;var s=!0}catch(p){}var c=rg.call(e);return s&&(o?e[nn]=t:delete e[nn]),c}Pc.exports=ag});var Nc=S((b2,Cc)=>{n();var ig=Object.prototype,sg=ig.toString;function cg(e){return sg.call(e)}Cc.exports=cg});var Er=S((w2,Mc)=>{n();var Rc=Pr(),pg=Ec(),mg=Nc(),dg="[object Null]",lg="[object Undefined]",Dc=Rc?Rc.toStringTag:void 0;function _g(e){return e==null?e===void 0?lg:dg:Dc&&Dc in Object(e)?pg(e):mg(e)}Mc.exports=_g});var Fc=S((v2,Oc)=>{n();var ug=Array.isArray;Oc.exports=ug});var Wc=S((I2,Bc)=>{n();function gg(e){return e!=null&&typeof e=="object"}Bc.exports=gg});var Gc=S((A2,Uc)=>{n();var fg=Er(),Tg=Fc(),hg=Wc(),xg="[object String]";function yg(e){return typeof e=="string"||!Tg(e)&&hg(e)&&fg(e)==xg}Uc.exports=yg});var Rr={};ne(Rr,{css:()=>zc,default:()=>Lg});var zc,Lg,Dr=T(()=>{"use strict";n();zc=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: calc(65 / 14)rem;
  height: calc(65 / 14)rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: calc(24 / 14)rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: calc(42 / 14)rem;
  margin-right: calc(24 / 14)rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}

.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}

.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}

.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}

.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}

.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}

.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}

.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}

.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}

.dapp-core-component__main__btn-link:hover {
  color: rgb(0, 86.1, 178.5);
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}

.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}

.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}

.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}

.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: pre-wrap;
  word-break: break-all;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}

.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}

.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}

.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  height: 100%;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}

.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}
.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}
@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}
.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}
.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}

.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}
.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}
.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}

.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}

.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}

.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: rgb(69.1465517241, 77.125, 85.1034482759);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}

.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(zc));Lg={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"}});var Or={};ne(Or,{css:()=>Yc,default:()=>Cg});var Yc,Cg,Fr=T(()=>{"use strict";n();Yc=`.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast {
  padding: 8px;
  display: flex;
  flex-direction: column;
  gap: 8px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading {
  display: flex;
  gap: 8px;
  align-items: center;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon {
  color: #d9534f;
  font-size: 20px;
}
.dapp-core-component__guardianWarningToastStyles__guardian-warning-toast .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading .dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text {
  color: #d9534f;
  font-size: 16px;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Yc));Cg={"guardian-warning-toast":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast",guardianWarningToast:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast","guardian-warning-toast-heading":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading",guardianWarningToastHeading:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading","guardian-warning-toast-heading-icon":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon",guardianWarningToastHeadingIcon:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-icon","guardian-warning-toast-heading-text":"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text",guardianWarningToastHeadingText:"dapp-core-component__guardianWarningToastStyles__guardian-warning-toast-heading-text"}});var gt=S((J3,ep)=>{n();function Kg(e){var o=typeof e;return e!=null&&(o=="object"||o=="function")}ep.exports=Kg});var np=S((Q3,op)=>{n();var Xg=Er(),Yg=gt(),Zg="[object AsyncFunction]",Jg="[object Function]",Qg="[object GeneratorFunction]",ef="[object Proxy]";function of(e){if(!Yg(e))return!1;var o=Xg(e);return o==Jg||o==Qg||o==Zg||o==ef}op.exports=of});var rp=S((eO,tp)=>{n();var nf=on(),tf=nf["__core-js_shared__"];tp.exports=tf});var sp=S((oO,ip)=>{n();var Wr=rp(),ap=function(){var e=/[^.]+$/.exec(Wr&&Wr.keys&&Wr.keys.IE_PROTO||"");return e?"Symbol(src)_1."+e:""}();function rf(e){return!!ap&&ap in e}ip.exports=rf});var pp=S((nO,cp)=>{n();var af=Function.prototype,sf=af.toString;function cf(e){if(e!=null){try{return sf.call(e)}catch(o){}try{return e+""}catch(o){}}return""}cp.exports=cf});var dp=S((tO,mp)=>{n();var pf=np(),mf=sp(),df=gt(),lf=pp(),_f=/[\\^$.*+?()[\]{}|]/g,uf=/^\[object .+?Constructor\]$/,gf=Function.prototype,ff=Object.prototype,Tf=gf.toString,hf=ff.hasOwnProperty,xf=RegExp("^"+Tf.call(hf).replace(_f,"\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g,"$1.*?")+"$");function yf(e){if(!df(e)||mf(e))return!1;var o=pf(e)?xf:uf;return o.test(lf(e))}mp.exports=yf});var _p=S((rO,lp)=>{n();function Sf(e,o){return e==null?void 0:e[o]}lp.exports=Sf});var ft=S((aO,up)=>{n();var bf=dp(),wf=_p();function vf(e,o){var t=wf(e,o);return bf(t)?t:void 0}up.exports=vf});var an=S((iO,gp)=>{n();var If=ft(),Af=If(Object,"create");gp.exports=Af});var hp=S((sO,Tp)=>{n();var fp=an();function kf(){this.__data__=fp?fp(null):{},this.size=0}Tp.exports=kf});var yp=S((cO,xp)=>{n();function Lf(e){var o=this.has(e)&&delete this.__data__[e];return this.size-=o?1:0,o}xp.exports=Lf});var bp=S((pO,Sp)=>{n();var Pf=an(),Ef="__lodash_hash_undefined__",Cf=Object.prototype,Nf=Cf.hasOwnProperty;function Rf(e){var o=this.__data__;if(Pf){var t=o[e];return t===Ef?void 0:t}return Nf.call(o,e)?o[e]:void 0}Sp.exports=Rf});var vp=S((mO,wp)=>{n();var Df=an(),Mf=Object.prototype,Of=Mf.hasOwnProperty;function Ff(e){var o=this.__data__;return Df?o[e]!==void 0:Of.call(o,e)}wp.exports=Ff});var Ap=S((dO,Ip)=>{n();var Bf=an(),Wf="__lodash_hash_undefined__";function Uf(e,o){var t=this.__data__;return this.size+=this.has(e)?0:1,t[e]=Bf&&o===void 0?Wf:o,this}Ip.exports=Uf});var Lp=S((lO,kp)=>{n();var Gf=hp(),Hf=yp(),$f=bp(),Vf=vp(),qf=Ap();function Ro(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Ro.prototype.clear=Gf;Ro.prototype.delete=Hf;Ro.prototype.get=$f;Ro.prototype.has=Vf;Ro.prototype.set=qf;kp.exports=Ro});var Ep=S((_O,Pp)=>{n();function jf(){this.__data__=[],this.size=0}Pp.exports=jf});var Np=S((uO,Cp)=>{n();function zf(e,o){return e===o||e!==e&&o!==o}Cp.exports=zf});var sn=S((gO,Rp)=>{n();var Kf=Np();function Xf(e,o){for(var t=e.length;t--;)if(Kf(e[t][0],o))return t;return-1}Rp.exports=Xf});var Mp=S((fO,Dp)=>{n();var Yf=sn(),Zf=Array.prototype,Jf=Zf.splice;function Qf(e){var o=this.__data__,t=Yf(o,e);if(t<0)return!1;var s=o.length-1;return t==s?o.pop():Jf.call(o,t,1),--this.size,!0}Dp.exports=Qf});var Fp=S((TO,Op)=>{n();var eT=sn();function oT(e){var o=this.__data__,t=eT(o,e);return t<0?void 0:o[t][1]}Op.exports=oT});var Wp=S((hO,Bp)=>{n();var nT=sn();function tT(e){return nT(this.__data__,e)>-1}Bp.exports=tT});var Gp=S((xO,Up)=>{n();var rT=sn();function aT(e,o){var t=this.__data__,s=rT(t,e);return s<0?(++this.size,t.push([e,o])):t[s][1]=o,this}Up.exports=aT});var $p=S((yO,Hp)=>{n();var iT=Ep(),sT=Mp(),cT=Fp(),pT=Wp(),mT=Gp();function Do(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Do.prototype.clear=iT;Do.prototype.delete=sT;Do.prototype.get=cT;Do.prototype.has=pT;Do.prototype.set=mT;Hp.exports=Do});var qp=S((SO,Vp)=>{n();var dT=ft(),lT=on(),_T=dT(lT,"Map");Vp.exports=_T});var Kp=S((bO,zp)=>{n();var jp=Lp(),uT=$p(),gT=qp();function fT(){this.size=0,this.__data__={hash:new jp,map:new(gT||uT),string:new jp}}zp.exports=fT});var Yp=S((wO,Xp)=>{n();function TT(e){var o=typeof e;return o=="string"||o=="number"||o=="symbol"||o=="boolean"?e!=="__proto__":e===null}Xp.exports=TT});var cn=S((vO,Zp)=>{n();var hT=Yp();function xT(e,o){var t=e.__data__;return hT(o)?t[typeof o=="string"?"string":"hash"]:t.map}Zp.exports=xT});var Qp=S((IO,Jp)=>{n();var yT=cn();function ST(e){var o=yT(this,e).delete(e);return this.size-=o?1:0,o}Jp.exports=ST});var om=S((AO,em)=>{n();var bT=cn();function wT(e){return bT(this,e).get(e)}em.exports=wT});var tm=S((kO,nm)=>{n();var vT=cn();function IT(e){return vT(this,e).has(e)}nm.exports=IT});var am=S((LO,rm)=>{n();var AT=cn();function kT(e,o){var t=AT(this,e),s=t.size;return t.set(e,o),this.size+=t.size==s?0:1,this}rm.exports=kT});var sm=S((PO,im)=>{n();var LT=Kp(),PT=Qp(),ET=om(),CT=tm(),NT=am();function Mo(e){var o=-1,t=e==null?0:e.length;for(this.clear();++o<t;){var s=e[o];this.set(s[0],s[1])}}Mo.prototype.clear=LT;Mo.prototype.delete=PT;Mo.prototype.get=ET;Mo.prototype.has=CT;Mo.prototype.set=NT;im.exports=Mo});var pm=S((EO,cm)=>{n();var RT="__lodash_hash_undefined__";function DT(e){return this.__data__.set(e,RT),this}cm.exports=DT});var dm=S((CO,mm)=>{n();function MT(e){return this.__data__.has(e)}mm.exports=MT});var _m=S((NO,lm)=>{n();var OT=sm(),FT=pm(),BT=dm();function Tt(e){var o=-1,t=e==null?0:e.length;for(this.__data__=new OT;++o<t;)this.add(e[o])}Tt.prototype.add=Tt.prototype.push=FT;Tt.prototype.has=BT;lm.exports=Tt});var gm=S((RO,um)=>{n();function WT(e,o,t,s){for(var c=e.length,p=t+(s?1:-1);s?p--:++p<c;)if(o(e[p],p,e))return p;return-1}um.exports=WT});var Tm=S((DO,fm)=>{n();function UT(e){return e!==e}fm.exports=UT});var xm=S((MO,hm)=>{n();function GT(e,o,t){for(var s=t-1,c=e.length;++s<c;)if(e[s]===o)return s;return-1}hm.exports=GT});var Sm=S((OO,ym)=>{n();var HT=gm(),$T=Tm(),VT=xm();function qT(e,o,t){return o===o?VT(e,o,t):HT(e,$T,t)}ym.exports=qT});var wm=S((FO,bm)=>{n();var jT=Sm();function zT(e,o){var t=e==null?0:e.length;return!!t&&jT(e,o,0)>-1}bm.exports=zT});var Im=S((BO,vm)=>{n();function KT(e,o,t){for(var s=-1,c=e==null?0:e.length;++s<c;)if(t(o,e[s]))return!0;return!1}vm.exports=KT});var km=S((WO,Am)=>{n();function XT(e,o){return e.has(o)}Am.exports=XT});var Pm=S((UO,Lm)=>{n();var YT=ft(),ZT=on(),JT=YT(ZT,"Set");Lm.exports=JT});var Cm=S((GO,Em)=>{n();function QT(){}Em.exports=QT});var Ur=S((HO,Nm)=>{n();function eh(e){var o=-1,t=Array(e.size);return e.forEach(function(s){t[++o]=s}),t}Nm.exports=eh});var Dm=S(($O,Rm)=>{n();var Gr=Pm(),oh=Cm(),nh=Ur(),th=1/0,rh=Gr&&1/nh(new Gr([,-0]))[1]==th?function(e){return new Gr(e)}:oh;Rm.exports=rh});var Om=S((VO,Mm)=>{n();var ah=_m(),ih=wm(),sh=Im(),ch=km(),ph=Dm(),mh=Ur(),dh=200;function lh(e,o,t){var s=-1,c=ih,p=e.length,m=!0,l=[],_=l;if(t)m=!1,c=sh;else if(p>=dh){var f=o?null:ph(e);if(f)return mh(f);m=!1,c=ch,_=new ah}else _=o?[]:l;e:for(;++s<p;){var x=e[s],b=o?o(x):x;if(x=t||x!==0?x:0,m&&b===b){for(var P=_.length;P--;)if(_[P]===b)continue e;o&&_.push(b),l.push(x)}else c(_,b,t)||(_!==l&&_.push(b),l.push(x))}return l}Mm.exports=lh});var Bm=S((qO,Fm)=>{n();var _h=Om();function uh(e){return e&&e.length?_h(e):[]}Fm.exports=uh});var Ee={};ne(Ee,{css:()=>ed,default:()=>Kh});var ed,Kh,Ce=T(()=>{"use strict";n();ed=`.dapp-core-component__dappModalStyles__dappModal {
  position: fixed;
  z-index: 999;
  padding-top: 100px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
}

.dapp-core-component__dappModalStyles__dappModalContent {
  background: #fff;
  border: 2px solid #aaa;
  border-radius: 5px;
  z-index: 999;
  max-width: 500px;
  margin: auto;
  padding: 0;
  position: relative;
  width: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.dapp-core-component__dappModalStyles__dappModalHeader {
  padding: 2px 16px;
  background-color: #fff;
  font-size: 1.25rem;
}

.dapp-core-component__dappModalStyles__dappModalHeaderText {
  padding: 0 1rem;
}

.dapp-core-component__dappModalStyles__dappModalBody {
  padding: 16px 0;
}

.dapp-core-component__dappModalStyles__dappModalFooter {
  background-color: #fff;
  padding: 1rem 0.5rem;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton {
  position: absolute;
  top: 0;
  right: 0;
  float: right;
  font-weight: bold;
  border-radius: 5px;
  font-size: 1rem;
  padding: 0.5rem 1rem;
  border: 0;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton:hover,
.dapp-core-component__dappModalStyles__dappModalCloseButton:focus {
  text-decoration: none;
  cursor: pointer;
  color: #212529;
  background-color: #dee2e6;
  border-color: #e9ecef;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ed));Kh={dappModal:"dapp-core-component__dappModalStyles__dappModal",dappModalContent:"dapp-core-component__dappModalStyles__dappModalContent",dappModalHeader:"dapp-core-component__dappModalStyles__dappModalHeader",dappModalHeaderText:"dapp-core-component__dappModalStyles__dappModalHeaderText",dappModalBody:"dapp-core-component__dappModalStyles__dappModalBody",dappModalFooter:"dapp-core-component__dappModalStyles__dappModalFooter",dappModalCloseButton:"dapp-core-component__dappModalStyles__dappModalCloseButton"}});var dn={};ne(dn,{css:()=>ld,default:()=>lx});var ld,lx,ln=T(()=>{"use strict";n();ld=`.dapp-core-component__transactionStatusToastStyles__transactions-status-toast {
  position: fixed;
  bottom: 1rem;
  right: 1rem;
  z-index: 9999;
}

.dapp-core-component__transactionStatusToastStyles__transactions-status-toast-content {
  display: flex;
  flex-direction: row;
  gap: 10px;
  justify-content: left;
  align-items: center;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
}

.dapp-core-component__transactionStatusToastStyles__transactions-status-toast-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1rem;
  width: 4.5rem;
  height: 4.5rem;
  border-radius: 50%;
}
.dapp-core-component__transactionStatusToastStyles__transactions-status-toast-icon .dapp-core-component__transactionStatusToastStyles__svg {
  height: auto;
  width: 40px;
}
.dapp-core-component__transactionStatusToastStyles__transactions-status-toast-icon .dapp-core-component__transactionStatusToastStyles__svg path {
  fill: #fff;
}

.dapp-core-component__transactionStatusToastStyles__toasts {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
  font-family: "Helvetica", sans-serif;
}
.dapp-core-component__transactionStatusToastStyles__toasts.dapp-core-component__transactionStatusToastStyles__toastWrapper {
  border-radius: 0.25rem;
  box-shadow: 0 0.25rem 0.75rem rgba(0, 0, 0, 0.1);
  margin-bottom: 0.7rem;
  padding: 0.5rem;
  background-color: #fff;
  position: relative;
}
@media (min-width: 768px) {
  .dapp-core-component__transactionStatusToastStyles__toasts {
    right: 2rem;
    bottom: 2rem;
    width: 22.5rem;
    left: auto;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ld));lx={"transactions-status-toast":"dapp-core-component__transactionStatusToastStyles__transactions-status-toast",transactionsStatusToast:"dapp-core-component__transactionStatusToastStyles__transactions-status-toast","transactions-status-toast-content":"dapp-core-component__transactionStatusToastStyles__transactions-status-toast-content",transactionsStatusToastContent:"dapp-core-component__transactionStatusToastStyles__transactions-status-toast-content","transactions-status-toast-icon":"dapp-core-component__transactionStatusToastStyles__transactions-status-toast-icon",transactionsStatusToastIcon:"dapp-core-component__transactionStatusToastStyles__transactions-status-toast-icon",svg:"dapp-core-component__transactionStatusToastStyles__svg",toasts:"dapp-core-component__transactionStatusToastStyles__toasts",toastWrapper:"dapp-core-component__transactionStatusToastStyles__toastWrapper"}});var hx={};ne(hx,{DeviceConfirmationScreen:()=>Tx});module.exports=Y(hx);n();var ta=h(require("react"));n();n();n();var ys=require("react"),Io=require("react-redux");n();var pr=require("@reduxjs/toolkit"),hs=require("react-redux/lib/utils/Subscription");k();n();var Qi=h(require("lodash.throttle"));A();k();Kt();Ke();ze();var zl=[Ln],Yn=!1,Kl=(0,Qi.default)(()=>{qo(Vo())},5e3),es=e=>o=>t=>{if(zl.includes(t.type))return o(t);let s=e.getState(),c=Oe.local.getItem(be.loginExpiresAt);if(!Boolean(s==null?void 0:s.account.address))return o(t);if(c==null)return qo(Vo());let m=Date.now();return c-m<0&&!Yn?setTimeout(()=>{Yn=!0,e.dispatch(Ti())},1e3):(Yn&&(Yn=!1),Kl()),o(t)};n();n();function Zn(){return typeof sessionStorage!="undefined"}var gs=Zn()?(is(),Y(as)).default:(cs(),Y(ss)).default,fs=Zn()?(ms(),Y(ps)).default:[],Ts=Zn()?(us(),Y(_s)).default:e=>e;Jn();var w=(0,pr.configureStore)({reducer:gs,middleware:e=>e({serializableCheck:{ignoredActions:[...fs,zt.type,te.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(es)}),xs=(0,hs.createSubscription)(w),I0=Ts(w),A0=(0,pr.configureStore)({reducer:Ye});var T_={store:w,subscription:xs},mr=(0,ys.createContext)(T_),E0=(0,Io.createStoreHook)(mr),F=(0,Io.createDispatchHook)(mr),I=(0,Io.createSelectorHook)(mr);n();n();k();n();var Ss=h(require("lodash.isequal")),Qn=require("reselect"),y=(0,Qn.createSelectorCreator)(Qn.defaultMemoize,Ss.default);var ue=e=>e.account,Ue=y(ue,e=>e.address),Ae=y(ue,Ue,(e,o)=>o in e.accounts?e.accounts[o]:ee),h_=y(ue,Ae,(e,o)=>{let c=e,{accounts:t}=c,s=Ne(c,["accounts"]);return v(u({},s),{address:o.address,account:o})}),M0=y(Ae,e=>e.balance),bs=y(Ae,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),O0=y(ue,e=>e.shard),x_=y(ue,e=>e.ledgerAccount),F0=y(ue,e=>e.walletConnectAccount),B0=y(ue,e=>e.isAccountLoading),W0=y(ue,e=>e.accountLoadingError),y_=y(ue,e=>e.websocketEvent),S_=y(ue,e=>e.websocketBatchEvent);n();var Yo=e=>e.dappConfig,H0=y(Yo,e=>e.shouldUseWebViewProvider);n();var Z=e=>e.loginInfo,b_=y(Z,e=>e.loginMethod),et=y(Z,Ue,(e,o)=>Boolean(o)),j0=y(Z,e=>e.walletConnectLogin),w_=y(Z,e=>e.ledgerLogin),v_=y(Z,e=>e.walletLogin),z0=y(Z,e=>e.isLoginSessionInvalid),dr=y(Z,e=>e.tokenLogin),ws=y(Z,e=>e.logoutRoute),I_=y(Z,e=>e.isWalletConnectV2Initialized);n();var vs=e=>e.modals,Y0=y(vs,e=>e.txSubmittedModal),A_=y(vs,e=>e.notificationModal);n();var ge=e=>e.networkConfig,ot=y(ge,e=>e.network.chainId),k_=y(ge,e=>e.network.roundDuration),Q0=y(ge,e=>e.network.walletConnectBridgeAddress),L_=y(ge,e=>e.network.walletConnectV2RelayAddress),P_=y(ge,e=>e.network.walletConnectV2ProjectId),E_=y(ge,e=>e.network.walletConnectV2Options),C_=y(ge,e=>e.network.walletConnectDeepLink),K=y(ge,e=>e.network),lr=y(K,e=>e.apiAddress),Is=y(K,e=>e.explorerAddress),As=y(K,e=>{var o;return(o=e.customWalletAddress)!=null?o:e.walletAddress}),N_=y(K,e=>e.xAliasAddress),ks=y(K,e=>e.egldLabel);n();var nt=e=>e.signedMessageInfo,nw=y(nt,e=>e.isSigning),tw=y(nt,e=>e.errorMessage),rw=y(nt,e=>{let o=Object.keys(e.signedSessions),t=o.length;return e.signedSessions[o[t-1]]}),aw=y(nt,e=>{let o=Object.keys(e.signedSessions),t=o.length;return o.length>0?o[t-1]:""});n();var Ls=e=>e.toasts,R_=y(Ls,e=>e.customToasts),Ps=y(Ls,e=>e.transactionToasts);n();k();var Es={errorMessage:or,successMessage:nr,processingMessage:tr},Cs=e=>e.transactionsInfo,D_=y(Cs,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Es);n();n();var Je=require("@multiversx/sdk-core");A();n();n();n();function tt(e){try{let o=atob(e),t=btoa(o),s=r.Buffer.from(e,"base64").toString(),c=r.Buffer.from(s).toString("base64"),p=e===t||t.startsWith(e),m=e===c||c.startsWith(e);if(p&&m)return!0}catch(o){return!1}return!1}function Ze(e){return tt(e)?atob(e):e}n();n();n();n();var Ns=e=>{let o=e!=null?e:"";return tt(o)?r.Buffer.from(r.Buffer.from(o,"base64").toString().trim()):r.Buffer.from(o.trim())};n();A();var Ao=({data:e,onlySetGuardian:o})=>e?o?e.startsWith("SetGuardian"):Object.values(Dt).some(t=>e.startsWith(t)):!1;function Zo(e){var s,c,p;let o=u({},e);Ao({data:o.data,onlySetGuardian:!0})&&(delete o.senderUsername,delete o.receiverUsername);let t=new Je.Transaction(u(u(v(u(v(u(u({value:BigInt(o.value.valueOf()),data:Ns(o.data),nonce:BigInt(o.nonce.valueOf()),receiver:new Je.Address(o.receiver)},o.receiverUsername?{receiverUsername:o.receiverUsername}:{}),o.relayer?{relayer:new Je.Address(o.relayer)}:{}),{sender:new Je.Address(o.sender)}),o.senderUsername?{senderUsername:o.senderUsername}:{}),{gasLimit:BigInt((s=o.gasLimit.valueOf())!=null?s:An),gasPrice:BigInt((c=o.gasPrice.valueOf())!=null?c:kn),chainID:o.chainID.valueOf(),version:(p=o.version)!=null?p:oi}),o.options?{options:o.options}:{}),o.guardian?{guardian:new Je.Address(o.guardian)}:{}));return o.relayerSignature&&(t.relayerSignature=r.Buffer.from(o.relayerSignature,"hex")),o.guardianSignature&&(t.guardianSignature=r.Buffer.from(o.guardianSignature,"hex")),o.signature&&(t.signature=r.Buffer.from(o.signature,"hex")),t}n();n();A();n();n();n();var F_=require("@multiversx/sdk-core");n();A();n();var W_=require("@multiversx/sdk-core");M();n();var B_=require("@multiversx/sdk-core");n();n();n();var U_=require("@multiversx/sdk-core"),G_=h(require("bignumber.js"));E();n();n();n();var ko="accounts";var ur="blocks";n();n();n();n();q();n();n();q();n();n();n();var X_=h(require("bignumber.js"));A();n();var K_=require("@multiversx/sdk-core"),gr=h(require("bignumber.js"));A();n();var j_=h(require("bignumber.js"));n();gr.default.config({ROUNDING_MODE:gr.default.ROUND_FLOOR});n();n();n();A();n();n();n();n();A();n();A();n();var Y_=require("@multiversx/sdk-core"),Z_=h(require("bignumber.js"));A();n();var Tr=require("@multiversx/sdk-core"),eu=h(require("bignumber.js"));A();n();n();var J_=h(require("bignumber.js"));A();n();A();n();n();n();n();n();n();A();n();A();n();A();n();q();var nu=["reDelegateRewards","claimRewards","unBond"],tu=["wrapEgld","unwrapEgld"],ru=["unStake"],au=["unDelegate"];n();n();n();A();n();var pu=h(require("bignumber.js"));n();n();q();n();var du=h(require("bignumber.js"));n();n();n();n();var uu=h(require("bignumber.js"));M();n();n();var Fs=require("@multiversx/sdk-core");n();n();E();function it(e){let t=Object.getPrototypeOf(e).toPlainObject!=null?e:Zo(e),s=new Fs.TransactionComputer,c=v(u({},t.toPlainObject()),{hash:s.computeTransactionHash(t),senderUsername:t.senderUsername,receiverUsername:t.receiverUsername,status:"pending"});return Ao({data:c.data,onlySetGuardian:!0})&&(delete c.senderUsername,delete c.receiverUsername),c}n();var Us=require("@multiversx/sdk-web-wallet-provider");A();n();var Ws=h(require("qs"));n();U();var st=e=>{let o=new URLSearchParams(e).toString(),{pathname:t,hash:s}=N(),c=o?`?${o}`:"",p=t?`${t}${c}${s}`:"./";setTimeout(()=>{window==null||window.history.replaceState({},document==null?void 0:document.title,p)})};ho();n();ho();var gu={search:we()?window.location.search:"",removeParams:[]},Bs=(e,o=gu)=>{var m;let t={},s=we()?window.location.search:"",c=(m=o.search)!=null?m:s;if(c){let l=c?new URLSearchParams(c):[];t=Object.fromEntries(l)}let p={};return e.forEach(l=>{p[l]=t[l],delete t[l]}),o.removeParams!=null&&Object.keys(t).forEach(l=>{var f,x;let[_]=l.split("[");(((f=o.removeParams)==null?void 0:f.includes(l))||((x=o.removeParams)==null?void 0:x.includes(_)))&&delete t[l]}),{remainingParams:p,params:t,clearNavigationHistory:()=>st(t)}};var ct=({removeParams:e,search:o})=>{let t=we()?window.location.search:"",s=o!=null?o:t;if(!s)return{};let c=Ws.default.parse(s.replace("?","")),p=Object.keys(c).filter(l=>!e.includes(l)),{remainingParams:m}=Bs(p,{search:o,removeParams:e});return st(m),m};var Gs=({transaction:e,search:o})=>ct({removeParams:[...Object.keys(e),Us.WALLET_PROVIDER_CALLBACK_PARAM,Me,Ut],search:o});n();n();n();W();n();q();n();n();W();n();var fu=h(require("linkifyjs"));n();A();n();var Tu=h(require("bignumber.js"));n();M();n();n();E();n();E();n();n();n();q();n();q();n();var hu=h(require("bignumber.js"));A();q();n();q();n();var $s=require("react");M();n();n();q();n();n();var xu=require("@multiversx/sdk-core"),yu=h(require("bignumber.js"));q();n();M();n();n();M();var EP=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];n();var Vs=require("react");M();W();n();var bu=require("react");q();var BP=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];wo();n();var pt=e=>e.transactions,Lo=y(pt,e=>e.signedTransactions),qs=y(pt,e=>e.signTransactionsError),xr=y(pt,e=>e.signTransactionsCancelMessage),mt=e=>o=>Object.entries(o).reduce((t,[s,c])=>(e(c.status)&&(t[s]=c),t),{}),js=y(Lo,mt(yo)),zs=y(Lo,mt(So)),Ks=y(Lo,mt(bo)),wu=y(Lo,mt(rr)),yr=y(pt,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:v(u({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(t=>Zo(t)))||[]})}),vu=y(Lo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});n();var Xs=require("react");n();k();n();n();n();n();n();n();n();n();var Zs=require("@multiversx/sdk-extension-provider"),Js=require("@multiversx/sdk-hw-provider"),Qs=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),ec=require("@multiversx/sdk-opera-provider"),oc=require("@multiversx/sdk-passkey-provider/out"),nc=require("@multiversx/sdk-web-wallet-provider");A();Re();n();var Le=require("@multiversx/sdk-web-wallet-iframe-provider/out"),Ys=require("@multiversx/sdk-web-wallet-iframe-provider/out/constants");E();n();var Sr=Po;var Qe=e=>{switch(e==null?void 0:e.constructor){case nc.WalletProvider:return"wallet";case Sr:return"walletconnectv2";case Js.HWProvider:return"ledger";case Zs.ExtensionProvider:return"extension";case oc.PasskeyProvider:return"passkey";case Qs.MetamaskProvider:return"metamask";case ec.OperaProvider:return"opera";case le.CrossWindowProvider:return"crossWindow";case Le.IframeProvider:return"iframe";case Po:return"";default:return"extra"}};var re=e=>`Unable to perform ${e}, Provider not initialized`,Po=class{init(){return Promise.resolve(!1)}getAccount(){throw new Error(re("getAccount"))}setAccount(o){throw new Error(re(`setAccount: ${o}`))}login(o){throw new Error(re(`login with options: ${o}`))}logout(o){throw new Error(re(`logout with options: ${o}`))}getAddress(){throw new Error(re("getAddress"))}isInitialized(){return!1}isConnected(){return!1}sendTransaction(o,t){throw new Error(re(`sendTransaction with transactions: ${o} options: ${t}`))}signTransaction(o,t){throw new Error(re(`signTransaction with transactions: ${o} options: ${t}`))}signTransactions(o,t){throw new Error(re(`signTransactions with transactions: ${o} options: ${t}`))}signMessage(o,t){throw new Error(re(`signTransactions with ${o} and options ${t}`))}sendCustomMessage({method:o,params:t}){throw new Error(re(`sendCustomMessage with method: ${o} params: ${t}`))}sendCustomRequest(o){throw new Error(re(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},dt=new Po;var rc=dt,tc=null;function ac(){tc!=null&&(rc=tc)}function fe(){return rc||dt}k();n();n();n();n();n();var ic=require("@lifeomic/axios-fetch"),br=h(require("axios")),wr=(0,ic.buildAxiosFetch)(br.default),vr=(e,o)=>g(void 0,null,function*(){if(!e.ok)throw new Error(`HTTP error! Status: ${e.status}`);let s=e.clone().json(),[c]=yield Promise.all([s]);return{data:c,status:e.status,statusText:e.statusText,headers:e.headers,config:o}});function Iu(e,o,t){return g(this,null,function*(){try{let s=yield wr(e,u({method:"POST",body:o?JSON.stringify(o):void 0,headers:u({"Content-Type":"application/json"},(t==null?void 0:t.headers)||{})},t));return vr(s,t)}catch(s){throw console.error("Fetch Error:",s),s}})}function Au(e,o){return g(this,null,function*(){try{let t=yield wr(e,o);if(!t.ok)throw new Error(`HTTP error! Status: ${t.status}`);return vr(t,o)}catch(t){throw console.error("Fetch Error:",t),t}})}function ku(e,o,t){return g(this,null,function*(){try{let s=yield wr(e,u({method:"PATCH",body:o?JSON.stringify(o):void 0,headers:(t==null?void 0:t.headers)||{}},t));if(!s.ok)throw new Error(`HTTP error! Status: ${s.status}`);return vr(s,t)}catch(s){throw console.error("Fetch Error:",s),s}})}var Ge=br.default.create();Ge.get=Au;Ge.post=Iu;Ge.patch=ku;var sc=e=>Ge.get(e).then(o=>o.data);n();var eo=e=>{let o=e!=null?e:lr(w.getState());return o.endsWith("/")?o.slice(0,-1):o};n();var Lu=h(require("axios"));n();var cc=h(require("swr"));var Eo=({apiEndpoint:e})=>{let o=eo(),t=e?`${o}/${e}`:null;return(0,cc.default)(t,sc)};var Pu=(e,o)=>{var c;let s=`${(c=o==null?void 0:o.apiAddress)!=null?c:eo()}/${ko}/${e}?withGuardianInfo=true`;return Ge.get(s)},pc=(e,o)=>g(void 0,null,function*(){if(!e)return null;try{let{data:t}=yield Pu(e,o);return t}catch(t){console.error("error fetching configuration for ",e,t)}return null});var Co=(e,o)=>pc(e,o);n();E();n();n();n();W();n();function en(e){let o=w.getState(),t=bs(o);return e?t&&!isNaN(t)?Math.max(t,e.nonce):e.nonce:t}wo();n();n();n();var Eu=h(require("axios"));n();var Nu=h(require("axios"));go();n();A();n();var Ru=h(require("axios"));n();var Mu=h(require("axios"));n();n();var Ou=h(require("axios"));n();var Fu=h(require("axios"));n();n();k();E();n();n();n();n();W();n();R();M();n();k();n();var _c=require("@multiversx/sdk-core");M();U();n();wo();n();k();E();n();k();M();n();n();n();M();n();Ho();n();var gc=()=>I(qs);n();n();n();var Uu=h(require("swr"));n();n();n();n();var Tc=require("react");n();n();var hc=require("react"),xc=require("@multiversx/sdk-web-wallet-provider"),yc=require("@multiversx/sdk-web-wallet-provider"),Sc=h(require("qs"));A();k();E();Ft();W();var Gu=N();function Ar(e){let o=I(K),t=F(),{shouldFetchWalletUrlOnLoad:s}=I(Yo),{search:c}=s?Gu:N();(0,hc.useEffect)(()=>{if(c!=null){let p=Sc.default.parse(c.replace("?",""));if(p&&Me in p){let m=String(p[Me]),l=new xc.WalletProvider(`${o.walletAddress}${ni}`).getTransactionsFromWalletUrl(c);if(p.status==="cancelled"){t(Ko({sessionId:m,status:"cancelled"})),e(),ct({removeParams:[...Object.keys(Ot),yc.WALLET_PROVIDER_CALLBACK_PARAM,Me,"address"],search:c}),t(Xe(In));return}if(l.length>0){t(Ko({sessionId:m,status:"signed",transactions:l.map(f=>it(f))}));let[_]=l;Gs({transaction:_,search:c})}}}},[c])}n();var Qc=require("react"),ut=require("@multiversx/sdk-core");n();var _t=e=>Eo({apiEndpoint:e?`${ko}/${e}?withGuardianInfo=true`:null});A();Yt();n();n();n();n();n();n();var kr=({accountNonce:e,transactionNonce:o})=>o&&o>e?o:e;var bc=({latestNonce:e,transactions:o})=>o.length===0?o:o.map((t,s)=>{let c=e+s,p=t.nonce,m=kr({accountNonce:c,transactionNonce:Number(p)});return t.nonce=BigInt(m),t});var Lr=()=>{let e=I(Ue),{data:o}=_t(e);return t=>g(void 0,null,function*(){let s=o!=null?o:yield Co(e),c=en(s);return bc({latestNonce:c,transactions:t})})};n();n();n();n();go();n();var Vu=h(require("axios"));n();var ju=h(require("axios"));n();var zu=h(require("axios"));n();go();n();go();n();n();n();n();var Xu=require("@multiversx/sdk-opera-provider");n();var Yu=require("@multiversx/sdk-extension-provider");n();Re();n();n();n();n();var Hc=h(Gc());var tn=e=>{if(!e||!(0,Hc.default)(e))return null;let o=e.split(".");if(o.length!==4)return null;try{let[t,s,c,p]=o,m=JSON.parse(Ze(p)),l=Ze(t);return{ttl:Number(c),extraInfo:m,origin:l,blockHash:s}}catch(t){return console.error(`Error trying to decode ${e}:`,t),null}};n();var vg=require("@multiversx/sdk-native-auth-client");n();var qc=h(require("axios"));n();n();n();function $c(e){return g(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}n();var Vc=(e,o,t,s=0)=>g(void 0,null,function*(){try{return yield e(...t)}catch(c){return s<o.retries?((o==null?void 0:o.delay)!=null&&(yield $c(o.delay)),yield Vc(e,o,t,s+1)):null}}),Cr=(e,o={retries:5,delay:500})=>(...t)=>g(void 0,null,function*(){return yield Vc(e,o,t)});var Sg=4;var B2=Cr((e,o,t)=>g(void 0,null,function*(){if(t){let p=Math.floor(Date.now()/1e3);return{hash:yield t(),timestamp:p}}let{data:s}=yield qc.default.get(`${e}/${ur}?from=${Sg}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[c]=s;return c}));n();n();Pn();n();function bg(e){return Object.prototype.toString.call(e)==="[object String]"}var Nr=e=>{var t;if(!e||!bg(e))return null;let o=e.split(".");if(o.length!==3)return console.error("Invalid nativeAuthToken. You may be trying to decode a loginToken. Try using decodeLoginToken method instead"),null;try{let[s,c,p]=o,m=Ze(s),l=Ze(c),_=tn(l);if(!_)return{address:m,body:l,signature:p,blockHash:"",origin:"",ttl:0};let f=v(u({},_),{address:m,body:l,signature:p});return(t=_.extraInfo)!=null&&t.timestamp||delete f.extraInfo,f}catch(s){return null}};n();W();var K2={origin:N().origin,apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};n();var Ag=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");n();var kg=require("@multiversx/sdk-passkey-provider/out");n();var Og=h(require("react")),Fg=require("@fortawesome/free-solid-svg-icons");n();n();var No=h(require("react")),Zc=require("@fortawesome/free-solid-svg-icons"),Jc=require("@fortawesome/react-fontawesome");n();var Xc=h(require("react"));n();var rn=h(require("react"));n();ho();var jc=()=>!we();var Pg=()=>g(void 0,null,function*(){return yield Promise.resolve().then(()=>(Dr(),Rr))}),Eg=()=>(Dr(),Y(Rr)).default,Mr=jc();function Kc({ssrGlobalImportCallback:e,ssrImportCallback:o,clientImportCallback:t}){let[s,c]=rn.default.useState(Mr?void 0:Eg()),[p,m]=rn.default.useState(Mr||t==null?void 0:t()),l=()=>g(this,null,function*(){(e?e():Pg()).then(_=>c(_.default)),o==null||o().then(_=>m(_.default))});return(0,rn.useEffect)(()=>{Mr&&l()},[]),{globalStyles:s,styles:p}}function X(e,o){return t=>{let{globalStyles:s,styles:c}=Kc({ssrGlobalImportCallback:o.ssrGlobalStyles,ssrImportCallback:o.ssrStyles,clientImportCallback:o.clientStyles});return Xc.default.createElement(e,v(u({},t),{globalStyles:s!=null?s:{},styles:c!=null?c:{}}))}}var Ng=({styles:e})=>No.default.createElement("div",{className:e==null?void 0:e.guardianWarningToast},No.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeading},No.default.createElement(Jc.FontAwesomeIcon,{icon:Zc.faWarning,className:e==null?void 0:e.guardianWarningToastHeadingIcon}),No.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastHeadingText},"Account at risk!")),No.default.createElement("div",{className:e==null?void 0:e.guardianWarningToastDescription},"Your account has an unconfirmed guardian change, which could be caused by a possible breach! Be cautious before making any transaction.")),Rg=X(Ng,{ssrStyles:()=>Promise.resolve().then(()=>(Fr(),Or)),clientStyles:()=>(Fr(),Y(Or)).default});Ke();ze();n();n();k();var Dg={},Mg={};var Br=e=>{if(e.component!=null){let o=e,{component:t,onClose:s}=o,c=Ne(o,["component","onClose"]),p=w.dispatch(Qt(v(u({},c),{component:null,onClose:void 0})));return Dg[p==null?void 0:p.payload.toastId]=t,s&&(Mg[p==null?void 0:p.payload.toastId]=s),p==null?void 0:p.payload}return w.dispatch(Qt(e)).payload};var MM=24*60*60;A();W();n();n();n();var $g=require("react"),Vg=require("@multiversx/sdk-hw-provider");n();fo();R();M();n();jo();Ke();ze();U();n();var He=()=>{let e=fe(),o=Qe(e);return{provider:e,providerType:o}};n();n();var Vm=require("react"),qm=require("@multiversx/sdk-core"),bh=require("@multiversx/sdk-extension-provider"),wh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),vh=require("@multiversx/sdk-passkey-provider/out"),Ih=h(Bm());A();n();n();var Pe=()=>I(Ae);n();n();n();n();n();n();n();n();var pn=h(require("react"));var s5=(0,pn.createContext)({}),c5=w.getState();n();var Um=h(require("react"));n();var Hr=h(require("react")),gh=h(require("axios"));n();n();Re();n();n();var Hm=require("@multiversx/sdk-dapp-utils/out/helpers/providerNotInitializedError"),$m=require("@multiversx/sdk-webview-provider/out/WebviewProvider");R();n();R();k();M();function Gm(e,o=w.dispatch){let t=Nr(e);if(t==null)return;let{signature:s,address:c}=t;s&&e&&c&&(o(Xt({loginToken:e,signature:s,nativeAuthToken:e})),o(j({address:c,loginMethod:"extra"})))}n();n();k();function $r(){w.dispatch(vo())}n();U();n();k();M();n();n();var Th=require("@multiversx/sdk-core"),hh=h(require("bignumber.js"));A();n();var xh=h(require("bignumber.js"));A();var ae=class{constructor(){this.init=()=>g(this,null,function*(){return yield this._provider.init()});this.login=()=>g(this,null,function*(){var t;let o=yield this._provider.login();return{address:(t=o==null?void 0:o.address)!=null?t:""}});this.logout=()=>g(this,null,function*(){return w.dispatch(C()),yield this._provider.logout()});this.relogin=()=>g(this,null,function*(){let o=yield this._provider.relogin();return o?(Gm(o),ac(),o):(console.error("Unable to re-login. Missing accessToken."),null)});this.signTransactions=o=>g(this,null,function*(){let t=yield this._provider.signTransactions(o);return t||($r(),this._provider.cancelAction(),null)});this.signTransaction=o=>g(this,null,function*(){return yield this._provider.signTransaction(o)});this.signMessage=o=>g(this,null,function*(){return yield this._provider.signMessage(o)});this.cancelAction=()=>g(this,null,function*(){return yield this._provider.cancelAction()});this.isInitialized=()=>this._provider.isInitialized();this.isConnected=()=>this._provider.isConnected();this.sendCustomRequest=o=>g(this,null,function*(){this._provider.sendPostMessage({type:o.request.method,payload:o.request.params})});this.getAddress=(0,Hm.providerNotInitializedError)("getAddress");this._provider=$m.WebviewProvider.getInstance({resetStateCallback:()=>w.dispatch(C())})}static getInstance(){return ae._instance||(ae._instance=new ae),ae._instance}getAccount(){var t;let o=this._provider.getAccount();return{address:(t=o==null?void 0:o.address)!=null?t:""}}setAccount(o){this._provider.setAccount(o)}};k();E();U();W();n();var Oo=require("react"),Vr=require("@multiversx/sdk-extension-provider"),qr=require("@multiversx/sdk-metamask-provider/out/metamaskProvider"),jr=require("@multiversx/sdk-passkey-provider/out");Re();k();var ht=()=>{let e=F(),{provider:o}=He(),{nonce:t}=Pe(),[s,c]=(0,Oo.useState)(null),[p,m]=(0,Oo.useState)(),[l,_]=(0,Oo.useState)([]),f=Lr(),x=I(yr),b=I(xr),P=()=>g(void 0,null,function*(){var Te,he;let $=Boolean(x==null?void 0:x.transactions),J=(Te=x==null?void 0:x.transactions)!=null?Te:[];if($){let H=(he=x==null?void 0:x.customTransactionInformation)!=null&&he.skipUpdateNonces?J:yield f(J);_(H)}m($)});(0,Oo.useEffect)(()=>{P()},[x,t]);let L=()=>{c(null),e(Xe(null))},G=$=>{L(),ie($)};Ar(G);function ie($){var so,un,co,gn,fn,ra,Tn,aa,hn,ia,xn,sa;let J=o instanceof Vr.ExtensionProvider,Te=o instanceof jr.PasskeyProvider,he=o instanceof le.CrossWindowProvider,H=o instanceof Le.IframeProvider,_n=o instanceof qr.MetamaskProvider,St=o instanceof ae;e(vo()),e($n($)),!(!J&&!he&&!H&&!Te&&!_n)&&(L(),J&&((un=(so=Vr.ExtensionProvider.getInstance())==null?void 0:so.cancelAction)==null||un.call(so)),Te&&((gn=(co=jr.PasskeyProvider.getInstance())==null?void 0:co.cancelAction)==null||gn.call(co)),_n&&((ra=(fn=qr.MetamaskProvider.getInstance())==null?void 0:fn.cancelAction)==null||ra.call(fn)),he&&((aa=(Tn=le.CrossWindowProvider.getInstance())==null?void 0:Tn.cancelAction)==null||aa.call(Tn)),H&&((ia=(hn=Le.IframeProvider.getInstance())==null?void 0:hn.cancelAction)==null||ia.call(hn)),St&&((sa=(xn=ae.getInstance())==null?void 0:xn.cancelAction)==null||sa.call(xn)))}return{error:s,canceledTransactionsMessage:b,clearTransactionStatusMessage:L,onAbort:G,setError:c,hasTransactions:p,transactionsToSign:x&&v(u({},x),{transactions:l})}};n();k();M();jo();W();n();de();k();n();n();var kh=require("react");n();n();E();wo();n();n();var Oh=require("react");n();n();var jm=require("react");n();var Lh=h(require("axios"));n();var Nh=require("react");n();var Kr=require("react"),Ch=require("socket.io-client");k();n();var zr=class{constructor(){this.current=null;this._status="NOT_INITIALIZED";this.listeners=[]}get status(){return this._status}set status(o){this._status!==o&&(this._status=o,this.notifyListeners())}subscribe(o){return this.listeners.push(o),()=>this.unsubscribe(o)}unsubscribe(o){this.listeners=this.listeners.filter(t=>t!==o)}notifyListeners(){this.listeners.forEach(o=>o(this._status))}},zm=new zr;n();var Dh=require("react");Ho();n();n();var Xr=require("react");n();var Mh=require("react");n();var Xm=require("react");n();n();var Bh=require("react"),Wh=require("@multiversx/sdk-extension-provider");de();R();k();E();n();ve();U();U();n();n();Re();M();n();var Fh=require("react"),Ym=require("@multiversx/sdk-core");k();n();var Uh=require("react"),Gh=require("@multiversx/sdk-opera-provider");de();R();k();E();ve();U();W();n();var Hh=require("react");Wt();de();Re();R();k();k();E();ve();W();n();var jh=require("react");de();k();n();n();Gt();n();n();var $h=h(require("platform"));ho();n();n();n();n();n();n();n();fo();E();n();n();var Vh=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");n();n();jo();Ke();n();n();U();n();n();var qh=h(gt());fo();var Zm,Jm,Qm,KG=(Qm=(Jm=(Zm=De.safeWindow)==null?void 0:Zm.location)==null?void 0:Jm.origin)==null?void 0:Qm.includes("localhost");n();W();n();n();var ox=require("react");A();n();n();A();n();n();n();var zh=h(require("bignumber.js"));n();n();n();var $e=h(require("react")),id=require("react"),sd=require("react"),ea=h(require("classnames")),cd=require("react-dom");A();n();var od=h(require("react")),nd=h(require("classnames"));var Xh=({className:e,children:o,styles:t})=>od.default.createElement("div",{className:(0,nd.default)(t==null?void 0:t.dappModalBody,e)},o),Yr=X(Xh,{ssrStyles:()=>Promise.resolve().then(()=>(Ce(),Ee)),clientStyles:()=>(Ce(),Y(Ee)).default});n();var Zr=h(require("react")),td=h(require("classnames"));var Yh=({visible:e,customFooter:o,className:t,footerText:s,styles:c})=>e?Zr.default.createElement("div",{className:(0,td.default)(c==null?void 0:c.dappModalFooter,t)},o!=null?o:Zr.default.createElement("div",null,s)):null,Jr=X(Yh,{ssrStyles:()=>Promise.resolve().then(()=>(Ce(),Ee)),clientStyles:()=>(Ce(),Y(Ee)).default});n();var Wo=h(require("react")),rd=require("@fortawesome/free-solid-svg-icons"),ad=require("@fortawesome/react-fontawesome"),mn=h(require("classnames"));var Zh=({visible:e,headerText:o,customHeader:t,className:s,closeButtonClassName:c,headerTextClassName:p,onHide:m,globalStyles:l,styles:_})=>e?t?Wo.default.createElement("div",{className:(0,mn.default)(_==null?void 0:_.dappModalHeader,s)},t):Wo.default.createElement("div",{className:(0,mn.default)(_==null?void 0:_.dappModalHeader,s)},Wo.default.createElement("div",{className:(0,mn.default)(_==null?void 0:_.dappModalHeaderText,p)},o),Wo.default.createElement("button",{onClick:m,className:(0,mn.default)(_==null?void 0:_.dappModalCloseButton,l==null?void 0:l.btn,l==null?void 0:l.btnLight,c)},Wo.default.createElement(ad.FontAwesomeIcon,{size:"lg",icon:rd.faTimes}))):null,Qr=X(Zh,{ssrStyles:()=>Promise.resolve().then(()=>(Ce(),Ee)),clientStyles:()=>(Ce(),Y(Ee)).default});var Jh={showHeader:!0,showFooter:!1,headerText:"",footerText:""},Qh=({"data-testid":e="dappModal",children:o,className:t="dapp-modal-dialog-wrapper",closeOnEscape:s,config:c=Jh,id:p="dapp-modal",onHide:m,parentElement:l,visible:_,styles:f})=>{let[x,b]=(0,id.useState)(!1);if((0,sd.useEffect)(()=>{b(!0)},[]),!_)return null;let{showHeader:P,showFooter:L,headerText:G,footerText:ie,modalDialogClassName:$="dapp-modal-dialog",modalContentClassName:J="dapp-modal-dialog-content",modalHeaderClassName:Te="dapp-modal-dialog-header",modalHeaderTextClassName:he="dapp-modal-dialog-header-text",modalCloseButtonClassName:H="dapp-modal-dialog-close-button",modalBodyClassName:_n="dapp-modal-dialog-content-body",modalFooterClassName:St="dapp-modal-dialog-footer",customModalHeader:so,customModalFooter:un}=c,co=gn=>{gn.key==="Escape"&&s&&(m==null||m())};return $e.default.createElement($e.default.Fragment,null,x&&(0,cd.createPortal)($e.default.createElement("div",{id:p,role:"dialog","aria-modal":"true",className:(0,ea.default)($,f==null?void 0:f.dappModal,t),"data-testid":e,onKeyDown:co},$e.default.createElement("div",{className:(0,ea.default)(f==null?void 0:f.dappModalContent,J)},$e.default.createElement(Qr,{visible:P,headerText:G,customHeader:so,className:Te,headerTextClassName:he,closeButtonClassName:H,onHide:m}),$e.default.createElement(Yr,{className:_n},o),$e.default.createElement(Jr,{visible:L,customFooter:un,footerText:ie,className:St}))),l!=null?l:document==null?void 0:document.body))},ex=X(Qh,{ssrStyles:()=>Promise.resolve().then(()=>(Ce(),Ee)),clientStyles:()=>(Ce(),Y(Ee)).default});n();n();var pd=require("react");n();n();var md=require("react"),tx=require("@multiversx/sdk-hw-provider");de();R();k();E();ve();n();var nx=require("react");n();n();var oa=require("react");R();k();E();ve();W();n();var ax=require("react"),ix=require("@multiversx/sdk-metamask-provider/out/metamaskProvider");de();R();k();E();ve();U();U();n();var sx=require("react"),cx=require("@multiversx/sdk-passkey-provider/out");de();R();k();E();ve();U();U();n();n();var px=require("react");M();n();n();n();var dx=h(require("swr"));n();n();n();n();var mx=h(require("axios"));n();U();n();var io=h(require("react")),gd=require("@fortawesome/free-solid-svg-icons");n();var na=h(require("react"));n();var ao=h(require("react")),Uo=require("@fortawesome/free-solid-svg-icons"),xt=require("@fortawesome/react-fontawesome"),yt=h(require("classnames"));n();var ux=({type:e,globalStyles:o,styles:t})=>{switch(e){case"info":return ao.default.createElement("div",{className:(0,yt.default)(t==null?void 0:t.transactionsStatusToastIcon,o==null?void 0:o.success)},ao.default.createElement(xt.FontAwesomeIcon,{icon:Uo.faInfo,className:t==null?void 0:t.svg,size:"5x"}));case"warning":return ao.default.createElement("div",{className:(0,yt.default)(t==null?void 0:t.transactionsStatusToastIcon,o==null?void 0:o.warning,t==null?void 0:t.warningIcon)},ao.default.createElement(xt.FontAwesomeIcon,{icon:Uo.faWarning,className:t==null?void 0:t.svg,size:"5x"}));case"error":return ao.default.createElement("div",{className:(0,yt.default)(t==null?void 0:t.transactionsStatusToastIcon,o==null?void 0:o.danger,t==null?void 0:t.errorIcon)},ao.default.createElement(xt.FontAwesomeIcon,{icon:Uo.faTimes,className:t==null?void 0:t.svg,size:"5x"}));default:return null}},_d=X(ux,{ssrStyles:()=>Promise.resolve().then(()=>(ln(),dn)),clientStyles:()=>(ln(),Y(dn)).default});var gx=({type:e,message:o,styles:t})=>na.default.createElement("div",{className:t==null?void 0:t.transactionsStatusToastContent},na.default.createElement(_d,{type:e}),o),ud=X(gx,{ssrStyles:()=>Promise.resolve().then(()=>(ln(),dn)),clientStyles:()=>(ln(),Y(dn)).default});A();var fd=({signError:e,canceledTransactionsMessage:o,onDelete:t})=>{let s=I(Yo),c=(0,io.useMemo)(()=>e||o||Da,[e,o]),p=(0,io.useMemo)(()=>e?"error":o?"warning":"info",[e,o]);return(0,io.useEffect)(()=>{var m;Br({toastId:`${Za}-${Date.now()}`,title:"Transaction canceled",duration:(m=s.cancelTransactionToastDuration)!=null?m:2e4,component:()=>io.default.createElement(ud,{type:p,message:c}),onClose:()=>t==null?void 0:t(),icon:gd.faWarning})},[]),null};var Tx=({Screen:e,GuardianScreen:o,verifyReceiverScam:t,className:s})=>{var J;if(!e)return null;let{transactionsToSign:c,error:p,hasTransactions:m,onAbort:l,canceledTransactionsMessage:_}=ht(),f=gc(),{providerType:x}=He(),{isGuarded:b}=Pe(),P=()=>{l(c==null?void 0:c.sessionId)},L=p||f,G={handleClose:P,GuardianScreen:o,isGuarded:b,error:L,sessionId:c==null?void 0:c.sessionId,transactions:(J=c==null?void 0:c.transactions)!=null?J:[],providerType:x,callbackRoute:(c==null?void 0:c.callbackRoute)||Be(),className:s,verifyReceiverScam:t},ie={signError:L,canceledTransactionsMessage:_,onDelete:P},$=Boolean(L)||Boolean(_);return!$&&!m?null:$?ta.default.createElement(fd,u({},ie)):ta.default.createElement(e,u({},G))};0&&(module.exports={DeviceConfirmationScreen});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
